---
title: "CATALASE ARCHIVE"
output: html_document
date: "2024-07-19"
---

```{r}
library(tidyverse)
library(readxl)
```

```{r}
 ##other catalase
abundance_matrix_KatG <- read.delim("~/Desktop/Tara Oceans hmm/KatG/OceanGeneAtlas_Data_20240307-03_13_KatG_ArcticmetaG/abundance_matrix.csv")

abundance_matrix_KatG <- janitor::row_to_names(abundance_matrix_KatG, 1) 
#
catalase_copynum <- abundance_matrix_KatG %>%
  pivot_longer(3:182, names_to = "sample_ID", values_to = "copynum") 

#environmental
environmental_parameters_KatG <- read.delim("~/Desktop/Tara Oceans hmm/KatG/OceanGeneAtlas_Data_20240307-03_13_KatG_ArcticmetaG/environmental_parameters.csv")

#merging and filtering for surface and size fraction

catalase_copynum_enviro <- left_join(catalase_copynum,
                                     environmental_parameters_KatG,
                                     by = "sample_ID")

catalase_copynum_enviro <- catalase_copynum_enviro %>%
        filter(depth == "SRF" | depth == "DCM",
         size.fraction == "[0.22-3µm]" | size.fraction == "[0.45-0.8µm]" | size.fraction == "[0.22-0.45µm]")

#doing things on not a regional level
#turning samples with copy numbers lower than 0.0001
catalase_copynum_enviro_filt <- catalase_copynum_enviro

# catalase_copynum_enviro_filt$copynum[catalase_copynum_enviro_filt$copynum < 0.00001] <- 0

#calculating the total number of copies in each sample 
catalase_copynum_enviro_filt_totalbysample <- catalase_copynum_enviro_filt %>%
  transform(copynum = as.numeric(copynum)) %>%
  group_by(sample_ID) %>%
  summarise(copynum = sum(copynum))

# catalase_copynum_enviro_filt_meanbysample <- catalase_copynum_enviro_filt %>%
#   group_by(sample_ID) %>%
#   summarise(copynummeadn = mean(copynum))


enviro_by_sample <- environmental_parameters_KatG %>%
  rename("Temperature" = "Temperature...C.",
         "Salinity" = "Salinity..PSU.",
         "Chlorophyll_A" = "Chlorophyll_A..mg.m..3.",
         "Depth_m" = "Depth..m.",
         "O2" = "O2..µmol.kg.",
         "POC" = "POC...µg.kg.",
         "Alkalinity" = "Alkalinity..µmol.kg.",
         "CO2" = "CO2..µmol.kg.",
         "PO4" = "PO4..µmol.l.",
         "NPP" = "NPP_C...mg.m..2.day.",
         "Coast_km" = "Distance_coast..km.",
         "PAR" = "PAR..mol.quanta.m..2.day.",
         "NO3_NO2" = "NO3_NO2..µmol.l.",
         "Iron_5m" = "Iron_5m...µmol.l.") %>%
  transform(Temperature = as.numeric(Temperature),
          Salinity = as.numeric(Salinity),
          Chlorophyll_A = as.numeric(Chlorophyll_A),
          Alkalinity = as.numeric(Alkalinity),
          PAR = as.numeric(PAR),
          O2 = as.numeric(O2),
          POC = as.numeric(POC),
          CO2 = as.numeric(CO2),
          PO4 = as.numeric(PO4),
          NPP = as.numeric(NPP),
          NO3_NO2 = as.numeric(NO3_NO2),
          Iron_5m = as.numeric(Iron_5m),
          Coast_km = as.numeric(Coast_km)) 

copynumbysample_enviro <- left_join(catalase_copynum_enviro_filt_totalbysample,
                                    enviro_by_sample,
                                    by = "sample_ID")



#
############################################################################################
#repeating for rbcL
abundance_matrix_rbcLmetaG <- read.delim("~/Desktop/Tara Oceans hmm/rbcL/OceanGeneAtlas_Data_20240307-03_54_Rub_ArcticmetaG/abundance_matrix.csv")

#making the new first row into the column names 
abundance_matrix_rbcLmetaG <- janitor::row_to_names(abundance_matrix_rbcLmetaG, 1) 
#reformat so the sample IDs are rows so can merge with environmental
rubisco_copynum <- abundance_matrix_rbcLmetaG %>%
  pivot_longer(3:182, names_to = "sample_ID", values_to = "copynum_rub") 

#environmental
environmental_parameters_rbcLmetaG <- environmental_parameters <- read.delim("~/Desktop/Tara Oceans hmm/rbcL/OceanGeneAtlas_Data_20240307-03_54_Rub_ArcticmetaG/environmental_parameters.csv")
#merging and filtering for surface and size fraction

rubisco_copynum_enviro <- left_join(rubisco_copynum,
                                     environmental_parameters_rbcLmetaG,
                                     by = "sample_ID")

rubisco_copynum_enviro <- rubisco_copynum_enviro %>%
        filter(depth == "SRF" | depth == "DCM",
         size.fraction == "[0.22-3µm]" | size.fraction == "[0.45-0.8µm]" | size.fraction == "[0.22-0.45µm]")

#doing things on not a regional level
#turning samples with copy numbers lower than 0.0001
rubisco_copynum_enviro_filt <- rubisco_copynum_enviro 

#calculating the total number of copies in each sample 
rubisco_copynum_enviro_filt_totalbysample <- rubisco_copynum_enviro_filt %>%
  transform(copynum_rub = as.numeric(copynum_rub)) %>%
  group_by(sample_ID) %>%
  summarise(copynum_rub = sum(copynum_rub))

# rubisco_copynum_enviro_filt_meanbysample <- rubisco_copynum_enviro_filt %>%
#   transform(copynum_rub = as.numeric(copynum_rub)) %>%
#   group_by(sample_ID) %>%
#   summarise(copynum_rub_mean = mean(copynum_rub))


enviro_by_sample_rub <- environmental_parameters_rbcLmetaG %>%
  rename("Temperature" = "Temperature...C.",
         "Salinity" = "Salinity..PSU.",
         "Chlorophyll_A" = "Chlorophyll_A..mg.m..3.",
         "Depth_m" = "Depth..m.",
         "O2" = "O2..µmol.kg.",
         "POC" = "POC...µg.kg.",
         "Alkalinity" = "Alkalinity..µmol.kg.",
         "CO2" = "CO2..µmol.kg.",
         "PO4" = "PO4..µmol.l.",
         "NPP" = "NPP_C...mg.m..2.day.",
         "Coast_km" = "Distance_coast..km.",
         "PAR" = "PAR..mol.quanta.m..2.day.",
         "NO3_NO2" = "NO3_NO2..µmol.l.",
         "Iron_5m" = "Iron_5m...µmol.l.") %>%
  transform(Temperature = as.numeric(Temperature),
          Salinity = as.numeric(Salinity),
          Chlorophyll_A = as.numeric(Chlorophyll_A),
          Alkalinity = as.numeric(Alkalinity),
          PAR = as.numeric(PAR),
          O2 = as.numeric(O2),
          POC = as.numeric(POC),
          CO2 = as.numeric(CO2),
          PO4 = as.numeric(PO4),
          NPP = as.numeric(NPP),
          NO3_NO2 = as.numeric(NO3_NO2),
          Iron_5m = as.numeric(Iron_5m),
          Coast_km = as.numeric(Coast_km)) 

copynumbysample_rubenviro <- left_join(rubisco_copynum_enviro_filt_totalbysample,
                                    enviro_by_sample_rub,
                                    by = "sample_ID")

#gene by gene: what are the best predictors of each gene?
#starting with catalase

#importing other catalase files
abundance_matrix_KatE <- read.delim("~/Desktop/Tara Oceans hmm/KatE/OceanGeneAtlas_Data_20240307-03_24_KatE_ArcticmetaG/abundance_matrix.csv")
environmental_parameters_KatE <- read.delim("~/Desktop/Tara Oceans hmm/KatE/OceanGeneAtlas_Data_20240307-03_24_KatE_ArcticmetaG/environmental_parameters.csv")

abundance_matrix_CatMn <- read.delim("~/Desktop/Tara Oceans hmm/Manganese_catalase/OceanGeneAtlas_Data_20240307-03_45_CatMn_ArcticmetaG/abundance_matrix.csv")
environmental_parameters_CatMn <- read.delim("~/Desktop/Tara Oceans hmm/Manganese_catalase/OceanGeneAtlas_Data_20240307-03_45_CatMn_ArcticmetaG/environmental_parameters.csv")

##################
KatE_copynum <- abundance_matrix_KatE %>%
  janitor::row_to_names(1) %>%
  pivot_longer(3:182, names_to = "sample_ID", values_to = "copynum_KatE") %>%
  mutate(copynum_KatE = as.numeric(copynum_KatE))

#environmental

#merging and filtering for surface and size fraction

KatE_copynum_enviro <- left_join(KatE_copynum,
                                     environmental_parameters_KatE,
                                     by = "sample_ID")

KatE_copynum_enviro <- KatE_copynum_enviro %>%
        filter(depth == "SRF" | depth == "DCM",
                 size.fraction == "[0.22-3µm]" | size.fraction == "[0.45-0.8µm]" | size.fraction == "[0.22-0.45µm]")
  
       

# KatE_copynum_enviro2 <- KatE_copynum_enviro %>%
#         filter(depth == "SRF" | depth == "DCM",
#                  size.fraction == "[0.22-3µm]" | size.fraction == "[0.45-0.8µm]" | size.fraction == "[0.22-0.45µm]")
#doing things on not a regional level
#turning samples with copy numbers lower than 0.0001
KatE_copynum_enviro_filt <- KatE_copynum_enviro 
# KatE_copynum_enviro_filt <- KatE_copynum_enviro2


# KatE_copynum_enviro_filt$copynum_KatE[KatE_copynum_enviro_filt$copynum_KatE < 0.00001] <- 0

#calculating the total number of copies in each sample 
KatE_copynum_enviro_filt_totalbysample <- KatE_copynum_enviro_filt %>%
  group_by(sample_ID) %>%
  summarise(copynum_KatE = sum(copynum_KatE))

# catalase_copynum_enviro_filt_meanbysample <- catalase_copynum_enviro_filt %>%
#   group_by(sample_ID) %>%
#   summarise(copynummean = mean(copynum))


enviro_by_sample_KatE <- environmental_parameters_KatE %>%
  rename("Temperature" = "Temperature...C.",
         "Salinity" = "Salinity..PSU.",
         "Chlorophyll_A" = "Chlorophyll_A..mg.m..3.",
         "Depth_m" = "Depth..m.",
         "O2" = "O2..µmol.kg.",
         "POC" = "POC...µg.kg.",
         "Alkalinity" = "Alkalinity..µmol.kg.",
         "CO2" = "CO2..µmol.kg.",
         "PO4" = "PO4..µmol.l.",
         "NPP" = "NPP_C...mg.m..2.day.",
         "Coast_km" = "Distance_coast..km.",
         "PAR" = "PAR..mol.quanta.m..2.day.",
         "NO3_NO2" = "NO3_NO2..µmol.l.",
         "Iron_5m" = "Iron_5m...µmol.l.") %>%
  transform(Temperature = as.numeric(Temperature),
          Salinity = as.numeric(Salinity),
          Chlorophyll_A = as.numeric(Chlorophyll_A),
          Alkalinity = as.numeric(Alkalinity),
          PAR = as.numeric(PAR),
          O2 = as.numeric(O2),
          POC = as.numeric(POC),
          CO2 = as.numeric(CO2),
          PO4 = as.numeric(PO4),
          NPP = as.numeric(NPP),
          NO3_NO2 = as.numeric(NO3_NO2),
          Iron_5m = as.numeric(Iron_5m),
          Coast_km = as.numeric(Coast_km)) 

copynum_KatE_bysample_enviro <- left_join(KatE_copynum_enviro_filt_totalbysample,
                                    enviro_by_sample_KatE,
                                    by = "sample_ID")

###repeating for Mn catalase
##################
CatMn_copynum <- abundance_matrix_CatMn %>%
  janitor::row_to_names(1) %>%
  pivot_longer(3:145, names_to = "sample_ID", values_to = "copynum_CatMn") %>%
  mutate(copynum_CatMn = as.numeric(copynum_CatMn))

#environmental

#merging and filtering for surface and size fraction

CatMn_copynum_enviro <- left_join(CatMn_copynum,
                                     environmental_parameters_CatMn,
                                     by = "sample_ID")
       

CatMn_copynum_enviro <- CatMn_copynum_enviro %>%
        filter(depth == "SRF" | depth == "DCM",
                 size.fraction == "[0.22-3µm]" | size.fraction == "[0.45-0.8µm]" | size.fraction == "[0.22-0.45µm]")
#doing things on not a regional level
#turning samples with copy numbers lower than 0.0001
CatMn_copynum_enviro_filt <- CatMn_copynum_enviro 



# KatE_copynum_enviro_filt$copynum_KatE[KatE_copynum_enviro_filt$copynum_KatE < 0.00001] <- 0

#calculating the total number of copies in each sample 
CatMn_copynum_enviro_filt_totalbysample <- CatMn_copynum_enviro_filt %>%
  group_by(sample_ID) %>%
  summarise(copynum_CatMn = sum(copynum_CatMn))

# catalase_copynum_enviro_filt_meanbysample <- catalase_copynum_enviro_filt %>%
#   group_by(sample_ID) %>%
#   summarise(copynummean = mean(copynum))


enviro_by_sample_CatMn <- environmental_parameters_CatMn %>%
  rename("Temperature" = "Temperature...C.",
         "Salinity" = "Salinity..PSU.",
         "Chlorophyll_A" = "Chlorophyll_A..mg.m..3.",
         "Depth_m" = "Depth..m.",
         "O2" = "O2..µmol.kg.",
         "POC" = "POC...µg.kg.",
         "Alkalinity" = "Alkalinity..µmol.kg.",
         "CO2" = "CO2..µmol.kg.",
         "PO4" = "PO4..µmol.l.",
         "NPP" = "NPP_C...mg.m..2.day.",
         "Coast_km" = "Distance_coast..km.",
         "PAR" = "PAR..mol.quanta.m..2.day.",
         "NO3_NO2" = "NO3_NO2..µmol.l.",
         "Iron_5m" = "Iron_5m...µmol.l.") %>%
  transform(Temperature = as.numeric(Temperature),
          Salinity = as.numeric(Salinity),
          Chlorophyll_A = as.numeric(Chlorophyll_A),
          Alkalinity = as.numeric(Alkalinity),
          PAR = as.numeric(PAR),
          O2 = as.numeric(O2),
          POC = as.numeric(POC),
          CO2 = as.numeric(CO2),
          PO4 = as.numeric(PO4),
          NPP = as.numeric(NPP),
          NO3_NO2 = as.numeric(NO3_NO2),
          Iron_5m = as.numeric(Iron_5m),
          Coast_km = as.numeric(Coast_km)) 

copynum_CatMn_bysample_enviro <- left_join(CatMn_copynum_enviro_filt_totalbysample,
                                    enviro_by_sample_CatMn,
                                    by = "sample_ID")

#preparing for models with all datasets combined
#adding another gene
#starting with catalase

#importing other catalase files
abundance_matrix_psbO <- read.delim("~/Desktop/Tara Oceans hmm/psbO/abundance_matrix.csv")
environmental_parameters_psbO <- read.delim("~/Desktop/Tara Oceans hmm/psbO/environmental_parameters.csv")

##################
psbO_copynum <- abundance_matrix_psbO %>%
  janitor::row_to_names(1) %>%
  pivot_longer(3:182, names_to = "sample_ID", values_to = "copynum_psbO") %>%
  mutate(copynum_psbO = as.numeric(copynum_psbO))

#environmental

#merging and filtering for surface and size fraction

psbO_copynum_enviro <- left_join(psbO_copynum,
                                     environmental_parameters_psbO,
                                     by = "sample_ID")

psbO_copynum_enviro <- psbO_copynum_enviro %>%
        filter(depth == "SRF" | depth == "DCM",
                 size.fraction == "[0.22-3µm]" | size.fraction == "[0.45-0.8µm]" | size.fraction == "[0.22-0.45µm]")
  
       

# KatE_copynum_enviro2 <- KatE_copynum_enviro %>%
#         filter(depth == "SRF" | depth == "DCM",
#                  size.fraction == "[0.22-3µm]" | size.fraction == "[0.45-0.8µm]" | size.fraction == "[0.22-0.45µm]")
#doing things on not a regional level
#turning samples with copy numbers lower than 0.0001
psbO_copynum_enviro_filt <- psbO_copynum_enviro 
# KatE_copynum_enviro_filt <- KatE_copynum_enviro2


# KatE_copynum_enviro_filt$copynum_KatE[KatE_copynum_enviro_filt$copynum_KatE < 0.00001] <- 0

#calculating the total number of copies in each sample 
psbO_copynum_enviro_filt_totalbysample <- psbO_copynum_enviro_filt %>%
  group_by(sample_ID) %>%
  summarise(copynum_psbO = sum(copynum_psbO))

# catalase_copynum_enviro_filt_meanbysample <- catalase_copynum_enviro_filt %>%
#   group_by(sample_ID) %>%
#   summarise(copynummean = mean(copynum))


enviro_by_sample_psbO <- environmental_parameters_psbO %>%
  rename("Temperature" = "Temperature...C.",
         "Salinity" = "Salinity..PSU.",
         "Chlorophyll_A" = "Chlorophyll_A..mg.m..3.",
         "Depth_m" = "Depth..m.",
         "O2" = "O2..µmol.kg.",
         "POC" = "POC...µg.kg.",
         "Alkalinity" = "Alkalinity..µmol.kg.",
         "CO2" = "CO2..µmol.kg.",
         "PO4" = "PO4..µmol.l.",
         "NPP" = "NPP_C...mg.m..2.day.",
         "Coast_km" = "Distance_coast..km.",
         "PAR" = "PAR..mol.quanta.m..2.day.",
         "NO3_NO2" = "NO3_NO2..µmol.l.",
         "Iron_5m" = "Iron_5m...µmol.l.") %>%
  transform(Temperature = as.numeric(Temperature),
          Salinity = as.numeric(Salinity),
          Chlorophyll_A = as.numeric(Chlorophyll_A),
          Alkalinity = as.numeric(Alkalinity),
          PAR = as.numeric(PAR),
          O2 = as.numeric(O2),
          POC = as.numeric(POC),
          CO2 = as.numeric(CO2),
          PO4 = as.numeric(PO4),
          NPP = as.numeric(NPP),
          NO3_NO2 = as.numeric(NO3_NO2),
          Iron_5m = as.numeric(Iron_5m),
          Coast_km = as.numeric(Coast_km)) 

copynum_psbO_bysample_enviro <- left_join(psbO_copynum_enviro_filt_totalbysample,
                                    enviro_by_sample_psbO,
                                    by = "sample_ID")

#try with light protein
abundance_matrix_light <- read.delim("~/Desktop/Tara Oceans hmm/light/abundance_matrix.csv")
environmental_parameters_light <- read.delim("~/Desktop/Tara Oceans hmm/light/environmental_parameters.csv")
##################
light_copynum <- abundance_matrix_light %>%
  janitor::row_to_names(1) %>%
  pivot_longer(3:143, names_to = "sample_ID", values_to = "copynum_light") %>%
  mutate(copynum_light = as.numeric(copynum_light))

#environmental

#merging and filtering for surface and size fraction

light_copynum_enviro <- left_join(light_copynum,
                                     environmental_parameters_light,
                                     by = "sample_ID")

light_copynum_enviro <- light_copynum_enviro %>%
        filter(depth == "SRF" | depth == "DCM",
                 size.fraction == "[0.22-3µm]" | size.fraction == "[0.45-0.8µm]" | size.fraction == "[0.22-0.45µm]")
  
       

# KatE_copynum_enviro2 <- KatE_copynum_enviro %>%
#         filter(depth == "SRF" | depth == "DCM",
#                  size.fraction == "[0.22-3µm]" | size.fraction == "[0.45-0.8µm]" | size.fraction == "[0.22-0.45µm]")
#doing things on not a regional level
#turning samples with copy numbers lower than 0.0001
light_copynum_enviro_filt <- light_copynum_enviro 
# KatE_copynum_enviro_filt <- KatE_copynum_enviro2


# KatE_copynum_enviro_filt$copynum_KatE[KatE_copynum_enviro_filt$copynum_KatE < 0.00001] <- 0

#calculating the total number of copies in each sample 
light_copynum_enviro_filt_totalbysample <- light_copynum_enviro_filt %>%
  group_by(sample_ID) %>%
  summarise(copynum_light = sum(copynum_light))

# catalase_copynum_enviro_filt_meanbysample <- catalase_copynum_enviro_filt %>%
#   group_by(sample_ID) %>%
#   summarise(copynummean = mean(copynum))


enviro_by_sample_light <- environmental_parameters_light %>%
  rename("Temperature" = "Temperature...C.",
         "Salinity" = "Salinity..PSU.",
         "Chlorophyll_A" = "Chlorophyll_A..mg.m..3.",
         "Depth_m" = "Depth..m.",
         "O2" = "O2..µmol.kg.",
         "POC" = "POC...µg.kg.",
         "Alkalinity" = "Alkalinity..µmol.kg.",
         "CO2" = "CO2..µmol.kg.",
         "PO4" = "PO4..µmol.l.",
         "NPP" = "NPP_C...mg.m..2.day.",
         "Coast_km" = "Distance_coast..km.",
         "PAR" = "PAR..mol.quanta.m..2.day.",
         "NO3_NO2" = "NO3_NO2..µmol.l.",
         "Iron_5m" = "Iron_5m...µmol.l.") %>%
  transform(Temperature = as.numeric(Temperature),
          Salinity = as.numeric(Salinity),
          Chlorophyll_A = as.numeric(Chlorophyll_A),
          Alkalinity = as.numeric(Alkalinity),
          PAR = as.numeric(PAR),
          O2 = as.numeric(O2),
          POC = as.numeric(POC),
          CO2 = as.numeric(CO2),
          PO4 = as.numeric(PO4),
          NPP = as.numeric(NPP),
          NO3_NO2 = as.numeric(NO3_NO2),
          Iron_5m = as.numeric(Iron_5m),
          Coast_km = as.numeric(Coast_km)) 

copynum_light_bysample_enviro <- left_join(light_copynum_enviro_filt_totalbysample,
                                    enviro_by_sample_light,
                                    by = "sample_ID")

#merging these together 
#copynumbysample_enviro copynumbysample_rubenviro copynum_KatE_bysample_enviro copynum_CatMn_bysample_enviro
#selecting only the sample ID and the copy number columns
copynumbysample_enviro_clean <- copynumbysample_enviro %>%
  select(sample_ID, copynum)
copynumbysample_rubenviro_clean <- copynumbysample_rubenviro %>%
  select(sample_ID, copynum_rub)
copynum_KatE_bysample_enviro_clean <- copynum_KatE_bysample_enviro %>%
  select(sample_ID, copynum_KatE)
copynum_CatMn_bysample_enviro_clean <- copynum_CatMn_bysample_enviro %>%
  select(sample_ID, copynum_CatMn)
copynum_psbO_bysample_enviro_clean <- copynum_psbO_bysample_enviro %>%
  select(sample_ID, copynum_psbO)
copynum_light_bysample_enviro_clean <- copynum_light_bysample_enviro %>%
  select(sample_ID, copynum_light)


KatG_rubisco_models <- merge(copynumbysample_enviro_clean,
                  copynumbysample_rubenviro_clean,
                  by = "sample_ID",
                  all.x = TRUE,
                  all.y = TRUE)

KatG_rubisco_KatE_models <- merge(KatG_rubisco_models,
                  copynum_KatE_bysample_enviro_clean,
                  by = "sample_ID",
                  all.x = TRUE,
                  all.y = TRUE)

KatG_rubisco_KatE_CatMn_models <- merge(KatG_rubisco_KatE_models,
                  copynum_CatMn_bysample_enviro_clean,
                  by = "sample_ID",
                  all.x = TRUE,
                  all.y = TRUE)

KatG_rubisco_KatE_CatMn_psbO_models <- merge(KatG_rubisco_KatE_CatMn_models,
                  copynum_psbO_bysample_enviro_clean,
                  by = "sample_ID",
                  all.x = TRUE,
                  all.y = TRUE)

KatG_rubisco_KatE_CatMn_psbO_light_models <- merge(KatG_rubisco_KatE_CatMn_psbO_models,
                  copynum_light_bysample_enviro_clean,
                  by = "sample_ID",
                  all.x = TRUE,
                  all.y = TRUE)



KatG_rubisco_KatE_CatMn_psbO_light_models[is.na(KatG_rubisco_KatE_CatMn_psbO_light_models)] <- 0

##preparing SCGs
SCG1_metaG <- read.delim("~/Desktop/Tara Oceans hmm/SCG1_synt_2d/OceanGeneAtlas_Data_20240306-22_43_SCG1_ArcticmetaG/abundance_matrix.csv")

#making the new first row into the column names 
SCG1_metaG <- janitor::row_to_names(SCG1_metaG, 1) 
#reformat so the sample IDs are rows so can merge with environmental
SCG1_copynum <- SCG1_metaG %>%
  pivot_longer(3:182, names_to = "sample_ID", values_to = "copynum_SCG1") 

#environmental
environmental_parameters_SCG1metaG <- read.delim("~/Desktop/Tara Oceans hmm/SCG1_synt_2d/OceanGeneAtlas_Data_20240306-22_43_SCG1_ArcticmetaG/environmental_parameters.csv")
#merging and filtering for surface and size fraction

SCG1_copynum_enviro <- left_join(SCG1_copynum,
                                     environmental_parameters_SCG1metaG ,
                                     by = "sample_ID")

SCG1_copynum_enviro <- SCG1_copynum_enviro %>%
        filter(depth == "SRF" | depth == "DCM",
         size.fraction == "[0.22-3µm]" | size.fraction == "[0.45-0.8µm]" | size.fraction == "[0.22-0.45µm]")

#doing things on not a regional level
#turning samples with copy numbers lower than 0.0001
SCG1_copynum_enviro_filt <- SCG1_copynum_enviro 

#calculating the total number of copies in each sample 
SCG1_copynum_enviro_filt_totalbysample <- SCG1_copynum_enviro_filt %>%
  transform(copynum_SCG1 = as.numeric(copynum_SCG1)) %>%
  group_by(sample_ID) %>%
  summarise(copynum_SCG1 = sum(copynum_SCG1))

#####

SCG2_metaG <- read.delim("~/Desktop/Tara Oceans hmm/SCG2_Seryl_tRNA_N/OceanGeneAtlas_Data_20240306-22_51_SCG2_ArcticmetaG/abundance_matrix.csv")

#making the new first row into the column names 
SCG2_metaG <- janitor::row_to_names(SCG2_metaG, 1) 
#reformat so the sample IDs are rows so can merge with environmental
SCG2_copynum <- SCG2_metaG %>%
  pivot_longer(3:182, names_to = "sample_ID", values_to = "copynum_SCG2") 

#environmental
environmental_parameters_SCG2metaG <- read.delim("~/Desktop/Tara Oceans hmm/SCG2_Seryl_tRNA_N/OceanGeneAtlas_Data_20240306-22_51_SCG2_ArcticmetaG/environmental_parameters.csv")
#merging and filtering for surface and size fraction

SCG2_copynum_enviro <- left_join(SCG2_copynum,
                                     environmental_parameters_SCG2metaG ,
                                     by = "sample_ID")

SCG2_copynum_enviro <- SCG2_copynum_enviro %>%
        filter(depth == "SRF" | depth == "DCM",
         size.fraction == "[0.22-3µm]" | size.fraction == "[0.45-0.8µm]" | size.fraction == "[0.22-0.45µm]")

#doing things on not a regional level
#turning samples with copy numbers lower than 0.0001
SCG2_copynum_enviro_filt <- SCG2_copynum_enviro 

#calculating the total number of copies in each sample 
SCG2_copynum_enviro_filt_totalbysample <- SCG2_copynum_enviro_filt %>%
  transform(copynum_SCG2 = as.numeric(copynum_SCG2)) %>%
  group_by(sample_ID) %>%
  summarise(copynum_SCG2 = sum(copynum_SCG2))

##############
SCG3_metaG <- read.delim("~/Desktop/Tara Oceans hmm/SCG3_synt_1e/OceanGeneAtlas_Data_20240306-22_57_SCG3_ArcticmetaG/abundance_matrix.csv")

#making the new first row into the column names 
SCG3_metaG <- janitor::row_to_names(SCG3_metaG, 1) 
#reformat so the sample IDs are rows so can merge with environmental
SCG3_copynum <- SCG3_metaG %>%
  pivot_longer(3:182, names_to = "sample_ID", values_to = "copynum_SCG3") 

#environmental
environmental_parameters_SCG3metaG <- read.delim("~/Desktop/Tara Oceans hmm/SCG3_synt_1e/OceanGeneAtlas_Data_20240306-22_57_SCG3_ArcticmetaG/environmental_parameters.csv")
#merging and filtering for surface and size fraction

SCG3_copynum_enviro <- left_join(SCG3_copynum,
                                     environmental_parameters_SCG3metaG ,
                                     by = "sample_ID")

SCG3_copynum_enviro <- SCG3_copynum_enviro %>%
        filter(depth == "SRF" | depth == "DCM",
         size.fraction == "[0.22-3µm]" | size.fraction == "[0.45-0.8µm]" | size.fraction == "[0.22-0.45µm]")

#doing things on not a regional level
#turning samples with copy numbers lower than 0.0001
SCG3_copynum_enviro_filt <- SCG3_copynum_enviro 

#calculating the total number of copies in each sample 
SCG3_copynum_enviro_filt_totalbysample <- SCG3_copynum_enviro_filt %>%
  transform(copynum_SCG3 = as.numeric(copynum_SCG3)) %>%
  group_by(sample_ID) %>%
  summarise(copynum_SCG3 = sum(copynum_SCG3))

##########

SCG4_metaG <- read.delim("~/Downloads/OceanGeneAtlas_Data_20240306-23_00_SCG4_ArcticmetaG/abundance_matrix.csv")

#making the new first row into the column names 
SCG4_metaG <- janitor::row_to_names(SCG4_metaG, 1) 
#reformat so the sample IDs are rows so can merge with environmental
SCG4_copynum <- SCG4_metaG %>%
  pivot_longer(3:182, names_to = "sample_ID", values_to = "copynum_SCG4") 

#environmental
environmental_parameters_SCG4metaG <- read.delim("~/Downloads/OceanGeneAtlas_Data_20240306-23_00_SCG4_ArcticmetaG/environmental_parameters.csv")
#merging and filtering for surface and size fraction

SCG4_copynum_enviro <- left_join(SCG4_copynum,
                                     environmental_parameters_SCG4metaG ,
                                     by = "sample_ID")

SCG4_copynum_enviro <- SCG4_copynum_enviro %>%
        filter(depth == "SRF" | depth == "DCM",
         size.fraction == "[0.22-3µm]" | size.fraction == "[0.45-0.8µm]" | size.fraction == "[0.22-0.45µm]")

#doing things on not a regional level
#turning samples with copy numbers lower than 0.0001
SCG4_copynum_enviro_filt <- SCG4_copynum_enviro 

#calculating the total number of copies in each sample 
SCG4_copynum_enviro_filt_totalbysample <- SCG4_copynum_enviro_filt %>%
  transform(copynum_SCG4 = as.numeric(copynum_SCG4)) %>%
  group_by(sample_ID) %>%
  summarise(copynum_SCG4 = sum(copynum_SCG4))

################
SCG5_metaG <- read.delim("~/Desktop/Tara Oceans hmm/SCG5_synt_1_2/OceanGeneAtlas_Data_20240307-02_07_SCG5_ArcticmetaG/abundance_matrix.csv")

#making the new first row into the column names 
SCG5_metaG <- janitor::row_to_names(SCG5_metaG, 1) 
#reformat so the sample IDs are rows so can merge with environmental
SCG5_copynum <- SCG5_metaG %>%
  pivot_longer(3:182, names_to = "sample_ID", values_to = "copynum_SCG5") 

#environmental
environmental_parameters_SCG5metaG <- read.delim("~/Desktop/Tara Oceans hmm/SCG5_synt_1_2/OceanGeneAtlas_Data_20240307-02_07_SCG5_ArcticmetaG/environmental_parameters.csv")
#merging and filtering for surface and size fraction

SCG5_copynum_enviro <- left_join(SCG5_copynum,
                                     environmental_parameters_SCG5metaG ,
                                     by = "sample_ID")

SCG5_copynum_enviro <- SCG5_copynum_enviro %>%
        filter(depth == "SRF" | depth == "DCM",
         size.fraction == "[0.22-3µm]" | size.fraction == "[0.45-0.8µm]" | size.fraction == "[0.22-0.45µm]")

#doing things on not a regional level
#turning samples with copy numbers lower than 0.0001
SCG5_copynum_enviro_filt <- SCG5_copynum_enviro 

#calculating the total number of copies in each sample 
SCG5_copynum_enviro_filt_totalbysample <- SCG5_copynum_enviro_filt %>%
  transform(copynum_SCG5 = as.numeric(copynum_SCG5)) %>%
  group_by(sample_ID) %>%
  summarise(copynum_SCG5 = sum(copynum_SCG5))

###########
SCG6_metaG <- read.delim("~/Desktop/Tara Oceans hmm/SCG6_tRNA_synt_1/OceanGeneAtlas_Data_20240307-02_21_SCG6_ArcticmetaG/abundance_matrix.csv")

#making the new first row into the column names 
SCG6_metaG <- janitor::row_to_names(SCG6_metaG, 1) 
#reformat so the sample IDs are rows so can merge with environmental
SCG6_copynum <- SCG6_metaG %>%
  pivot_longer(3:182, names_to = "sample_ID", values_to = "copynum_SCG6") 

#environmental
environmental_parameters_SCG6metaG <- read.delim("~/Desktop/Tara Oceans hmm/SCG6_tRNA_synt_1/OceanGeneAtlas_Data_20240307-02_21_SCG6_ArcticmetaG/environmental_parameters.csv")
#merging and filtering for surface and size fraction

SCG6_copynum_enviro <- left_join(SCG6_copynum,
                                     environmental_parameters_SCG6metaG ,
                                     by = "sample_ID")

SCG6_copynum_enviro <- SCG6_copynum_enviro %>%
        filter(depth == "SRF" | depth == "DCM",
         size.fraction == "[0.22-3µm]" | size.fraction == "[0.45-0.8µm]" | size.fraction == "[0.22-0.45µm]")

#doing things on not a regional level
#turning samples with copy numbers lower than 0.0001
SCG6_copynum_enviro_filt <- SCG6_copynum_enviro 

#calculating the total number of copies in each sample 
SCG6_copynum_enviro_filt_totalbysample <- SCG6_copynum_enviro_filt %>%
  transform(copynum_SCG6 = as.numeric(copynum_SCG6)) %>%
  group_by(sample_ID) %>%
  summarise(copynum_SCG6 = sum(copynum_SCG6))
##########
SCG7_metaG <- read.delim("~/Desktop/Tara Oceans hmm/SCG7_TsaD/OceanGeneAtlas_Data_20240307-02_38_SCG7_ArcticmetaG/abundance_matrix.csv")


#making the new first row into the column names 
SCG7_metaG <- janitor::row_to_names(SCG7_metaG, 1) 
#reformat so the sample IDs are rows so can merge with environmental
SCG7_copynum <- SCG7_metaG %>%
  pivot_longer(3:182, names_to = "sample_ID", values_to = "copynum_SCG7") 

#environmental
environmental_parameters_SCG7metaG <- read.delim("~/Desktop/Tara Oceans hmm/SCG7_TsaD/OceanGeneAtlas_Data_20240307-02_38_SCG7_ArcticmetaG/environmental_parameters.csv")
#merging and filtering for surface and size fraction

SCG7_copynum_enviro <- left_join(SCG7_copynum,
                                     environmental_parameters_SCG7metaG ,
                                     by = "sample_ID")

SCG7_copynum_enviro <- SCG7_copynum_enviro %>%
        filter(depth == "SRF" | depth == "DCM",
         size.fraction == "[0.22-3µm]" | size.fraction == "[0.45-0.8µm]" | size.fraction == "[0.22-0.45µm]")

#doing things on not a regional level
#turning samples with copy numbers lower than 0.0001
SCG7_copynum_enviro_filt <- SCG7_copynum_enviro 

#calculating the total number of copies in each sample 
SCG7_copynum_enviro_filt_totalbysample <- SCG7_copynum_enviro_filt %>%
  transform(copynum_SCG7 = as.numeric(copynum_SCG7)) %>%
  group_by(sample_ID) %>%
  summarise(copynum_SCG7 = sum(copynum_SCG7))
#########
SCG8_metaG <- read.delim("~/Desktop/Tara Oceans hmm/SCG8_SRP19/OceanGeneAtlas_Data_20240307-02_44_SCG8_ArcticmetaG/abundance_matrix.csv")

#making the new first row into the column names 
SCG8_metaG <- janitor::row_to_names(SCG8_metaG, 1) 
#reformat so the sample IDs are rows so can merge with environmental
SCG8_copynum <- SCG8_metaG %>%
  pivot_longer(3:182, names_to = "sample_ID", values_to = "copynum_SCG8") 

#environmental
environmental_parameters_SCG8metaG <- read.delim("~/Desktop/Tara Oceans hmm/SCG8_SRP19/OceanGeneAtlas_Data_20240307-02_44_SCG8_ArcticmetaG/environmental_parameters.csv")
#merging and filtering for surface and size fraction

SCG8_copynum_enviro <- left_join(SCG8_copynum,
                                     environmental_parameters_SCG8metaG ,
                                     by = "sample_ID")

SCG8_copynum_enviro <- SCG8_copynum_enviro %>%
        filter(depth == "SRF" | depth == "DCM",
         size.fraction == "[0.22-3µm]" | size.fraction == "[0.45-0.8µm]" | size.fraction == "[0.22-0.45µm]")

#doing things on not a regional level
#turning samples with copy numbers lower than 0.0001
SCG8_copynum_enviro_filt <- SCG8_copynum_enviro 

#calculating the total number of copies in each sample 
SCG8_copynum_enviro_filt_totalbysample <- SCG8_copynum_enviro_filt %>%
  transform(copynum_SCG8 = as.numeric(copynum_SCG8)) %>%
  group_by(sample_ID) %>%
  summarise(copynum_SCG8 = sum(copynum_SCG8))
##############
SCG9_metaG <- read.delim("~/Desktop/Tara Oceans hmm/SCG9_SRP54/OceanGeneAtlas_Data_20240307-02_59_SCG9_ArcticmetaG/abundance_matrix.csv")

#making the new first row into the column names 
SCG9_metaG <- janitor::row_to_names(SCG9_metaG, 1) 
#reformat so the sample IDs are rows so can merge with environmental
SCG9_copynum <- SCG9_metaG %>%
  pivot_longer(3:182, names_to = "sample_ID", values_to = "copynum_SCG9") 

#environmental
environmental_parameters_SCG9metaG <- read.delim("~/Desktop/Tara Oceans hmm/SCG9_SRP54/OceanGeneAtlas_Data_20240307-02_59_SCG9_ArcticmetaG/environmental_parameters.csv")
#merging and filtering for surface and size fraction

SCG9_copynum_enviro <- left_join(SCG9_copynum,
                                     environmental_parameters_SCG9metaG ,
                                     by = "sample_ID")

SCG9_copynum_enviro <- SCG9_copynum_enviro %>%
        filter(depth == "SRF" | depth == "DCM",
         size.fraction == "[0.22-3µm]" | size.fraction == "[0.45-0.8µm]" | size.fraction == "[0.22-0.45µm]")

#doing things on not a regional level
#turning samples with copy numbers lower than 0.0001
SCG9_copynum_enviro_filt <- SCG9_copynum_enviro 

#calculating the total number of copies in each sample 
SCG9_copynum_enviro_filt_totalbysample <- SCG9_copynum_enviro_filt %>%
  transform(copynum_SCG9 = as.numeric(copynum_SCG9)) %>%
  group_by(sample_ID) %>%
  summarise(copynum_SCG9 = sum(copynum_SCG9))



#merging SCGs and finding median

SCG12_fornorm_metaG <- merge(SCG1_copynum_enviro_filt_totalbysample,
                  SCG2_copynum_enviro_filt_totalbysample,
                  by = "sample_ID",
                  all.x = TRUE,
                  all.y = TRUE)

SCG123_fornorm_metaG <- merge(SCG12_fornorm_metaG,
                  SCG3_copynum_enviro_filt_totalbysample,
                  by = "sample_ID",
                  all.x = TRUE,
                  all.y = TRUE)


SCG1234_fornorm_metaG <- merge(SCG123_fornorm_metaG,
                  SCG4_copynum_enviro_filt_totalbysample,
                  by = "sample_ID",
                  all.x = TRUE,
                  all.y = TRUE)

SCG12345_fornorm_metaG <- merge(SCG1234_fornorm_metaG,
                  SCG5_copynum_enviro_filt_totalbysample,
                  by = "sample_ID",
                  all.x = TRUE,
                  all.y = TRUE)


SCG123456_fornorm_metaG <- merge(SCG12345_fornorm_metaG,
                  SCG6_copynum_enviro_filt_totalbysample,
                  by = "sample_ID",
                  all.x = TRUE,
                  all.y = TRUE)

SCG1234567_fornorm_metaG <- merge(SCG123456_fornorm_metaG,
                  SCG7_copynum_enviro_filt_totalbysample,
                  by = "sample_ID",
                  all.x = TRUE,
                  all.y = TRUE)

SCG12345678_fornorm_metaG <- merge(SCG1234567_fornorm_metaG,
                  SCG8_copynum_enviro_filt_totalbysample,
                  by = "sample_ID",
                  all.x = TRUE,
                  all.y = TRUE)

SCG123456789_fornorm_metaG <- merge(SCG12345678_fornorm_metaG,
                  SCG9_copynum_enviro_filt_totalbysample,
                  by = "sample_ID",
                  all.x = TRUE,
                  all.y = TRUE) %>%
    rowwise() %>%
  mutate(row_median = median(c_across(where(is.numeric)), na.rm=TRUE))


#prepping gene dataset

#preparing for merging and normalization
SCG_med_metaG <- SCG123456789_fornorm_metaG %>%
  select(sample_ID, row_median)

SCG_catalases_metaG <- left_join(KatG_rubisco_KatE_CatMn_psbO_light_models,
                                 SCG_med_metaG,
                                 by = "sample_ID")

enviro_by_sample_tobind <- enviro_by_sample %>%
  select(sample_ID, station, depth, size.fraction, Region,
         CO2, O2, Iron_5m, PAR, Salinity, POC, Alkalinity, PO4, NO3_NO2,
         Coast_km, Chlorophyll_A, Temperature, NPP, longitude, latitude)
enviro_by_sample_KatE_tobind <- enviro_by_sample_KatE %>%
  select(sample_ID, station, depth, size.fraction, Region,
         CO2, O2, Iron_5m, PAR, Salinity, POC, Alkalinity, PO4, NO3_NO2,
         Coast_km, Chlorophyll_A, Temperature, NPP,longitude, latitude)
enviro_by_sample_rub_tobind <- enviro_by_sample_rub %>%
  select(sample_ID, station, depth, size.fraction, Region,
         CO2, O2, Iron_5m, PAR, Salinity, POC, Alkalinity, PO4, NO3_NO2,
         Coast_km, Chlorophyll_A, Temperature, NPP,longitude, latitude)
enviro_by_sample_CatMn_tobind <- enviro_by_sample_CatMn  %>%
  select(sample_ID, station, depth, size.fraction, Region,
         CO2, O2, Iron_5m, PAR, Salinity, POC, Alkalinity, PO4, NO3_NO2,
         Coast_km, Chlorophyll_A, Temperature, NPP,longitude, latitude) 

#rbinding

enviro_KatG_KatE <- rbind(enviro_by_sample_tobind, enviro_by_sample_KatE_tobind)
enviro_KatG_KatE_rub <- rbind(enviro_KatG_KatE, enviro_by_sample_rub_tobind)
enviro_KatG_KatE_CatMn_rub <- rbind(enviro_KatG_KatE_rub, enviro_by_sample_CatMn_tobind) %>%
  unique()

#normalizing 
SCG_catalases_metaG_norm <- SCG_catalases_metaG %>%
  mutate(normKatG = copynum/row_median,
         normrub = copynum_rub/row_median,
         normKatE = copynum_KatE/row_median,
         normCatMn = copynum_CatMn/row_median,
         normpsbO = copynum_psbO/row_median,
         normlight = copynum_light/row_median)

#enviro

catmodels_norm <- left_join(SCG_catalases_metaG_norm, enviro_KatG_KatE_CatMn_rub,
                       by = "sample_ID")
write_csv(catmodels_norm, "catmodels_norm.csv")




####using the single copy genes size fraction to calculate relative abundance

SCG1_metaG <- read.delim("~/Desktop/Tara Oceans hmm/SCG1_synt_2d/OceanGeneAtlas_Data_20240306-22_43_SCG1_ArcticmetaG/abundance_matrix.csv")

#making the new first row into the column names 
SCG1_metaG <- janitor::row_to_names(SCG1_metaG, 1) 
#reformat so the sample IDs are rows so can merge with environmental
SCG1_copynum <- SCG1_metaG %>%
  pivot_longer(3:182, names_to = "sample_ID", values_to = "copynum_SCG1") 

#environmental
environmental_parameters_SCG1metaG <- read.delim("~/Desktop/Tara Oceans hmm/SCG1_synt_2d/OceanGeneAtlas_Data_20240306-22_43_SCG1_ArcticmetaG/environmental_parameters.csv")
#merging and filtering for surface and size fraction

SCG1_copynum_enviro <- left_join(SCG1_copynum,
                                     environmental_parameters_SCG1metaG ,
                                     by = "sample_ID")

SCG1_copynum_enviro <- SCG1_copynum_enviro %>%
        filter(depth == "SRF" | depth == "DCM",
         size.fraction == "[0.22-3µm]" | size.fraction == "[0.45-0.8µm]" | size.fraction == "[0.22-0.45µm]")

#separating the taxa
SCG1_copynum_enviro_taxa <- SCG1_copynum_enviro %>%
   separate(col=taxonomy, into=c('Biota', 'Taxa1', 'Taxa2', 'Taxa3', 'Taxa4', 'Taxa5', 'Taxa6', 'Taxa7', 'Taxa8', 'Taxa9'), sep=';')

SCG1_copynum_enviro_taxa_filt <- SCG1_copynum_enviro_taxa[2:13] %>%
  filter(Taxa1 != " Eukaryota")

SCG1_copynum_enviro_taxa_filt_totalreadsbysample <- SCG1_copynum_enviro_taxa_filt %>%
  select(Taxa4, sample_ID, copynum_SCG1) %>%
  transform(copynum_SCG1 = as.numeric(copynum_SCG1)) %>%
  group_by(sample_ID) %>%
  dplyr::summarise(totalreadsbysample = sum(copynum_SCG1))

  
SCG1_copynum_enviro_taxa_filt_order <- SCG1_copynum_enviro_taxa_filt %>%
  select(Taxa4, sample_ID, copynum_SCG1) %>%
  drop_na() 

SCG1_copynum_enviro_taxa_filt_order_bysample <- SCG1_copynum_enviro_taxa_filt_order %>%
  transform(copynum_SCG1 = as.numeric(copynum_SCG1)) %>%
  group_by(Taxa4, sample_ID) %>%
  dplyr::summarise(totalreads = sum(copynum_SCG1))

#merging the all taxa dataset with the total reads by sample

SCG1_taxa_norm <- left_join(SCG1_copynum_enviro_taxa_filt_order_bysample,
                            SCG1_copynum_enviro_taxa_filt_totalreadsbysample,
                            by = "sample_ID") %>%
  mutate(normreads = totalreads/totalreadsbysample)

SCG1_taxa_norm_taxaavgbysample <- SCG1_taxa_norm %>%
  group_by(Taxa4) %>%
  dplyr::summarise(avgnormreads_SCG1 = mean(normreads))


#####

SCG2_metaG <- read.delim("~/Desktop/Tara Oceans hmm/SCG2_Seryl_tRNA_N/OceanGeneAtlas_Data_20240306-22_51_SCG2_ArcticmetaG/abundance_matrix.csv")

#making the new first row into the column names 
SCG2_metaG <- janitor::row_to_names(SCG2_metaG, 1) 
#reformat so the sample IDs are rows so can merge with environmental
SCG2_copynum <- SCG2_metaG %>%
  pivot_longer(3:182, names_to = "sample_ID", values_to = "copynum_SCG2") 

#environmental
environmental_parameters_SCG2metaG <- read.delim("~/Desktop/Tara Oceans hmm/SCG2_Seryl_tRNA_N/OceanGeneAtlas_Data_20240306-22_51_SCG2_ArcticmetaG/environmental_parameters.csv")
#merging and filtering for surface and size fraction

SCG2_copynum_enviro <- left_join(SCG2_copynum,
                                     environmental_parameters_SCG2metaG ,
                                     by = "sample_ID")

SCG2_copynum_enviro <- SCG2_copynum_enviro %>%
        filter(depth == "SRF" | depth == "DCM",
         size.fraction == "[0.22-3µm]" | size.fraction == "[0.45-0.8µm]" | size.fraction == "[0.22-0.45µm]")

SCG2_copynum_enviro_taxa <- SCG2_copynum_enviro %>%
   separate(col=taxonomy, into=c('Biota', 'Taxa1', 'Taxa2', 'Taxa3', 'Taxa4', 'Taxa5', 'Taxa6', 'Taxa7', 'Taxa8', 'Taxa9'), sep=';')

SCG2_copynum_enviro_taxa_filt <- SCG2_copynum_enviro_taxa[2:13] %>%
  filter(Taxa1 != " Eukaryota")

SCG2_copynum_enviro_taxa_filt_totalreadsbysample <- SCG2_copynum_enviro_taxa_filt %>%
  select(Taxa4, sample_ID, copynum_SCG2) %>%
  transform(copynum_SCG2 = as.numeric(copynum_SCG2)) %>%
  group_by(sample_ID) %>%
  dplyr::summarise(totalreadsbysample = sum(copynum_SCG2))

  
SCG2_copynum_enviro_taxa_filt_order <- SCG2_copynum_enviro_taxa_filt %>%
  select(Taxa4, sample_ID, copynum_SCG2) %>%
  drop_na() 

SCG2_copynum_enviro_taxa_filt_order_bysample <- SCG2_copynum_enviro_taxa_filt_order %>%
  transform(copynum_SCG2 = as.numeric(copynum_SCG2)) %>%
  group_by(Taxa4, sample_ID) %>%
  dplyr::summarise(totalreads = sum(copynum_SCG2))

#merging the all taxa dataset with the total reads by sample

SCG2_taxa_norm <- left_join(SCG2_copynum_enviro_taxa_filt_order_bysample,
                            SCG2_copynum_enviro_taxa_filt_totalreadsbysample,
                            by = "sample_ID") %>%
  mutate(normreads = totalreads/totalreadsbysample)

SCG2_taxa_norm_taxaavgbysample <- SCG2_taxa_norm %>%
  group_by(Taxa4) %>%
  dplyr::summarise(avgnormreads_SCG2 = mean(normreads))


##############
SCG3_metaG <- read.delim("~/Desktop/Tara Oceans hmm/SCG3_synt_1e/OceanGeneAtlas_Data_20240306-22_57_SCG3_ArcticmetaG/abundance_matrix.csv")

#making the new first row into the column names 
SCG3_metaG <- janitor::row_to_names(SCG3_metaG, 1) 
#reformat so the sample IDs are rows so can merge with environmental
SCG3_copynum <- SCG3_metaG %>%
  pivot_longer(3:182, names_to = "sample_ID", values_to = "copynum_SCG3") 

#environmental
environmental_parameters_SCG3metaG <- read.delim("~/Desktop/Tara Oceans hmm/SCG3_synt_1e/OceanGeneAtlas_Data_20240306-22_57_SCG3_ArcticmetaG/environmental_parameters.csv")
#merging and filtering for surface and size fraction

SCG3_copynum_enviro <- left_join(SCG3_copynum,
                                     environmental_parameters_SCG3metaG ,
                                     by = "sample_ID")

SCG3_copynum_enviro <- SCG3_copynum_enviro %>%
        filter(depth == "SRF" | depth == "DCM",
         size.fraction == "[0.22-3µm]" | size.fraction == "[0.45-0.8µm]" | size.fraction == "[0.22-0.45µm]")

SCG3_copynum_enviro_taxa <- SCG3_copynum_enviro %>%
   separate(col=taxonomy, into=c('Biota', 'Taxa1', 'Taxa2', 'Taxa3', 'Taxa4', 'Taxa5', 'Taxa6', 'Taxa7', 'Taxa8', 'Taxa9'), sep=';')

SCG3_copynum_enviro_taxa_filt <- SCG3_copynum_enviro_taxa[2:13] %>%
  filter(Taxa1 != " Eukaryota")

SCG3_copynum_enviro_taxa_filt_totalreadsbysample <- SCG3_copynum_enviro_taxa_filt %>%
  select(Taxa4, sample_ID, copynum_SCG3) %>%
  transform(copynum_SCG3 = as.numeric(copynum_SCG3)) %>%
  group_by(sample_ID) %>%
  dplyr::summarise(totalreadsbysample = sum(copynum_SCG3))

  
SCG3_copynum_enviro_taxa_filt_order <- SCG3_copynum_enviro_taxa_filt %>%
  select(Taxa4, sample_ID, copynum_SCG3) %>%
  drop_na() 

SCG3_copynum_enviro_taxa_filt_order_bysample <- SCG3_copynum_enviro_taxa_filt_order %>%
  transform(copynum_SCG3 = as.numeric(copynum_SCG3)) %>%
  group_by(Taxa4, sample_ID) %>%
  dplyr::summarise(totalreads = sum(copynum_SCG3))

#merging the all taxa dataset with the total reads by sample

SCG3_taxa_norm <- left_join(SCG3_copynum_enviro_taxa_filt_order_bysample,
                            SCG3_copynum_enviro_taxa_filt_totalreadsbysample,
                            by = "sample_ID") %>%
  mutate(normreads = totalreads/totalreadsbysample)

SCG3_taxa_norm_taxaavgbysample <- SCG3_taxa_norm %>%
  group_by(Taxa4) %>%
  dplyr::summarise(avgnormreads_SCG3 = mean(normreads))



##########

SCG4_metaG <- read.delim("~/Downloads/OceanGeneAtlas_Data_20240306-23_00_SCG4_ArcticmetaG/abundance_matrix.csv")

#making the new first row into the column names 
SCG4_metaG <- janitor::row_to_names(SCG4_metaG, 1) 
#reformat so the sample IDs are rows so can merge with environmental
SCG4_copynum <- SCG4_metaG %>%
  pivot_longer(3:182, names_to = "sample_ID", values_to = "copynum_SCG4") 

#environmental
environmental_parameters_SCG4metaG <- read.delim("~/Downloads/OceanGeneAtlas_Data_20240306-23_00_SCG4_ArcticmetaG/environmental_parameters.csv")
#merging and filtering for surface and size fraction

SCG4_copynum_enviro <- left_join(SCG4_copynum,
                                     environmental_parameters_SCG4metaG ,
                                     by = "sample_ID")

SCG4_copynum_enviro <- SCG4_copynum_enviro %>%
        filter(depth == "SRF" | depth == "DCM",
         size.fraction == "[0.22-3µm]" | size.fraction == "[0.45-0.8µm]" | size.fraction == "[0.22-0.45µm]")

SCG4_copynum_enviro_taxa <- SCG4_copynum_enviro %>%
   separate(col=taxonomy, into=c('Biota', 'Taxa1', 'Taxa2', 'Taxa3', 'Taxa4', 'Taxa5', 'Taxa6', 'Taxa7', 'Taxa8', 'Taxa9'), sep=';')

SCG4_copynum_enviro_taxa_filt <- SCG4_copynum_enviro_taxa[2:13] %>%
  filter(Taxa1 != " Eukaryota")

SCG4_copynum_enviro_taxa_filt_totalreadsbysample <- SCG4_copynum_enviro_taxa_filt %>%
  select(Taxa4, sample_ID, copynum_SCG4) %>%
  transform(copynum_SCG4 = as.numeric(copynum_SCG4)) %>%
  group_by(sample_ID) %>%
  dplyr::summarise(totalreadsbysample = sum(copynum_SCG4))

  
SCG4_copynum_enviro_taxa_filt_order <- SCG4_copynum_enviro_taxa_filt %>%
  select(Taxa4, sample_ID, copynum_SCG4) %>%
  drop_na() 

SCG4_copynum_enviro_taxa_filt_order_bysample <- SCG4_copynum_enviro_taxa_filt_order %>%
  transform(copynum_SCG4 = as.numeric(copynum_SCG4)) %>%
  group_by(Taxa4, sample_ID) %>%
  dplyr::summarise(totalreads = sum(copynum_SCG4))

#merging the all taxa dataset with the total reads by sample

SCG4_taxa_norm <- left_join(SCG4_copynum_enviro_taxa_filt_order_bysample,
                            SCG4_copynum_enviro_taxa_filt_totalreadsbysample,
                            by = "sample_ID") %>%
  mutate(normreads = totalreads/totalreadsbysample)

SCG4_taxa_norm_taxaavgbysample <- SCG4_taxa_norm %>%
  group_by(Taxa4) %>%
  dplyr::summarise(avgnormreads_SCG4 = mean(normreads))


################
SCG5_metaG <- read.delim("~/Desktop/Tara Oceans hmm/SCG5_synt_1_2/OceanGeneAtlas_Data_20240307-02_07_SCG5_ArcticmetaG/abundance_matrix.csv")

#making the new first row into the column names 
SCG5_metaG <- janitor::row_to_names(SCG5_metaG, 1) 
#reformat so the sample IDs are rows so can merge with environmental
SCG5_copynum <- SCG5_metaG %>%
  pivot_longer(3:182, names_to = "sample_ID", values_to = "copynum_SCG5") 

#environmental
environmental_parameters_SCG5metaG <- read.delim("~/Desktop/Tara Oceans hmm/SCG5_synt_1_2/OceanGeneAtlas_Data_20240307-02_07_SCG5_ArcticmetaG/environmental_parameters.csv")
#merging and filtering for surface and size fraction

SCG5_copynum_enviro <- left_join(SCG5_copynum,
                                     environmental_parameters_SCG5metaG ,
                                     by = "sample_ID")

SCG5_copynum_enviro <- SCG5_copynum_enviro %>%
        filter(depth == "SRF" | depth == "DCM",
         size.fraction == "[0.22-3µm]" | size.fraction == "[0.45-0.8µm]" | size.fraction == "[0.22-0.45µm]")

SCG5_copynum_enviro_taxa <- SCG5_copynum_enviro %>%
   separate(col=taxonomy, into=c('Biota', 'Taxa1', 'Taxa2', 'Taxa3', 'Taxa4', 'Taxa5', 'Taxa6', 'Taxa7', 'Taxa8', 'Taxa9'), sep=';')

SCG5_copynum_enviro_taxa_filt <- SCG5_copynum_enviro_taxa[2:13] %>%
  filter(Taxa1 != " Eukaryota")

SCG5_copynum_enviro_taxa_filt_totalreadsbysample <- SCG5_copynum_enviro_taxa_filt %>%
  select(Taxa4, sample_ID, copynum_SCG5) %>%
  transform(copynum_SCG5 = as.numeric(copynum_SCG5)) %>%
  group_by(sample_ID) %>%
  dplyr::summarise(totalreadsbysample = sum(copynum_SCG5))

  
SCG5_copynum_enviro_taxa_filt_order <- SCG5_copynum_enviro_taxa_filt %>%
  select(Taxa4, sample_ID, copynum_SCG5) %>%
  drop_na() 

SCG5_copynum_enviro_taxa_filt_order_bysample <- SCG5_copynum_enviro_taxa_filt_order %>%
  transform(copynum_SCG5 = as.numeric(copynum_SCG5)) %>%
  group_by(Taxa4, sample_ID) %>%
  dplyr::summarise(totalreads = sum(copynum_SCG5))

#merging the all taxa dataset with the total reads by sample

SCG5_taxa_norm <- left_join(SCG5_copynum_enviro_taxa_filt_order_bysample,
                            SCG5_copynum_enviro_taxa_filt_totalreadsbysample,
                            by = "sample_ID") %>%
  mutate(normreads = totalreads/totalreadsbysample)

SCG5_taxa_norm_taxaavgbysample <- SCG5_taxa_norm %>%
  group_by(Taxa4) %>%
  dplyr::summarise(avgnormreads_SCG5 = mean(normreads))



###########
SCG6_metaG <- read.delim("~/Desktop/Tara Oceans hmm/SCG6_tRNA_synt_1/OceanGeneAtlas_Data_20240307-02_21_SCG6_ArcticmetaG/abundance_matrix.csv")

#making the new first row into the column names 
SCG6_metaG <- janitor::row_to_names(SCG6_metaG, 1) 
#reformat so the sample IDs are rows so can merge with environmental
SCG6_copynum <- SCG6_metaG %>%
  pivot_longer(3:182, names_to = "sample_ID", values_to = "copynum_SCG6") 

#environmental
environmental_parameters_SCG6metaG <- read.delim("~/Desktop/Tara Oceans hmm/SCG6_tRNA_synt_1/OceanGeneAtlas_Data_20240307-02_21_SCG6_ArcticmetaG/environmental_parameters.csv")
#merging and filtering for surface and size fraction

SCG6_copynum_enviro <- left_join(SCG6_copynum,
                                     environmental_parameters_SCG6metaG ,
                                     by = "sample_ID")

SCG6_copynum_enviro <- SCG6_copynum_enviro %>%
        filter(depth == "SRF" | depth == "DCM",
         size.fraction == "[0.22-3µm]" | size.fraction == "[0.45-0.8µm]" | size.fraction == "[0.22-0.45µm]")

SCG6_copynum_enviro_taxa <- SCG6_copynum_enviro %>%
   separate(col=taxonomy, into=c('Biota', 'Taxa1', 'Taxa2', 'Taxa3', 'Taxa4', 'Taxa5', 'Taxa6', 'Taxa7', 'Taxa8', 'Taxa9'), sep=';')

SCG6_copynum_enviro_taxa_filt <- SCG6_copynum_enviro_taxa[2:13] %>%
  filter(Taxa1 != " Eukaryota")

SCG6_copynum_enviro_taxa_filt_totalreadsbysample <- SCG6_copynum_enviro_taxa_filt %>%
  select(Taxa4, sample_ID, copynum_SCG6) %>%
  transform(copynum_SCG6 = as.numeric(copynum_SCG6)) %>%
  group_by(sample_ID) %>%
  dplyr::summarise(totalreadsbysample = sum(copynum_SCG6))

  
SCG6_copynum_enviro_taxa_filt_order <- SCG6_copynum_enviro_taxa_filt %>%
  select(Taxa4, sample_ID, copynum_SCG6) %>%
  drop_na() 

SCG6_copynum_enviro_taxa_filt_order_bysample <- SCG6_copynum_enviro_taxa_filt_order %>%
  transform(copynum_SCG6 = as.numeric(copynum_SCG6)) %>%
  group_by(Taxa4, sample_ID) %>%
  dplyr::summarise(totalreads = sum(copynum_SCG6))

#merging the all taxa dataset with the total reads by sample

SCG6_taxa_norm <- left_join(SCG6_copynum_enviro_taxa_filt_order_bysample,
                            SCG6_copynum_enviro_taxa_filt_totalreadsbysample,
                            by = "sample_ID") %>%
  mutate(normreads = totalreads/totalreadsbysample)

SCG6_taxa_norm_taxaavgbysample <- SCG6_taxa_norm %>%
  group_by(Taxa4) %>%
  dplyr::summarise(avgnormreads_SCG6 = mean(normreads))


##########
SCG7_metaG <- read.delim("~/Desktop/Tara Oceans hmm/SCG7_TsaD/OceanGeneAtlas_Data_20240307-02_38_SCG7_ArcticmetaG/abundance_matrix.csv")


#making the new first row into the column names 
SCG7_metaG <- janitor::row_to_names(SCG7_metaG, 1) 
#reformat so the sample IDs are rows so can merge with environmental
SCG7_copynum <- SCG7_metaG %>%
  pivot_longer(3:182, names_to = "sample_ID", values_to = "copynum_SCG7") 

#environmental
environmental_parameters_SCG7metaG <- read.delim("~/Desktop/Tara Oceans hmm/SCG7_TsaD/OceanGeneAtlas_Data_20240307-02_38_SCG7_ArcticmetaG/environmental_parameters.csv")
#merging and filtering for surface and size fraction

SCG7_copynum_enviro <- left_join(SCG7_copynum,
                                     environmental_parameters_SCG7metaG ,
                                     by = "sample_ID")

SCG7_copynum_enviro <- SCG7_copynum_enviro %>%
        filter(depth == "SRF" | depth == "DCM",
         size.fraction == "[0.22-3µm]" | size.fraction == "[0.45-0.8µm]" | size.fraction == "[0.22-0.45µm]")

SCG7_copynum_enviro_taxa <- SCG7_copynum_enviro %>%
   separate(col=taxonomy, into=c('Biota', 'Taxa1', 'Taxa2', 'Taxa3', 'Taxa4', 'Taxa5', 'Taxa6', 'Taxa7', 'Taxa8', 'Taxa9'), sep=';')

SCG7_copynum_enviro_taxa_filt <- SCG7_copynum_enviro_taxa[2:13] %>%
  filter(Taxa1 != " Eukaryota")

SCG7_copynum_enviro_taxa_filt_totalreadsbysample <- SCG7_copynum_enviro_taxa_filt %>%
  select(Taxa4, sample_ID, copynum_SCG7) %>%
  transform(copynum_SCG7 = as.numeric(copynum_SCG7)) %>%
  group_by(sample_ID) %>%
  dplyr::summarise(totalreadsbysample = sum(copynum_SCG7))

  
SCG7_copynum_enviro_taxa_filt_order <- SCG7_copynum_enviro_taxa_filt %>%
  select(Taxa4, sample_ID, copynum_SCG7) %>%
  drop_na() 

SCG7_copynum_enviro_taxa_filt_order_bysample <- SCG7_copynum_enviro_taxa_filt_order %>%
  transform(copynum_SCG7 = as.numeric(copynum_SCG7)) %>%
  group_by(Taxa4, sample_ID) %>%
  dplyr::summarise(totalreads = sum(copynum_SCG7))

#merging the all taxa dataset with the total reads by sample

SCG7_taxa_norm <- left_join(SCG7_copynum_enviro_taxa_filt_order_bysample,
                            SCG7_copynum_enviro_taxa_filt_totalreadsbysample,
                            by = "sample_ID") %>%
  mutate(normreads = totalreads/totalreadsbysample)

SCG7_taxa_norm_taxaavgbysample <- SCG7_taxa_norm %>%
  group_by(Taxa4) %>%
  dplyr::summarise(avgnormreads_SCG7 = mean(normreads))

#making plot

#########
SCG8_metaG <- read.delim("~/Desktop/Tara Oceans hmm/SCG8_SRP19/OceanGeneAtlas_Data_20240307-02_44_SCG8_ArcticmetaG/abundance_matrix.csv")

#making the new first row into the column names 
SCG8_metaG <- janitor::row_to_names(SCG8_metaG, 1) 
#reformat so the sample IDs are rows so can merge with environmental
SCG8_copynum <- SCG8_metaG %>%
  pivot_longer(3:182, names_to = "sample_ID", values_to = "copynum_SCG8") 

#environmental
environmental_parameters_SCG8metaG <- read.delim("~/Desktop/Tara Oceans hmm/SCG8_SRP19/OceanGeneAtlas_Data_20240307-02_44_SCG8_ArcticmetaG/environmental_parameters.csv")
#merging and filtering for surface and size fraction

SCG8_copynum_enviro <- left_join(SCG8_copynum,
                                     environmental_parameters_SCG8metaG ,
                                     by = "sample_ID")

SCG8_copynum_enviro <- SCG8_copynum_enviro %>%
        filter(depth == "SRF" | depth == "DCM",
         size.fraction == "[0.22-3µm]" | size.fraction == "[0.45-0.8µm]" | size.fraction == "[0.22-0.45µm]")

SCG8_copynum_enviro_taxa <- SCG8_copynum_enviro %>%
   separate(col=taxonomy, into=c('Biota', 'Taxa1', 'Taxa2', 'Taxa3', 'Taxa4', 'Taxa5', 'Taxa6', 'Taxa7', 'Taxa8', 'Taxa9'), sep=';')

SCG8_copynum_enviro_taxa_filt <- SCG8_copynum_enviro_taxa[2:13] %>%
  filter(Taxa1 != " Eukaryota")

SCG8_copynum_enviro_taxa_filt_totalreadsbysample <- SCG8_copynum_enviro_taxa_filt %>%
  select(Taxa4, sample_ID, copynum_SCG8) %>%
  transform(copynum_SCG8 = as.numeric(copynum_SCG8)) %>%
  group_by(sample_ID) %>%
  dplyr::summarise(totalreadsbysample = sum(copynum_SCG8))

  
SCG8_copynum_enviro_taxa_filt_order <- SCG8_copynum_enviro_taxa_filt %>%
  select(Taxa4, sample_ID, copynum_SCG8) %>%
  drop_na() 

SCG8_copynum_enviro_taxa_filt_order_bysample <- SCG8_copynum_enviro_taxa_filt_order %>%
  transform(copynum_SCG8 = as.numeric(copynum_SCG8)) %>%
  group_by(Taxa4, sample_ID) %>%
  dplyr::summarise(totalreads = sum(copynum_SCG8))

#merging the all taxa dataset with the total reads by sample

SCG8_taxa_norm <- left_join(SCG8_copynum_enviro_taxa_filt_order_bysample,
                            SCG8_copynum_enviro_taxa_filt_totalreadsbysample,
                            by = "sample_ID") %>%
  mutate(normreads = totalreads/totalreadsbysample)


SCG8_taxa_norm$normreads[SCG8_taxa_norm$normreads == "NaN"] <- 0


SCG8_taxa_norm_taxaavgbysample <- SCG8_taxa_norm %>%
  group_by(Taxa4) %>%
  dplyr::summarise(avgnormreads_SCG8 = mean(normreads))


##############
SCG9_metaG <- read.delim("~/Desktop/Tara Oceans hmm/SCG9_SRP54/OceanGeneAtlas_Data_20240307-02_59_SCG9_ArcticmetaG/abundance_matrix.csv")

#making the new first row into the column names 
SCG9_metaG <- janitor::row_to_names(SCG9_metaG, 1) 
#reformat so the sample IDs are rows so can merge with environmental
SCG9_copynum <- SCG9_metaG %>%
  pivot_longer(3:182, names_to = "sample_ID", values_to = "copynum_SCG9") 

#environmental
environmental_parameters_SCG9metaG <- read.delim("~/Desktop/Tara Oceans hmm/SCG9_SRP54/OceanGeneAtlas_Data_20240307-02_59_SCG9_ArcticmetaG/environmental_parameters.csv")
#merging and filtering for surface and size fraction

SCG9_copynum_enviro <- left_join(SCG9_copynum,
                                     environmental_parameters_SCG9metaG ,
                                     by = "sample_ID")

SCG9_copynum_enviro <- SCG9_copynum_enviro %>%
        filter(depth == "SRF" | depth == "DCM",
         size.fraction == "[0.22-3µm]" | size.fraction == "[0.45-0.8µm]" | size.fraction == "[0.22-0.45µm]")

SCG9_copynum_enviro_taxa <- SCG9_copynum_enviro %>%
   separate(col=taxonomy, into=c('Biota', 'Taxa1', 'Taxa2', 'Taxa3', 'Taxa4', 'Taxa5', 'Taxa6', 'Taxa7', 'Taxa8', 'Taxa9'), sep=';')

SCG9_copynum_enviro_taxa_filt <- SCG9_copynum_enviro_taxa[2:13] %>%
  filter(Taxa1 != " Eukaryota")

SCG9_copynum_enviro_taxa_filt_totalreadsbysample <- SCG9_copynum_enviro_taxa_filt %>%
  select(Taxa4, sample_ID, copynum_SCG9) %>%
  transform(copynum_SCG9 = as.numeric(copynum_SCG9)) %>%
  group_by(sample_ID) %>%
  dplyr::summarise(totalreadsbysample = sum(copynum_SCG9))

  
SCG9_copynum_enviro_taxa_filt_order <- SCG9_copynum_enviro_taxa_filt %>%
  select(Taxa4, sample_ID, copynum_SCG9) %>%
  drop_na() 

SCG9_copynum_enviro_taxa_filt_order_bysample <- SCG9_copynum_enviro_taxa_filt_order %>%
  transform(copynum_SCG9 = as.numeric(copynum_SCG9)) %>%
  group_by(Taxa4, sample_ID) %>%
  dplyr::summarise(totalreads = sum(copynum_SCG9))

####
SCG9_taxa_norm <- left_join(SCG9_copynum_enviro_taxa_filt_order_bysample,
                            SCG9_copynum_enviro_taxa_filt_totalreadsbysample,
                            by = "sample_ID") %>%
  mutate(normreads = totalreads/totalreadsbysample)

SCG9_taxa_norm_taxaavgbysample <- SCG9_taxa_norm %>%
  group_by(Taxa4) %>%
  dplyr::summarise(avgnormreads_SCG9 = mean(normreads))


#merging the all taxa dataset with the total reads by sample


#merging SCGs and finding median

SCG12_taxa <- merge(SCG1_taxa_norm_taxaavgbysample,
                  SCG2_taxa_norm_taxaavgbysample,
                  by = "Taxa4",
                  all.x = TRUE,
                  all.y = TRUE)

SCG123_taxa <- merge(SCG12_taxa,
                  SCG3_taxa_norm_taxaavgbysample,
                  by = "Taxa4",
                  all.x = TRUE,
                  all.y = TRUE)


SCG1234_taxa <- merge(SCG123_taxa,
                  SCG4_taxa_norm_taxaavgbysample,
                  by = "Taxa4",
                  all.x = TRUE,
                  all.y = TRUE)

SCG12345_taxa <- merge(SCG1234_taxa,
                  SCG5_taxa_norm_taxaavgbysample,
                  by = "Taxa4",
                  all.x = TRUE,
                  all.y = TRUE)


SCG123456_taxa <- merge(SCG12345_taxa,
                  SCG6_taxa_norm_taxaavgbysample,
                  by = "Taxa4",
                  all.x = TRUE,
                  all.y = TRUE)

SCG1234567_taxa <- merge(SCG123456_taxa,
                  SCG7_taxa_norm_taxaavgbysample,
                  by = "Taxa4",
                  all.x = TRUE,
                  all.y = TRUE)

SCG12345678_taxa <- merge(SCG1234567_taxa,
                  SCG8_taxa_norm_taxaavgbysample,
                  by = "Taxa4",
                  all.x = TRUE,
                  all.y = TRUE)

SCG123456789_taxa <- merge(SCG12345678_taxa,
                            SCG9_taxa_norm_taxaavgbysample,
                  by = "Taxa4",
                  all.x = TRUE,
                  all.y = TRUE) 



#repeat process for taxa group 5 for the taxa that I get more specific with
SCG1_copynum_enviro_taxa_filt_totalreadsbysample5 <- SCG1_copynum_enviro_taxa_filt %>%
    select(Taxa5, sample_ID, copynum_SCG1) %>%
    transform(copynum_SCG1 = as.numeric(copynum_SCG1)) %>%
    group_by(sample_ID) %>%
    dplyr::summarise(totalreadsbysample = sum(copynum_SCG1))


SCG1_copynum_enviro_taxa_filt_order5 <- SCG1_copynum_enviro_taxa_filt %>%
    select(Taxa5, sample_ID, copynum_SCG1) %>%
    drop_na() 

SCG1_copynum_enviro_taxa_filt_order_bysample5 <- SCG1_copynum_enviro_taxa_filt_order5 %>%
    transform(copynum_SCG1 = as.numeric(copynum_SCG1)) %>%
    group_by(Taxa5, sample_ID) %>%
    dplyr::summarise(totalreads = sum(copynum_SCG1))

#merging the all taxa dataset with the total reads by sample

SCG1_taxa_norm5 <- left_join(SCG1_copynum_enviro_taxa_filt_order_bysample5,
                            SCG1_copynum_enviro_taxa_filt_totalreadsbysample5,
                            by = "sample_ID") %>%
    mutate(normreads = totalreads/totalreadsbysample)

SCG1_taxa_norm_taxaavgbysample5 <- SCG1_taxa_norm5 %>%
    group_by(Taxa5) %>%
    dplyr::summarise(avgnormreads_SCG1 = mean(normreads))


#######

SCG2_copynum_enviro_taxa_filt_totalreadsbysample5 <- SCG2_copynum_enviro_taxa_filt %>%
    select(Taxa5, sample_ID, copynum_SCG2) %>%
    transform(copynum_SCG2 = as.numeric(copynum_SCG2)) %>%
    group_by(sample_ID) %>%
    dplyr::summarise(totalreadsbysample = sum(copynum_SCG2))


SCG2_copynum_enviro_taxa_filt_order5 <- SCG2_copynum_enviro_taxa_filt %>%
    select(Taxa5, sample_ID, copynum_SCG2) %>%
    drop_na() 

SCG2_copynum_enviro_taxa_filt_order_bysample5 <- SCG2_copynum_enviro_taxa_filt_order5 %>%
    transform(copynum_SCG2 = as.numeric(copynum_SCG2)) %>%
    group_by(Taxa5, sample_ID) %>%
    dplyr::summarise(totalreads = sum(copynum_SCG2))

#merging the all taxa dataset with the total reads by sample

SCG2_taxa_norm5 <- left_join(SCG2_copynum_enviro_taxa_filt_order_bysample5,
                            SCG2_copynum_enviro_taxa_filt_totalreadsbysample5,
                            by = "sample_ID") %>%
    mutate(normreads = totalreads/totalreadsbysample)

SCG2_taxa_norm_taxaavgbysample5 <- SCG2_taxa_norm5 %>%
    group_by(Taxa5) %>%
    dplyr::summarise(avgnormreads_SCG2 = mean(normreads))


#########

SCG3_copynum_enviro_taxa_filt_totalreadsbysample5 <- SCG3_copynum_enviro_taxa_filt5 %>%
    select(Taxa4, sample_ID, copynum_SCG3) %>%
    transform(copynum_SCG3 = as.numeric(copynum_SCG3)) %>%
    group_by(sample_ID) %>%
    dplyr::summarise(totalreadsbysample = sum(copynum_SCG3))


SCG3_copynum_enviro_taxa_filt_order5 <- SCG3_copynum_enviro_taxa_filt5 %>%
    select(Taxa4, sample_ID, copynum_SCG3) %>%
    drop_na() 

SCG3_copynum_enviro_taxa_filt_order_bysample5 <- SCG3_copynum_enviro_taxa_filt_order5 %>%
    transform(copynum_SCG3 = as.numeric(copynum_SCG3)) %>%
    group_by(Taxa4, sample_ID) %>%
    dplyr::summarise(totalreads = sum(copynum_SCG3))

#merging the all taxa dataset with the total reads by sample

SCG3_taxa_norm <- left_join(SCG3_copynum_enviro_taxa_filt_order_bysample,
                            SCG3_copynum_enviro_taxa_filt_totalreadsbysample,
                            by = "sample_ID") %>%
    mutate(normreads = totalreads/totalreadsbysample)

SCG3_taxa_norm_taxaavgbysample <- SCG3_taxa_norm %>%
    group_by(Taxa4) %>%
    dplyr::summarise(avgnormreads_SCG3 = mean(normreads))

SCG4_copynum_enviro_taxa_filt_totalreadsbysample <- SCG4_copynum_enviro_taxa_filt %>%
    select(Taxa4, sample_ID, copynum_SCG4) %>%
    transform(copynum_SCG4 = as.numeric(copynum_SCG4)) %>%
    group_by(sample_ID) %>%
    dplyr::summarise(totalreadsbysample = sum(copynum_SCG4))


SCG4_copynum_enviro_taxa_filt_order <- SCG4_copynum_enviro_taxa_filt %>%
    select(Taxa4, sample_ID, copynum_SCG4) %>%
    drop_na() 

SCG4_copynum_enviro_taxa_filt_order_bysample <- SCG4_copynum_enviro_taxa_filt_order %>%
    transform(copynum_SCG4 = as.numeric(copynum_SCG4)) %>%
    group_by(Taxa4, sample_ID) %>%
    dplyr::summarise(totalreads = sum(copynum_SCG4))

#merging the all taxa dataset with the total reads by sample

SCG4_taxa_norm <- left_join(SCG4_copynum_enviro_taxa_filt_order_bysample,
                            SCG4_copynum_enviro_taxa_filt_totalreadsbysample,
                            by = "sample_ID") %>%
    mutate(normreads = totalreads/totalreadsbysample)

SCG4_taxa_norm_taxaavgbysample <- SCG4_taxa_norm %>%
    group_by(Taxa4) %>%
    dplyr::summarise(avgnormreads_SCG4 = mean(normreads))
SCG5_copynum_enviro_taxa_filt_totalreadsbysample <- SCG5_copynum_enviro_taxa_filt %>%
    select(Taxa4, sample_ID, copynum_SCG5) %>%
    transform(copynum_SCG5 = as.numeric(copynum_SCG5)) %>%
    group_by(sample_ID) %>%
    dplyr::summarise(totalreadsbysample = sum(copynum_SCG5))


SCG5_copynum_enviro_taxa_filt_order <- SCG5_copynum_enviro_taxa_filt %>%
    select(Taxa4, sample_ID, copynum_SCG5) %>%
    drop_na() 

SCG5_copynum_enviro_taxa_filt_order_bysample <- SCG5_copynum_enviro_taxa_filt_order %>%
    transform(copynum_SCG5 = as.numeric(copynum_SCG5)) %>%
    group_by(Taxa4, sample_ID) %>%
    dplyr::summarise(totalreads = sum(copynum_SCG5))

#merging the all taxa dataset with the total reads by sample

SCG5_taxa_norm <- left_join(SCG5_copynum_enviro_taxa_filt_order_bysample,
                            SCG5_copynum_enviro_taxa_filt_totalreadsbysample,
                            by = "sample_ID") %>%
    mutate(normreads = totalreads/totalreadsbysample)

SCG5_taxa_norm_taxaavgbysample <- SCG5_taxa_norm %>%
    group_by(Taxa4) %>%
    dplyr::summarise(avgnormreads_SCG5 = mean(normreads))

SCG6_copynum_enviro_taxa_filt_totalreadsbysample <- SCG6_copynum_enviro_taxa_filt %>%
    select(Taxa4, sample_ID, copynum_SCG6) %>%
    transform(copynum_SCG6 = as.numeric(copynum_SCG6)) %>%
    group_by(sample_ID) %>%
    dplyr::summarise(totalreadsbysample = sum(copynum_SCG6))


SCG6_copynum_enviro_taxa_filt_order <- SCG6_copynum_enviro_taxa_filt %>%
    select(Taxa4, sample_ID, copynum_SCG6) %>%
    drop_na() 

SCG6_copynum_enviro_taxa_filt_order_bysample <- SCG6_copynum_enviro_taxa_filt_order %>%
    transform(copynum_SCG6 = as.numeric(copynum_SCG6)) %>%
    group_by(Taxa4, sample_ID) %>%
    dplyr::summarise(totalreads = sum(copynum_SCG6))

#merging the all taxa dataset with the total reads by sample

SCG6_taxa_norm <- left_join(SCG6_copynum_enviro_taxa_filt_order_bysample,
                            SCG6_copynum_enviro_taxa_filt_totalreadsbysample,
                            by = "sample_ID") %>%
    mutate(normreads = totalreads/totalreadsbysample)

SCG6_taxa_norm_taxaavgbysample <- SCG6_taxa_norm %>%
    group_by(Taxa4) %>%
    dplyr::summarise(avgnormreads_SCG6 = mean(normreads))
SCG7_copynum_enviro_taxa_filt_totalreadsbysample <- SCG7_copynum_enviro_taxa_filt %>%
    select(Taxa4, sample_ID, copynum_SCG7) %>%
    transform(copynum_SCG7 = as.numeric(copynum_SCG7)) %>%
    group_by(sample_ID) %>%
    dplyr::summarise(totalreadsbysample = sum(copynum_SCG7))


SCG7_copynum_enviro_taxa_filt_order <- SCG7_copynum_enviro_taxa_filt %>%
    select(Taxa4, sample_ID, copynum_SCG7) %>%
    drop_na() 

SCG7_copynum_enviro_taxa_filt_order_bysample <- SCG7_copynum_enviro_taxa_filt_order %>%
    transform(copynum_SCG7 = as.numeric(copynum_SCG7)) %>%
    group_by(Taxa4, sample_ID) %>%
    dplyr::summarise(totalreads = sum(copynum_SCG7))

#merging the all taxa dataset with the total reads by sample

SCG7_taxa_norm <- left_join(SCG7_copynum_enviro_taxa_filt_order_bysample,
                            SCG7_copynum_enviro_taxa_filt_totalreadsbysample,
                            by = "sample_ID") %>%
    mutate(normreads = totalreads/totalreadsbysample)

SCG7_taxa_norm_taxaavgbysample <- SCG7_taxa_norm %>%
    group_by(Taxa4) %>%
    dplyr::summarise(avgnormreads_SCG7 = mean(normreads))
SCG8_copynum_enviro_taxa_filt_totalreadsbysample <- SCG8_copynum_enviro_taxa_filt %>%
    select(Taxa4, sample_ID, copynum_SCG8) %>%
    transform(copynum_SCG8 = as.numeric(copynum_SCG8)) %>%
    group_by(sample_ID) %>%
    dplyr::summarise(totalreadsbysample = sum(copynum_SCG8))


SCG8_copynum_enviro_taxa_filt_order <- SCG8_copynum_enviro_taxa_filt %>%
    select(Taxa4, sample_ID, copynum_SCG8) %>%
    drop_na() 

SCG8_copynum_enviro_taxa_filt_order_bysample <- SCG8_copynum_enviro_taxa_filt_order %>%
    transform(copynum_SCG8 = as.numeric(copynum_SCG8)) %>%
    group_by(Taxa4, sample_ID) %>%
    dplyr::summarise(totalreads = sum(copynum_SCG8))

#merging the all taxa dataset with the total reads by sample

SCG8_taxa_norm <- left_join(SCG8_copynum_enviro_taxa_filt_order_bysample,
                            SCG8_copynum_enviro_taxa_filt_totalreadsbysample,
                            by = "sample_ID") %>%
    mutate(normreads = totalreads/totalreadsbysample)


SCG8_taxa_norm$normreads[SCG8_taxa_norm$normreads == "NaN"] <- 0


SCG8_taxa_norm_taxaavgbysample <- SCG8_taxa_norm %>%
    group_by(Taxa4) %>%
    dplyr::summarise(avgnormreads_SCG8 = mean(normreads))
SCG9_copynum_enviro_taxa_filt_totalreadsbysample <- SCG9_copynum_enviro_taxa_filt %>%
    select(Taxa4, sample_ID, copynum_SCG9) %>%
    transform(copynum_SCG9 = as.numeric(copynum_SCG9)) %>%
    group_by(sample_ID) %>%
    dplyr::summarise(totalreadsbysample = sum(copynum_SCG9))


SCG9_copynum_enviro_taxa_filt_order <- SCG9_copynum_enviro_taxa_filt %>%
    select(Taxa4, sample_ID, copynum_SCG9) %>%
    drop_na() 

SCG9_copynum_enviro_taxa_filt_order_bysample <- SCG9_copynum_enviro_taxa_filt_order %>%
    transform(copynum_SCG9 = as.numeric(copynum_SCG9)) %>%
    group_by(Taxa4, sample_ID) %>%
    dplyr::summarise(totalreads = sum(copynum_SCG9))

####
SCG9_taxa_norm <- left_join(SCG9_copynum_enviro_taxa_filt_order_bysample,
                            SCG9_copynum_enviro_taxa_filt_totalreadsbysample,
                            by = "sample_ID") %>%
    mutate(normreads = totalreads/totalreadsbysample)

SCG9_taxa_norm_taxaavgbysample <- SCG9_taxa_norm %>%
    group_by(Taxa4) %>%
    dplyr::summarise(avgnormreads_SCG9 = mean(normreads))

   
############
library(readxl)
library(writexl)

write_xlsx(SCG123456789_taxa, "SCG123456789_taxa.xlsx")
#exporting into excel so that I can match taxa names with the names from the figure
SCG123456789_taxa_updated <- read_excel("SCG123456789_taxa_updated.xlsx")

SCG123456789_taxa_updated$avgnormreads_SCG9[is.na(SCG123456789_taxa_updated$avgnormreads_SCG9)] <- 0
SCG123456789_taxa_updated$avgnormreads_SCG4[is.na(SCG123456789_taxa_updated$avgnormreads_SCG4)] <- 0
SCG123456789_taxa_updated$avgnormreads_SCG5[is.na(SCG123456789_taxa_updated$avgnormreads_SCG5)] <- 0

#importing edited table
SCG123456789_taxa_updated_filt <- SCG123456789_taxa_updated %>%
  drop_na(...11) %>%
  select(-avgnormreads_SCG8) %>%
 rowwise() %>%
  mutate(row_median = median(c_across(where(is.numeric)), na.rm=TRUE))


########repeating for the larger size fraction from Iva

TARA_ANNOTATIONS_WITH_EVALUE_1e10_coverage_table <- read.delim("~/Downloads/Catalase/otherscores/EVALUE_THRESHOLD_RESULTS/TARA_ANNOTATIONS_WITH_EVALUE_1e10/TARA_ANNOTATIONS_WITH_EVALUE_1e10_coverage_table.txt")


SRA_metadata_large <- read.delim("~/Downloads/Catalase/otherscores/EVALUE_THRESHOLD_RESULTS/TARA_METADATA/SRA_metadata_large.txt") 

SRA_metadata_large_tomerge <- SRA_metadata_large %>%
  dplyr::rename("sample" = "Sample.Name")

metaG_large_sizefraction <- left_join(SRA_metadata_large_tomerge,
                                      TARA_ANNOTATIONS_WITH_EVALUE_1e10_coverage_table, 
                                      by = "sample")

TARA_NUM_READS_MAPPED <- read.delim("~/Downloads/Catalase/otherscores/EVALUE_THRESHOLD_RESULTS/TARA_NUM_READS_MAPPED.txt")

####working on data
metaG_large_sizefraction_readsmapped <- left_join(metaG_large_sizefraction,
                                      TARA_NUM_READS_MAPPED, 
                                      by = "sample")


metaG_large_sizefraction_readsmapped_trans <- metaG_large_sizefraction_readsmapped %>%
  mutate(reads_trans = coverage/total_reads_mapped) %>%
  filter(Depth == 5)

#assuming that the concatenated annotations means that the reads were identical
library(splitstackshape)
metaG_large_sizefraction_readsmapped_trans_notconcatenated <- metaG_large_sizefraction_readsmapped_trans %>%
  cSplit("annotation", ",", "long")
#first filter out the not large size fractions
#pooling by sample and gene
metaG_large_sizefraction_bygenebysample <- metaG_large_sizefraction_readsmapped_trans_notconcatenated %>%
  group_by(annotation, sample) %>%
  dplyr::summarise(total_reads = sum(reads_trans)) %>%
  drop_na()

#now go through and separate the catalases from the single copy genes
onlySCGs <- metaG_large_sizefraction_bygenebysample %>%
  filter(annotation == "SRP19" | annotation == "SRP54" | annotation == "Seryl_tRNA_N" | annotation == "TsaD" | annotation == "tRNA-synt_1" | annotation == "tRNA-synt_1_2" | annotation == "tRNA-synt_1e" | annotation == "tRNA-synt_2" | annotation == "tRNA-synt_2d")

#find the median reads/mapped reads in each sample 
onlySCGs_medians <- onlySCGs %>%
  group_by(sample) %>%
  dplyr::summarise(median_reads = median(total_reads))

###################selecting only the catalases and changing their names
catalases_large <- metaG_large_sizefraction_bygenebysample %>%
  filter(annotation == "PTHR30555.orig.30.pir" | annotation == "Mn_catalase" | annotation == "PTHR42821.SF1.pir") %>%
  mutate(catalase = case_when(annotation == "PTHR30555.orig.30.pir" ~ "katG",
       annotation == "Mn_catalase" ~ "manganese_catalase",
       annotation == "PTHR42821.SF1.pir" ~ "katE"))

catalases_large_trans <- left_join(catalases_large,
                                   onlySCGs_medians,
                                   by = "sample") %>%
  mutate(reads_trans = total_reads/median_reads)

#catalases_large_trans_with enviro info
catalases_large_trans_enviro <- left_join(catalases_large_trans, SRA_metadata_large_tomerge,
by = "sample")

write.csv(catalases_large_trans_enviro, "catalases_large_trans_enviro.csv")

catmodels_norm <- read.csv("~/Desktop/Dyhrman Lab/Desktop_DyhrmanLab/catmodels_norm.csv")
#turning copy numbers to 0
catmodels_norm_Arctic <- catmodels_norm %>%
  filter(latitude >= 50)

catmodels_norm_notArctic <- catmodels_norm %>%
  filter(latitude < 50)

#models separated by latitude,


#removing CO2
catmodels_Arctic <- catmodels_norm_Arctic %>%
  select(-CO2, -NPP, -POC, -Alkalinity, -Chlorophyll_A)


catmodels_notArctic <- catmodels_norm_notArctic %>%
  select(-CO2, -Alkalinity, -NPP, -Chlorophyll_A)

#dropping NA for arctic an not arctic
#Arctic
catmodels_clean_Arctic <- catmodels_Arctic %>%
  drop_na()

catmodels_Arctic_clean_SRF <- catmodels_clean_Arctic %>%
  filter(depth == "SRF")

catmodels_Arctic_clean_DCM <- catmodels_clean_Arctic %>%
  filter(depth == "DCM")

#NotArctic
catmodels_clean_notArctic <- catmodels_notArctic %>%
  filter(Region != "[SO] Southern Ocean (MRGID:1907)") %>%
  drop_na()

catmodels_notArctic_clean_SRF <- catmodels_clean_notArctic %>%
  filter(depth == "SRF") %>%
  mutate(NtoP = NO3_NO2/PO4)

catmodels_notArctic_clean_SRF$NtoP[catmodels_notArctic_clean_SRF$NtoP == "NaN"] <- 0
catmodels_notArctic_clean_SRF$NtoP[catmodels_notArctic_clean_SRF$NtoP == "Inf"] <- 0

catmodels_notArctic_clean_SRF <- catmodels_notArctic_clean_SRF %>%
  transform(NtoP = as.numeric(NtoP))


catmodels_notArctic_clean_DCM <- catmodels_clean_notArctic %>%
  filter(depth == "DCM")
# ##############################trying to gam
# #doing big models with psbO
# light01 <- mgcv::gam(normlight ~  s(normKatG, bs = "ts") + s(normKatE, bs = "ts") + s(normCatMn, bs = "ts") + s(Temperature, bs = "ts") + s(Salinity, bs = "ts") + s(Coast_km, bs = "ts") + s(PAR, bs = "ts") + s(NO3_NO2, bs = "ts") + s(PO4, bs = "ts") + s(Iron_5m, bs = "ts"), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# light02 <- mgcv::gam(normlight ~  s(normKatG, bs = "ts") + s(normCatMn, bs = "ts") + s(Temperature, bs = "ts") + s(Salinity, bs = "ts") + s(Coast_km, bs = "ts") + s(PAR, bs = "ts") + s(NO3_NO2, bs = "ts") + s(PO4, bs = "ts") + s(Iron_5m, bs = "ts"), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# light03 <- mgcv::gam(normlight ~  s(normKatG, bs = "ts") + s(Temperature, bs = "ts") + s(Salinity, bs = "ts") + s(Coast_km, bs = "ts") + s(PAR, bs = "ts") + s(NO3_NO2, bs = "ts") + s(PO4, bs = "ts") + s(Iron_5m, bs = "ts"), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# light04 <- mgcv::gam(normlight ~  s(normKatG, bs = "ts") + s(Salinity, bs = "ts") + s(Coast_km, bs = "ts") + s(PAR, bs = "ts") + s(NO3_NO2, bs = "ts") + s(PO4, bs = "ts") + s(Iron_5m, bs = "ts"), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# light05 <- mgcv::gam(normlight ~  s(normKatG, bs = "ts") + s(Coast_km, bs = "ts") + s(PAR, bs = "ts") + s(NO3_NO2, bs = "ts") + s(PO4, bs = "ts") + s(Iron_5m, bs = "ts"), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# light06 <- mgcv::gam(normlight ~  s(normKatG, bs = "ts") + s(PAR, bs = "ts") + s(NO3_NO2, bs = "ts") + s(PO4, bs = "ts") + s(Iron_5m, bs = "ts"), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# light07 <- mgcv::gam(normlight ~  s(normKatG, bs = "ts") + s(NO3_NO2, bs = "ts") + s(PO4, bs = "ts") + s(Iron_5m, bs = "ts"), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# light08 <- mgcv::gam(normlight ~  s(normKatG, bs = "ts") + s(PO4, bs = "ts") + s(Iron_5m, bs = "ts"), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# light09 <- mgcv::gam(normlight ~  s(normKatG, bs = "ts")  + s(Iron_5m, bs = "ts"), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# light10 <- mgcv::gam(normlight ~  s(normKatG, bs = "ts"), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# #nothing was significant
# 
# #running katG and katE
# 
# catmodels_notArctic_clean_SRF_KatE <- catmodels_notArctic_clean_SRF %>%
#   arrange(desc(normKatE))
#   
# catmodels_notArctic_clean_SRF_KatE <- catmodels_notArctic_clean_SRF_KatE[-c(1),]
# 
# plot(ggeffects::ggpredict(light06, terms = "PAR"), show_data = TRUE, colors = "blue", dot_size = 4, line_size = 2) + theme_classic() + ggtitle("") + theme(text=element_text(size=24)) + xlab("Iron concentration (µm/L) at 5 meters") + ylab("katG copy number per genome")
# 
# #######modeling katE with the light gene
# 
# KatE_light <- mgcv::gam(log10(normKatE) ~ s(normKatG) + s(normlight) + s(normCatMn) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF_KatE, method = "REML")
# 
# KatE_light2 <- mgcv::gam(log10(normKatE) ~  s(normlight) + s(normCatMn) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF_KatE, method = "REML")
# 
# KatE_light3 <- mgcv::gam(log10(normKatE) ~  s(normlight) + s(normCatMn) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF_KatE, method = "REML")
# 
# KatE_light4 <- mgcv::gam(log10(normKatE) ~  s(normlight) + s(normCatMn) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(PO4), data=catmodels_notArctic_clean_SRF_KatE, method = "REML")
# 
# KatE_light5 <- mgcv::gam(log10(normKatE) ~  normlight + s(normCatMn) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(PO4), data=catmodels_notArctic_clean_SRF_KatE, method = "REML")
# 
# KatE_light6 <- mgcv::gam(log10(normKatE) ~  normlight + s(normCatMn) + Temperature + s(Salinity) + s(Coast_km) + s(PAR) + s(PO4), data=catmodels_notArctic_clean_SRF_KatE, method = "REML")
# 
# KatE_light7 <- mgcv::gam(log10(normKatE) ~  normlight + s(normCatMn) + Temperature + Salinity + s(Coast_km) + s(PAR) + s(PO4), data=catmodels_notArctic_clean_SRF_KatE, method = "REML")
# 
# KatE_light8 <- mgcv::gam(log10(normKatE) ~  normlight + s(normCatMn) + Temperature + Salinity + Coast_km + s(PAR) + s(PO4), data=catmodels_notArctic_clean_SRF_KatE, method = "REML")
# 
# KatE_light9 <- mgcv::gam(log10(normKatE) ~  normlight + s(normCatMn) + Temperature + Salinity + Coast_km + s(PAR) + PO4, data=catmodels_notArctic_clean_SRF_KatE, method = "REML")
# 
# # KatE_light <- mgcv::gam(log10(normKatE) ~ s(normKatG, bs = "ts") + s(normlight, bs = "ts") + s(normCatMn, bs = "ts") + s(Temperature, bs = "ts") + s(Salinity, bs = "ts") + s(Coast_km, bs = "ts") + s(PAR, bs = "ts") + s(NO3_NO2, bs = "ts") + s(PO4, bs = "ts") + s(Iron_5m, bs = "ts"), data=catmodels_notArctic_clean_SRF_KatE, method = "REML")
# # 
# # KatE_light2 <- mgcv::gam(log10(normKatE) ~ s(normlight, bs = "ts") + s(normCatMn, bs = "ts") + s(Temperature, bs = "ts") + s(Salinity, bs = "ts") + s(Coast_km, bs = "ts") + s(PAR, bs = "ts") + s(NO3_NO2, bs = "ts") + s(PO4, bs = "ts") + s(Iron_5m, bs = "ts"), data=catmodels_notArctic_clean_SRF_KatE, method = "REML")
# # 
# # KatE_light3 <- mgcv::gam(log10(normKatE) ~ s(normlight, bs = "ts") + s(normCatMn, bs = "ts") + s(Temperature, bs = "ts") + s(Salinity, bs = "ts") + s(Coast_km, bs = "ts") + s(PAR, bs = "ts") + s(NO3_NO2, bs = "ts") + s(Iron_5m, bs = "ts"), data=catmodels_notArctic_clean_SRF_KatE, method = "REML")
# # 
# # KatE_light4 <- mgcv::gam(log10(normKatE) ~ s(normlight, bs = "ts") + s(normCatMn, bs = "ts") + s(Temperature, bs = "ts") + s(Salinity, bs = "ts") + s(Coast_km, bs = "ts") + s(PAR, bs = "ts") + s(NO3_NO2, bs = "ts") + s(Iron_5m, bs = "ts"), data=catmodels_notArctic_clean_SRF_KatE, method = "REML")
# # 
# # KatE_light5 <- mgcv::gam(log10(normKatE) ~ s(normlight, bs = "ts") + s(normCatMn, bs = "ts") + s(Temperature, bs = "ts") + s(Salinity, bs = "ts") + s(Coast_km, bs = "ts") + s(PAR, bs = "ts") + s(NO3_NO2, bs = "ts"), data=catmodels_notArctic_clean_SRF_KatE, method = "REML")
# # 
# # KatE_light6 <- mgcv::gam(log10(normKatE) ~ normlight + s(normCatMn, bs = "ts") + s(Temperature, bs = "ts") + s(Salinity, bs = "ts") + s(Coast_km, bs = "ts") + s(PAR, bs = "ts") + s(NO3_NO2, bs = "ts"), data=catmodels_notArctic_clean_SRF_KatE, method = "REML")
# # 
# # KatE_light7 <- mgcv::gam(log10(normKatE) ~ normlight + s(normCatMn, bs = "ts") + Temperature + s(Salinity, bs = "ts") + s(Coast_km, bs = "ts") + s(PAR, bs = "ts") + s(NO3_NO2, bs = "ts"), data=catmodels_notArctic_clean_SRF_KatE, method = "REML")
# # 
# # KatE_light8 <- mgcv::gam(log10(normKatE) ~ normlight + s(normCatMn, bs = "ts") + Temperature + Salinity + s(Coast_km, bs = "ts") + s(PAR, bs = "ts") + s(NO3_NO2, bs = "ts"), data=catmodels_notArctic_clean_SRF_KatE, method = "REML")
# # 
# # KatE_light9 <- mgcv::gam(log10(normKatE) ~ normlight + s(normCatMn, bs = "ts") + Temperature + Salinity + Coast_km + s(PAR, bs = "ts") + s(NO3_NO2, bs = "ts"), data=catmodels_notArctic_clean_SRF_KatE, method = "REML")
# # 
# # KatE_light10 <- mgcv::gam(log10(normKatE) ~ normlight + s(normCatMn, bs = "ts") + Temperature + Salinity + Coast_km + s(PAR, bs = "ts") + NO3_NO2, data=catmodels_notArctic_clean_SRF_KatE, method = "REML")
# 
# catmodels_notArctic_clean_SRF_KatE <- catmodels_notArctic_clean_SRF_KatE %>%
#   arrange(desc(normCatMn))
# 
# catmodels_notArctic_clean_SRF_KatE_CatMn <- catmodels_notArctic_clean_SRF_KatE[-c(1,2),]
# 
# KatE_light10 <- mgcv::gam(log10(normKatE) ~  normlight + s(normCatMn) + Temperature + Salinity + Coast_km + s(PAR) + PO4, data=catmodels_notArctic_clean_SRF_KatE_CatMn, method = "REML")
# 
# # KatE_light11 <- mgcv::gam(log10(normKatE) ~ normlight + s(normCatMn, bs = "ts") + Temperature + Salinity + Coast_km + s(PAR, bs = "ts") + NO3_NO2, data=catmodels_notArctic_clean_SRF_KatE_CatMn, method = "REML")
# 
# plot(ggeffects::ggpredict(KatE_light10, terms = "normCatMn"), show_data = TRUE, colors = "blue", dot_size = 4, line_size = 2) + theme_classic() + ggtitle("") + theme(text=element_text(size=24)) + xlab("manganese catalase copy number per genome") + ylab("katE copy number per genome")
# 
# ###########################
# #Salinity
# catmodels_notArctic_clean_SRF <- catmodels_notArctic_clean_SRF %>%
#   arrange(Salinity)
# 
# catmodels_notArctic_clean_SRF_Sal <- catmodels_notArctic_clean_SRF[-c(1),]
# 
# 
# 
# 
# 
# #katG light model
# KatG_light <- mgcv::gam(log10(normKatG) ~ s(normKatE) + s(normlight) + s(normCatMn) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# KatG_light2 <- mgcv::gam(log10(normKatG) ~ s(normKatE) + s(normCatMn) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# KatG_light3 <- mgcv::gam(log10(normKatG) ~ s(normKatE) + s(normCatMn) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(Iron_5m), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# KatG_light4 <- mgcv::gam(log10(normKatG) ~ normKatE + s(normCatMn) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(Iron_5m), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# KatG_light5 <- mgcv::gam(log10(normKatG) ~ normKatE + s(normCatMn) + s(Temperature) + s(Salinity) + Coast_km + s(PAR) + s(NO3_NO2) + s(Iron_5m), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# KatG_light6 <- mgcv::gam(log10(normKatG) ~ normKatE + s(normCatMn) + s(Temperature) + s(Salinity) + Coast_km + s(PAR) + NO3_NO2 + s(Iron_5m), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# KatG_light7 <- mgcv::gam(log10(normKatG) ~ normKatE + s(normCatMn) + s(Temperature) + s(Salinity) + Coast_km + s(PAR) + NO3_NO2 + Iron_5m, data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# KatG_light8 <- mgcv::gam(log10(normKatG) ~ normKatE + s(normCatMn) + s(Temperature) + s(Salinity) + Coast_km + s(PAR) + NO3_NO2 + Iron_5m, data=catmodels_notArctic_clean_SRF_KatE_CatMn, method = "REML")
# 
# ####################
# KatG_light <- mgcv::gam(log10(normKatG) ~ s(normKatE) + s(normlight) + s(normCatMn) + s(Temperature) + s(Salinity, bs = "ts") + s(Coast_km, bs = "ts") + s(PAR, bs = "ts") + s(NO3_NO2, bs = "ts") + s(PO4, bs = "ts") + s(Iron_5m, bs = "ts"), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# KatG_light2 <- mgcv::gam(log10(normKatG) ~ s(normKatE) + s(normCatMn) + s(Temperature) + s(Salinity, bs = "ts") + s(Coast_km, bs = "ts") + s(PAR, bs = "ts") + s(NO3_NO2, bs = "ts") + s(PO4, bs = "ts") + s(Iron_5m, bs = "ts"), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# KatG_light3 <- mgcv::gam(log10(normKatG) ~ s(normKatE) + s(normCatMn) + s(Temperature) + s(Coast_km, bs = "ts") + s(PAR, bs = "ts") + s(NO3_NO2, bs = "ts") + s(PO4, bs = "ts") + s(Iron_5m, bs = "ts"), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# KatG_light4 <- mgcv::gam(log10(normKatG) ~ s(normKatE) + s(normCatMn) + s(Temperature) + s(PAR, bs = "ts") + s(NO3_NO2, bs = "ts") + s(PO4, bs = "ts") + s(Iron_5m, bs = "ts"), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# KatG_light5 <- mgcv::gam(log10(normKatG) ~ s(normKatE) + s(normCatMn) + s(Temperature) + s(NO3_NO2, bs = "ts") + s(PO4, bs = "ts") + s(Iron_5m, bs = "ts"), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# KatG_light6 <- mgcv::gam(log10(normKatG) ~ s(normKatE) + s(normCatMn) + s(Temperature) + s(PO4, bs = "ts") + s(Iron_5m, bs = "ts"), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# KatG_light7 <- mgcv::gam(log10(normKatG) ~ s(normKatE) + s(normCatMn) + s(Temperature) + s(Iron_5m, bs = "ts"), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# KatG_light8 <- mgcv::gam(log10(normKatG) ~ s(normKatE) + s(normCatMn) + s(Temperature) + s(Iron_5m, bs = "ts"), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# KatG_light9 <- mgcv::gam(log10(normKatG) ~ normKatE + s(normCatMn) + s(Temperature) + s(Iron_5m, bs = "ts"), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# KatG_light10 <- mgcv::gam(log10(normKatG) ~ normKatE + s(normCatMn) + s(Temperature) + Iron_5m, data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# KatG_light10 <- mgcv::gam(log10(normKatG) ~ normKatE + s(normCatMn) + s(Temperature) + Iron_5m, data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# KatG_light10 <- mgcv::gam(log10(normKatG) ~ normKatE + s(normCatMn) + s(Temperature) + Iron_5m, data=catmodels_notArctic_clean_SRF_KatE, method = "REML")
# 
# #CatMn light model
# #cant do log transformation with 0s
# 
# CatMn_light <- mgcv::gam(normCatMn ~ s(normKatE) + s(normlight) + s(normKatG) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# catmodels_notArctic_clean_SRF_forlogcatmn <- catmodels_notArctic_clean_SRF
# 
# catmodels_notArctic_clean_SRF_forlogcatmn$normCatMn[catmodels_notArctic_clean_SRF_forlogcatmn$normCatMn == 0] <- 0.0000000001
# 
# catmodels_notArctic_clean_SRF_forlogcatmn_CatMn <- catmodels_notArctic_clean_SRF_forlogcatmn %>%
#   arrange(desc(normCatMn))
# 
# catmodels_notArctic_clean_SRF_forlogcatmn_CatMn <- catmodels_notArctic_clean_SRF_forlogcatmn_CatMn[-c(1,2),]
# 
# CatMn_light <- mgcv::gam(log10(normCatMn) ~ s(normKatE) + s(normlight) + s(normKatG) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF_forlogcatmn_CatMn, method = "REML")
# 
# CatMn_light2 <-  mgcv::gam(log10(normCatMn) ~ s(normKatE) + s(normKatG) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF_forlogcatmn_CatMn, method = "REML")
# 
# CatMn_light3 <-  mgcv::gam(log10(normCatMn) ~ s(normKatE) + s(normKatG) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF_forlogcatmn_CatMn, method = "REML")
# 
# CatMn_light4 <-  mgcv::gam(log10(normCatMn) ~ s(normKatE) + s(normKatG) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF_forlogcatmn_CatMn, method = "REML")
# 
# CatMn_light5 <-  mgcv::gam(log10(normCatMn) ~ s(normKatE) + s(normKatG) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF_forlogcatmn_CatMn, method = "REML")
# 
# CatMn_light6 <-  mgcv::gam(log10(normCatMn) ~ s(normKatE) + s(normKatG) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF_forlogcatmn_CatMn, method = "REML")
# 
# CatMn_light7 <-  mgcv::gam(log10(normCatMn) ~ s(normKatE) + s(normKatG) + s(NO3_NO2) + s(Iron_5m), data=catmodels_notArctic_clean_SRF_forlogcatmn_CatMn, method = "REML")
# 
# CatMn_light8 <-  mgcv::gam(log10(normCatMn) ~ s(normKatE) + s(normKatG) + s(NO3_NO2), data=catmodels_notArctic_clean_SRF_forlogcatmn_CatMn, method = "REML")
# 
# CatMn_light9 <-  mgcv::gam(log10(normCatMn) ~ normKatE + s(normKatG) + s(NO3_NO2), data=catmodels_notArctic_clean_SRF_forlogcatmn_CatMn, method = "REML")
# 
# ####psbO katE
# 
# #modeling psbO
# psbO1 <- mgcv::gam(log10(normpsbO) ~ s(normKatE) + s(normCatMn) + s(normKatG) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# psbO2 <- mgcv::gam(log10(normpsbO) ~ s(normKatE) + s(normKatG) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# psbO3 <- mgcv::gam(log10(normpsbO) ~ s(normKatE) + s(normKatG) + s(Temperature) + s(Salinity) + s(PAR) + s(NO3_NO2) + s(PO4), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# psbO4 <- mgcv::gam(log10(normpsbO) ~ s(normKatE) + normKatG + s(Temperature) + s(Salinity) + s(PAR) + s(NO3_NO2) + s(PO4), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# psbO5 <- mgcv::gam(log10(normpsbO) ~ s(normKatE) + normKatG + Temperature + s(Salinity) + s(PAR) + s(NO3_NO2) + s(PO4), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# psbO6 <- mgcv::gam(log10(normpsbO) ~ s(normKatE) + normKatG + Temperature + Salinity + s(PAR) + s(NO3_NO2) + s(PO4), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# psbO7 <- mgcv::gam(log10(normpsbO) ~ s(normKatE) + normKatG + Temperature + Salinity + PAR + s(NO3_NO2) + s(PO4), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# psbO8 <- mgcv::gam(log10(normpsbO) ~ s(normKatE) + normKatG + Temperature + Salinity + PAR + NO3_NO2 + s(PO4), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# psbO9 <- mgcv::gam(log10(normpsbO) ~ s(normKatE) + normKatG + Temperature + Salinity + PAR + NO3_NO2 + PO4, data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# psb10 <- mgcv::gam(log10(normpsbO) ~ s(normKatE) + normKatG + Temperature + Salinity + PAR + NO3_NO2 + PO4, data=catmodels_notArctic_clean_SRF_KatE, method = "REML")
# 
# ##########psbO katE
# KatE_psbO1 <- mgcv::gam(log10(normKatE) ~ s(normpsbO) + s(normCatMn) + s(normKatG) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF_KatE, method = "REML")
# 
# KatE_psbO2 <- mgcv::gam(log10(normKatE) ~ s(normpsbO) + s(normCatMn) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF_KatE, method = "REML")
# 
# KatE_psbO3 <- mgcv::gam(log10(normKatE) ~ s(normpsbO) + s(normCatMn) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(Iron_5m), data=catmodels_notArctic_clean_SRF_KatE, method = "REML")
# 
# KatE_psbO4 <- mgcv::gam(log10(normKatE) ~ s(normpsbO) + s(normCatMn) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2), data=catmodels_notArctic_clean_SRF_KatE, method = "REML")
# 
# KatE_psbO5 <- mgcv::gam(log10(normKatE) ~ s(normpsbO) + s(normCatMn) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2), data=catmodels_notArctic_clean_SRF_KatE, method = "REML")
# 
# KatE_psbO6 <- mgcv::gam(log10(normKatE) ~ s(normpsbO) + s(normCatMn) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2), data=catmodels_notArctic_clean_SRF_KatE, method = "REML")
# 
# KatE_psbO6 <- mgcv::gam(log10(normKatE) ~ s(normpsbO) + s(normCatMn) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2), data=catmodels_notArctic_clean_SRF_KatE, method = "REML")
# 
# KatE_psbO7 <- mgcv::gam(log10(normKatE) ~ s(normpsbO) + s(normCatMn) + Temperature + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2), data=catmodels_notArctic_clean_SRF_KatE, method = "REML")
# 
# KatE_psbO8 <- mgcv::gam(log10(normKatE) ~ s(normpsbO) + s(normCatMn) + Temperature + Salinity + s(Coast_km) + s(PAR) + s(NO3_NO2), data=catmodels_notArctic_clean_SRF_KatE, method = "REML")
# 
# KatE_psbO9 <- mgcv::gam(log10(normKatE) ~ s(normpsbO) + s(normCatMn) + Temperature + Salinity + Coast_km + s(PAR) + s(NO3_NO2), data=catmodels_notArctic_clean_SRF_KatE, method = "REML")
# 
# KatE_psbO10 <- mgcv::gam(log10(normKatE) ~ s(normpsbO) + s(normCatMn) + Temperature + Salinity + Coast_km + s(PAR) + NO3_NO2, data=catmodels_notArctic_clean_SRF_KatE_CatMn, method = "REML")
# 
# ####################
# #psbO katG
# 
# KatG_psbO1 <- mgcv::gam(log10(normKatG) ~ s(normpsbO) + s(normCatMn) + s(normKatE) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# KatG_psbO2 <- mgcv::gam(log10(normKatG) ~ s(normpsbO) + s(normCatMn) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# KatG_psbO3 <- mgcv::gam(log10(normKatG) ~ s(normpsbO) + s(normCatMn) + s(Temperature) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# KatG_psbO4 <- mgcv::gam(log10(normKatG) ~ s(normpsbO) + s(normCatMn) + s(Temperature) + s(Coast_km) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# KatG_psbO5 <- mgcv::gam(log10(normKatG) ~ s(normpsbO) + s(normCatMn) + s(Temperature) + s(Coast_km) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# KatG_psbO6 <- mgcv::gam(log10(normKatG) ~ s(normpsbO) + s(normCatMn) + s(Temperature) + s(Coast_km) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# KatG_psbO7 <- mgcv::gam(log10(normKatG) ~ normpsbO + s(normCatMn) + s(Temperature) + s(Coast_km) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# KatG_psbO8 <- mgcv::gam(log10(normKatG) ~ normpsbO + s(normCatMn) + Temperature + s(Coast_km) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# KatG_psbO9 <- mgcv::gam(log10(normKatG) ~ normpsbO + s(normCatMn) + Temperature + Coast_km + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# KatG_psb10 <- mgcv::gam(log10(normKatG) ~ normpsbO + s(normCatMn) + Temperature + Coast_km + PO4 + s(Iron_5m), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# KatG_psb11 <- mgcv::gam(log10(normKatG) ~ normpsbO + s(normCatMn) + Temperature + Coast_km + PO4 + Iron_5m, data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# catmodels_notArctic_clean_SRF_KatE <- catmodels_notArctic_clean_SRF %>%
#   arrange(desc(normCatMn))
# 
# catmodels_notArctic_clean_SRF_KatE <- catmodels_notArctic_clean_SRF_KatE[-c(1,2),]
# 
# KatG_psb11 <- mgcv::gam(log10(normKatG) ~ normpsbO + s(normCatMn) + Temperature + Coast_km + PO4 + Iron_5m, data=catmodels_notArctic_clean_SRF_KatE, method = "REML")
# #########
# #catmn
# catmodels_notArctic_clean_SRF_CatMn <- catmodels_notArctic_clean_SRF %>%
#   arrange(desc(normCatMn))
# 
# catmodels_notArctic_clean_SRF_CatMn <- catmodels_notArctic_clean_SRF_CatMn[-c(1,2),]
# 
# CatMn_psbO1 <- mgcv::gam(log10(normCatMn) ~ s(normpsbO) + s(normKatG) + s(normKatE) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF_CatMn, method = "REML")
# 
# catmodels_notArctic_clean_SRF_forlogcatmn <- catmodels_notArctic_clean_SRF
# 
# catmodels_notArctic_clean_SRF_forlogcatmn$normCatMn[catmodels_notArctic_clean_SRF_forlogcatmn$normCatMn == 0] <- 0.0000000001
# 
# catmodels_notArctic_clean_SRF_forlogcatmn_CatMn <- catmodels_notArctic_clean_SRF_forlogcatmn %>%
#   arrange(desc(normCatMn))
# 
# catmodels_notArctic_clean_SRF_forlogcatmn_CatMn <- catmodels_notArctic_clean_SRF_forlogcatmn_CatMn[-c(1,2),]
# 
# CatMn_psbO1 <- mgcv::gam(log10(normCatMn) ~ s(normpsbO) + s(normKatG) + s(normKatE) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF_forlogcatmn_CatMn, method = "REML")
# 
# CatMn_psbO2 <- mgcv::gam(log10(normCatMn) ~ s(normKatG) + s(normKatE) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF_forlogcatmn_CatMn, method = "REML")
# 
# CatMn_psbO3 <- mgcv::gam(log10(normCatMn) ~ s(normKatG) + s(normKatE) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF_forlogcatmn_CatMn, method = "REML")
# 
# CatMn_psbO4 <- mgcv::gam(log10(normCatMn) ~ s(normKatG) + s(normKatE) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF_forlogcatmn_CatMn, method = "REML")
# 
# CatMn_psbO5 <- mgcv::gam(log10(normCatMn) ~ s(normKatG) + s(normKatE) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF_forlogcatmn_CatMn, method = "REML")
# 
# CatMn_psbO6 <- mgcv::gam(log10(normCatMn) ~ s(normKatG) + s(normKatE) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF_forlogcatmn_CatMn, method = "REML")
# 
# CatMn_psbO7 <-  mgcv::gam(log10(normCatMn) ~ s(normKatG) + s(normKatE) + s(NO3_NO2) + s(Iron_5m), data=catmodels_notArctic_clean_SRF_forlogcatmn_CatMn, method = "REML")
# 
# CatMn_psbO8 <- mgcv::gam(log10(normCatMn) ~ s(normKatG) + s(normKatE) + s(NO3_NO2), data=catmodels_notArctic_clean_SRF_forlogcatmn_CatMn, method = "REML")
# 
# CatMn_psbO9 <- mgcv::gam(log10(normCatMn) ~ s(normKatG) + normKatE + s(NO3_NO2), data=catmodels_notArctic_clean_SRF_forlogcatmn_CatMn, method = "REML")
# 
# CatMn_psb10 <- mgcv::gam(log10(normCatMn) ~ s(normKatG) + normKatE + s(NO3_NO2), data=catmodels_notArctic_clean_SRF_forlogcatmn_CatMn, method = "REML")
# 
# #############
# catmodels_notArctic_clean_SRF_KatE
# psbO2 <- mgcv::gam(normpsbO ~  s(normKatG, bs = "ts") + s(normCatMn, bs = "ts") + s(Temperature, bs = "ts") + s(Salinity, bs = "ts") + s(Coast_km, bs = "ts") + s(PAR, bs = "ts") + s(NO3_NO2, bs = "ts") + s(PO4, bs = "ts") + s(Iron_5m, bs = "ts"), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# 
# catmodels_notArctic_clean_SRF_CatMn <- catmodels_notArctic_clean_SRF %>%
#   arrange(desc(normCatMn))
# 
# catmodels_notArctic_clean_SRF_CatMn <- catmodels_notArctic_clean_SRF_CatMn[-c(1,2),]
# 
# catmodels_notArctic_clean_SRF_CatMn_KatE <- catmodels_notArctic_clean_SRF %>%
#   arrange(desc(normKatE))
#   
# catmodels_notArctic_clean_SRF_CatMn_KatE <- catmodels_notArctic_clean_SRF_CatMn_KatE[-c(1),]
# 
# 
# # catmodels_notArctic_clean_SRF_Sal <- catmodels_notArctic_clean_SRF %>%
# #   arrange(desc(Salinity))
# # 
# # catmodels_notArctic_clean_SRF_Sal <- catmodels_notArctic_clean_SRF_Sal[-c(1),]
# 
# 
# #models from scratch
# 
# KatG1 <- mgcv::gam(normKatG ~  s(normrub, bs = "ts") + s(normKatE, bs = "ts") + s(normCatMn, bs = "ts") + s(Temperature, bs = "ts") + s(Salinity, bs = "ts") + s(Coast_km, bs = "ts") + s(PAR, bs = "ts") + s(NO3_NO2, bs = "ts") + s(PO4, bs = "ts") + s(Iron_5m, bs = "ts"), data=catmodels_notArctic_clean_SRF_Sal, method = "REML")
# 
# KatG2 <- mgcv::gam(normKatG ~  s(normrub, bs = "ts") + s(normKatE, bs = "ts") + s(normCatMn, bs = "ts") + s(Salinity, bs = "ts") + s(Coast_km, bs = "ts") + s(PAR, bs = "ts") + s(NO3_NO2, bs = "ts") + s(PO4, bs = "ts") + s(Iron_5m, bs = "ts"), data=catmodels_notArctic_clean_SRF_Sal, method = "REML")
# 
# KatG3 <- mgcv::gam(normKatG ~  s(normrub, bs = "ts") + s(normKatE, bs = "ts") + s(normCatMn, bs = "ts") + s(Coast_km, bs = "ts") + s(PAR, bs = "ts") + s(NO3_NO2, bs = "ts") + s(PO4, bs = "ts") + s(Iron_5m, bs = "ts"), data=catmodels_notArctic_clean_SRF_Sal, method = "REML")
# 
# KatG4 <- mgcv::gam(normKatG ~  normrub + s(normKatE, bs = "ts") + s(normCatMn, bs = "ts") + s(Coast_km, bs = "ts") + s(PAR, bs = "ts") + s(NO3_NO2, bs = "ts") + s(PO4, bs = "ts") + s(Iron_5m, bs = "ts"), data=catmodels_notArctic_clean_SRF_Sal, method = "REML")
# 
# KatG5 <- mgcv::gam(normKatG ~  normrub + s(normKatE, bs = "ts") + s(normCatMn, bs = "ts") + s(Coast_km, bs = "ts") + s(PAR, bs = "ts") + s(NO3_NO2, bs = "ts") + s(PO4, bs = "ts") + s(Iron_5m, bs = "ts"), data=catmodels_notArctic_clean_SRF_Sal, method = "REML")
# 
# KatG6 <- mgcv::gam(normKatG ~  normrub + s(normKatE, bs = "ts") + normCatMn + s(Coast_km, bs = "ts") + s(PAR, bs = "ts") + s(NO3_NO2, bs = "ts") + s(PO4, bs = "ts") + s(Iron_5m, bs = "ts"), data=catmodels_notArctic_clean_SRF_Sal, method = "REML")
# 
# KatG7 <- mgcv::gam(normKatG ~  normrub + s(normKatE, bs = "ts") + normCatMn +  s(PAR, bs = "ts") + s(NO3_NO2, bs = "ts") + s(PO4, bs = "ts") + s(Iron_5m, bs = "ts"), data=catmodels_notArctic_clean_SRF_Sal, method = "REML")
# 
# KatG8 <- mgcv::gam(normKatG ~  normrub + s(normKatE, bs = "ts") + normCatMn +  PAR + s(NO3_NO2, bs = "ts") + s(PO4, bs = "ts") + s(Iron_5m, bs = "ts"), data=catmodels_notArctic_clean_SRF_Sal, method = "REML")
# 
# KatG9 <- mgcv::gam(normKatG ~  normrub + s(normKatE, bs = "ts") + normCatMn +  PAR + s(NO3_NO2, bs = "ts") + PO4 + s(Iron_5m, bs = "ts"), data=catmodels_notArctic_clean_SRF_Sal, method = "REML")
# 
# KatG10 <- mgcv::gam(normKatG ~  normrub + s(normKatE, bs = "ts") + normCatMn +  PAR + s(NO3_NO2, bs = "ts") + PO4 + Iron_5m, data=catmodels_notArctic_clean_SRF_Sal, method = "REML")
# 
# KatG11 <- mgcv::gam(normKatG ~  normrub + s(normKatE) + normCatMn +  PAR + s(NO3_NO2) + PO4 + Iron_5m, data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# KatG12 <- mgcv::gam(normKatG ~  normrub + s(normKatE) + normCatMn +  PAR + s(NO3_NO2) + PO4 + Iron_5m, data=catmodels_notArctic_clean_SRF_CatMn, method = "REML")
# 
# KatG13 <- mgcv::gam(normKatG ~  normrub + s(normKatE) + normCatMn +  PAR + s(NO3_NO2) + PO4 + Iron_5m, data=catmodels_notArctic_clean_SRF_CatMn_KatE, method = "REML")
# 
# # KatG14 <- mgcv::gam(normKatG ~  normrub + s(normKatE, bs = "ts") + normCatMn +  PAR + s(NO3_NO2, bs = "ts") + PO4 + Iron_5m, data=catmodels_notArctic_clean_SRF_CatMn_KatE, method = "REML")
# 
# 
# 
# plot(ggeffects::ggpredict(KatG13, terms = "Iron_5m"), show_data = TRUE, colors = "blue", dot_size = 4, line_size = 2) + theme_classic() + ggtitle("") + theme(text=element_text(size=24)) + xlab("Iron concentration (µm/L) at 5 meters") + ylab("katG copy number per genome")
# 
# #combine DCM and SRF and see if I get the same predictors
# 
# KatG1_alldepths <- mgcv::gam(normKatG ~  s(normrub) + s(normKatE) + s(normCatMn) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_clean_notArctic, method = "REML")
# 
# KatG2_alldepths <- mgcv::gam(normKatG ~  s(normrub) + s(normCatMn) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_clean_notArctic, method = "REML")
# 
# KatG3_alldepths <- mgcv::gam(normKatG ~  s(normrub) + s(normCatMn) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_clean_notArctic, method = "REML")
# 
# KatG4_alldepths <- mgcv::gam(normKatG ~  s(normrub) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_clean_notArctic, method = "REML")
# 
# KatG5_alldepths <- mgcv::gam(normKatG ~  s(normrub) + s(Salinity) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_clean_notArctic, method = "REML")
# 
# KatG5_alldepths <- mgcv::gam(normKatG ~  s(normrub) + s(Salinity) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_clean_notArctic, method = "REML")
# 
# KatG6_alldepths <- mgcv::gam(normKatG ~  s(normrub) + s(Salinity) + s(PAR) + s(PO4) + s(Iron_5m), data=catmodels_clean_notArctic, method = "REML")
# 
# KatG7_alldepths <-  mgcv::gam(normKatG ~  s(normrub) + s(Salinity) + s(PAR) + s(PO4) + s(Iron_5m), data=catmodels_clean_notArctic, method = "REML")
# 
# KatG8_alldepths <-  mgcv::gam(normKatG ~  normrub + s(Salinity) + s(PAR) + s(PO4) + s(Iron_5m), data=catmodels_clean_notArctic, method = "REML")
# 
# KatG9_alldepths <-  mgcv::gam(normKatG ~  normrub + s(Salinity) + PAR + s(PO4) + s(Iron_5m), data=catmodels_clean_notArctic, method = "REML")
# 
# KatG10_alldepths <-  mgcv::gam(normKatG ~  normrub + s(Salinity) + PAR + PO4 + s(Iron_5m), data=catmodels_clean_notArctic, method = "REML")
# 
# KatG11_alldepths <-  mgcv::gam(normKatG ~  normrub + s(Salinity) + PAR + PO4 + Iron_5m, data=catmodels_clean_notArctic, method = "REML")
# 
# KatG12_alldepths <-  mgcv::gam(normKatG ~  normrub + s(Salinity) + PAR + PO4 + Iron_5m, data=catmodels_clean_notArctic, method = "REML")
#     
# 
# catmodels_clean_notArctic_KatG <- catmodels_clean_notArctic %>%
#   arrange(normKatG)
# 
# catmodels_clean_notArctic_KatG <- catmodels_clean_notArctic_KatG[-c(1),]
# 
# catmodels_clean_notArctic_KatG_Sal <- catmodels_clean_notArctic_KatG %>%
#   arrange(Salinity)
# 
# catmodels_clean_notArctic_KatG_Sal <- catmodels_clean_notArctic_KatG_Sal[-c(1),]
# 
# catmodels_clean_notArctic_KatG_Sal_PO4 <- catmodels_clean_notArctic_KatG_Sal %>%
#   arrange(desc(PO4))
# 
# catmodels_clean_notArctic_KatG_Sal_PO4 <- catmodels_clean_notArctic_KatG_Sal_PO4[-c(1),]
# 
# 
# KatG12_alldepths <-  mgcv::gam(normKatG ~  normrub + s(Salinity) + PAR + PO4 + Iron_5m, data=catmodels_clean_notArctic_KatG_Sal, method = "REML")
# 
# #KatE
# 
# KatE1 <- mgcv::gam(normKatE ~  s(normrub) + s(normKatG) + s(normCatMn) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# KatE2 <- mgcv::gam(normKatE ~ s(normKatG) + s(normCatMn) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# KatE3 <- mgcv::gam(normKatE ~ s(normKatG) + s(normCatMn) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# KatE4 <- mgcv::gam(normKatE ~ normKatG + s(normCatMn) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# KatE5 <- mgcv::gam(normKatE ~ normKatG + s(normCatMn) + Temperature + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# KatE6 <- mgcv::gam(normKatE ~ normKatG + s(normCatMn) + Temperature + s(Salinity) + s(Coast_km) + s(PAR) + NO3_NO2 + s(PO4), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# catmodels_notArctic_clean_SRF_katE <- catmodels_notArctic_clean_SRF %>%
#   arrange(desc(normKatE))
# 
# catmodels_notArctic_clean_SRF_katE <- catmodels_notArctic_clean_SRF_katE[-c(1),]
# 
# catmodels_notArctic_clean_SRF_katE_Sal <- catmodels_notArctic_clean_SRF_katE %>%
#   arrange((Salinity))
# 
# catmodels_notArctic_clean_SRF_katE_Sal <- catmodels_notArctic_clean_SRF_katE_Sal[-c(1),]
# 
# KatE6 <- mgcv::gam(normKatE ~ normKatG + s(normCatMn) + Temperature + s(Salinity) + s(Coast_km) + s(PAR) + NO3_NO2 + s(PO4), data=catmodels_notArctic_clean_SRF_katE_Sal, method = "REML")
# 
# ##running the model with log transformed KatE
# 
# KatE1_log <- mgcv::gam(log10(normKatE) ~  s(normrub) + s(normKatG) + s(normCatMn) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# KatE2_log <- mgcv::gam(log10(normKatE) ~  s(normKatG) + s(normCatMn) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# KatE3_log <- mgcv::gam(log10(normKatE) ~  s(normKatG) + s(normCatMn) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# KatE4_log <- mgcv::gam(log10(normKatE) ~  normKatG + s(normCatMn) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# KatE5_log <- mgcv::gam(log10(normKatE) ~  normKatG + s(normCatMn) + Temperature + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# KatE6_log <- mgcv::gam(log10(normKatE) ~  normKatG + s(normCatMn) + Temperature + Salinity + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# KatE7_log <- mgcv::gam(log10(normKatE) ~  normKatG + s(normCatMn) + Temperature + Salinity + Coast_km + s(PAR) + s(NO3_NO2) + s(PO4), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# KatE8_log <- mgcv::gam(log10(normKatE) ~  normKatG + s(normCatMn) + Temperature + Salinity + Coast_km + s(PAR) + NO3_NO2 + s(PO4), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# KatE9_log <- mgcv::gam(log10(normKatE) ~  normKatG + s(normCatMn) + Temperature + Salinity + Coast_km + s(PAR) + NO3_NO2 + PO4, data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# 
# plot(ggeffects::ggpredict(KatE3_log, terms = "PAR"), show_data = TRUE, colors = "blue", dot_size = 4, line_size = 2) + theme_classic() + ggtitle("") + theme(text=element_text(size=24)) + xlab("Distance from the coast") + ylab("katE copy number per genome")
# 
# catmodels_notArctic_clean_SRF_katE <- catmodels_notArctic_clean_SRF %>%
#   arrange(desc(normKatE))
# 
# catmodels_notArctic_clean_SRF_katE <- catmodels_notArctic_clean_SRF_katE[-c(1),]
# 
# catmodels_notArctic_clean_SRF_katE_Sal <- catmodels_notArctic_clean_SRF_katE %>%
#   arrange((Salinity))
# 
# catmodels_notArctic_clean_SRF_katE_Sal <- catmodels_notArctic_clean_SRF_katE_Sal[-c(1),]
# 
# catmodels_notArctic_clean_SRF_katE_Sal_CatMn <- catmodels_notArctic_clean_SRF_katE_Sal %>%
#   arrange(desc(normCatMn))
# 
# catmodels_notArctic_clean_SRF_katE_Sal_CatMn <- catmodels_notArctic_clean_SRF_katE_Sal_CatMn[-c(1),]
# 
# KatE9_log_nooutliers <- mgcv::gam(log10(normKatE) ~  normKatG + s(normCatMn) + Temperature + Salinity + Coast_km + s(PAR) + NO3_NO2 + PO4, data=catmodels_notArctic_clean_SRF_katE_Sal_CatMn, method = "REML")
# 
# plot(ggeffects::ggpredict(KatE9_log_nooutliers, terms = "normCatMn"), show_data = TRUE, colors = "blue", dot_size = 4, line_size = 2) + theme_classic() + ggtitle("") + theme(text=element_text(size=24)) + xlab("manganese catalase copy number per genome") + ylab("katE copy number per genome")
# 
# #now combine SRF and DCM
# KatE1_log_alldepths <- mgcv::gam(log10(normKatE) ~  s(normrub) + s(normKatG) + s(normCatMn) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_clean_notArctic, method = "REML")
# 
# KatE2_log_alldepths <- mgcv::gam(log10(normKatE) ~ s(normrub) + s(normCatMn) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_clean_notArctic, method = "REML")
# 
# KatE3_log_alldepths <- mgcv::gam(log10(normKatE) ~ s(normrub) + s(normCatMn) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_clean_notArctic, method = "REML")
# 
# KatE4_log_alldepths <- mgcv::gam(log10(normKatE) ~ s(normrub) + s(normCatMn) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4), data=catmodels_clean_notArctic, method = "REML")
# 
# KatE5_log_alldepths <- mgcv::gam(log10(normKatE) ~ normrub + s(normCatMn) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4), data=catmodels_clean_notArctic, method = "REML")
# 
# KatE6_log_alldepths <- mgcv::gam(log10(normKatE) ~ normrub + s(normCatMn) + Salinity + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4), data=catmodels_clean_notArctic, method = "REML")
# 
# KatE7_log_alldepths <- mgcv::gam(log10(normKatE) ~ normrub + s(normCatMn) + Salinity + Coast_km + s(PAR) + s(NO3_NO2) + s(PO4), data=catmodels_clean_notArctic, method = "REML")
# 
# KatE8_log_alldepths <- mgcv::gam(log10(normKatE) ~ normrub + s(normCatMn) + Salinity + Coast_km + s(PAR) + NO3_NO2 + s(PO4), data=catmodels_clean_notArctic, method = "REML")
# 
# KatE9_log_alldepths <- mgcv::gam(log10(normKatE) ~ normrub + s(normCatMn) + Salinity + Coast_km + s(PAR) + NO3_NO2 + PO4, data=catmodels_clean_notArctic, method = "REML")
# 
# plot(ggeffects::ggpredict(KatE9_log_alldepths, terms = "PAR"), show_data = TRUE, colors = "blue", dot_size = 4, line_size = 2) + theme_classic() + ggtitle("") + theme(text=element_text(size=24)) + xlab("PAR") + ylab("katE copy number per genome")
# 
# catmodels_clean_notArctic_ordered <- catmodels_clean_notArctic %>%
#   arrange(desc(normKatE))
# # KatE62<- mgcv::gam(normKatE ~ normKatG + s(normCatMn) + Temperature + s(Salinity) + s(Coast_km) + s(PAR) + NO3_NO2 + s(PO4), family = tw(link=log), data = catmodels_notArctic_clean_SRF_katE_Sal, method = "REML")
# # 
# 
# # KatE12 <- mgcv::gam(normKatE ~  s(normrub) + s(normKatG) + s(normCatMn) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m),family = tw(link=log), data=catmodels_notArctic_clean_SRF, method = "REML")
# # 
# # KatE13 <- mgcv::gam(normKatE ~ s(normKatG) + s(normCatMn) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m),family = tw(link=log), data=catmodels_notArctic_clean_SRF, method = "REML")
# # 
# # KatE15 <- mgcv::gam(normKatE ~ s(normKatG) + s(normCatMn) + s(Temperature) + s(Salinity) + s(NO3_NO2) + s(PO4) + s(Iron_5m),family = tw(link=log), data=catmodels_notArctic_clean_SRF, method = "REML")
# # 
# # KatE16 <- mgcv::gam(normKatE ~ s(normKatG) + s(normCatMn) + s(Temperature) + s(Salinity) + s(PO4) + s(Iron_5m),family = tw(link=log), data=catmodels_notArctic_clean_SRF, method = "REML")
# # 
# # KatE16 <- mgcv::gam(normKatE ~ s(normKatG) + s(normCatMn) + s(Temperature) + s(Salinity) + s(PO4) + s(Iron_5m),family = tw(link=log), data=catmodels_notArctic_clean_SRF, method = "REML")
# # 
# # KatE16 <- mgcv::gam(normKatE ~ s(normKatG) + s(normCatMn) + s(Temperature, k = 20) + s(Salinity) + s(PO4, k = 20) + s(Iron_5m),family = tw(link=log), data=catmodels_notArctic_clean_SRF_katE_Sal, method = "REML")
# # 
# # plot(ggeffects::ggpredict(KatE16, terms = "PO4"), show_data = TRUE, colors = "blue", dot_size = 4, line_size = 2) + theme_classic() + ggtitle("") + theme(text=element_text(size=24)) + xlab("Distance from the coast") + ylab("katE copy number per genome")
# 
# 
# #doing k check
# 
# #combining SRF and DCM data
# 
# #manganese catalase
# 
# 
# CatMn1 <- mgcv::gam(normCatMn ~  s(normrub) + s(normKatG) + s(normKatE) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# catmodels_notArctic_clean_SRF <- catmodels_notArctic_clean_SRF %>%
#   arrange(desc(normCatMn))
# 
# catmodels_notArctic_clean_SRF_CatMn <- catmodels_notArctic_clean_SRF[-c(1:2),]
# 
# CatMn1 <- mgcv::gam(normCatMn ~  s(normrub) + s(normKatG) + s(normKatE) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF_CatMn, method = "REML")
# 
# CatMn2 <- mgcv::gam(normCatMn ~ s(normKatG) + s(normKatE) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF_CatMn, method = "REML")
# 
# CatMn3 <- mgcv::gam(normCatMn ~ s(normKatG) + s(normKatE) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF_CatMn, method = "REML")
# 
# CatMn4 <- mgcv::gam(normCatMn ~ s(normKatG) + s(normKatE) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF_CatMn, method = "REML")
# 
# CatMn5 <- mgcv::gam(normCatMn ~ s(normKatG) + s(normKatE) + s(Coast_km) + s(PAR) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF_CatMn, method = "REML")
# 
# CatMn6 <- mgcv::gam(normCatMn ~ s(normKatG) + s(normKatE) + s(Coast_km) + s(PAR) + s(PO4), data=catmodels_notArctic_clean_SRF_CatMn, method = "REML")
# 
# CatMn6 <- mgcv::gam(normCatMn ~ s(normKatG) + s(normKatE) + s(Coast_km) + s(PAR) + s(PO4), data=catmodels_notArctic_clean_SRF_CatMn, method = "REML")
# 
# CatMn6 <- mgcv::gam(normCatMn ~ s(normKatG) + s(normKatE) + s(Coast_km) + s(PAR) + s(PO4), data=catmodels_notArctic_clean_SRF_CatMn, method = "REML")
# 
# CatMn7 <- mgcv::gam(normCatMn ~ s(normKatG) + s(normKatE) + s(Coast_km) + s(PAR) + s(PO4), data=catmodels_notArctic_clean_SRF_CatMn, method = "REML")
# 
# catmodels_notArctic_clean_SRF_CatMn <- catmodels_notArctic_clean_SRF_CatMn %>%
#   arrange(desc(normKatE))
# 
# catmodels_notArctic_clean_SRF_CatMn_KatE <- catmodels_notArctic_clean_SRF_CatMn[-c(1),]
# 
# CatMn7 <- mgcv::gam(normCatMn ~ s(normKatG) + s(normKatE) + s(Coast_km) + s(PAR) + s(PO4, bs = "ts"), data=catmodels_notArctic_clean_SRF_CatMn_KatE, method = "REML")
# 
# 
# plot(ggeffects::ggpredict(CatMn7, terms = "PAR"), show_data = TRUE, colors = "blue", dot_size = 4, line_size = 2) + theme_classic() + ggtitle("") + theme(text=element_text(size=24)) + xlab("PAR") + ylab("manganese catalase copy number per genome")
# 
# #SRF and DCM
# CatMn1_alldepths <- mgcv::gam(normCatMn ~  s(normrub) + s(normKatG) + s(normKatE) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_clean_notArctic, method = "REML")
# 
# catmodels_clean_notArctic <- catmodels_clean_notArctic %>%
#   arrange(desc(normCatMn))
# 
# catmodels_clean_notArctic_CatMn <- catmodels_clean_notArctic[-c(1),]
# 
# CatMn1_alldepths <- mgcv::gam(normCatMn ~  s(normrub) + s(normKatG) + s(normKatE) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_clean_notArctic_CatMn, method = "REML")
# 
# CatMn2_alldepths <- mgcv::gam(normCatMn ~  s(normrub) + s(normKatG) + s(normKatE) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_clean_notArctic_CatMn, method = "REML")
# 
# CatMn3_alldepths <- mgcv::gam(normCatMn ~  s(normrub) + s(normKatG) + s(normKatE) + s(Salinity)  + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_clean_notArctic_CatMn, method = "REML")
# 
# CatMn4_alldepths <- mgcv::gam(normCatMn ~  s(normrub) + s(normKatG) + s(normKatE) + s(Salinity) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_clean_notArctic_CatMn, method = "REML")
# 
# CatMn5_alldepths <- mgcv::gam(normCatMn ~  s(normrub) + s(normKatG) + s(normKatE) + s(Salinity) + s(PO4) + s(Iron_5m), data=catmodels_clean_notArctic_CatMn, method = "REML")
# 
# CatMn6_alldepths <- mgcv::gam(normCatMn ~  s(normrub) + s(normKatG) + s(normKatE) + s(Salinity) + s(Iron_5m), data=catmodels_clean_notArctic_CatMn, method = "REML")
# 
# CatMn7_alldepths <- mgcv::gam(normCatMn ~  s(normrub) + s(normKatG) + s(normKatE) + s(Salinity), data=catmodels_clean_notArctic_CatMn, method = "REML")
# 
# CatMn8_alldepths <- mgcv::gam(normCatMn ~  normrub + s(normKatG) + s(normKatE) + s(Salinity), data=catmodels_clean_notArctic_CatMn, method = "REML")
# 
# CatMn9_alldepths <- mgcv::gam(normCatMn ~  normrub + s(normKatG) + s(normKatE) + s(Salinity), data=catmodels_clean_notArctic_CatMn, method = "REML")
# 
# 
# catmodels_clean_notArctic <- catmodels_clean_notArctic %>%
#   arrange(desc(normCatMn))
# 
# catmodels_clean_notArctic_CatMn_more <- catmodels_clean_notArctic[-c(1),]
# 
# catmodels_clean_notArctic_CatMn_Sal <- catmodels_clean_notArctic_CatMn_more %>%
#   arrange((normKatG))
# 
# catmodels_clean_notArctic_CatMn_Sal_katG <- catmodels_clean_notArctic_CatMn_Sal[-c(1),]
# 
# CatMn9_alldepths <- mgcv::gam(normCatMn ~  normrub + s(normKatG) + s(normKatE) + s(Salinity), data=catmodels_clean_notArctic_CatMn_Sal_katG, method = "REML")
# 
# CatMn9_alldepths <- mgcv::gam(normCatMn ~  normrub + s(normKatG) + s(normKatE) + s(Salinity), data=catmodels_clean_notArctic_CatMn_Sal_katG, method = "REML")
# 
# 
# ggeffects::ggpredict(rub13, terms = "normKatE"))
# 
# 
# , show_data = TRUE, colors = "blue", dot_size = 4, line_size = 2)) + theme_classic() + ggtitle("") + theme(text=element_text(size=24)) + xlab("rubisco copy number per genome") + ylab("rubisco copy number per genome") + ggplot(mtcars, aes(x=wt, y=mpg, shape=cyl, color=cyl)) +
#   geom_point()
# 
# 
# CatMn3_alldepths <- mgcv::gam(normCatMn ~  s(normrub) + s(normKatG) + s(normKatE) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_clean_notArctic, method = "REML")
# 
# CatMn4_alldepths <- mgcv::gam(normCatMn ~  s(normrub) + s(normKatG) + s(normKatE) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_clean_notArctic, method = "REML")
# 
# CatMn5_alldepths <- mgcv::gam(normCatMn ~  s(normrub) + s(normKatG) + s(normKatE) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_clean_notArctic, method = "REML")
# 
# 
# CatMn2_alldepths <- mgcv::gam(normCatMn ~  s(normrub) + s(normKatG) + s(normKatE) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_clean_notArctic_CatMn, method = "REML")
# 
# #rubisco
# rub1 <- mgcv::gam(normrub ~  s(normCatMn) + s(normKatG) + s(normKatE) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# rub2<- mgcv::gam(normrub ~  s(normKatG) + s(normKatE) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# rub3 <- mgcv::gam(normrub ~  s(normKatG) + s(normKatE) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# rub4 <- mgcv::gam(normrub ~  s(normKatG) + s(normKatE) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# rub5 <- mgcv::gam(normrub ~  s(normKatG) + s(normKatE) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# 
# catmodels_notArctic_clean_SRF <- catmodels_notArctic_clean_SRF %>%
#   arrange(desc(normKatE))
# 
# catmodels_notArctic_clean_SRF_katE <- catmodels_notArctic_clean_SRF[-c(1),]
# 
# rub5 <- mgcv::gam(normrub ~  s(normKatG) + s(normKatE) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF_katE, method = "REML")
# 
# rub6 <- mgcv::gam(normrub ~  normKatG + s(normKatE) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF_katE, method = "REML")
# 
# rub7 <- mgcv::gam(normrub ~  normKatG + normKatE + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF_katE, method = "REML")
# 
# rub8 <- mgcv::gam(normrub ~  normKatG + normKatE + Temperature + s(Salinity) + s(Coast_km) + s(PAR) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF_katE, method = "REML")
# 
# rub9 <- mgcv::gam(normrub ~  normKatG + normKatE + Temperature + Salinity + s(Coast_km) + s(PAR) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF_katE, method = "REML")
# 
# rub10 <- mgcv::gam(normrub ~  normKatG + normKatE + Temperature + Salinity + Coast_km + s(PAR) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF_katE, method = "REML")
# 
# rub11 <- mgcv::gam(normrub ~  normKatG + normKatE + Temperature + Salinity + Coast_km + PAR + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF_katE, method = "REML")
# 
# rub12 <- mgcv::gam(normrub ~  normKatG + normKatE + Temperature + Salinity + Coast_km + PAR + PO4 + s(Iron_5m), data=catmodels_notArctic_clean_SRF_katE, method = "REML")
# 
# rub13 <- mgcv::gam(normrub ~  normKatG + normKatE + Temperature + Salinity + Coast_km + PAR + PO4 + s(Iron_5m, k = 6), data=catmodels_notArctic_clean_SRF_katE, method = "REML")
# 
# newdata <- data.frame(Petal.Length = iris$Petal.Length,
#                       Sepal.Width = samp_vals[i],
#                       Petal.Width = mean(iris$Petal.Width))
# ## <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
# 
# 
# 
# newdata$y <- predict(m, newdata = newdata)
# newdata$dots <- newdata$y + resid(m)
# 
# 
# library(ggplot2)
# 
# library(ggeffects)
# 
# model <- mgcv::gam(normKatE ~ s(latitude), data=catmodels_notArctic_clean_SRF_katE, method = "REML")
# model <- mgcv::gam(normKatG ~ s(latitude), data=catmodels_notArctic_clean_SRF_katE, method = "REML")
# 
# model <- mgcv::gam(normKatG ~ s(latitude), data=catmodels_notArctic_clean_SRF_CatMn_KatE, method = "REML")
# 
# model <- mgcv::gam(normCatMn ~ s(latitude), data=catmodels_notArctic_clean_SRF_CatMn_KatE, method = "REML")
# 
# 
# CatMn7 catmodels_notArctic_clean_SRF_CatMn_KatE
# 
# gge <- ggeffects::ggpredict(rub13, "normKatE")
# gge <- ggeffects::ggpredict(KatE_enviro4, "Coast_km")
# gge <- ggeffects::ggpredict(model, "latitude")
# gge <- ggeffects::ggpredict(model, "latitude")
# KatG13
# gge <- ggeffects::ggpredict(model, "latitude")
# gge <- ggeffects::ggpredict(CatMn7, "PAR")
# gge <- ggeffects::ggpredict(KatG_enviro6, "Iron_5m")
# 
# 
# 
# 
# 
# a <- ggplot(gge, aes(x, predicted)) +
#   geom_ribbon(aes(ymin = conf.low, ymax = conf.high), 
#               color = NA, alpha = 0.2) +
#   geom_line() + geom_point(aes(x=Iron_5m, y=normKatG, color=Region), data = catmodels_notArctic_clean_SRF)
# 
# 
# 
# ggplot(newdata, aes(Petal.Length)) + 
#   facet_wrap(~Sepal.Width) + 
#   geom_line(aes(y = y)) + 
#   geom_point(aes(y = dots))
# 
# newdata$y <- predict(m, newdata = catmodels_notArctic_clean_SRF_katE)
# newdata$dots <- newdata$y + resid(m)
# 
# 
# library(ggplot2)
# ggplot(newdata, aes(Petal.Length)) + 
#   facet_wrap(~Sepal.Width) + 
#   geom_line(aes(y = y)) + 
#   geom_point(aes(y = dots))
# 
# ggplot(catmodels_notArctic_clean_SRF_katE, aes(x=normKatE, y=normrub, color=Region)) + geom_point() + plot(ggeffects::predict_response(rub13, terms = c("normKatE")) + theme_classic() + ggtitle("") + theme(text=element_text(size=24)) + xlab("rubisco copy number per genome") + ylab("rubisco copy number per genome"))
# 
# rub_katG <- plot(ggeffects::ggpredict(rub13, terms = "normKatG"), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12),  panel.background = element_blank(),
# panel.grid.major = element_blank(), 
#         panel.grid.minor = element_blank(),
#         panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("katG copy number per genome") + ylab("rubisco\ncopy number per genome")
# 
# rub_katE <- plot(ggeffects::ggpredict(rub13, terms = "normKatE"), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12),  panel.background = element_blank(),
# panel.grid.major = element_blank(), 
#         panel.grid.minor = element_blank(),
#         panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("katE copy number per genome") + ylab("rubisco\ncopy number per genome")
# 
# rub1_alldepths <- mgcv::gam(normrub ~ s(normCatMn) + s(normKatG) + s(normKatE) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_clean_notArctic, method = "REML")
# 
# catmodels_clean_notArctic <- catmodels_clean_notArctic %>%
#   arrange(desc(normKatE))
# 
# catmodels_notArctic_clean_katE <- catmodels_clean_notArctic[-c(1),]
# 
# 
# rub2_alldepths <- mgcv::gam(normrub ~ s(normKatG) + s(normKatE) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_katE, method = "REML")
# 
# rub3_alldepths <- mgcv::gam(normrub ~ s(normKatG) + s(normKatE) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_clean_notArctic, method = "REML")
# 
# rub4_alldepths <- mgcv::gam(normrub ~ s(normKatG) + s(normKatE) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(PO4) + s(Iron_5m), data=catmodels_clean_notArctic, method = "REML")
# 
# rub4_alldepths <- mgcv::gam(normrub ~ s(normKatG) + s(normKatE) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_katE, method = "REML")
# 
# rub5_alldepths <- mgcv::gam(normrub ~ normKatG + s(normKatE) + s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_katE, method = "REML")
# 
# rub6_alldepths <- mgcv::gam(normrub ~ normKatG + s(normKatE) + Temperature + s(Salinity) + s(Coast_km) + s(PAR) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_katE, method = "REML")
# 
# rub7_alldepths <- mgcv::gam(normrub ~ normKatG + s(normKatE) + Temperature + Salinity + s(Coast_km) + s(PAR) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_katE, method = "REML")
# 
# rub8_alldepths <- mgcv::gam(normrub ~ normKatG + s(normKatE) + Temperature + Salinity + s(Coast_km) + s(PAR) + s(PO4) + s(Iron_5m, k = 10), data=catmodels_notArctic_clean_katE, method = "REML")
# 
# catmodels_notArctic_clean_katE <- catmodels_notArctic_clean_katE %>%
#   arrange((normKatG))
# 
# catmodels_notArctic_clean_katE_katG <- catmodels_notArctic_clean_katE[-c(1),]
# 
# rub8_alldepths <- mgcv::gam(normrub ~ normKatG + s(normKatE) + Temperature + Salinity + s(Coast_km) + s(PAR) + s(PO4) + s(Iron_5m, k = 10), data=catmodels_notArctic_clean_katE_katG, method = "REML")
# 
# #environmental predictors
# KatG_enviro1 <- mgcv::gam(normKatG ~ s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# # catmodels_notArctic_clean_SRF <- catmodels_notArctic_clean_SRF %>%
# #   arrange((Salinity))
# # 
# # catmodels_notArctic_clean_SRF_Sal  <- catmodels_notArctic_clean_SRF[-c(1),]
# 
# KatG_enviro2 <- mgcv::gam(normKatG ~ s(Temperature) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# KatG_enviro3 <- mgcv::gam(normKatG ~ s(Temperature) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# KatG_enviro4 <- mgcv::gam(normKatG ~ s(Temperature) + s(PAR) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# KatG_enviro5 <- mgcv::gam(normKatG ~ s(Temperature) + s(PAR) + s(Iron_5m), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# KatG_enviro6 <- mgcv::gam(normKatG ~ s(Temperature) + s(PAR) + Iron_5m, data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# plot(ggeffects::ggpredict(KatG_enviro6, terms = "Iron_5m"), show_data = TRUE, colors = "blue", dot_size = 4, line_size = 2) + theme_classic() + ggtitle("") + theme(text=element_text(size=24)) + xlab("Iron concentration (µg/L) at 5 meters") + ylab("katG copy number per genome")
# 
# #KatE
# KatE_enviro1 <- mgcv::gam(normKatE ~ s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# #definitely need to remove this katE outlier before running the model selection method.
# #log transformed based off of the residuals
# catmodels_notArctic_clean_SRF <- catmodels_notArctic_clean_SRF %>%
#   arrange(desc(normKatE))
# 
# catmodels_notArctic_clean_SRF_katE  <- catmodels_notArctic_clean_SRF[-c(1),]
# 
# KatE_enviro1 <- mgcv::gam(log10(normKatE) ~ s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF_katE, method = "REML")
# 
# KatE_enviro2 <- mgcv::gam(log10(normKatE) ~ s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF_katE, method = "REML")
# 
# KatE_enviro3 <- mgcv::gam(log10(normKatE) ~ s(Salinity) + s(Coast_km) + s(PAR) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF_katE, method = "REML")
# 
# KatE_enviro4 <- mgcv::gam(log10(normKatE) ~ s(Salinity) + s(Coast_km) + s(PAR) + s(Iron_5m), data=catmodels_notArctic_clean_SRF_katE, method = "REML")
# 
# KatE_enviro5 <- mgcv::gam(log10(normKatE) ~ s(Salinity) + s(Coast_km) + s(PAR), data=catmodels_notArctic_clean_SRF_katE, method = "REML")
# 
# KatE_enviro6 <- mgcv::gam(log10(normKatE) ~ s(Salinity) + Coast_km + s(PAR), data=catmodels_notArctic_clean_SRF_katE, method = "REML")
# 
# KatE_enviro7 <- mgcv::gam(log10(normKatE) ~ s(Salinity) + s(Coast_km) + s(PAR), data=catmodels_notArctic_clean_SRF_katE, method = "REML")
# 
# 
# ##fitting a model without log transformation
# KatE_enviro1 <- mgcv::gam((normKatE) ~ s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF_katE, method = "REML")
# 
# KatE_enviro2 <- mgcv::gam((normKatE) ~ s(Temperature) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF_katE, method = "REML")
# 
# KatE_enviro3 <- mgcv::gam(normKatE ~ s(Temperature) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(Iron_5m), data=catmodels_notArctic_clean_SRF_katE, method = "REML")
# 
# KatE_enviro4 <- mgcv::gam(normKatE ~ s(Temperature) + s(Coast_km) + s(PAR) + s(NO3_NO2), data=catmodels_notArctic_clean_SRF_katE, method = "REML")
# 
# 
# ##catmn
# catmn_enviro1 <- mgcv::gam(normCatMn ~ s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# catmodels_notArctic_clean_SRF <- catmodels_notArctic_clean_SRF %>%
#   arrange(desc(normCatMn))
# 
# catmodels_notArctic_clean_SRF_CatMn  <- catmodels_notArctic_clean_SRF[-c(1:2),]
# 
# catmn_enviro1 <- mgcv::gam(normCatMn ~ s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF_CatMn, method = "REML")
# 
# catmn_enviro2 <- mgcv::gam(normCatMn ~ s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF_CatMn, method = "REML")
# 
# catmn_enviro3 <- mgcv::gam(normCatMn ~ s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF_CatMn, method = "REML")
# 
# catmn_enviro4 <- mgcv::gam(normCatMn ~ s(Coast_km) + s(PAR) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF_CatMn, method = "REML")
# 
# catmn_enviro5 <- mgcv::gam(normCatMn ~ s(Coast_km) + s(PAR) + s(Iron_5m), data=catmodels_notArctic_clean_SRF_CatMn, method = "REML")
# 
# catmn_enviro6 <- mgcv::gam(normCatMn ~ s(Coast_km) + s(PAR), data=catmodels_notArctic_clean_SRF_CatMn, method = "REML")
# 
# catmn_enviro7 <- mgcv::gam(normCatMn ~ s(PAR), data=catmodels_notArctic_clean_SRF_CatMn, method = "REML")
# 
# catmn_enviro8 <- mgcv::gam(normCatMn ~ s(PAR), data=catmodels_notArctic_clean_SRF_CatMn, method = "REML")
# 
# ##rub
# rub_enviro1 <- mgcv::gam(normrub ~ s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(NO3_NO2) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# rub_enviro2 <- mgcv::gam(normrub ~ s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# rub_enviro3 <- mgcv::gam(normrub ~ s(Temperature) + s(Salinity) + s(Coast_km) + s(PAR) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# rub_enviro4 <- mgcv::gam(normrub ~ s(Temperature) + Salinity + s(Coast_km) + s(PAR) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# rub_enviro5 <- mgcv::gam(normrub ~ s(Temperature) + Salinity + Coast_km + s(PAR) + s(PO4) + s(Iron_5m), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# rub_enviro6 <- mgcv::gam(normrub ~ s(Temperature) + Salinity + Coast_km + s(PAR) + PO4 + s(Iron_5m), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# rub_enviro7 <- mgcv::gam(normrub ~ s(Temperature) + Salinity + Coast_km + s(PAR) + PO4 + s(Iron_5m), data=catmodels_notArctic_clean_SRF, method = "REML")
# 
# catmodels_notArctic_clean_SRF <- catmodels_notArctic_clean_SRF %>%
#   arrange((Salinity))
# 
# catmodels_notArctic_clean_SRF_sal  <- catmodels_notArctic_clean_SRF[-c(1),]
# 
# rub_enviro7 <- mgcv::gam(normrub ~ s(Temperature) + Salinity + Coast_km + s(PAR) + PO4 + s(Iron_5m), data=catmodels_notArctic_clean_SRF_sal, method = "REML")

#choosing plots to arrange into a figure
katG_Iron <- plot(ggeffects::ggpredict(KatG_enviro6, terms = "Iron_5m"), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("Iron concentration (µg/L) at 5 meters") + ylab("katG\ncopy number per genome")
katG_temp <- plot(ggeffects::ggpredict(KatG_enviro6, terms = "Temperature"), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("Temperature (°C)") + ylab("katG\ncopy number per genome")
katE_dcoast <- plot(ggeffects::ggpredict(KatE_enviro7, terms = "Coast_km"), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("Distance from the coast (km)") + ylab("katE\ncopy number per genome")
katE_PAR <- plot(ggeffects::ggpredict(KatE_enviro7, terms = "PAR"), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("PAR (mol quanta/m2 per day)") + ylab("katE\ncopy number per genome")
catmn_PAR <- plot(ggeffects::ggpredict(catmn_enviro8, terms = "PAR"), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("PAR (mol quanta/m2 per day)") + ylab("manganese catalase\ncopy number per genome")

library(gridExtra)
grid.arrange(katG_Iron, katG_temp, katE_dcoast, katE_PAR, catmn_PAR, ncol=2)

KatG_psbO <- plot(ggeffects::ggpredict(KatG_psb11, terms = "normpsbO"), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("psbO copy number per genome") + ylab("katG\ncopy number per genome")


#modeling with model
library(tidyverse)
library(mgcv)
library(readxl)

catmodels_norm <- read.csv("~/Desktop/Dyhrman Lab/Desktop_DyhrmanLab/catmodels_norm.csv")

modeled_TARAmetadata <- read_excel("modeled_TARAmetadata.xlsx") %>%
  filter(type == "metagenome") %>%
  rename("sample_ID" = "Sampleme")

catmodels_modeleddata <- left_join(catmodels_norm,
                                   modeled_TARAmetadata,
                                   by = "sample_ID") %>%
  distinct(sample_ID, .keep_all = TRUE) 

catmodels_modeleddata_clean <- catmodels_modeleddata[-c(34:41)]

#doing some coalescing
catmodels_modeleddata_cleaner <- catmodels_modeleddata_clean %>%
mutate(CO2_co = coalesce(CO2, CO2..umol_kg.1.),
       NO3_NO2_co = coalesce(NO3_NO2, NO3_NO2_modeled),
       PO4_co = coalesce(PO4, phosphate_dissolved_umol.kg),
       Temp_co = coalesce(Temperature, temperature_dissolved_deg.c),
       Sal_co = coalesce(Salinity, salinity_dissolved_PSS.1978)) %>%
  select(sample_ID, station, latitude.x, longitude.x, depth.x, normKatG, normrub, normKatE, normCatMn, normpsbO, normlight,
         Region, Iron_5m, POironDarwin_dissolved_nmol.kg, PAR, POC, synDarwin_umolC.kg, proDarwin_umolC.kg, synDarwin_umolC.kg, pro_HLI, pro_HLII,
         sar11, NO3_NO2_modeled, NO3_NO2_co, PO4_co, Temp_co, Sal_co)

#newmodels
catmodels_modeleddata_Arctic <- catmodels_modeleddata_cleaner %>%
  filter(latitude.x >= 50)

catmodels_modeleddata_notArctic_SRF <- catmodels_modeleddata_cleaner %>%
  filter(latitude.x < 50, depth.x == "SRF", Region != "[SO] Southern Ocean (MRGID:1907)") 

#models separated by latitude,


#removing CO2
catmodels_notArctic_new <- catmodels_modeleddata_notArctic_SRF %>%
  select(-proDarwin_umolC.kg, -synDarwin_umolC.kg, -pro_HLI, -pro_HLII, -sar11, -NO3_NO2_modeled)

catmodels_notArctic_extra <- catmodels_modeleddata_notArctic_SRF %>%
  select(-NO3_NO2_modeled)

#dropping NA for arctic an not arctic
#Arctic
catmodels_notArctic_new_clean_SRF <- catmodels_notArctic_new %>%
  drop_na()

catmodels_notArctic_new_clean_SRF_extra <- catmodels_notArctic_extra %>%
  drop_na()

write.csv(catmodels_notArctic_new_clean_SRF_extra, "catmodels_notArctic_new_clean_SRF_extra.csv")

catmodels_notArctic_new_clean_SRF_extra <- read.csv("~/Desktop/Dyhrman Lab/Desktop_DyhrmanLab/catmodels_notArctic_new_clean_SRF_extra.csv")

#modeling each gene
#modeling again

#model gene with POiron
# normKatG1_POiron <- mgcv::gam(log10(normKatG) ~ s(Temp_co) + s(Sal_co) + s(PAR) + s(NO3_NO2_co) + s(PO4_co) + s(Iron_5m) + s(POC) + s(POironDarwin_dissolved_nmol.kg), data=catmodels_notArctic_new_clean_SRF_extra, method = "REML")
# 
# normKatG2_POiron <- mgcv::gam(log10(normKatG) ~ s(Temp_co) + s(Sal_co) + s(NO3_NO2_co) + s(PO4_co) + s(Iron_5m) + s(POC) + s(POironDarwin_dissolved_nmol.kg), data=catmodels_notArctic_new_clean_SRF_extra, method = "REML")
# 
# normKatG3_POiron <- mgcv::gam(log10(normKatG) ~ s(Temp_co) + s(Sal_co) + s(PO4_co) + s(Iron_5m) + s(POC) + s(POironDarwin_dissolved_nmol.kg), data=catmodels_notArctic_new_clean_SRF_extra, method = "REML")
# 
# normKatG4_POiron <- mgcv::gam(log10(normKatG) ~ s(Temp_co) + s(Sal_co) + s(PO4_co) + s(POC) + s(POironDarwin_dissolved_nmol.kg), data=catmodels_notArctic_new_clean_SRF_extra, method = "REML")
# 
# normKatG5_POiron <- mgcv::gam(log10(normKatG) ~ s(Temp_co) + s(Sal_co) + s(PO4_co) + s(POC) + s(POironDarwin_dissolved_nmol.kg), data=catmodels_notArctic_new_clean_SRF_extra, method = "REML")

#removing outliers
catmodels_notArctic_new_clean_SRF_extra_Sal <- catmodels_notArctic_new_clean_SRF_extra %>%
arrange((Sal_co))
#
catmodels_notArctic_new_clean_SRF_extra_Sal <- catmodels_notArctic_new_clean_SRF_extra_Sal[-c(1),]
#

catmodels_notArctic_new_clean_SRF_extra_Sal_POC <- catmodels_notArctic_new_clean_SRF_extra_Sal %>%
  arrange(desc(POC))

catmodels_notArctic_new_clean_SRF_extra_Sal_POC <- catmodels_notArctic_new_clean_SRF_extra_Sal_POC[-c(1:2),]
# 
# catmodels_notArctic_new_clean_SRF_extra_Sal_POC_POFe <- catmodels_notArctic_new_clean_SRF_extra_Sal_POC %>%
#   arrange(desc(POironDarwin_dissolved_nmol.kg))
# 
# catmodels_notArctic_new_clean_SRF_extra_Sal_POC_POFe <- catmodels_notArctic_new_clean_SRF_extra_Sal_POC_POFe[-c(1:2),]
# 
# 
# normKatG5_POiron <- mgcv::gam(log10(normKatG) ~ s(Temp_co) + s(Sal_co) + s(PO4_co) + s(POC) + s(POironDarwin_dissolved_nmol.kg), data=catmodels_notArctic_new_clean_SRF_extra_Sal_POC_POFe, method = "REML")
# 
# normKatG6_POiron <- mgcv::gam(log10(normKatG) ~ s(Temp_co) + s(Sal_co) + PO4_co + s(POC) + s(POironDarwin_dissolved_nmol.kg), data=catmodels_notArctic_new_clean_SRF_extra_Sal_POC_POFe, method = "REML")
# 
# normKatG7_POiron <- mgcv::gam(log10(normKatG) ~ s(Temp_co) + s(Sal_co) + PO4_co + s(POC) + s(POironDarwin_dissolved_nmol.kg), data=catmodels_notArctic_new_clean_SRF_extra_Sal_POC_POFe, method = "REML")
# 
# normKatG8_POiron <- mgcv::gam(log10(normKatG) ~ s(Temp_co) + s(Sal_co) + PO4_co + POC + s(POironDarwin_dissolved_nmol.kg), data=catmodels_notArctic_new_clean_SRF_extra_Sal_POC_POFe, method = "REML")
# 
# normKatG9_POiron <- mgcv::gam(log10(normKatG) ~ s(Temp_co) + s(Sal_co) + PO4_co + POC + POironDarwin_dissolved_nmol.kg, data=catmodels_notArctic_new_clean_SRF_extra_Sal_POC_POFe, method = "REML")


#model gene with sar11
normKatG1_sar11 <- mgcv::gam(log10(normKatG) ~ s(Temp_co) + s(Sal_co) + s(PAR) + s(NO3_NO2_co) + s(PO4_co) + s(Iron_5m) + s(POC) + s(sar11), data=catmodels_notArctic_new_clean_SRF_extra, method = "REML")

normKatG2_sar11 <- mgcv::gam(log10(normKatG) ~ s(Temp_co) + s(Sal_co) + s(PAR) + s(NO3_NO2_co) + s(Iron_5m) + s(POC) + s(sar11), data=catmodels_notArctic_new_clean_SRF_extra, method = "REML")

normKatG3_sar11 <- mgcv::gam(log10(normKatG) ~ s(Temp_co) + s(Sal_co) + s(NO3_NO2_co) + s(Iron_5m) + s(POC) + s(sar11), data=catmodels_notArctic_new_clean_SRF_extra, method = "REML")

normKatG4_sar11 <- mgcv::gam(log10(normKatG) ~ s(Temp_co) + s(Sal_co) + s(Iron_5m) + s(POC) + s(sar11), data=catmodels_notArctic_new_clean_SRF_extra, method = "REML")

normKatG5_sar11 <- mgcv::gam(log10(normKatG) ~ s(Temp_co) + s(Sal_co) + s(Iron_5m) + s(POC) + s(sar11), data=catmodels_notArctic_new_clean_SRF_extra_Sal_POC, method = "REML")

normKatG6_sar11 <- mgcv::gam(log10(normKatG) ~ Temp_co + s(Sal_co) + s(Iron_5m) + s(POC) + s(sar11), data=catmodels_notArctic_new_clean_SRF_extra_Sal_POC, method = "REML")

normKatG7_sar11 <- mgcv::gam(log10(normKatG) ~ Temp_co + s(Sal_co) + s(Iron_5m) + s(POC) + s(sar11), data=catmodels_notArctic_new_clean_SRF_extra_Sal_POC, method = "REML")

#######
#model gene with pro
normKatG1_pro <- mgcv::gam(log10(normKatG) ~ s(Temp_co) + s(Sal_co) + s(PAR) + s(NO3_NO2_co) + s(PO4_co) + s(Iron_5m) + s(POC) + s(proDarwin_umolC.kg), data=catmodels_notArctic_new_clean_SRF_extra, method = "REML")

normKatG2_pro <- mgcv::gam(log10(normKatG) ~ s(Temp_co) + s(Sal_co) + s(NO3_NO2_co) + s(PO4_co) + s(Iron_5m) + s(POC) + s(proDarwin_umolC.kg), data=catmodels_notArctic_new_clean_SRF_extra, method = "REML")

normKatG3_pro <- mgcv::gam(log10(normKatG) ~ s(Temp_co) + s(Sal_co) + s(PO4_co) + s(Iron_5m) + s(POC) + s(proDarwin_umolC.kg), data=catmodels_notArctic_new_clean_SRF_extra, method = "REML")

normKatG4_pro <- mgcv::gam(log10(normKatG) ~ s(Temp_co) + s(Sal_co) + s(PO4_co) + s(POC) + s(proDarwin_umolC.kg), data=catmodels_notArctic_new_clean_SRF_extra, method = "REML")

#plots

plot(ggeffects::ggpredict(normKatG4_pro, terms = "POC"), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("POC") + ylab("katG\ncopy number per genome")


#remove two POC


#

catmodels_notArctic_new_clean_SRF_extra_POC <- catmodels_notArctic_new_clean_SRF_extra %>%
  arrange(desc(POC))

catmodels_notArctic_new_clean_SRF_extra_POC <- catmodels_notArctic_new_clean_SRF_extra_POC[-c(1:2),]

##########
catmodels_notArctic_new_clean_SRF_extra_POC_Sal <- catmodels_notArctic_new_clean_SRF_extra_POC %>%
arrange((Sal_co))
#
catmodels_notArctic_new_clean_SRF_extra_POC_Sal <- catmodels_notArctic_new_clean_SRF_extra_POC_Sal[-c(1),]

###########
normKatG5_pro <- mgcv::gam(log10(normKatG) ~ s(Temp_co) + s(Sal_co) + s(PO4_co) + s(Iron_5m) + s(POC) + s(proDarwin_umolC.kg), data=catmodels_notArctic_new_clean_SRF_extra_POC, method = "REML")






plot(ggeffects::ggpredict(normKatG5_pro, terms = "PO4_co"), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("Sal") + ylab("katG\ncopy number per genome")


normKatG6_pro <- mgcv::gam(log10(normKatG) ~ s(Temp_co) + s(Sal_co) + PO4_co + s(Iron_5m) + s(POC) + s(proDarwin_umolC.kg), data=catmodels_notArctic_new_clean_SRF_extra_POC_Sal, method = "REML")

plot(ggeffects::ggpredict(normKatG6_pro, terms = "proDarwin_umolC.kg"), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("Sal") + ylab("katG\ncopy number per genome")



normKatG7_pro <- mgcv::gam(log10(normKatG) ~ s(Temp_co) + s(Sal_co) + PO4_co + s(Iron_5m) + POC + s(proDarwin_umolC.kg), data=catmodels_notArctic_new_clean_SRF_extra_Sal_POC, method = "REML")

normKatG8_pro <- mgcv::gam(log10(normKatG) ~ s(Temp_co) + s(Sal_co) + PO4_co + s(Iron_5m) + POC + proDarwin_umolC.kg, data=catmodels_notArctic_new_clean_SRF_extra_Sal_POC, method = "REML")

#model gene with syn
normKatG1_syn <- mgcv::gam(log10(normKatG) ~ s(Temp_co) + s(Sal_co) + s(PAR) + s(NO3_NO2_co) + s(PO4_co) + s(Iron_5m) + s(POC) + s(synDarwin_umolC.kg), data=catmodels_notArctic_new_clean_SRF_extra, method = "REML")

normKatG2_syn <- mgcv::gam(log10(normKatG) ~ s(Temp_co) + s(Sal_co) + s(NO3_NO2_co) + s(PO4_co) + s(Iron_5m) + s(POC) + s(synDarwin_umolC.kg), data=catmodels_notArctic_new_clean_SRF_extra, method = "REML")

normKatG3_syn <- mgcv::gam(log10(normKatG) ~ s(Temp_co) + s(Sal_co) + s(PO4_co) + s(Iron_5m) + s(POC) + s(synDarwin_umolC.kg), data=catmodels_notArctic_new_clean_SRF_extra, method = "REML")

normKatG5_syn <- mgcv::gam(log10(normKatG) ~ s(Temp_co) + s(Sal_co) + s(PO4_co) + s(POC) + s(synDarwin_umolC.kg), data=catmodels_notArctic_new_clean_SRF_extra, method = "REML")

normKatG6_syn <-  mgcv::gam(log10(normKatG) ~ s(Temp_co) + s(Sal_co) + s(PO4_co) + s(POC) + s(synDarwin_umolC.kg), data=catmodels_notArctic_new_clean_SRF_extra_Sal_POC, method = "REML")

normKatG7_syn <-  mgcv::gam(log10(normKatG) ~ s(Temp_co) + s(Sal_co) + s(PO4_co) + POC + s(synDarwin_umolC.kg), data=catmodels_notArctic_new_clean_SRF_extra_Sal_POC, method = "REML")

normKatG8_syn <-  mgcv::gam(log10(normKatG) ~ s(Temp_co) + s(Sal_co) + s(PO4_co) + POC + synDarwin_umolC.kg, data=catmodels_notArctic_new_clean_SRF_extra_Sal_POC, method = "REML")


#model with HLI and HLII

normKatG1_HL <- mgcv::gam(log10(normKatG) ~ s(Temp_co) + s(Sal_co) + s(PAR) + s(NO3_NO2_co) + s(PO4_co) + s(Iron_5m) + s(POC) + s(pro_HLII) + s(pro_HLI) + s(proDarwin_umolC.kg), data=catmodels_notArctic_new_clean_SRF_extra, method = "REML")

normKatG2_HL <- mgcv::gam(log10(normKatG) ~ s(Temp_co) + s(Sal_co) + s(NO3_NO2_co) + s(PO4_co) + s(Iron_5m) + s(POC) + s(pro_HLII) + s(pro_HLI) + s(proDarwin_umolC.kg), data=catmodels_notArctic_new_clean_SRF_extra, method = "REML")

normKatG3_HL <- mgcv::gam(log10(normKatG) ~ s(Temp_co) + s(Sal_co) + s(PO4_co) + s(Iron_5m) + s(POC) + s(pro_HLII) + s(pro_HLI) + s(proDarwin_umolC.kg), data=catmodels_notArctic_new_clean_SRF_extra, method = "REML")

normKatG4_HL <- mgcv::gam(log10(normKatG) ~ s(Temp_co) + s(Sal_co) + s(PO4_co) + s(POC) + s(pro_HLII) + s(pro_HLI) + s(proDarwin_umolC.kg), data=catmodels_notArctic_new_clean_SRF_extra, method = "REML")

normKatG5_HL <- mgcv::gam(log10(normKatG) ~ s(Temp_co) + s(Sal_co) + s(PO4_co) + s(POC) + s(pro_HLI) + s(proDarwin_umolC.kg), data=catmodels_notArctic_new_clean_SRF_extra, method = "REML")

normKatG6_HL <- mgcv::gam(log10(normKatG) ~ s(Temp_co) + s(Sal_co) + s(PO4_co) + s(POC) + s(pro_HLI), data=catmodels_notArctic_new_clean_SRF_extra, method = "REML")

normKatG7_HL <- mgcv::gam(log10(normKatG) ~ s(Temp_co) + s(Sal_co) + s(PO4_co) + s(POC) + s(pro_HLI) + s(proDarwin_umolC.kg), data=catmodels_notArctic_new_clean_SRF_extra_Sal_POC, method = "REML")

normKatG8_HL <- mgcv::gam(log10(normKatG) ~ s(Temp_co) + s(Sal_co) + s(PO4_co) + s(POC) + pro_HLI + s(proDarwin_umolC.kg), data=catmodels_notArctic_new_clean_SRF_extra_Sal_POC, method = "REML")

normKatG9_HL <- mgcv::gam(log10(normKatG) ~ s(Temp_co) + s(Sal_co) + PO4_co + s(POC) + pro_HLI + s(proDarwin_umolC.kg), data=catmodels_notArctic_new_clean_SRF_extra_Sal_POC, method = "REML")

normKatG10_HL <- mgcv::gam(log10(normKatG) ~ s(Temp_co) + s(Sal_co) + PO4_co + POC + pro_HLI + s(proDarwin_umolC.kg), data=catmodels_notArctic_new_clean_SRF_extra_Sal_POC, method = "REML")

#THIS IS THE RIGHT MODEL!!!!!!!!!!!!!!!! 
normKatG11_HL <- mgcv::gam(log10(normKatG) ~ s(Temp_co) + s(Sal_co) + PO4_co + POC + pro_HLI + proDarwin_umolC.kg, data=catmodels_notArctic_new_clean_SRF_extra_Sal_POC, method = "REML")

#repeating analysis
#######
normKatE1_POiron <- mgcv::gam(log10(normKatE) ~ s(Temp_co) + s(Sal_co) + s(PAR) + s(NO3_NO2_co) + s(PO4_co) + s(Iron_5m) + s(POC) + s(POironDarwin_dissolved_nmol.kg), data=catmodels_notArctic_new_clean_SRF_extra, method = "REML")

#should remove the katE outlier first before fitting
catmodels_notArctic_new_clean_SRF_extra_katE <- catmodels_notArctic_new_clean_SRF_extra %>%
  arrange(desc(normKatE))

catmodels_notArctic_new_clean_SRF_extra_katE <- catmodels_notArctic_new_clean_SRF_extra_katE[-c(1),]

# normKatE1_POiron <- mgcv::gam(log10(normKatE) ~ s(Temp_co) + s(Sal_co) + s(PAR) + s(NO3_NO2_co) + s(PO4_co) + s(Iron_5m) + s(POC) + s(POironDarwin_dissolved_nmol.kg), data=catmodels_notArctic_new_clean_SRF_extra_katE, method = "REML")
# 
# normKatE2_POiron <- mgcv::gam(log10(normKatE) ~  s(Sal_co) + s(PAR) + s(NO3_NO2_co) + s(PO4_co) + s(Iron_5m) + s(POC) + s(POironDarwin_dissolved_nmol.kg), data=catmodels_notArctic_new_clean_SRF_extra_katE, method = "REML")
# 
# normKatE3_POiron <- mgcv::gam(log10(normKatE) ~  s(Sal_co) + s(PAR) + s(NO3_NO2_co) + s(PO4_co) + s(POC) + s(POironDarwin_dissolved_nmol.kg), data=catmodels_notArctic_new_clean_SRF_extra_katE, method = "REML")
# 
# normKatE4_POiron <- mgcv::gam(log10(normKatE) ~  s(Sal_co) + s(PAR) + s(NO3_NO2_co) + s(PO4_co) + s(POC) + s(POironDarwin_dissolved_nmol.kg), data=catmodels_notArctic_new_clean_SRF_extra_katE, method = "REML")

catmodels_notArctic_new_clean_SRF_extra_katE_POC <- catmodels_notArctic_new_clean_SRF_extra_katE %>%
  arrange(desc(POC))

catmodels_notArctic_new_clean_SRF_extra_katE_POC <- catmodels_notArctic_new_clean_SRF_extra_katE_POC[-c(1:2),]

catmodels_notArctic_new_clean_SRF_extra_katE_POC_Sal <- catmodels_notArctic_new_clean_SRF_extra_katE_POC %>%
  arrange(Sal_co)

catmodels_notArctic_new_clean_SRF_extra_katE_POC_Sal <- catmodels_notArctic_new_clean_SRF_extra_katE_POC_Sal[-c(1),]

# normKatE4_POiron <- mgcv::gam(log10(normKatE) ~  s(Sal_co) + s(PAR) + s(NO3_NO2_co) + s(PO4_co) + s(POC) + s(POironDarwin_dissolved_nmol.kg), data=catmodels_notArctic_new_clean_SRF_extra_katE_POC_Sal, method = "REML")
# 
# normKatE5_POiron <- mgcv::gam(log10(normKatE) ~  s(Sal_co) + PAR + s(NO3_NO2_co) + s(PO4_co) + s(POC) + s(POironDarwin_dissolved_nmol.kg), data=catmodels_notArctic_new_clean_SRF_extra_katE_POC_Sal, method = "REML")
# 
# normKatE6_POiron <- mgcv::gam(log10(normKatE) ~  s(Sal_co) + PAR + NO3_NO2_co + s(PO4_co) + s(POC) + s(POironDarwin_dissolved_nmol.kg), data=catmodels_notArctic_new_clean_SRF_extra_katE_POC_Sal, method = "REML")
# 
# normKatE7_POiron <- mgcv::gam(log10(normKatE) ~  s(Sal_co) + PAR + NO3_NO2_co + PO4_co + s(POC) + s(POironDarwin_dissolved_nmol.kg), data=catmodels_notArctic_new_clean_SRF_extra_katE_POC_Sal, method = "REML")
# 
# normKatE8_POiron <- mgcv::gam(log10(normKatE) ~  s(Sal_co) + PAR + NO3_NO2_co + PO4_co + s(POC) + POironDarwin_dissolved_nmol.kg, data=catmodels_notArctic_new_clean_SRF_extra_katE_POC_Sal, method = "REML")

# katG_large_Fe <- plot(ggeffects::ggpredict(normKatG6_large_enviro, terms = "Iron_5m"), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
#                                                                          
# panel.grid.major = element_blank(), 
#         panel.grid.minor = element_blank(),
#         panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("Iron concentration at 5 meters") + ylab("katG\ncopy number per genome")
# 
# katG_large_POC <- plot(ggeffects::ggpredict(normKatG7_large_enviro, terms = c("POC [20,180]", "Size_Fraction_Upper_Threshold"), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1)) 
# 
# katG_large_TSS <- plot(ggeffects::ggpredict(normKatG6_large_enviro, terms = "TSS"), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
# panel.grid.major = element_blank(), 
#         panel.grid.minor = element_blank(),
#         panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("Total suspended matter") + ylab("katG\ncopy number per genome")
# 


#########################################repeat for syn
normKatE1_sar11 <- mgcv::gam(log10(normKatE) ~ s(Temp_co) + s(Sal_co) + s(PAR) + s(NO3_NO2_co) + s(PO4_co) + s(Iron_5m) + s(POC) + s(sar11), data=catmodels_notArctic_new_clean_SRF_extra_katE, method = "REML")

normKatE2_sar11 <- mgcv::gam(log10(normKatE) ~ s(Temp_co) + s(Sal_co) + s(NO3_NO2_co) + s(PO4_co) + s(Iron_5m) + s(POC) + s(sar11), data=catmodels_notArctic_new_clean_SRF_extra_katE, method = "REML")

normKatE3_sar11 <- mgcv::gam(log10(normKatE) ~ s(Temp_co) + s(Sal_co) + s(NO3_NO2_co) + s(Iron_5m) + s(POC) + s(sar11), data=catmodels_notArctic_new_clean_SRF_extra_katE, method = "REML")

normKatE4_sar11 <- mgcv::gam(log10(normKatE) ~ s(Temp_co) + s(Sal_co) + s(NO3_NO2_co) + s(Iron_5m) + s(POC) + s(sar11), data=catmodels_notArctic_new_clean_SRF_extra_katE, method = "REML")

normKatE5_sar11 <- mgcv::gam(log10(normKatE) ~ s(Temp_co) + s(Sal_co) + s(NO3_NO2_co) + s(Iron_5m) + s(POC) + s(sar11), data=catmodels_notArctic_new_clean_SRF_extra_katE_POC_Sal, method = "REML")

normKatE5_sar11 <- mgcv::gam(log10(normKatE) ~ s(Temp_co) + s(Sal_co) + s(NO3_NO2_co) + s(Iron_5m) + s(POC) + s(sar11), data=catmodels_notArctic_new_clean_SRF_extra_katE_POC_Sal, method = "REML")

normKatE6_sar11 <- mgcv::gam(log10(normKatE) ~ Temp_co + s(Sal_co) + s(NO3_NO2_co) + s(Iron_5m) + s(POC) + s(sar11), data=catmodels_notArctic_new_clean_SRF_extra_katE_POC_Sal, method = "REML")

normKatE7_sar11 <- mgcv::gam(log10(normKatE) ~ Temp_co + s(Sal_co) + s(NO3_NO2_co) + s(Iron_5m) + POC + s(sar11), data=catmodels_notArctic_new_clean_SRF_extra_katE_POC_Sal, method = "REML")

normKatE8_sar11 <- mgcv::gam(log10(normKatE) ~ Temp_co + s(Sal_co) + s(NO3_NO2_co) + s(Iron_5m) + POC + s(sar11), data=catmodels_notArctic_new_clean_SRF_extra_katE_POC_Sal, method = "REML")

######################################repeat for pro stuff
normKatE1_HL <- mgcv::gam(log10(normKatE) ~ s(Temp_co) + s(Sal_co) + s(PAR) + s(NO3_NO2_co) + s(PO4_co) + s(Iron_5m) + s(POC) + s(pro_HLII) + s(pro_HLI) + s(proDarwin_umolC.kg), data=catmodels_notArctic_new_clean_SRF_extra_katE, method = "REML")

normKatE2_HL <- mgcv::gam(log10(normKatE) ~ s(Sal_co) + s(PAR) + s(NO3_NO2_co) + s(PO4_co) + s(Iron_5m) + s(POC) + s(pro_HLII) + s(pro_HLI) + s(proDarwin_umolC.kg), data=catmodels_notArctic_new_clean_SRF_extra_katE, method = "REML")

normKatE3_HL <- mgcv::gam(log10(normKatE) ~ s(Sal_co) + s(PAR) + s(NO3_NO2_co) + s(PO4_co) + s(POC) + s(pro_HLII) + s(pro_HLI) + s(proDarwin_umolC.kg), data=catmodels_notArctic_new_clean_SRF_extra_katE, method = "REML")

normKatE4_HL <- mgcv::gam(log10(normKatE) ~ s(Sal_co) + s(PAR) + s(NO3_NO2_co) + s(PO4_co) + s(POC) + s(pro_HLII) + s(proDarwin_umolC.kg), data=catmodels_notArctic_new_clean_SRF_extra_katE, method = "REML")

normKatE5_HL <- mgcv::gam(log10(normKatE) ~ s(Sal_co) + s(PAR) + s(NO3_NO2_co) + s(PO4_co) + s(POC) + s(pro_HLII) + s(proDarwin_umolC.kg), data=catmodels_notArctic_new_clean_SRF_extra_katE, method = "REML")

normKatE6_HL <- mgcv::gam(log10(normKatE) ~ s(Sal_co) + s(PAR) + s(NO3_NO2_co) + s(PO4_co) + s(POC) + s(pro_HLII) + s(proDarwin_umolC.kg), data=catmodels_notArctic_new_clean_SRF_extra_katE_POC_Sal, method = "REML")

normKatE7_HL <- mgcv::gam(log10(normKatE) ~ s(Sal_co) + PAR + s(NO3_NO2_co) + s(PO4_co) + s(POC) + s(pro_HLII) + s(proDarwin_umolC.kg), data=catmodels_notArctic_new_clean_SRF_extra_katE_POC_Sal, method = "REML")

normKatE8_HL <- mgcv::gam(log10(normKatE) ~ s(Sal_co) + PAR + NO3_NO2_co + s(PO4_co) + s(POC) + s(pro_HLII) + s(proDarwin_umolC.kg), data=catmodels_notArctic_new_clean_SRF_extra_katE_POC_Sal, method = "REML")

normKatE9_HL <- mgcv::gam(log10(normKatE) ~ s(Sal_co) + PAR + NO3_NO2_co + PO4_co + s(POC) + s(pro_HLII) + s(proDarwin_umolC.kg), data=catmodels_notArctic_new_clean_SRF_extra_katE_POC_Sal, method = "REML")

normKatE10_HL <- mgcv::gam(log10(normKatE) ~ s(Sal_co) + PAR + NO3_NO2_co + PO4_co + POC + s(pro_HLII) + s(proDarwin_umolC.kg), data=catmodels_notArctic_new_clean_SRF_extra_katE_POC_Sal, method = "REML")

normKatE11_HL <- mgcv::gam(log10(normKatE) ~ s(Sal_co) + PAR + NO3_NO2_co + PO4_co + POC + s(pro_HLII) + proDarwin_umolC.kg, data=catmodels_notArctic_new_clean_SRF_extra_katE_POC_Sal, method = "REML")

#synecococcus
KatE1_syn <- mgcv::gam(log10(normKatE) ~ s(Temp_co) + s(Sal_co) + s(PAR) + s(NO3_NO2_co) + s(PO4_co) + s(Iron_5m) + s(POC) + s(synDarwin_umolC.kg), data=catmodels_notArctic_new_clean_SRF_extra_katE, method = "REML")

KatE2_syn <- mgcv::gam(log10(normKatE) ~ s(Sal_co) + s(PAR) + s(NO3_NO2_co) + s(PO4_co) + s(Iron_5m) + s(POC) + s(synDarwin_umolC.kg), data=catmodels_notArctic_new_clean_SRF_extra_katE, method = "REML")

KatE3_syn <- mgcv::gam(log10(normKatE) ~ s(Sal_co) + s(NO3_NO2_co) + s(PO4_co) + s(Iron_5m) + s(POC) + s(synDarwin_umolC.kg), data=catmodels_notArctic_new_clean_SRF_extra_katE, method = "REML")

KatE4_syn <- mgcv::gam(log10(normKatE) ~ s(Sal_co) + s(NO3_NO2_co) + s(PO4_co) + s(POC) + s(synDarwin_umolC.kg), data=catmodels_notArctic_new_clean_SRF_extra_katE, method = "REML")

KatE5_syn <- mgcv::gam(log10(normKatE) ~ s(Sal_co) + s(NO3_NO2_co) + s(PO4_co) + s(POC) + s(synDarwin_umolC.kg), data=catmodels_notArctic_new_clean_SRF_extra_katE_POC_Sal, method = "REML")

#environmental parameters only
normKatG1_enviro <- mgcv::gam(log10(normKatG) ~ s(Temp_co) + s(Sal_co) + s(PAR) + s(NO3_NO2_co) + s(PO4_co) + s(Iron_5m) + s(POC), data=catmodels_notArctic_new_clean_SRF_extra, method = "REML")

normKatG2_enviro <- mgcv::gam(log10(normKatG) ~ s(Temp_co) + s(Sal_co) + s(NO3_NO2_co) + s(PO4_co) + s(Iron_5m) + s(POC), data=catmodels_notArctic_new_clean_SRF_extra, method = "REML")

normKatG3_enviro <- mgcv::gam(log10(normKatG) ~ s(Temp_co) + s(Sal_co) + s(PO4_co) + s(Iron_5m) + s(POC), data=catmodels_notArctic_new_clean_SRF_extra, method = "REML")

normKatG4_enviro <- mgcv::gam(log10(normKatG) ~ s(Temp_co) + s(Sal_co) + s(Iron_5m) + s(POC), data=catmodels_notArctic_new_clean_SRF_extra, method = "REML")

normKatG5_enviro <- mgcv::gam(log10(normKatG) ~ s(Temp_co) + s(Sal_co) + s(Iron_5m) + s(POC), data=catmodels_notArctic_new_clean_SRF_extra_POC_Sal, method = "REML")

normKatG6_enviro <- mgcv::gam(log10(normKatG) ~ s(Temp_co) + s(Sal_co) + s(Iron_5m) + POC, data=catmodels_notArctic_new_clean_SRF_extra_Sal_POC, method = "REML")

#####
normKatE1_enviro <- mgcv::gam(log10(normKatE) ~ s(Temp_co) + s(Sal_co) + s(PAR) + s(NO3_NO2_co) + s(PO4_co) + s(Iron_5m) + s(POC), data=catmodels_notArctic_new_clean_SRF_extra_katE, method = "REML")

normKatE2_enviro <- mgcv::gam(log10(normKatE) ~ s(Sal_co) + s(PAR) + s(NO3_NO2_co) + s(PO4_co) + s(Iron_5m) + s(POC), data=catmodels_notArctic_new_clean_SRF_extra_katE, method = "REML")

normKatE3_enviro <- mgcv::gam(log10(normKatE) ~ s(Sal_co) + s(PAR) + s(NO3_NO2_co) + s(PO4_co) + s(POC), data=catmodels_notArctic_new_clean_SRF_extra_katE, method = "REML")

normKatE4_enviro <- mgcv::gam(log10(normKatE) ~ s(Sal_co) + s(PAR) + s(NO3_NO2_co) + s(PO4_co) + s(POC), data=catmodels_notArctic_new_clean_SRF_extra_katE, method = "REML")

normKatE5_enviro <- mgcv::gam(log10(normKatE) ~ s(Sal_co) + s(PAR) + s(NO3_NO2_co) + s(PO4_co) + s(POC), data=catmodels_notArctic_new_clean_SRF_extra_katE_POC_Sal, method = "REML")

normKatE6_enviro <- mgcv::gam(log10(normKatE) ~ s(Sal_co) + PAR + s(NO3_NO2_co) + s(PO4_co) + s(POC), data=catmodels_notArctic_new_clean_SRF_extra_katE_POC_Sal, method = "REML")

normKatE7_enviro <- mgcv::gam(log10(normKatE) ~ s(Sal_co) + PAR + NO3_NO2_co + s(PO4_co) + s(POC), data=catmodels_notArctic_new_clean_SRF_extra_katE_POC_Sal, method = "REML")

normKatE8_enviro <- mgcv::gam(log10(normKatE) ~ s(Sal_co) + PAR + NO3_NO2_co + PO4_co + s(POC), data=catmodels_notArctic_new_clean_SRF_extra_katE_POC_Sal, method = "REML")

normKatE9_enviro <- mgcv::gam(log10(normKatE) ~ s(Sal_co) + PAR + NO3_NO2_co + PO4_co + s(POC), data=catmodels_notArctic_new_clean_SRF_extra_katE_POC_Sal, method = "REML")

#prepare figure for just enviro
katG_POC <- plot(ggeffects::ggpredict(normKatG6_enviro, terms = "POC"), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("POC") + ylab("katG\ncopy number per genome")

katG_Iron <- plot(ggeffects::ggpredict(normKatG6_enviro, terms = "Iron_5m"), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("Iron concentration (µmol/kg) at 5 meters") + ylab("katG\ncopy number per genome")

katG_Sal <- plot(ggeffects::ggpredict(normKatG6_enviro, terms = "Sal_co"), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("Salinity (PSU)") + ylab("katG\ncopy number per genome")

katE_PO4 <- plot(ggeffects::ggpredict(normKatE9_enviro, terms = "PO4_co"), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("Phosphate concentration (µmol/kg)") + ylab("katE\ncopy number per genome")

katE_Sal <- plot(ggeffects::ggpredict(normKatE9_enviro, terms = "Sal_co"), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("Salinity (PSU)") + ylab("katE\ncopy number per genome")


grid.arrange(katG_POC, katG_Iron, ncol=3)

##showing other things
katG_sar11 <- plot(ggeffects::ggpredict(normKatG7_sar11, terms = "sar11"), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("sar11 biomass") + ylab("katG\ncopy number per genome")

katE_sar11 <- plot(ggeffects::ggpredict(normKatE8_sar11, terms = "sar11"), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("sar11 biomass") + ylab("katE\ncopy number per genome")

katG_syn <- plot(ggeffects::ggpredict(normKatG8_syn, terms = "synDarwin_umolC.kg"), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("Synechococcus biomass") + ylab("katG\ncopy number per genome")

katG_pro <- plot(ggeffects::ggpredict(normKatG11_HL, terms = "proDarwin_umolC.kg"), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("Prochlorococcus biomass") + ylab("katG\ncopy number per genome")

katG_HLI <- plot(ggeffects::ggpredict(normKatG11_HL, terms = "pro_HLI"), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("Prochlorococcus HLI") + ylab("katG\ncopy number per genome")

grid.arrange(katG_sar11, katE_sar11, katG_pro, katG_HLI, katG_syn, ncol=2)

#visualizing katE models
library(ggplot2)
library(ggeffects)

normKatE9_enviro <- mgcv::gam(log10(normKatE) ~ s(Sal_co) + PAR + NO3_NO2_co + PO4_co + s(POC), data=catmodels_notArctic_new_clean_SRF_extra_katE_POC_Sal, method = "REML")


gge <- ggeffects::ggpredict(normKatE9_enviro, "PO4_co")
gge <- ggeffects::ggpredict(normKatE9_enviro, "Sal_co")
gge <- ggeffects::ggpredict(model, "latitude")
gge <- ggeffects::ggpredict(model, "latitude")
KatG13
gge <- ggeffects::ggpredict(model, "latitude")
gge <- ggeffects::ggpredict(CatMn7, "PAR")
gge <- ggeffects::ggpredict(normKatG6_enviro, "Iron_5m")
gge <- ggeffects::ggpredict(normKatG6_enviro, "POC")


normKatG6_enviro <- mgcv::gam(log10(normKatG) ~ s(Temp_co) + s(Sal_co) + s(Iron_5m) + POC, data=catmodels_notArctic_new_clean_SRF_extra_Sal_POC, method = "REML")

normKatG8_pro <- mgcv::gam(log10(normKatG) ~ s(Temp_co) + s(Sal_co) + PO4_co + s(Iron_5m) + POC + proDarwin_umolC.kg, data=catmodels_notArctic_new_clean_SRF_extra_Sal_POC, method = "REML")

a <- ggplot(gge, aes(x, predicted)) +
  geom_ribbon(aes(ymin = conf.low, ymax = conf.high),
              color = NA, alpha = 0.2) +
  geom_line() + geom_point(aes(x=POC, y=normKatG, color=Region), data = catmodels_notArctic_new_clean_SRF_extra_POC_Sal)



ggplot(newdata, aes(Petal.Length)) +
  facet_wrap(~Sepal.Width) +
  geom_line(aes(y = y)) +
  geom_point(aes(y = dots))

newdata$y <- predict(m, newdata = catmodels_notArctic_clean_SRF_katE)
newdata$dots <- newdata$y + resid(m)


library(ggplot2)
ggplot(newdata, aes(Petal.Length)) +
  facet_wrap(~Sepal.Width) +
  geom_line(aes(y = y)) +
  geom_point(aes(y = dots))

ggplot(catmodels_notArctic_clean_SRF_katE, aes(x=normKatE, y=normrub, color=Region)) + geom_point() + plot(ggeffects::predict_response(rub13, terms = c("normKatE")) + theme_classic() + ggtitle("") + theme(text=element_text(size=24)) + xlab("rubisco copy number per genome") + ylab("rubisco copy number per genome"))


##models for the larger size fraction
catalases_large_trans_enviro <- read.csv("~/Desktop/Dyhrman Lab/Desktop_DyhrmanLab/catalases_large_trans_enviro.csv")

catalases_large_trans_enviro_filtered <- catalases_large_trans_enviro %>%
  filter(size_fraction_lower_threshold != 180) %>%
  #rotate so that each gene reads can be a column and values in the cells are a number or a 0 for all of the samples
  #Use the TARA IDs so that can merge with the extra darwin modeled dataset
  select(Sample_Name, annotation, reads_trans, Latitude_Start, Longitude_Start,
         Nitrate_Sensor, Oxygen_Sensor, Salinity_Sensor, size_fraction_lower_threshold,
         Size_Fraction_Upper_Threshold, Temperature, Marine_Region) %>%
  mutate(gene_ID = paste(Sample_Name, annotation, sep = "_")) %>%
  distinct(gene_ID, .keep_all = TRUE) %>%
  select(-gene_ID) %>%
  group_by(Sample_Name) %>%
  pivot_wider(names_from = "annotation", values_from = "reads_trans")

modeled_TARAmetadata <- read_excel("modeled_TARAmetadata.xlsx") %>%
  filter(type == "metagenome") %>%
  dplyr::rename("Sample_Name" = "Sampleme")

modeleddata_large <- left_join(catalases_large_trans_enviro_filtered,
                          modeled_TARAmetadata,
                          by = "Sample_Name")

#selecting the same predictors as in the small size fraction

modeleddata_large_clean <- modeleddata_large %>%
  select(Sample_Name, size_fraction_lower_threshold, Size_Fraction_Upper_Threshold,
         Mn_catalase, PTHR30555.orig.30.pir, PTHR42821.SF1.pir, CO2..umol_kg.1.,
         NO3_NO2_modeled, phosphate_dissolved_umol.kg, temperature_dissolved_deg.c,
         salinity_dissolved_PSS.1978, latitude, longitude, Total_carbon..umol_kg.1., synDarwin_umolC.kg, proDarwin_umolC.kg, pro_HLI, pro_HLII, sar11) %>%
  dplyr::rename("catmn" = "Mn_catalase",
         "katE" = "PTHR42821.SF1.pir",
         "katG" = "PTHR30555.orig.30.pir")


TARA_SAMPLES_CONTEXT_ENV_WATERCOLUMN <- read_excel("~/Downloads/Catalase/TARA_SAMPLES_CONTEXT_ENV-WATERCOLUMN.xlsx")

TARA_SAMPLES_CONTEXT_ENV_WATERCOLUMN_tomerge <- TARA_SAMPLES_CONTEXT_ENV_WATERCOLUMN %>%
  dplyr::rename("Sample_Name" = "Sample ID (TARA_barcode#, registered at ...)")

modeleddata_large_clean_otherenviro <- left_join(modeleddata_large_clean,
                                                 TARA_SAMPLES_CONTEXT_ENV_WATERCOLUMN_tomerge,
                                                 by = "Sample_Name")

TARA_SAMPLES_CONTEXT_ENV_WATERCOLUMN_tomerge2 <- TARA_SAMPLES_CONTEXT_ENV_WATERCOLUMN %>%
  dplyr::rename("Sample_Name" = "Sample ID (ENA sample accession number (...)")

modeleddata_large_clean_otherenviro2 <- left_join(modeleddata_large_clean_otherenviro,
                                                 TARA_SAMPLES_CONTEXT_ENV_WATERCOLUMN_tomerge2,
                                                 by = "Sample_Name")


#coalescing
modeleddata_large_clean_otherenviro2_co <- modeleddata_large_clean_otherenviro2 %>%
mutate(NO3_co = coalesce(`[NO3]- [¬µmol/l] (mean value at a depth of 5 m ...).x`, `[NO3]- [¬µmol/l] (mean value at a depth of 5 m ...).y`),
       NO2_co = coalesce(`[NO2]- [¬µmol/l] (mean value at a depth of 5 m ...).x`, `[NO2]- [¬µmol/l] (mean value at a depth of 5 m ...).y`),
       PAR_co = coalesce(`PAR [mol quanta/m**2/day] (at the sampling location and ...).x`, `PAR [mol quanta/m**2/day] (at the sampling location and ...).y`),
       Iron_5m = coalesce(`Fe tot [¬µmol/l] (mean value at a depth of 5 m ...).x`, `Fe tot [¬µmol/l] (mean value at a depth of 5 m ...).y`),
       POC = coalesce(`POC [¬µg/kg] (at the sampling location for ...).x`, `POC [¬µg/kg] (at the sampling location for ...).y`),
       Region = coalesce(`OS region ([abbreviation] full name (MRG...).x`, `OS region ([abbreviation] full name (MRG...).y`),
       TSS = coalesce(`TSS [mg/l] (at the sampling location for ...).y`, `TSS [mg/l] (at the sampling location for ...).x`)) %>%
  dplyr::rename("station" = "Station (TARA_station#, registered at ...).y") %>%
  select(Sample_Name, station, size_fraction_lower_threshold, Size_Fraction_Upper_Threshold, catmn, katE, katG, CO2..umol_kg.1.,
         NO3_NO2_modeled, phosphate_dissolved_umol.kg, temperature_dissolved_deg.c, salinity_dissolved_PSS.1978,
         latitude, longitude, pro_HLI, pro_HLII, sar11, NO3_co, NO2_co, PAR_co, Iron_5m, POC, TSS, synDarwin_umolC.kg, proDarwin_umolC.kg, Region) %>%
  mutate(NO3_NO2_co = NO3_co + NO2_co) %>%
  select(-NO3_co, -NO2_co, -NO3_NO2_modeled) %>%
  filter(Region != "[SO] Southern Ocean (MRGID:1907)")
  
modeleddata_large_clean_otherenviro2_co$catmn[is.na(modeleddata_large_clean_otherenviro2_co$catmn)] <- 0
modeleddata_large_clean_otherenviro2_co$katG[is.na(modeleddata_large_clean_otherenviro2_co$katG)] <- 0
modeleddata_large_clean_otherenviro2_co$katE[is.na(modeleddata_large_clean_otherenviro2_co$katE)] <- 0

largedata_withCO2 <- modeleddata_large_clean_otherenviro2_co %>%
  mutate(log10katG = log10(katG),
         log10katE = log10(katE)) %>%
  drop_na()

largedata_withCO2$log10katG[largedata_withCO2$log10katG=="-Inf"]<- 0
largedata_withCO2$log10katE[largedata_withCO2$log10katE=="-Inf"]<- 0

largedata_noCO2 <- modeleddata_large_clean_otherenviro2_co %>%
  select(-CO2..umol_kg.1.) %>%
  mutate(log10katG = log10(katG),
          log10katE = log10(katE)) 

largedata_noCO2$log10katG[largedata_noCO2$log10katG=="-Inf"]<- 0
largedata_noCO2$log10katE[largedata_noCO2$log10katE=="-Inf"]<- 0
  
stationids <- TARA_SAMPLES_CONTEXT_ENV_WATERCOLUMN_tomerge %>%
  select(Sample_Name, `Station (TARA_station#, registered at ...)`) %>%
    dplyr::rename("station" = "Station (TARA_station#, registered at ...)")

largedata_noCO2_station <- left_join(largedata_noCO2,
                                      stationids,
                                      by = "Sample_Name") %>%
  mutate(station_co = coalesce(station.x, station.y)) %>%
  select(-station.x, station.y) %>%
  drop_na()


#large size fraction models
plot(ggeffects::ggpredict(normKatG2_large_enviro, terms = "salinity_dissolved_PSS.1978"), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("Prochlorococcus biomass") + ylab("katG\ncopy number per genome")
#enviro only
library(mgcv)
largedata_noCO2_trans <- largedata_noCO2_station %>%
  transform(Size_Fraction_Upper_Threshold = as.factor(Size_Fraction_Upper_Threshold))


normKatG1_large_enviro <- mgcv::gam(log10katG ~ s(temperature_dissolved_deg.c) + s(salinity_dissolved_PSS.1978) + s(PAR_co) + s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC), data=largedata_noCO2_trans, method = "REML")


normKatG1_large_enviro <- mgcv::gam(log10katG ~ s(temperature_dissolved_deg.c, by = Size_Fraction_Upper_Threshold) + s(salinity_dissolved_PSS.1978, by = Size_Fraction_Upper_Threshold) + s(PAR_co, by = Size_Fraction_Upper_Threshold) + s(NO3_NO2_co, by = Size_Fraction_Upper_Threshold) + s(phosphate_dissolved_umol.kg, by = Size_Fraction_Upper_Threshold) + s(Iron_5m, by = Size_Fraction_Upper_Threshold) + s(POC, by = Size_Fraction_Upper_Threshold) + Size_Fraction_Upper_Threshold, data=largedata_noCO2_trans, method = "REML")

normKatG2_large_enviro <- mgcv::gam(log10katG ~ s(temperature_dissolved_deg.c, by = Size_Fraction_Upper_Threshold) +  s(PAR_co, by = Size_Fraction_Upper_Threshold) + s(NO3_NO2_co, by = Size_Fraction_Upper_Threshold) + s(phosphate_dissolved_umol.kg, by = Size_Fraction_Upper_Threshold) + s(Iron_5m, by = Size_Fraction_Upper_Threshold) + s(POC, by = Size_Fraction_Upper_Threshold) + Size_Fraction_Upper_Threshold, data=largedata_noCO2_trans, method = "REML")


normKatG3_large_enviro <- mgcv::gam(log10katG ~ s(temperature_dissolved_deg.c, by = Size_Fraction_Upper_Threshold) + s(NO3_NO2_co, by = Size_Fraction_Upper_Threshold) + s(phosphate_dissolved_umol.kg, by = Size_Fraction_Upper_Threshold) + s(Iron_5m, by = Size_Fraction_Upper_Threshold) + s(POC, by = Size_Fraction_Upper_Threshold) + Size_Fraction_Upper_Threshold, data=largedata_noCO2_trans, method = "REML")

normKatG4_large_enviro <- mgcv::gam(log10katG ~ s(temperature_dissolved_deg.c, by = Size_Fraction_Upper_Threshold) + s(NO3_NO2_co, by = Size_Fraction_Upper_Threshold) + s(Iron_5m, by = Size_Fraction_Upper_Threshold) + s(POC, by = Size_Fraction_Upper_Threshold) + Size_Fraction_Upper_Threshold, data=largedata_noCO2_trans, method = "REML")

normKatG5_large_enviro <- mgcv::gam(log10katG ~ s(temperature_dissolved_deg.c, by = Size_Fraction_Upper_Threshold) + s(NO3_NO2_co, by = Size_Fraction_Upper_Threshold) + s(POC, by = Size_Fraction_Upper_Threshold) + Size_Fraction_Upper_Threshold, data=largedata_noCO2_trans, method = "REML")

normKatG6_large_enviro <- mgcv::gam(log10katG ~ s(temperature_dissolved_deg.c, by = Size_Fraction_Upper_Threshold) + s(NO3_NO2_co, by = Size_Fraction_Upper_Threshold) + s(POC, by = Size_Fraction_Upper_Threshold), data=largedata_noCO2_trans, method = "REML")

largedata_noCO2_trans_POC <- largedata_noCO2_trans %>%
  arrange(desc(POC))

largedata_noCO2_trans_POC <- largedata_noCO2_trans_POC[-c(1:4),] %>%
  filter(Size_Fraction_Upper_Threshold!="3")

normKatG6_large_enviro <- mgcv::gam(log10katG ~ s(temperature_dissolved_deg.c, by = Size_Fraction_Upper_Threshold) + s(NO3_NO2_co, by = Size_Fraction_Upper_Threshold) + s(POC, by = Size_Fraction_Upper_Threshold), data=largedata_noCO2_trans_POC, method = "REML")




######
largedata_noCO2_trans <- largedata_noCO2_trans %>%
  filter(Size_Fraction_Upper_Threshold!="3")

normKatE1_large_enviro <- mgcv::gam(log10katE~ s(temperature_dissolved_deg.c) + s(salinity_dissolved_PSS.1978) + s(PAR_co) + s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + Size_Fraction_Upper_Threshold, data=largedata_noCO2_trans, method = "REML")

normKatE2_large_enviro <- mgcv::gam(log10katE ~ s(salinity_dissolved_PSS.1978) + s(PAR_co) + s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + Size_Fraction_Upper_Threshold, data=largedata_noCO2_trans, method = "REML")

normKatE3_large_enviro <- mgcv::gam(log10katE ~  s(PAR_co) + s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + Size_Fraction_Upper_Threshold, data=largedata_noCO2_trans, method = "REML")

normKatE4_large_enviro <- mgcv::gam(log10katE ~ s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + Size_Fraction_Upper_Threshold, data=largedata_noCO2_trans, method = "REML")

normKatE5_large_enviro <- mgcv::gam(log10katE ~ s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + Size_Fraction_Upper_Threshold, data=largedata_noCO2_trans, method = "REML")

normKatE6_large_enviro <- mgcv::gam(log10katE ~ s(Iron_5m) + s(POC) + Size_Fraction_Upper_Threshold, data=largedata_noCO2_trans, method = "REML")

normKatE7_large_enviro <- mgcv::gam(log10katE ~ s(POC) + Size_Fraction_Upper_Threshold, data=largedata_noCO2_trans, method = "REML")

normKatE8_large_enviro <- mgcv::gam(log10katE ~ s(POC) + Size_Fraction_Upper_Threshold, data=largedata_noCO2_trans_POC, method = "REML")



###try repeating with sar11
normKatE1_large_enviro <- mgcv::gam(log10katE ~ s(temperature_dissolved_deg.c) + s(salinity_dissolved_PSS.1978) + s(PAR_co) + s(NO3_NO2_co, by = Size_Fraction_Upper_Threshold) + s(phosphate_dissolved_umol.kg, by = Size_Fraction_Upper_Threshold) + s(Iron_5m, by = Size_Fraction_Upper_Threshold) + s(POC, by = Size_Fraction_Upper_Threshold) + Size_Fraction_Upper_Threshold + s(sar11, by = Size_Fraction_Upper_Threshold), data=largedata_noCO2_trans, method = "REML")

normKatE2_large_enviro <- mgcv::gam(log10katE ~ s(temperature_dissolved_deg.c) + s(salinity_dissolved_PSS.1978) + s(NO3_NO2_co, by = Size_Fraction_Upper_Threshold) + s(phosphate_dissolved_umol.kg, by = Size_Fraction_Upper_Threshold) + s(Iron_5m, by = Size_Fraction_Upper_Threshold) + s(POC, by = Size_Fraction_Upper_Threshold) + Size_Fraction_Upper_Threshold + s(sar11, by = Size_Fraction_Upper_Threshold), data=largedata_noCO2_trans, method = "REML")

normKatE3_large_enviro <- mgcv::gam(log10katE ~ s(temperature_dissolved_deg.c) + s(salinity_dissolved_PSS.1978) + s(NO3_NO2_co, by = Size_Fraction_Upper_Threshold) + s(phosphate_dissolved_umol.kg, by = Size_Fraction_Upper_Threshold) + s(Iron_5m, by = Size_Fraction_Upper_Threshold) + s(POC, by = Size_Fraction_Upper_Threshold) + Size_Fraction_Upper_Threshold + s(sar11, by = Size_Fraction_Upper_Threshold), data=largedata_noCO2_trans, method = "REML")

normKatE3_large_enviro <- mgcv::gam(log10katE ~ s(temperature_dissolved_deg.c) + s(salinity_dissolved_PSS.1978) + s(NO3_NO2_co, by = Size_Fraction_Upper_Threshold) + s(phosphate_dissolved_umol.kg, by = Size_Fraction_Upper_Threshold) + s(Iron_5m, by = Size_Fraction_Upper_Threshold) + s(POC, by = Size_Fraction_Upper_Threshold) + Size_Fraction_Upper_Threshold + s(sar11, by = Size_Fraction_Upper_Threshold), data=largedata_noCO2_trans_POC, method = "REML")




#################
normKatG1_large_enviro <- mgcv::gam(log10katG~ s(temperature_dissolved_deg.c) + s(salinity_dissolved_PSS.1978) + s(PAR_co) + s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + Size_Fraction_Upper_Threshold, data=largedata_noCO2_trans, method = "REML")

normKatG2_large_enviro <- mgcv::gam(log10katG~ s(salinity_dissolved_PSS.1978) + s(PAR_co) + s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + Size_Fraction_Upper_Threshold, data=largedata_noCO2_trans, method = "REML")

normKatG3_large_enviro <- mgcv::gam(log10katG~  s(PAR_co) + s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + Size_Fraction_Upper_Threshold, data=largedata_noCO2_trans, method = "REML")

normKatG4_large_enviro <- mgcv::gam(log10katG~  s(PAR_co) + s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(POC) + Size_Fraction_Upper_Threshold, data=largedata_noCO2_trans, method = "REML")

normKatG5_large_enviro <- mgcv::gam(log10katG ~  s(PAR_co) + s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(POC) + Size_Fraction_Upper_Threshold, data=largedata_noCO2_trans_POC, method = "REML")



#####

largedata_noCO2_trans <- largedata_noCO2_trans %>%
  transform(size_fraction_lower_threshold = as.factor(size_fraction_lower_threshold))

normKatE1_large_enviro <- mgcv::gam(log10katE ~ s(temperature_dissolved_deg.c) + s(salinity_dissolved_PSS.1978) + s(NO3_NO2_co, by = size_fraction_lower_threshold) + s(PAR_co) + s(phosphate_dissolved_umol.kg, by = size_fraction_lower_threshold) + s(Iron_5m, by = size_fraction_lower_threshold) + + s(POC, by = size_fraction_lower_threshold)+ size_fraction_lower_threshold, data=largedata_noCO2_trans, method = "REML")

normKatE2_large_enviro <- mgcv::gam(log10katE ~ s(salinity_dissolved_PSS.1978) + s(NO3_NO2_co, by = size_fraction_lower_threshold) + s(PAR_co) + s(phosphate_dissolved_umol.kg, by = size_fraction_lower_threshold) + s(Iron_5m, by = size_fraction_lower_threshold) + s(POC, by = size_fraction_lower_threshold)+ size_fraction_lower_threshold, data=largedata_noCO2_trans, method = "REML")

normKatE3_large_enviro <- mgcv::gam(log10katE ~ s(NO3_NO2_co, by = size_fraction_lower_threshold) + s(PAR_co) + s(phosphate_dissolved_umol.kg, by = size_fraction_lower_threshold) + s(Iron_5m, by = size_fraction_lower_threshold) + s(POC, by = size_fraction_lower_threshold)+ size_fraction_lower_threshold, data=largedata_noCO2_trans, method = "REML")

normKatE4_large_enviro <- mgcv::gam(log10katE ~ s(PAR_co) + s(phosphate_dissolved_umol.kg, by = size_fraction_lower_threshold) + s(Iron_5m, by = size_fraction_lower_threshold) + s(POC, by = size_fraction_lower_threshold)+ size_fraction_lower_threshold, data=largedata_noCO2_trans, method = "REML")

normKatE5_large_enviro <- mgcv::gam(log10katE ~ s(PAR_co) + s(phosphate_dissolved_umol.kg, by = size_fraction_lower_threshold) + s(POC, by = size_fraction_lower_threshold)+ size_fraction_lower_threshold, data=largedata_noCO2_trans, method = "REML")

normKatE6_large_enviro <- mgcv::gam(log10katE ~ s(PAR_co) + s(phosphate_dissolved_umol.kg, by = size_fraction_lower_threshold) + s(POC, by = size_fraction_lower_threshold)+ size_fraction_lower_threshold, data=largedata_noCO2_trans_POC, method = "REML")

#############


normKatG1_large_enviro <- mgcv::gam(log10katG ~ s(temperature_dissolved_deg.c) + s(salinity_dissolved_PSS.1978) + s(NO3_NO2_co, by = size_fraction_lower_threshold) + s(PAR_co) + s(phosphate_dissolved_umol.kg, by = size_fraction_lower_threshold) + s(Iron_5m, by = size_fraction_lower_threshold) + + s(POC, by = size_fraction_lower_threshold)+ size_fraction_lower_threshold, data=largedata_noCO2_trans, method = "REML")

normKatG2_large_enviro <- mgcv::gam(log10katG ~  s(salinity_dissolved_PSS.1978) + s(NO3_NO2_co, by = size_fraction_lower_threshold) + s(PAR_co) + s(phosphate_dissolved_umol.kg, by = size_fraction_lower_threshold) + s(Iron_5m, by = size_fraction_lower_threshold) + s(POC, by = size_fraction_lower_threshold)+ size_fraction_lower_threshold, data=largedata_noCO2_trans, method = "REML")

normKatG3_large_enviro <- mgcv::gam(log10katG ~ s(NO3_NO2_co, by = size_fraction_lower_threshold) + s(PAR_co) + s(phosphate_dissolved_umol.kg, by = size_fraction_lower_threshold) + s(Iron_5m, by = size_fraction_lower_threshold) + s(POC, by = size_fraction_lower_threshold)+ size_fraction_lower_threshold, data=largedata_noCO2_trans, method = "REML")

normKatG4_large_enviro <- mgcv::gam(log10katG ~ s(NO3_NO2_co, by = size_fraction_lower_threshold) + s(phosphate_dissolved_umol.kg, by = size_fraction_lower_threshold) + s(Iron_5m, by = size_fraction_lower_threshold) + s(POC, by = size_fraction_lower_threshold)+ size_fraction_lower_threshold, data=largedata_noCO2_trans, method = "REML")

normKatG5_large_enviro <- mgcv::gam(log10katG ~ s(NO3_NO2_co, by = size_fraction_lower_threshold) + s(phosphate_dissolved_umol.kg, by = size_fraction_lower_threshold) + s(Iron_5m, by = size_fraction_lower_threshold) + s(POC, by = size_fraction_lower_threshold)+ size_fraction_lower_threshold, data=largedata_noCO2_trans_POC, method = "REML")



###sar11
normKatE1_large_enviro <- mgcv::gam(log10katE ~ s(temperature_dissolved_deg.c) + s(salinity_dissolved_PSS.1978) + s(NO3_NO2_co, by = size_fraction_lower_threshold) + s(PAR_co) + s(phosphate_dissolved_umol.kg, by = size_fraction_lower_threshold) + s(Iron_5m, by = size_fraction_lower_threshold) + s(POC, by = size_fraction_lower_threshold)+ size_fraction_lower_threshold + s(sar11, by = size_fraction_lower_threshold), data=largedata_noCO2_trans, method = "REML")

normKatE2_large_enviro <- mgcv::gam(log10katE ~ s(salinity_dissolved_PSS.1978) + s(NO3_NO2_co, by = size_fraction_lower_threshold) + s(PAR_co) + s(phosphate_dissolved_umol.kg, by = size_fraction_lower_threshold) + s(Iron_5m, by = size_fraction_lower_threshold) + s(POC, by = size_fraction_lower_threshold)+ size_fraction_lower_threshold + s(sar11, by = size_fraction_lower_threshold), data=largedata_noCO2_trans, method = "REML")


normKatE3_large_enviro <- mgcv::gam(log10katE ~ s(NO3_NO2_co, by = size_fraction_lower_threshold) + s(PAR_co) + s(phosphate_dissolved_umol.kg, by = size_fraction_lower_threshold) + s(Iron_5m, by = size_fraction_lower_threshold) + s(POC, by = size_fraction_lower_threshold)+ size_fraction_lower_threshold + s(sar11, by = size_fraction_lower_threshold), data=largedata_noCO2_trans, method = "REML")

normKatE4_large_enviro <- mgcv::gam(log10katE ~ s(PAR_co) + s(phosphate_dissolved_umol.kg, by = size_fraction_lower_threshold) + s(Iron_5m, by = size_fraction_lower_threshold) + s(POC, by = size_fraction_lower_threshold)+ size_fraction_lower_threshold + s(sar11, by = size_fraction_lower_threshold), data=largedata_noCO2_trans, method = "REML")

normKatE5_large_enviro <- mgcv::gam(log10katE ~ s(phosphate_dissolved_umol.kg, by = size_fraction_lower_threshold) + s(Iron_5m, by = size_fraction_lower_threshold) + s(POC, by = size_fraction_lower_threshold)+ size_fraction_lower_threshold + s(sar11, by = size_fraction_lower_threshold), data=largedata_noCO2_trans, method = "REML")

normKatE6_large_enviro <- mgcv::gam(log10katE ~ s(phosphate_dissolved_umol.kg, by = size_fraction_lower_threshold) + s(Iron_5m, by = size_fraction_lower_threshold) + s(POC, by = size_fraction_lower_threshold) + size_fraction_lower_threshold + s(sar11, by = size_fraction_lower_threshold), data=largedata_noCO2_trans_POC, method = "REML")


normKatE3_large_enviro_poc_sar11 <- plot(ggeffects::ggpredict(normKatE6_large_enviro, terms = c("POC", "size_fraction_lower_threshold")), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("POC") + ylab("katG\ncopy number per genome")

normKatE3_large_enviro_sar11 <- plot(ggeffects::ggpredict(normKatE6_large_enviro, terms = c("sar11", "size_fraction_lower_threshold")), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("sar11") + ylab("katE\ncopy number per genome")




#########

normKatG1_large_enviro <- mgcv::gam(log10katG ~ s(temperature_dissolved_deg.c) + s(salinity_dissolved_PSS.1978) + s(NO3_NO2_co, by = size_fraction_lower_threshold) + s(PAR_co) + s(phosphate_dissolved_umol.kg, by = size_fraction_lower_threshold) + s(Iron_5m, by = size_fraction_lower_threshold) + s(POC, by = size_fraction_lower_threshold)+ size_fraction_lower_threshold + s(sar11, by = size_fraction_lower_threshold), data=largedata_noCO2_trans, method = "REML")

normKatG2_large_enviro <- mgcv::gam(log10katG ~ s(salinity_dissolved_PSS.1978) + s(NO3_NO2_co, by = size_fraction_lower_threshold) + s(PAR_co) + s(phosphate_dissolved_umol.kg, by = size_fraction_lower_threshold) + s(Iron_5m, by = size_fraction_lower_threshold) + s(POC, by = size_fraction_lower_threshold)+ size_fraction_lower_threshold + s(sar11, by = size_fraction_lower_threshold), data=largedata_noCO2_trans, method = "REML")

normKatG3_large_enviro <- mgcv::gam(log10katG ~ s(salinity_dissolved_PSS.1978) + s(PAR_co) + s(phosphate_dissolved_umol.kg, by = size_fraction_lower_threshold) + s(Iron_5m, by = size_fraction_lower_threshold) + s(POC, by = size_fraction_lower_threshold)+ size_fraction_lower_threshold + s(sar11, by = size_fraction_lower_threshold), data=largedata_noCO2_trans, method = "REML")

normKatG4_large_enviro <- mgcv::gam(log10katG ~ s(salinity_dissolved_PSS.1978) + s(phosphate_dissolved_umol.kg, by = size_fraction_lower_threshold) + s(Iron_5m, by = size_fraction_lower_threshold) + s(POC, by = size_fraction_lower_threshold)+ size_fraction_lower_threshold + s(sar11, by = size_fraction_lower_threshold), data=largedata_noCO2_trans, method = "REML")

normKatG5_large_enviro <- mgcv::gam(log10katG ~ s(salinity_dissolved_PSS.1978) + s(phosphate_dissolved_umol.kg, by = size_fraction_lower_threshold) + s(POC, by = size_fraction_lower_threshold)+ size_fraction_lower_threshold + s(sar11, by = size_fraction_lower_threshold), data=largedata_noCO2_trans, method = "REML")

normKatG6_large_enviro <- mgcv::gam(log10katG ~ s(salinity_dissolved_PSS.1978) + s(phosphate_dissolved_umol.kg, by = size_fraction_lower_threshold) + s(POC, by = size_fraction_lower_threshold)+ size_fraction_lower_threshold + s(sar11, by = size_fraction_lower_threshold), data=largedata_noCO2_trans_POC, method = "REML")

normKatG6_large_enviro_phosphate_sar11 <- plot(ggeffects::ggpredict(normKatG6_large_enviro, terms = c("phosphate_dissolved_umol.kg", "size_fraction_lower_threshold")), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("Phosphate concentration") + ylab("katG\ncopy number per genome")

normKatG6_large_enviro_poc_sar11 <- plot(ggeffects::ggpredict(normKatG6_large_enviro, terms = c("POC", "size_fraction_lower_threshold")), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("POC") + ylab("katG\ncopy number per genome")



normKatG1_large_enviro_pro <- mgcv::gam(log10katG ~ s(temperature_dissolved_deg.c) + s(salinity_dissolved_PSS.1978) + s(NO3_NO2_co, by = size_fraction_lower_threshold) + s(PAR_co) + s(phosphate_dissolved_umol.kg, by = size_fraction_lower_threshold) + s(Iron_5m, by = size_fraction_lower_threshold) + s(POC, by = size_fraction_lower_threshold)+ size_fraction_lower_threshold + s(proDarwin_umolC.kg, by = size_fraction_lower_threshold) + s(pro_HLI, by = size_fraction_lower_threshold) + s(pro_HLII, by = size_fraction_lower_threshold), data=largedata_noCO2_trans, method = "REML")

normKatG2_large_enviro_pro <- mgcv::gam(log10katG ~  s(salinity_dissolved_PSS.1978) + s(NO3_NO2_co, by = size_fraction_lower_threshold) + s(PAR_co) + s(phosphate_dissolved_umol.kg, by = size_fraction_lower_threshold) + s(Iron_5m, by = size_fraction_lower_threshold) + s(POC, by = size_fraction_lower_threshold)+ size_fraction_lower_threshold + s(proDarwin_umolC.kg, by = size_fraction_lower_threshold) + s(pro_HLI, by = size_fraction_lower_threshold) + s(pro_HLII, by = size_fraction_lower_threshold), data=largedata_noCO2_trans, method = "REML")

normKatG3_large_enviro_pro <- mgcv::gam(log10katG ~  s(salinity_dissolved_PSS.1978) + s(PAR_co) + s(phosphate_dissolved_umol.kg, by = size_fraction_lower_threshold) + s(Iron_5m, by = size_fraction_lower_threshold) + s(POC, by = size_fraction_lower_threshold)+ size_fraction_lower_threshold + s(proDarwin_umolC.kg, by = size_fraction_lower_threshold) + s(pro_HLI, by = size_fraction_lower_threshold) + s(pro_HLII, by = size_fraction_lower_threshold), data=largedata_noCO2_trans, method = "REML")

normKatG4_large_enviro_pro <- mgcv::gam(log10katG ~  s(PAR_co) + s(phosphate_dissolved_umol.kg, by = size_fraction_lower_threshold) + s(Iron_5m, by = size_fraction_lower_threshold) + s(POC, by = size_fraction_lower_threshold)+ size_fraction_lower_threshold + s(proDarwin_umolC.kg, by = size_fraction_lower_threshold) + s(pro_HLI, by = size_fraction_lower_threshold) + s(pro_HLII, by = size_fraction_lower_threshold), data=largedata_noCO2_trans, method = "REML")

normKatG5_large_enviro_pro <- mgcv::gam(log10katG ~  s(PAR_co) + s(phosphate_dissolved_umol.kg, by = size_fraction_lower_threshold) + s(Iron_5m, by = size_fraction_lower_threshold) + s(POC, by = size_fraction_lower_threshold) + size_fraction_lower_threshold + s(proDarwin_umolC.kg, by = size_fraction_lower_threshold) + s(pro_HLI, by = size_fraction_lower_threshold) + s(pro_HLII, by = size_fraction_lower_threshold), data=largedata_noCO2_trans, method = "REML")

normKatG6_large_enviro_pro <- mgcv::gam(log10katG ~  s(PAR_co) + s(phosphate_dissolved_umol.kg, by = size_fraction_lower_threshold) + s(Iron_5m, by = size_fraction_lower_threshold) + s(POC, by = size_fraction_lower_threshold) + size_fraction_lower_threshold + s(proDarwin_umolC.kg, by = size_fraction_lower_threshold) + s(pro_HLI, by = size_fraction_lower_threshold) + s(pro_HLII, by = size_fraction_lower_threshold), data=largedata_noCO2_trans_POC, method = "REML")

normKatG6_large_enviro_poc_pro <- plot(ggeffects::ggpredict(normKatG6_large_enviro_pro, terms = c("POC", "size_fraction_lower_threshold")), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("POC") + ylab("katG\ncopy number per genome")

########
normKatE1_large_enviro_pro <- mgcv::gam(log10katE ~ s(temperature_dissolved_deg.c) + s(salinity_dissolved_PSS.1978) + s(NO3_NO2_co, by = size_fraction_lower_threshold) + s(PAR_co) + s(phosphate_dissolved_umol.kg, by = size_fraction_lower_threshold) + s(Iron_5m, by = size_fraction_lower_threshold) + s(POC, by = size_fraction_lower_threshold)+ size_fraction_lower_threshold + s(proDarwin_umolC.kg, by = size_fraction_lower_threshold) + s(pro_HLI, by = size_fraction_lower_threshold) + s(pro_HLII, by = size_fraction_lower_threshold), data=largedata_noCO2_trans, method = "REML")

normKatE2_large_enviro_pro <- mgcv::gam(log10katE ~  s(salinity_dissolved_PSS.1978) + s(NO3_NO2_co, by = size_fraction_lower_threshold) + s(PAR_co) + s(phosphate_dissolved_umol.kg, by = size_fraction_lower_threshold) + s(Iron_5m, by = size_fraction_lower_threshold) + s(POC, by = size_fraction_lower_threshold)+ size_fraction_lower_threshold + s(proDarwin_umolC.kg, by = size_fraction_lower_threshold) + s(pro_HLI, by = size_fraction_lower_threshold) + s(pro_HLII, by = size_fraction_lower_threshold), data=largedata_noCO2_trans, method = "REML")

normKatE3_large_enviro_pro <- mgcv::gam(log10katE ~  s(salinity_dissolved_PSS.1978)  + s(PAR_co) + s(phosphate_dissolved_umol.kg, by = size_fraction_lower_threshold) + s(Iron_5m, by = size_fraction_lower_threshold) + s(POC, by = size_fraction_lower_threshold)+ size_fraction_lower_threshold + s(proDarwin_umolC.kg, by = size_fraction_lower_threshold) + s(pro_HLI, by = size_fraction_lower_threshold) + s(pro_HLII, by = size_fraction_lower_threshold), data=largedata_noCO2_trans, method = "REML")

normKatE4_large_enviro_pro <- mgcv::gam(log10katE ~ s(PAR_co) + s(phosphate_dissolved_umol.kg, by = size_fraction_lower_threshold) + s(Iron_5m, by = size_fraction_lower_threshold) + s(POC, by = size_fraction_lower_threshold)+ size_fraction_lower_threshold + s(proDarwin_umolC.kg, by = size_fraction_lower_threshold) + s(pro_HLI, by = size_fraction_lower_threshold) + s(pro_HLII, by = size_fraction_lower_threshold), data=largedata_noCO2_trans, method = "REML")

normKatE4_large_enviro_pro <- mgcv::gam(log10katE ~ s(PAR_co) + s(phosphate_dissolved_umol.kg, by = size_fraction_lower_threshold) + s(Iron_5m, by = size_fraction_lower_threshold) + s(POC, by = size_fraction_lower_threshold)+ size_fraction_lower_threshold + s(proDarwin_umolC.kg, by = size_fraction_lower_threshold) + s(pro_HLI, by = size_fraction_lower_threshold) + s(pro_HLII, by = size_fraction_lower_threshold), data=largedata_noCO2_trans_POC, method = "REML")



normKatE4_large_enviro_hlii_pro <- plot(ggeffects::ggpredict(normKatE4_large_enviro_pro, terms = c("pro_HLII", "size_fraction_lower_threshold")), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("HLII pro biomass") + ylab("katE\ncopy number per genome")
############running models with only the largest size fracions

largedata_noCO2_trans_filtered <- largedata_noCO2_trans %>%
  filter(size_fraction_lower_threshold != "0.8")

largedata_noCO2_trans_filtered <- read.csv("~/Desktop/Dyhrman Lab/Desktop_DyhrmanLab/largedata_noCO2_trans_filt.csv")

normKatE1_large_enviro_filt <- mgcv::gam(log10katE ~ s(temperature_dissolved_deg.c) + s(salinity_dissolved_PSS.1978) + s(NO3_NO2_co) + s(PAR_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC), data=largedata_noCO2_trans_filtered, method = "REML")

normKatE2_large_enviro_filt <- mgcv::gam(log10katE ~ s(salinity_dissolved_PSS.1978) + s(NO3_NO2_co) + s(PAR_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC), data=largedata_noCO2_trans_filtered, method = "REML")

normKatE3_large_enviro_filt <- mgcv::gam(log10katE ~ s(NO3_NO2_co) + s(PAR_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC), data=largedata_noCO2_trans_filtered, method = "REML")

normKatE4_large_enviro_filt <- mgcv::gam(log10katE ~ s(PAR_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC), data=largedata_noCO2_trans_filtered, method = "REML")

normKatE5_large_enviro_filt <- mgcv::gam(log10katE ~  s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC), data=largedata_noCO2_trans_filtered, method = "REML")

normKatE6_large_enviro_filt <- mgcv::gam(log10katE ~ s(Iron_5m) + s(POC), data=largedata_noCO2_trans_filtered, method = "REML")

normKatE7_large_enviro_filt <- mgcv::gam(log10katE ~ s(Iron_5m), data=largedata_noCO2_trans_filtered, method = "REML")


normKatE4_large_enviro_iron <- plot(ggeffects::ggpredict(normKatE7_large_enviro_filt, terms = c("Iron_5m")), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("Iron concentration at 5 meters") + ylab("katE\ncopy number per genome")


###repeat for katG
normKatG1_large_enviro_filt <- mgcv::gam(log10katG ~ s(temperature_dissolved_deg.c) + s(salinity_dissolved_PSS.1978) + s(NO3_NO2_co) + s(PAR_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC), data=largedata_noCO2_trans_filtered, method = "REML")

normKatG2_large_enviro_filt <- mgcv::gam(log10katG ~ s(temperature_dissolved_deg.c) + s(NO3_NO2_co) + s(PAR_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC), data=largedata_noCO2_trans_filtered, method = "REML")

normKatG3_large_enviro_filt <- mgcv::gam(log10katG ~ s(temperature_dissolved_deg.c) + s(NO3_NO2_co) + s(PAR_co) + s(phosphate_dissolved_umol.kg) + s(POC), data=largedata_noCO2_trans_filtered, method = "REML")

normKatG4_large_enviro_filt <- mgcv::gam(log10katG ~ s(temperature_dissolved_deg.c) + s(NO3_NO2_co) + s(PAR_co) + s(phosphate_dissolved_umol.kg) + s(POC), data=largedata_noCO2_trans_filtered, method = "REML")

normKatG4_large_enviro_filt_POC <- plot(ggeffects::ggpredict(normKatG4_large_enviro_filt, terms = c("POC")), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("Iron concentrationat 5 meters") + ylab("katE\ncopy number per genome")


#POC
largedata_noCO2_trans_filtered <- largedata_noCO2_trans_filtered %>%
  arrange(desc(POC))
largedata_noCO2_trans_filtered_POC <- largedata_noCO2_trans_filtered[-c(1:3),]

normKatG4_large_enviro_filt <- mgcv::gam(log10katG) ~ s(temperature_dissolved_deg.c) + s(NO3_NO2_co) + s(PAR_co) + s(phosphate_dissolved_umol.kg) + s(POC), data=largedata_noCO2_trans_filtered_POC, method = "REML")

normKatG4_large_enviro_filt_POC <- plot(ggeffects::ggpredict(normKatG4_large_enviro_filt, terms = c("POC")), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("POC") + ylab("log10 katG\ncopy number per genome")

normKatG4_large_enviro_filt_N<- plot(ggeffects::ggpredict(normKatG4_large_enviro_filt, terms = c("NO3_NO2_co")), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("Inorganic nitrogen concentration") + ylab("log10 katG\ncopy number per genome")

normKatG4_large_enviro_filt_P <- plot(ggeffects::ggpredict(normKatG4_large_enviro_filt, terms = c("phosphate_dissolved_umol.kg")), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("Phosphate concentration") + ylab("log10 katG\ncopy number per genome")

grid.arrange(normKatG4_large_enviro_filt_POC, normKatG4_large_enviro_filt_N, normKatG4_large_enviro_filt_P, ncol = 1)


######sar11

normKatG1_large_enviro_filt <- mgcv::gam(log10katG ~ s(temperature_dissolved_deg.c) + s(salinity_dissolved_PSS.1978) + s(NO3_NO2_co) + s(PAR_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + s(sar11), data=largedata_noCO2_trans_filtered, method = "REML")

normKatG2_large_enviro_filt <- mgcv::gam(log10katG ~ s(salinity_dissolved_PSS.1978) + s(NO3_NO2_co) + s(PAR_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + s(sar11), data=largedata_noCO2_trans_filtered, method = "REML")

normKatG3_large_enviro_filt <- mgcv::gam(log10katG ~ s(NO3_NO2_co) + s(PAR_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + s(sar11), data=largedata_noCO2_trans_filtered, method = "REML")

normKatG4_large_enviro_filt <- mgcv::gam(log10katG ~  s(PAR_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + s(sar11), data=largedata_noCO2_trans_filtered, method = "REML")

normKatG5_large_enviro_filt <- mgcv::gam(log10katG ~ s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + s(sar11), data=largedata_noCO2_trans_filtered, method = "REML")

normKatG6_large_enviro_filt <- mgcv::gam(log10katG ~ s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(sar11), data=largedata_noCO2_trans_filtered, method = "REML")

normKatG7_large_enviro_filt <- mgcv::gam(log10katG ~ s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(sar11), data=largedata_noCO2_trans_filtered, method = "REML")


normKatG7_large_enviro_filt_iron <- plot(ggeffects::ggpredict(normKatG7_large_enviro_filt, terms = c("Iron_5m")), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("Iron concentration at 5 meters") + ylab("katG\ncopy number per genome")

normKatG7_large_enviro_filt_sar11 <- plot(ggeffects::ggpredict(normKatG7_large_enviro_filt, terms = c("sar11")), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("sar11 biomass") + ylab("log10 katG\ncopy number per genome")

#repeat for katE
normKatE1_large_enviro_filt <- mgcv::gam(log10katE ~ s(temperature_dissolved_deg.c) + s(salinity_dissolved_PSS.1978) + s(NO3_NO2_co) + s(PAR_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + s(sar11), data=largedata_noCO2_trans_filtered, method = "REML")

normKatE2_large_enviro_filt <- mgcv::gam(log10katE ~  s(salinity_dissolved_PSS.1978) + s(NO3_NO2_co) + s(PAR_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + s(sar11), data=largedata_noCO2_trans_filtered, method = "REML")

normKatE3_large_enviro_filt <- mgcv::gam(log10katE ~ s(NO3_NO2_co) + s(PAR_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + s(sar11), data=largedata_noCO2_trans_filtered, method = "REML")

normKatE4_large_enviro_filt <- mgcv::gam(log10katE ~ s(PAR_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + s(sar11), data=largedata_noCO2_trans_filtered, method = "REML")

normKatE5_large_enviro_filt <- mgcv::gam(log10katE ~ s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + s(sar11), data=largedata_noCO2_trans_filtered, method = "REML")

normKatE6_large_enviro_filt <- mgcv::gam(log10katE ~ s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC), data=largedata_noCO2_trans_filtered, method = "REML")

###
normKatE1_large_enviro_filt_pro <- mgcv::gam(log10katE ~ s(temperature_dissolved_deg.c) + s(salinity_dissolved_PSS.1978) + s(NO3_NO2_co) + s(PAR_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + s(proDarwin_umolC.kg) + s(pro_HLI) + s(pro_HLII), data=largedata_noCO2_trans_filtered, method = "REML")

normKatE2_large_enviro_filt_pro <- mgcv::gam(log10katE ~ s(salinity_dissolved_PSS.1978) + s(NO3_NO2_co) + s(PAR_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + s(proDarwin_umolC.kg) + s(pro_HLI) + s(pro_HLII), data=largedata_noCO2_trans_filtered, method = "REML")

normKatGE_large_enviro_filt_pro <- mgcv::gam(log10katE ~ s(NO3_NO2_co) + s(PAR_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + s(proDarwin_umolC.kg) + s(pro_HLI) + s(pro_HLII), data=largedata_noCO2_trans_filtered, method = "REML")

normKatE4_large_enviro_filt_pro <- mgcv::gam(log10katE ~ s(PAR_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + s(proDarwin_umolC.kg) + s(pro_HLI) + s(pro_HLII), data=largedata_noCO2_trans_filtered, method = "REML")

normKatE5_large_enviro_filt_pro <- mgcv::gam(log10katE ~  s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + s(proDarwin_umolC.kg) + s(pro_HLI) + s(pro_HLII), data=largedata_noCO2_trans_filtered, method = "REML")

normKatE6_large_enviro_filt_pro <- mgcv::gam(log10katE ~  s(phosphate_dissolved_umol.kg)  + s(POC) + s(proDarwin_umolC.kg) + s(pro_HLI) + s(pro_HLII), data=largedata_noCO2_trans_filtered, method = "REML")

normKatE7_large_enviro_filt_pro <- mgcv::gam(log10katE ~  s(phosphate_dissolved_umol.kg) + s(proDarwin_umolC.kg) + s(pro_HLI) + s(pro_HLII), data=largedata_noCO2_trans_filtered, method = "REML")

normKatE8_large_enviro_filt_pro <- mgcv::gam(log10katE ~  s(phosphate_dissolved_umol.kg) + s(pro_HLI) + s(pro_HLII), data=largedata_noCO2_trans_filtered, method = "REML")

normKatE9_large_enviro_filt_pro <- mgcv::gam(log10katE ~  s(phosphate_dissolved_umol.kg) + s(pro_HLII), data=largedata_noCO2_trans_filtered, method = "REML")

normKatE10_large_enviro_filt_pro <- mgcv::gam(log10katE ~  s(phosphate_dissolved_umol.kg), data=largedata_noCO2_trans_filtered, method = "REML")
####
normKatG1_large_enviro_filt_pro <- mgcv::gam(log10katG ~ s(temperature_dissolved_deg.c) + s(salinity_dissolved_PSS.1978) + s(NO3_NO2_co) + s(PAR_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + s(proDarwin_umolC.kg) + s(pro_HLI) + s(pro_HLII), data=largedata_noCO2_trans_filtered, method = "REML")

normKatG2_large_enviro_filt_pro <- mgcv::gam(log10katG ~ s(temperature_dissolved_deg.c) + s(NO3_NO2_co) + s(PAR_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + s(proDarwin_umolC.kg) + s(pro_HLI) + s(pro_HLII), data=largedata_noCO2_trans_filtered, method = "REML")

normKatG3_large_enviro_filt_pro <- mgcv::gam(log10katG ~ s(temperature_dissolved_deg.c) + s(PAR_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + s(proDarwin_umolC.kg) + s(pro_HLI) + s(pro_HLII), data=largedata_noCO2_trans_filtered, method = "REML")

normKatG4_large_enviro_filt_pro <- mgcv::gam(log10katG ~ s(temperature_dissolved_deg.c) + s(PAR_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC)  + s(pro_HLI) + s(pro_HLII), data=largedata_noCO2_trans_filtered, method = "REML")

normKatG5_large_enviro_filt_pro <- mgcv::gam(log10katG ~ s(temperature_dissolved_deg.c) + s(PAR_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC)  + s(pro_HLI) + s(pro_HLII), data=largedata_noCO2_trans_filtered_POC, method = "REML")


###syn

normKatG1_large_enviro_filt_syn <- mgcv::gam(log10katG ~ s(temperature_dissolved_deg.c) + s(salinity_dissolved_PSS.1978) + s(NO3_NO2_co) + s(PAR_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + s(synDarwin_umolC.kg), data=largedata_noCO2_trans_filtered, method = "REML")

normKatG2_large_enviro_filt_syn <- mgcv::gam(log10katG ~ s(temperature_dissolved_deg.c) + s(NO3_NO2_co) + s(PAR_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + s(synDarwin_umolC.kg), data=largedata_noCO2_trans_filtered, method = "REML")

normKatG3_large_enviro_filt_syn <- mgcv::gam(log10katG ~ s(temperature_dissolved_deg.c) + s(NO3_NO2_co) + s(PAR_co) + s(phosphate_dissolved_umol.kg) + s(POC) + s(synDarwin_umolC.kg), data=largedata_noCO2_trans_filtered, method = "REML")

normKatG4_large_enviro_filt_syn <- mgcv::gam(log10katG ~ s(temperature_dissolved_deg.c) + s(NO3_NO2_co) + s(PAR_co) + s(POC) + s(synDarwin_umolC.kg), data=largedata_noCO2_trans_filtered, method = "REML")

normKatG5_large_enviro_filt_syn <- mgcv::gam(log10katG ~ s(temperature_dissolved_deg.c) + s(NO3_NO2_co) + s(PAR_co) + s(POC) + s(synDarwin_umolC.kg), data=largedata_noCO2_trans_filtered_POC, method = "REML")

#syn\
normKatE1_large_enviro_filt_syn <- mgcv::gam(log10katE ~ s(temperature_dissolved_deg.c) + s(salinity_dissolved_PSS.1978) + s(NO3_NO2_co) + s(PAR_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + s(synDarwin_umolC.kg), data=largedata_noCO2_trans_filtered, method = "REML")

normKatE2_large_enviro_filt_syn <- mgcv::gam(log10katE ~ s(temperature_dissolved_deg.c) + s(NO3_NO2_co) + s(PAR_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + s(synDarwin_umolC.kg), data=largedata_noCO2_trans_filtered, method = "REML")

normKatE3_large_enviro_filt_syn <- mgcv::gam(log10katE ~ s(temperature_dissolved_deg.c) + s(PAR_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + s(synDarwin_umolC.kg), data=largedata_noCO2_trans_filtered, method = "REML")

normKatE4_large_enviro_filt_syn <- mgcv::gam(log10katE ~ s(temperature_dissolved_deg.c) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + s(synDarwin_umolC.kg), data=largedata_noCO2_trans_filtered, method = "REML")

normKatE5_large_enviro_filt_syn <- mgcv::gam(log10katE ~ s(temperature_dissolved_deg.c) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC), data=largedata_noCO2_trans_filtered, method = "REML")

#####
#some outliers
#remove 4 high POC

largedata_noCO2_POC <- largedata_noCO2 %>%
  arrange(desc(POC))

largedata_noCO2_POC <- largedata_noCO2_POC[-c(1:5),]



#remove 4 high TSS
#plotting

katG_large_NO3_NO2_co <- plot(ggeffects::ggpredict(normKatG5_large_enviro, terms = "NO3_NO2_co"), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("Inorganic nitrogen concentration (µm/kg)") + ylab("katG\ncopy number per genome")

katG_large_POC <- plot(ggeffects::ggpredict(normKatG6_large_enviro, terms = "POC"), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("POC") + ylab("katG\ncopy number per genome")

katG_large_temp <- plot(ggeffects::ggpredict(normKatG6_large_enviro, terms = "Size_Fraction_Upper_Threshold"), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("Size fraction") + ylab("katG\ncopy number per genome")

katE_large_PAR <- plot(ggeffects::ggpredict(normKatE2_large_enviro, terms = c("PAR_co", "Size_Fraction_Upper_Threshold")), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("PAR") + ylab("katE\ncopy number per genome")


katE_large_POC <- plot(ggeffects::ggpredict(normKatE8_large_enviro, terms =  "Size_Fraction_Upper_Threshold"), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("Upper size fraction") + ylab("katE\ncopy number per genome")

katG_large_POC <- plot(ggeffects::ggpredict(normKatG6_large_enviro, terms = c("NO3_NO2_co", "Size_Fraction_Upper_Threshold")), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("Inorganic nitrogen") + ylab("katG\ncopy number per genome")

katG_large_TSS <- plot(ggeffects::ggpredict(normKatG6_large_enviro, terms = "TSS"), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("Total suspended matter") + ylab("katG\ncopy number per genome")



largedata_noCO2_POC_katE <- largedata_noCO2 %>%
  arrange(desc(POC))

largedata_noCO2_POC_katE <- largedata_noCO2_POC_katE[-c(1:4),]


###########repeat for katE

normKatE1_large_enviro <- mgcv::gam(log10katE ~ s(CO2..umol_kg.1.) + s(temperature_dissolved_deg.c) + s(salinity_dissolved_PSS.1978) + s(PAR_co) + s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC), data=largedata_withCO2, method = "REML")

normKatE2_large_enviro <- mgcv::gam(log10katE ~ s(temperature_dissolved_deg.c) + s(salinity_dissolved_PSS.1978) + s(PAR_co) + s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC), data=largedata_withCO2, method = "REML")

##CO2 doesnt affect anything

normKatE1_large_enviro <- mgcv::gam(log10katE ~ s(temperature_dissolved_deg.c) + s(salinity_dissolved_PSS.1978) + s(PAR_co) + s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC), data=largedata_noCO2, method = "REML")


normKatE2_large_enviro <- mgcv::gam(log10katE ~ s(temperature_dissolved_deg.c) + s(PAR_co) + s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC), data=largedata_noCO2, method = "REML")

normKatE3_large_enviro <- mgcv::gam(log10katE ~ s(PAR_co) + s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC), data=largedata_noCO2, method = "REML")

normKatE4_large_enviro <- mgcv::gam(log10katE ~ s(PAR_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC), data=largedata_noCO2, method = "REML")

normKatE5_large_enviro <- mgcv::gam(log10katE ~ s(PAR_co) + s(phosphate_dissolved_umol.kg) + s(POC), data=largedata_noCO2, method = "REML")

normKatE6_large_enviro <- mgcv::gam(log10katE ~ s(PAR_co) + s(phosphate_dissolved_umol.kg), data=largedata_noCO2, method = "REML")

largedata_noCO2 <- largedata_noCO2 %>%
  transform(size_fraction_lower_threshold = as.character(size_fraction_lower_threshold))

normKatE7_large_enviro <- mgcv::gam(log10katE ~ s(phosphate_dissolved_umol.kg) + s(size_fraction_lower_threshold), data=largedata_noCO2, method = "REML")


largedata_noCO2_TSS_katE <- largedata_noCO2 %>%
  arrange(desc(TSS))

largedata_noCO2_TSS_katE <- largedata_noCO2_TSS_katE[-c(1:4),]

normKatE7_large_enviro <- mgcv::gam(log10katE ~ s(PAR_co) + s(phosphate_dissolved_umol.kg) + s(TSS), data=largedata_noCO2_TSS_katE, method = "REML")


katG_large_Fe <- plot(ggeffects::ggpredict(normKatE7_large_enviro, terms = "phosphate_dissolved_umol.kg"), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
                                                                         
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("Iron concentration at 5 meters") + ylab("katG\ncopy number per genome")

katG_large_POC <- plot(ggeffects::ggpredict(normKatG6_large_enviro, terms = "POC"), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("POC") + ylab("katG\ncopy number per genome")


################
normKatG1_sar11 <- mgcv::gam(log10katG ~ s(temperature_dissolved_deg.c) + s(salinity_dissolved_PSS.1978) + s(PAR_co) + s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + Size_Fraction_Upper_Threshold + s(sar11), data=largedata_noCO2_trans, method = "REML")

normKatG2_sar11 <- mgcv::gam(log10katG ~ s(salinity_dissolved_PSS.1978) + s(PAR_co) + s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + Size_Fraction_Upper_Threshold + s(sar11), data=largedata_noCO2_trans, method = "REML")

normKatG3_sar11 <- mgcv::gam(log10katG ~ s(PAR_co) + s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + Size_Fraction_Upper_Threshold + s(sar11), data=largedata_noCO2_trans, method = "REML")

normKatG4_sar11 <- mgcv::gam(log10katG ~ s(PAR_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + Size_Fraction_Upper_Threshold + s(sar11), data=largedata_noCO2_trans, method = "REML")

normKatG5_sar11 <- mgcv::gam(log10katG ~ s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + Size_Fraction_Upper_Threshold + s(sar11), data=largedata_noCO2_trans, method = "REML")

normKatG6_sar11 <- mgcv::gam(log10katG ~ s(phosphate_dissolved_umol.kg) + s(Iron_5m) + Size_Fraction_Upper_Threshold + s(sar11), data=largedata_noCO2_trans, method = "REML")

normKatG7_sar11 <- mgcv::gam(log10katG ~ s(phosphate_dissolved_umol.kg) + s(Iron_5m) + Size_Fraction_Upper_Threshold + s(sar11), data=largedata_noCO2_trans, method = "REML")

katG_large_Fe_sar11 <- plot(ggeffects::ggpredict(normKatG7_sar11, terms = "Iron_5m"), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("Iron concentration at 5 meters") + ylab("katG\ncopy number per genome")

katG_large_sar11 <- plot(ggeffects::ggpredict(normKatG7_sar11, terms = "sar11"), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("sar11 biomass") + ylab("katG\ncopy number per genome")

katG_large_size_fraction <- plot(ggeffects::ggpredict(normKatG7_sar11, terms = "Size_Fraction_Upper_Threshold"), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("Size fraction") + ylab("katG\ncopy number per genome")

##########
normKatE1_sar11 <- mgcv::gam(log10katE ~ s(temperature_dissolved_deg.c) + s(salinity_dissolved_PSS.1978) + s(PAR_co) + s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + Size_Fraction_Upper_Threshold + s(sar11), data=largedata_noCO2_trans, method = "REML")

normKatE2_sar11 <- mgcv::gam(log10katE ~ s(salinity_dissolved_PSS.1978) + s(PAR_co) + s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + Size_Fraction_Upper_Threshold + s(sar11), data=largedata_noCO2_trans, method = "REML")

normKatE3_sar11 <- mgcv::gam(log10katE ~ s(PAR_co) + s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + Size_Fraction_Upper_Threshold + s(sar11), data=largedata_noCO2_trans, method = "REML")

normKatE4_sar11 <- mgcv::gam(log10katE ~ s(PAR_co) + s(NO3_NO2_co) + s(Iron_5m) + s(POC) + Size_Fraction_Upper_Threshold + s(sar11), data=largedata_noCO2_trans, method = "REML")

normKatE5_sar11 <- mgcv::gam(log10katE ~ s(NO3_NO2_co) + s(Iron_5m) + s(POC) + Size_Fraction_Upper_Threshold + s(sar11), data=largedata_noCO2_trans, method = "REML")

normKatE6_sar11 <- mgcv::gam(log10katE ~ s(Iron_5m) + s(POC) + Size_Fraction_Upper_Threshold + s(sar11), data=largedata_noCO2_trans, method = "REML")

normKatE7_sar11 <- mgcv::gam(log10katE ~ s(POC) + Size_Fraction_Upper_Threshold + s(sar11), data=largedata_noCO2_trans, method = "REML")

normKatE8_sar11 <- mgcv::gam(log10katE ~ s(POC) + Size_Fraction_Upper_Threshold, data=largedata_noCO2_trans, method = "REML")

normKatE9_sar11 <- mgcv::gam(log10katE ~ s(POC) + Size_Fraction_Upper_Threshold, data=largedata_noCO2_trans, method = "REML")

########

katE_large_size_fraction <- plot(ggeffects::ggpredict(normKatE9_sar11, terms = "Size_Fraction_Upper_Threshold"), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("Size fraction") + ylab("katE\ncopy number per genome")



######
################
normKatG1_large_HL <- mgcv::gam(log10katG ~ s(temperature_dissolved_deg.c) + s(salinity_dissolved_PSS.1978) + s(PAR_co) + s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + Size_Fraction_Upper_Threshold + s(proDarwin_umolC.kg) + s(pro_HLI) + s(pro_HLII), data=largedata_noCO2_trans, method = "REML")

normKatG2_large_HL <- mgcv::gam(log10katG ~ s(salinity_dissolved_PSS.1978) + s(PAR_co) + s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + Size_Fraction_Upper_Threshold + s(proDarwin_umolC.kg) + s(pro_HLI) + s(pro_HLII), data=largedata_noCO2_trans, method = "REML")

normKatG3_large_HL <- mgcv::gam(log10katG ~ s(salinity_dissolved_PSS.1978) + s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + Size_Fraction_Upper_Threshold + s(proDarwin_umolC.kg) + s(pro_HLI) + s(pro_HLII), data=largedata_noCO2_trans, method = "REML")

normKatG4_large_HL <- mgcv::gam(log10katG ~ s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + Size_Fraction_Upper_Threshold + s(proDarwin_umolC.kg) + s(pro_HLI) + s(pro_HLII), data=largedata_noCO2_trans, method = "REML")

normKatG5_large_HL <- mgcv::gam(log10katG ~ s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + Size_Fraction_Upper_Threshold + s(proDarwin_umolC.kg) + s(pro_HLI) + s(pro_HLII), data=largedata_noCO2_trans, method = "REML")

normKatG6_large_HL <- mgcv::gam(log10katG ~ s(Iron_5m) + s(POC) + Size_Fraction_Upper_Threshold + s(proDarwin_umolC.kg) + s(pro_HLI) + s(pro_HLII), data=largedata_noCO2_trans_POC, method = "REML")

normKatG7_large_HL <- mgcv::gam(log10katG ~ s(Iron_5m) + s(POC) + Size_Fraction_Upper_Threshold + s(pro_HLI) + s(pro_HLII), data=largedata_noCO2_trans_POC, method = "REML")

normKatG8_large_HL <- mgcv::gam(log10katG ~ s(Iron_5m) + s(POC) + Size_Fraction_Upper_Threshold + s(pro_HLII), data=largedata_noCO2_trans_POC, method = "REML")

normKatG9_large_HL <- mgcv::gam(log10katG ~ s(Iron_5m) + s(POC) + Size_Fraction_Upper_Threshold, data=largedata_noCO2_trans_POC, method = "REML")

normKatG10_large_HL <- mgcv::gam(log10katG ~ s(Iron_5m) + s(POC) + s(pro_HLII) + Size_Fraction_Upper_Threshold, data=largedata_noCO2_trans, method = "REML")

katG_large_size_fraction_pro <- plot(ggeffects::ggpredict(normKatG10_large_HL, terms = "Size_Fraction_Upper_Threshold"), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("Size fraction") + ylab("katG\ncopy number per genome")


######katE and pro
normKatE1_large_HL <- mgcv::gam(log10katE ~ s(temperature_dissolved_deg.c) + s(salinity_dissolved_PSS.1978) + s(PAR_co) + s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + Size_Fraction_Upper_Threshold + s(proDarwin_umolC.kg) + s(pro_HLI) + s(pro_HLII), data=largedata_noCO2_trans, method = "REML")

normKatE2_large_HL <- mgcv::gam(log10katE ~ s(salinity_dissolved_PSS.1978) + s(PAR_co) + s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + Size_Fraction_Upper_Threshold + s(proDarwin_umolC.kg) + s(pro_HLI) + s(pro_HLII), data=largedata_noCO2_trans, method = "REML")

normKatE3_large_HL <- mgcv::gam(log10katE ~ s(PAR_co) + s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + Size_Fraction_Upper_Threshold + s(proDarwin_umolC.kg) + s(pro_HLI) + s(pro_HLII), data=largedata_noCO2_trans, method = "REML")

normKatE4_large_HL <- mgcv::gam(log10katE ~ s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + Size_Fraction_Upper_Threshold + s(proDarwin_umolC.kg) + s(pro_HLI) + s(pro_HLII), data=largedata_noCO2_trans, method = "REML")

normKatE5_large_HL <- mgcv::gam(log10katE ~ s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + Size_Fraction_Upper_Threshold + s(proDarwin_umolC.kg) + s(pro_HLI) + s(pro_HLII), data=largedata_noCO2_trans, method = "REML")

normKatE6_large_HL <- mgcv::gam(log10katE ~ s(Iron_5m) + s(POC) + Size_Fraction_Upper_Threshold + s(proDarwin_umolC.kg) + s(pro_HLI) + s(pro_HLII), data=largedata_noCO2_trans, method = "REML")

normKatE7_large_HL <- mgcv::gam(log10katE ~ s(POC) + Size_Fraction_Upper_Threshold + s(proDarwin_umolC.kg) + s(pro_HLI) + s(pro_HLII), data=largedata_noCO2_trans, method = "REML")

normKatE8_large_HL <- mgcv::gam(log10katE ~ s(POC) + Size_Fraction_Upper_Threshold + s(pro_HLI) + s(pro_HLII), data=largedata_noCO2_trans, method = "REML")

normKatE9_large_HL <- mgcv::gam(log10katE ~ s(POC) + Size_Fraction_Upper_Threshold + s(pro_HLII), data=largedata_noCO2_trans, method = "REML")

normKatE10_large_HL <- mgcv::gam(log10katE ~ s(POC) + Size_Fraction_Upper_Threshold, data=largedata_noCO2_trans, method = "REML")

katE_large_size_fraction_pro <- plot(ggeffects::ggpredict(normKatE10_large_HL, terms = "Size_Fraction_Upper_Threshold"), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("Size fraction") + ylab("katE\ncopy number per genome")

############

KatG1_syn <- mgcv::gam(log10katG ~ s(temperature_dissolved_deg.c) + s(salinity_dissolved_PSS.1978) + s(PAR_co) + s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + Size_Fraction_Upper_Threshold + s(synDarwin_umolC.kg), data=largedata_noCO2_trans, method = "REML")

KatG2_syn <- mgcv::gam(log10katG ~ s(temperature_dissolved_deg.c) +  s(PAR_co) + s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + Size_Fraction_Upper_Threshold + s(synDarwin_umolC.kg), data=largedata_noCO2_trans, method = "REML")

KatG3_syn <- mgcv::gam(log10katG ~ s(temperature_dissolved_deg.c) +  s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + Size_Fraction_Upper_Threshold + s(synDarwin_umolC.kg), data=largedata_noCO2_trans, method = "REML")

KatG4_syn <- mgcv::gam(log10katG ~ s(temperature_dissolved_deg.c) +  s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(POC) + Size_Fraction_Upper_Threshold + s(synDarwin_umolC.kg), data=largedata_noCO2_trans, method = "REML")

KatG5_syn <- mgcv::gam(log10katG ~ s(temperature_dissolved_deg.c) +  s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(POC) + Size_Fraction_Upper_Threshold, data=largedata_noCO2_trans, method = "REML")


KatG6_syn <- mgcv::gam(log10katG ~ s(temperature_dissolved_deg.c) +  s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(POC) + Size_Fraction_Upper_Threshold, data=largedata_noCO2_trans_POC, method = "REML")

katG_large_size_fraction_syn  <- plot(ggeffects::ggpredict(KatG6_syn, terms = "Size_Fraction_Upper_Threshold"), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("Size fraction") + ylab("katG\ncopy number per genome")

KatE1_syn <- mgcv::gam(log10katE ~ s(temperature_dissolved_deg.c) + s(salinity_dissolved_PSS.1978) + s(PAR_co) + s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + Size_Fraction_Upper_Threshold + s(synDarwin_umolC.kg), data=largedata_noCO2_trans, method = "REML")

KatE2_syn <- mgcv::gam(log10katE ~ s(salinity_dissolved_PSS.1978) + s(PAR_co) + s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + Size_Fraction_Upper_Threshold + s(synDarwin_umolC.kg), data=largedata_noCO2_trans, method = "REML")

KatE3_syn <- mgcv::gam(log10katE ~ s(PAR_co) + s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + Size_Fraction_Upper_Threshold + s(synDarwin_umolC.kg), data=largedata_noCO2_trans, method = "REML")

KatE4_syn <- mgcv::gam(log10katE ~ s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + Size_Fraction_Upper_Threshold + s(synDarwin_umolC.kg), data=largedata_noCO2_trans, method = "REML")

KatE5_syn <- mgcv::gam(log10katE ~  s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + Size_Fraction_Upper_Threshold + s(synDarwin_umolC.kg), data=largedata_noCO2_trans, method = "REML")

KatE6_syn <- mgcv::gam(log10katE ~ s(Iron_5m) + s(POC) + Size_Fraction_Upper_Threshold + s(synDarwin_umolC.kg), data=largedata_noCO2_trans, method = "REML")

KatE7_syn <- mgcv::gam(log10katE ~ s(POC) + Size_Fraction_Upper_Threshold + s(synDarwin_umolC.kg), data=largedata_noCO2_trans, method = "REML")

KatE8_syn <- mgcv::gam(log10katE ~ s(POC) + Size_Fraction_Upper_Threshold, data=largedata_noCO2_trans_POC, method = "REML")

katE_large_size_fraction_syn  <- plot(ggeffects::ggpredict(KatE8_syn, terms = "Size_Fraction_Upper_Threshold"), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("Size fraction") + ylab("katE\ncopy number per genome")


####



model <- mgcv::gam(log10katE ~ s(POC, by = Size_Fraction_Upper_Threshold), data = largedata_noCO2_trans_POC, method = "REML")
gge <- ggeffects::ggpredict(model, c("POC", "Size_Fraction_Upper_Threshold"))

model  <- plot(ggeffects::ggpredict(model, terms = c("POC", "Size_Fraction_Upper_Threshold")), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("Size fraction") + ylab("katE\ncopy number per genome")


a <- ggplot(gge, aes(x, predicted)) +
  geom_ribbon(aes(ymin = conf.low, ymax = conf.high),
              color = NA, alpha = 0.2) +
   geom_point(aes(x=POC, y=log10katE, color=Size_Fraction_Upper_Threshold), data = largedata_noCO2_trans_POC) + facet_wrap(~Size_Fraction_Upper_Threshold)

#####running models but for subset of the size fraction

largedata_noCO2_trans_filt <- largedata_noCO2_trans %>%
  filter(size_fraction_lower_threshold != 0.8)

#environmental models
normKatG_large_enviro_new <- mgcv::gam(log10katG ~ s(temperature_dissolved_deg.c) + s(salinity_dissolved_PSS.1978) + s(PAR_co) + s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC), data=largedata_noCO2_trans_filt, method = "REML")

normKatG_large_enviro_new2 <- mgcv::gam(log10katG ~ s(temperature_dissolved_deg.c) + s(PAR_co) + s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC), data=largedata_noCO2_trans_filt, method = "REML")

normKatG_large_enviro_new3 <- mgcv::gam(log10katG ~ s(temperature_dissolved_deg.c) + s(PAR_co) + s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(POC), data=largedata_noCO2_trans_filt, method = "REML")

normKatG_large_enviro_new4 <- mgcv::gam(log10katG ~ s(temperature_dissolved_deg.c) + s(PAR_co) + s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(POC), data=largedata_noCO2_trans_filt, method = "REML")

largedata_noCO2_trans_filt_POC <- largedata_noCO2_trans_filt %>%
  arrange(desc(POC))

largedata_noCO2_trans_filt_POC <- largedata_noCO2_trans_filt_POC[-c(1:2),]

normKatG_large_enviro_new4 <- mgcv::gam(log10katG ~ s(temperature_dissolved_deg.c) + s(PAR_co) + s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(POC), data=largedata_noCO2_trans_filt_POC, method = "REML")

#plotting and checking outliers
plot(ggeffects::ggpredict(normKatG_large_enviro_new4, terms = c("POC")), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("POC") + ylab("katG\ncopy number per genome")

plot(ggeffects::ggpredict(normKatG_large_enviro_new4, terms = c("NO3_NO2_co")), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("Inorganic nitrogen concentration") + ylab("katG\ncopy number per genome")

plot(ggeffects::ggpredict(normKatG_large_enviro_new4, terms = c("phosphate_dissolved_umol.kg")), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("Inorganic phosphate concentration") + ylab("katG\ncopy number per genome")

############katE
normKatE_large_enviro_new <- mgcv::gam(log10katE ~ s(temperature_dissolved_deg.c) + s(salinity_dissolved_PSS.1978) + s(PAR_co) + s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC), data=largedata_noCO2_trans_filt, method = "REML")

normKatE_large_enviro_new2 <- mgcv::gam(log10katE ~  s(salinity_dissolved_PSS.1978) + s(PAR_co) + s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC), data=largedata_noCO2_trans_filt, method = "REML")

normKatE_large_enviro_new3 <- mgcv::gam(log10katE ~ s(PAR_co) + s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC), data=largedata_noCO2_trans_filt, method = "REML")

normKatE_large_enviro_new4 <- mgcv::gam(log10katE ~ s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC), data=largedata_noCO2_trans_filt, method = "REML")

normKatE_large_enviro_new5 <- mgcv::gam(log10katE ~ s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC), data=largedata_noCO2_trans_filt, method = "REML")

normKatE_large_enviro_new6 <- mgcv::gam(log10katE ~ s(phosphate_dissolved_umol.kg) + s(POC), data=largedata_noCO2_trans_filt, method = "REML")

normKatE_large_enviro_new7<- mgcv::gam(log10katE ~ s(phosphate_dissolved_umol.kg) + s(POC), data=largedata_noCO2_trans_filt_POC, method = "REML")


#plotting
plot(ggeffects::ggpredict(normKatE_large_enviro_new7, terms = c("phosphate_dissolved_umol.kg")), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("Inorganic phosphate concentration") + ylab("katE\ncopy number per genome")

plot(ggeffects::ggpredict(normKatE_large_enviro_new7, terms = c("POC")), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("POC") + ylab("katE\ncopy number per genome")

#sar11
normKatG_large_sar11_new <- mgcv::gam(log10katG ~ s(temperature_dissolved_deg.c) + s(salinity_dissolved_PSS.1978) + s(PAR_co) + s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + s(sar11), data=largedata_noCO2_trans_filt, method = "REML")

normKatG_large_sar11_new2 <- mgcv::gam(log10katG ~ s(temperature_dissolved_deg.c) + s(salinity_dissolved_PSS.1978) + s(PAR_co) + s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + s(sar11), data=largedata_noCO2_trans_filt, method = "REML")

normKatG_large_sar11_new3 <- mgcv::gam(log10katG ~ s(temperature_dissolved_deg.c) + s(salinity_dissolved_PSS.1978) + s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + s(sar11), data=largedata_noCO2_trans_filt, method = "REML")

normKatG_large_sar11_new4 <- mgcv::gam(log10katG ~ s(salinity_dissolved_PSS.1978) + s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + s(sar11), data=largedata_noCO2_trans_filt, method = "REML")

normKatG_large_sar11_new5 <- mgcv::gam(log10katG ~ s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + s(sar11), data=largedata_noCO2_trans_filt, method = "REML")

normKatG_large_sar11_new6 <- mgcv::gam(log10katG ~ s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + s(sar11), data=largedata_noCO2_trans_filt, method = "REML")

normKatG_large_sar11_new7 <- mgcv::gam(log10katG ~ s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(sar11), data=largedata_noCO2_trans_filt, method = "REML")

normKatG_large_sar11_new8 <- mgcv::gam(log10katG ~ s(phosphate_dissolved_umol.kg) + s(Iron_5m), data=largedata_noCO2_trans_filt, method = "REML")

#dropped sar11
normKatE_large_sar11_new <- mgcv::gam(log10katE ~ s(temperature_dissolved_deg.c) + s(salinity_dissolved_PSS.1978) + s(PAR_co) + s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + s(sar11), data=largedata_noCO2_trans_filt, method = "REML")

normKatE_large_sar11_new2 <- mgcv::gam(log10katE ~ s(salinity_dissolved_PSS.1978) + s(PAR_co) + s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + s(sar11), data=largedata_noCO2_trans_filt, method = "REML")

normKatE_large_sar11_new3 <- mgcv::gam(log10katE ~ s(PAR_co) + s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + s(sar11), data=largedata_noCO2_trans_filt, method = "REML")

normKatE_large_sar11_new4 <- mgcv::gam(log10katE ~ s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + s(sar11), data=largedata_noCO2_trans_filt, method = "REML")

normKatE_large_sar11_new5 <- mgcv::gam(log10katE ~ s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC), data=largedata_noCO2_trans_filt, method = "REML")

#dropped sar11

#prochlorococcus
normKatG_large_pro_new <- mgcv::gam(log10katG ~ s(temperature_dissolved_deg.c) + s(salinity_dissolved_PSS.1978) + s(PAR_co) + s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + s(proDarwin_umolC.kg), data=largedata_noCO2_trans_filt, method = "REML")

normKatG_large_pro_new2 <- mgcv::gam(log10katG ~ s(temperature_dissolved_deg.c) + s(PAR_co) + s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + s(proDarwin_umolC.kg), data=largedata_noCO2_trans_filt, method = "REML")

normKatG_large_pro_new3 <- mgcv::gam(log10katG ~ s(temperature_dissolved_deg.c) + s(PAR_co) + s(NO3_NO2_co) + s(Iron_5m) + s(POC) + s(proDarwin_umolC.kg), data=largedata_noCO2_trans_filt, method = "REML")

normKatG_large_pro_new4 <- mgcv::gam(log10katG ~ s(temperature_dissolved_deg.c) + s(PAR_co) + s(NO3_NO2_co) + s(POC) + s(proDarwin_umolC.kg), data=largedata_noCO2_trans_filt, method = "REML")

normKatG_large_pro_new5 <- mgcv::gam(log10katG ~ s(temperature_dissolved_deg.c) + s(PAR_co) + s(NO3_NO2_co) + s(POC) + s(proDarwin_umolC.kg), data=largedata_noCO2_trans_filt_POC, method = "REML")

#not significant
normKatE_large_pro_new <- mgcv::gam(log10katE ~ s(temperature_dissolved_deg.c) + s(salinity_dissolved_PSS.1978) + s(PAR_co) + s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + s(proDarwin_umolC.kg), data=largedata_noCO2_trans_filt, method = "REML")

normKatE_large_pro_new2 <- mgcv::gam(log10katE ~ s(salinity_dissolved_PSS.1978) + s(PAR_co) + s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + s(proDarwin_umolC.kg), data=largedata_noCO2_trans_filt, method = "REML")

normKatE_large_pro_new3 <- mgcv::gam(log10katE ~ s(PAR_co) + s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + s(proDarwin_umolC.kg), data=largedata_noCO2_trans_filt, method = "REML")

normKatE_large_pro_new4 <- mgcv::gam(log10katE ~ s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + s(proDarwin_umolC.kg), data=largedata_noCO2_trans_filt, method = "REML")

normKatE_large_pro_new5 <- mgcv::gam(log10katE ~ s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC), data=largedata_noCO2_trans_filt, method = "REML")

#syn
normKatG_large_syn_new <- mgcv::gam(log10katG ~ s(temperature_dissolved_deg.c) + s(salinity_dissolved_PSS.1978) + s(PAR_co) + s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + s(synDarwin_umolC.kg), data=largedata_noCO2_trans_filt, method = "REML")

normKatG_large_syn_new2 <- mgcv::gam(log10katG ~ s(temperature_dissolved_deg.c) + s(PAR_co) + s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + s(synDarwin_umolC.kg), data=largedata_noCO2_trans_filt, method = "REML")

normKatG_large_syn_new3 <- mgcv::gam(log10katG ~ s(temperature_dissolved_deg.c) + s(PAR_co) + s(NO3_NO2_co) + s(Iron_5m) + s(POC) + s(synDarwin_umolC.kg), data=largedata_noCO2_trans_filt, method = "REML")

normKatG_large_syn_new4 <- mgcv::gam(log10katG ~ s(temperature_dissolved_deg.c) + s(PAR_co) + s(NO3_NO2_co) + s(Iron_5m) + s(POC) + s(synDarwin_umolC.kg), data=largedata_noCO2_trans_filt, method = "REML")

normKatG_large_syn_new5 <- mgcv::gam(log10katG ~ s(temperature_dissolved_deg.c) + s(PAR_co) + s(NO3_NO2_co) + s(POC) + s(synDarwin_umolC.kg), data=largedata_noCO2_trans_filt, method = "REML")

normKatG_large_syn_new6 <- mgcv::gam(log10katG ~ s(temperature_dissolved_deg.c) + s(PAR_co) + s(NO3_NO2_co) + s(POC) + s(synDarwin_umolC.kg), data=largedata_noCO2_trans_filt, method = "REML")

######
normKatE_large_syn_new <- mgcv::gam(log10katE ~ s(temperature_dissolved_deg.c) + s(salinity_dissolved_PSS.1978) + s(PAR_co) + s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + s(synDarwin_umolC.kg), data=largedata_noCO2_trans_filt, method = "REML")

normKatE_large_syn_new2 <- mgcv::gam(log10katE ~ s(salinity_dissolved_PSS.1978) + s(PAR_co) + s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + s(synDarwin_umolC.kg), data=largedata_noCO2_trans_filt, method = "REML")

normKatE_large_syn_new3 <- mgcv::gam(log10katE ~ s(PAR_co) + s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC) + s(synDarwin_umolC.kg), data=largedata_noCO2_trans_filt, method = "REML")

normKatE_large_syn_new4 <- mgcv::gam(log10katE ~ s(PAR_co) + s(NO3_NO2_co) + s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC), data=largedata_noCO2_trans_filt, method = "REML")

#making figure with all of the large size fractions
a <- plot(ggeffects::ggpredict(normKatE_large_enviro_new7, terms = c("phosphate_dissolved_umol.kg")), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("Inorganic phosphate concentration") + ylab("katE\ncopy number per genome")

b <- plot(ggeffects::ggpredict(normKatE_large_enviro_new7, terms = c("POC")), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("POC") + ylab("katE\ncopy number per genome")

c <- plot(ggeffects::ggpredict(normKatG_large_enviro_new4, terms = c("POC")), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("POC") + ylab("katG\ncopy number per genome")

d <- plot(ggeffects::ggpredict(normKatG_large_enviro_new4, terms = c("NO3_NO2_co")), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("Inorganic nitrogen concentration") + ylab("katG\ncopy number per genome")

e <- plot(ggeffects::ggpredict(normKatG_large_enviro_new4, terms = c("phosphate_dissolved_umol.kg")), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("Inorganic phosphate concentration") + ylab("katG\ncopy number per genome")

grid.arrange(a, b, c, d, e, ncol = 2)

write.csv(largedata_noCO2_trans_filt, "largedata_noCO2_trans_filt.csv")


###plotting both small and large size fraction using other method
library(ggeffects)

#small size fraction
#need to transform the katG copy number per genome to be log10 so nothing gets back transformed

catmodels_notArctic_new_clean_SRF_extra_Sal_POC_log <- catmodels_notArctic_new_clean_SRF_extra_Sal_POC %>%
  mutate(log10normKatG = log10(normKatG))
normKatG6_enviro <- mgcv::gam(log10normKatG ~ s(Temp_co) + s(Sal_co) + s(Iron_5m) + POC, data=catmodels_notArctic_new_clean_SRF_extra_Sal_POC_log, method = "REML")


normKatG8_pro <- mgcv::gam(log10(normKatG) ~ s(Temp_co) + s(Sal_co) + PO4_co + s(Iron_5m) + POC + proDarwin_umolC.kg, data=catmodels_notArctic_new_clean_SRF_extra_Sal_POC_log, method = "REML")

normKatG7_sar11 <- mgcv::gam(log10normKatG ~ Temp_co + s(Sal_co) + s(Iron_5m) + s(POC) + s(sar11), data=catmodels_notArctic_new_clean_SRF_extra_Sal_POC_log, method = "REML")

# gge_small_katG_POC <- ggeffects::ggpredict(normKatG6_enviro, "POC")
# gge_small_katG_iron <- ggeffects::ggpredict(normKatG6_enviro, "Iron_5m")
# 
# gge_small_katG_pro <- ggeffects::ggpredict(normKatG8_pro, "proDarwin_umolC.kg")
# gge_small_katG_sar11 <- ggeffects::ggpredict(normKatG7_sar11, "sar11")
# 
# 
# a <- ggplot(gge_small_katG_POC, aes(x, predicted)) +
#   geom_ribbon(aes(ymin = conf.low, ymax = conf.high),
#               color = NA, alpha = 0.2) +
#   geom_line() + geom_point(aes(x=POC, y=normKatG), data = catmodels_notArctic_new_clean_SRF_extra_Sal_POC) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
# panel.grid.major = element_blank(), 
#         panel.grid.minor = element_blank(),
#         panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("POC") + ylab("katG\ncopy number per genome")

a <- plot(ggeffects::ggpredict(normKatG6_enviro, terms = c("POC")), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("POC") + ylab("log10 katG\ncopy number per genome")

b <- plot(ggeffects::ggpredict(normKatG6_enviro, terms = c("Iron_5m")), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("Iron concentration at 5 meters") + ylab("log10 katG\ncopy number per genome")

c <- plot(ggeffects::ggpredict(normKatG8_pro, terms = c("POC")), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("Prochlorococcus biomass") + ylab("log10 katG\ncopy number per genome")


d <- plot(ggeffects::ggpredict(normKatG7_sar11, terms = c("sar11")), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("Sar11 biomass") + ylab("log10 katG\ncopy number per genome")


grid.arrange(a, b, c, d)
#large
largedata_noCO2_trans_filtered <- largedata_noCO2_trans_filtered %>%
  arrange(desc(POC))
largedata_noCO2_trans_filtered_POC <- largedata_noCO2_trans_filtered[-c(1:3),]

normKatE_large_enviro_new7 <- mgcv::gam(log10katE ~ s(phosphate_dissolved_umol.kg) + s(POC), data=largedata_noCO2_trans_filt_POC, method = "REML")

normKatG4_large_enviro_filt <- mgcv::gam(log10katG ~ s(temperature_dissolved_deg.c) + s(NO3_NO2_co) + s(PAR_co) + s(phosphate_dissolved_umol.kg) + s(POC), data=largedata_noCO2_trans_filtered_POC, method = "REML")

# normKatE6_large_enviro_filt <- mgcv::gam(log10katE ~ s(phosphate_dissolved_umol.kg) + s(Iron_5m) + s(POC), data=largedata_noCO2_trans_filtered, method = "REML")

# gge_large_katE_POC <- ggeffects::ggpredict(normKatE6_large_enviro_filt, "POC")
# 
# gge_small_katG_PO4 <- ggeffects::ggpredict(normKatG5_large_enviro, "phosphate_dissolved_umol.kg")


e <- plot(ggeffects::ggpredict(normKatG4_large_enviro_filt, terms = c("phosphate_dissolved_umol.kg")), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("Inorganic phosphate concentration") + ylab("log10 katG\ncopy number per genome")

f <- plot(ggeffects::ggpredict(normKatG4_large_enviro_filt, terms = c("POC")), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("POC") + ylab("log10 katG\ncopy number per genome")

g <- plot(ggeffects::ggpredict(normKatE_large_enviro_new7, terms = c("phosphate_dissolved_umol.kg")), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("Inorganic phosphate concentration") + ylab("log10 katE\ncopy number per genome")

h <- plot(ggeffects::ggpredict(normKatE_large_enviro_new7, terms = c("POC")), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("POC") + ylab("log10 katE\ncopy number per genome")

grid.arrange(e, f, g, h)
```

```{r}
#ggplot map

#load packages
library(ggplot2)
library(maps)



#use this dataset

catmodels_modeleddata_cleaner <- read.csv("~/Desktop/Dyhrman Lab/Desktop_DyhrmanLab/catmodels_modeleddata_cleaner.csv")

catmodels_modeleddata_Arctic_SRF <- catmodels_modeleddata_cleaner %>%
  filter(latitude.x >= 50, depth.x == "SRF")

catmodels_modeleddata_notArctic_SRF <- catmodels_modeleddata_cleaner %>%
  filter(latitude.x < 50, depth.x == "SRF", Region != "[SO] Southern Ocean (MRGID:1907)") 

catmodels_notArctic_extra <- catmodels_modeleddata_notArctic_SRF %>%
  select(-NO3_NO2_modeled) %>%
  drop_na()

#dropping NA for arctic an not arctic
#Arctic
# catmodels_notArctic_stations <- catmodels_notArctic_new %>%
#   select(sample_ID, station)
  

modeleddata_large_clean_otherenviro2_station <- modeleddata_large_clean_otherenviro2 %>%
  dplyr::rename("sample_ID" = "Sample_Name",
                "station" = "Station (TARA_station#, registered at ...).y") %>%
  select(sample_ID, station)
largedata_noCO2_trans_renamed <- largedata_noCO2_trans %>%
 dplyr::rename("sample_ID" = "Sample_Name")

large_set <- left_join(largedata_noCO2_trans_renamed, modeleddata_large_clean_otherenviro2_station, 
          by = "sample_ID")

small_set <- left_join(catmodels_notArctic_new_clean_SRF_extra, catmodels_notArctic_stations, 
          by = "sample_ID")

#now I can make 6 unique maps
largedata_noCO2_trans_filt <- read.csv("~/Desktop/Dyhrman Lab/Desktop_DyhrmanLab/largedata_noCO2_trans_filt.csv")

catalases_large_trans_enviro_bystation <- largedata_noCO2_trans_filt %>%
  pivot_longer(5:7, names_to = "catalase", values_to = "copynum")

catalases_large_trans_enviro_bystation_colnames <- catalases_large_trans_enviro_bystation %>%
  select(station_co, catalase, copynum, latitude, longitude, Region) %>%
  dplyr::rename("station" = "station_co",
               "reads" = "copynum") %>%
  group_by(station, catalase) %>%
   dplyr::reframe(reads = (mean(reads)),
            latitude = latitude,
            longitude = longitude,
            Region = Region) 


#now I'll want to separate by each gene and bind to other dataset
catalases_large_trans_enviro_bystation_catmn <- catalases_large_trans_enviro_bystation_colnames %>%
  filter(catalase == "catmn") %>%
  distinct(station, .keep_all = TRUE)

catalases_large_trans_enviro_bystation_katE <- catalases_large_trans_enviro_bystation_colnames %>%
  filter(catalase == "katE") %>%
  distinct(station, .keep_all = TRUE)

catalases_large_trans_enviro_bystation_katG <- catalases_large_trans_enviro_bystation_colnames %>%
  filter(catalase == "katG") %>%
 distinct(station, .keep_all = TRUE)


###############working with small size fraction data so it's in the same format as large data
catmodels_modeleddata_notArctic_SRF_newformat <- catmodels_modeleddata_notArctic_SRF %>%
  select(sample_ID, station, latitude.x, longitude.x, normKatG, normKatE, normCatMn, Region)

catmodels_modeleddata_notArctic_SRF_newformat_long <- catmodels_modeleddata_notArctic_SRF_newformat %>%
  pivot_longer(5:7, names_to = "catalase", values_to = "reads_trans")

catalases_small_trans_enviro_bystation <- catmodels_modeleddata_notArctic_SRF_newformat_long %>%
  dplyr::select(station, catalase, reads_trans, latitude.x, longitude.x, Region) %>%
  group_by(station, catalase) %>%
   dplyr::reframe(reads = (mean(reads_trans)),
            latitude = latitude.x,
            longitude = longitude.x,
            Region = Region)

#now I'll want to separate by each gene and bind to other dataset

catalases_small_trans_enviro_bystation_catmn <- catalases_small_trans_enviro_bystation %>%
  filter(catalase == "normCatMn") 

catalases_small_trans_enviro_bystation_katE <- catalases_small_trans_enviro_bystation %>%
  filter(catalase == "normKatE")

catalases_small_trans_enviro_bystation_katG <- catalases_small_trans_enviro_bystation %>%
  filter(catalase == "normKatG") 

#binding each gene to each other

all_manganese_catalases <- rbind(catalases_small_trans_enviro_bystation_catmn, catalases_large_trans_enviro_bystation_catmn) %>% mutate(size_fraction = case_when(catalase == "normCatMn" ~ "small",
       catalase == "catmn" ~ "large")) %>%
  filter(latitude > -50)

all_katE <- rbind(catalases_small_trans_enviro_bystation_katE, catalases_large_trans_enviro_bystation_katE) %>% 
  mutate(size_fraction = case_when(catalase == "normKatE" ~ "small",
       catalase == "katE" ~ "large"))
      
all_katG <- rbind(catalases_small_trans_enviro_bystation_katG, catalases_large_trans_enviro_bystation_katG) %>% 
  mutate(size_fraction = case_when(catalase == "normKatG" ~ "small",
       catalase == "katG" ~ "large"))
       
#making the maps

library(maps)
mapWorld <- map_data("world")
# make a base map
map4 <- ggplot() + geom_polygon(aes(x = long, y = lat, group = group),data = mapWorld, fill = "grey", colour = "grey") + scale_x_continuous(limits = c(-200, 200), breaks = c(-100, 0, 100), labels = c("100", "0", "-100")) + scale_y_continuous(limits = c(-100, 100), breaks = seq(-50, 50, by = 50)) 

########
mapKatG <- map4 + geom_point(aes(x = longitude, y = latitude, size = reads)),data = all_katG, alpha = 0.6, color = "blue")  + theme_bw() + theme(panel.ontop = FALSE) + theme(strip.text = element_blank(), panel.border = element_blank()) + guides(size=guide_legend(title="katG copy number per genome")) + xlab("Longitude") + ylab("Latitude") + theme(text=element_text(size=12)) + facet_wrap(~size_fraction)


# add projections and color
mapKatG2 <- mapKatG + coord_map("mollweide", xlim=c(-200, 200), ylim = c(-100, 100)) 

###repeat for KatE
mapKatE <- map4 + geom_point(aes(x = longitude, y = latitude, size = reads) ,data = all_katE, alpha = 0.6, color = "darkgreen")  + theme_bw() + theme(panel.ontop = FALSE) + theme(strip.text = element_blank(), panel.border = element_blank()) + guides(size=guide_legend(title="katE copy number per genome")) + xlab("Longitude") + ylab("Latitude") + theme(text=element_text(size=12)) + facet_wrap(~size_fraction)

# add projections and color
mapKatE2 <- mapKatE + coord_map("mollweide", xlim=c(-200, 200), ylim = c(-100, 100)) 

######
mapCatMn <- map4 + geom_point(aes(x = longitude, y = latitude, size = (reads)),data = all_manganese_catalases, alpha = 0.6, color = "red")  + theme_bw() + theme(panel.ontop = FALSE) + theme(strip.text = element_blank(), panel.border = element_blank()) + guides(size=guide_legend(title="katM copy number per genome")) + xlab("Longitude") + ylab("Latitude") + theme(text=element_text(size=12)) + facet_wrap(~size_fraction)


# add projections and color
mapCatMn2 <- mapCatMn + coord_map("mollweide", xlim=c(-200, 200), ylim = c(-100, 100)) 

# mapWorld <- map_data("world")
# 
# piemap_plot <- ggplot(data = mapdata_short, aes(x = longitude, y = latitude)) + geom_polygon(aes(x = long, y = lat, group = group),data = mapWorld, fill = "grey", colour = "grey") + 
#   geom_pie_glyph(aes(radius = totalcat), slices = c("normKatG", "normKatE", "normCatMn"))+
#   theme_bw() + theme(panel.ontop = FALSE) + theme(strip.text = element_blank(),
#                                                                               panel.border = element_blank()) + scale_fill_manual(values=c("blue", "green", "red"))
# 
# piemap_plot_projected <- piemap_plot + coord_map("mollweide", xlim=c(-200,200))

library(gridExtra)
grid.arrange(mapKatG2, mapKatE2, mapCatMn2, ncol=1)




#####
mapdata_summary_byregion <- mapdata_short %>%
  group_by(Region) %>%
  summarise(mean_KatE = mean(normKatE),
            mean_KatG = mean(normKatG),
            mean_CatMn = mean(normCatMn),
            sd_KatE = sd(normKatE),
            sd_KatG = sd(normKatG),
            sd_CatMn = sd(normCatMn),
            var_KatE = sd(normKatE)/mean(normKatE)*100,
            var_KatG = sd(normKatG)/mean(normKatG)*100,
            var_CatMn = sd(normCatMn)/mean(normCatMn)*100,
            n = n())

mapdata_summary_everything <- mapdata_short %>%
  summarise(mean_KatE = mean(normKatE),
            mean_KatG = mean(normKatG),
            mean_CatMn = mean(normCatMn),
            sd_KatE = sd(normKatE),
            sd_KatG = sd(normKatG),
            sd_CatMn = sd(normCatMn),
            var_KatE = sd(normKatE)/mean(normKatE)*100,
            var_KatG = sd(normKatG)/mean(normKatG)*100,
            var_CatMn = sd(normCatMn)/mean(normCatMn)*100,
            n = n())


#performing statistical tests to determine if means of each gene of interest are significantly different between regions
#ask if variables are normally distributed
library(ggpubr)
library(rstatix)
model_katG <- lm(normKatG ~ Region, data = mapdata_short)

#not normally distributed
shapiro_test(residuals(model_katG))
#p < 0.05 therefore the distribution is not normal and we have to use the kruskal test
kruskal.test(normKatG ~ Region, data = mapdata_short)

#finding out which groups
pairwise.wilcox.test(mapdata_short$normKatG, mapdata_short$Region,
                 p.adjust.method = "BH")

#katE
model_katE <- lm(normKatE ~ Region, data = mapdata_short)


#not normally distributed
shapiro_test(residuals(model_katE))

kruskal.test(normKatE ~ Region, data = mapdata_short)

#finding out which groups
pairwise.wilcox.test(mapdata_short$normKatE, mapdata_short$Region,
                 p.adjust.method = "BH")

#repeat for manganese catalase
model_CatMn <- lm(normCatMn ~ Region, data = mapdata_short)


#not normally distributed
shapiro_test(residuals(model_CatMn))

kruskal.test(normCatMn ~ Region, data = mapdata_short)

#finding out which groups
pairwise.wilcox.test(mapdata_short$normCatMn, mapdata_short$Region,
                 p.adjust.method = "BH")

####comparing between regions for genes of interest
mapdata_long <- mapdata_short %>%
pivot_longer(2:4, names_to = "gene", values_to = "copynum")

#NAO
mapdata_NAO <- mapdata_long %>%
filter(Region == "[NAO] North Atlantic Ocean (MRGID:1912)")


model_NAO <- lm(copynum ~ gene, data = mapdata_NAO)


#not normally distributed
shapiro_test(residuals(model_NAO))

kruskal.test(copynum ~ gene, data = mapdata_NAO)

#finding out which groups
pairwise.wilcox.test(mapdata_NAO$copynum, mapdata_NAO$gene,
                 p.adjust.method = "BH")

#MS
mapdata_MS <- mapdata_long %>%
filter(Region == "[MS] Mediterranean Sea (MRGID:1905)")

model_MS <- lm(copynum ~ gene, data = mapdata_MS)


#not normally distributed
shapiro_test(residuals(model_MS))

aov(copynum ~ gene, data = mapdata_MS) %>%
  tukey_hsd()

#finding out which groups
pairwise.wilcox.test(mapdata_NAO$copynum, mapdata_NAO$gene,
                 p.adjust.method = "BH")


################

mapdata_RS <- mapdata_long %>%
filter(Region == "[RS] Red Sea (MRGID:4264)")

model_RS <- lm(copynum ~ gene, data = mapdata_RS)

shapiro_test(residuals(model_RS))

kruskal.test(copynum ~ gene, data = mapdata_RS)

#finding out which groups
pairwise.wilcox.test(mapdata_RS$copynum, mapdata_RS$gene,
                 p.adjust.method = "BH")

##
mapdata_IO <- mapdata_long %>%
filter(Region == "[IO] Indian Ocean (MRGID:1904)")

model_IO <- lm(copynum ~ gene, data = mapdata_IO)

shapiro_test(residuals(model_IO))

kruskal.test(copynum ~ gene, data = mapdata_IO)

pairwise.wilcox.test(mapdata_IO$copynum, mapdata_IO$gene,
                 p.adjust.method = "BH")

#######
mapdata_SAO <- mapdata_long %>%
filter(Region == "[SAO] South Atlantic Ocean (MRGID:1914)")

model_SAO <- lm(copynum ~ gene, data = mapdata_SAO)

shapiro_test(residuals(model_SAO))

kruskal.test(copynum ~ gene, data = mapdata_SAO)

pairwise.wilcox.test(mapdata_SAO$copynum, mapdata_SAO$gene,
                 p.adjust.method = "BH")


###
mapdata_SPO <- mapdata_long %>%
filter(Region == "[SPO] South Pacific Ocean (MRGID:1910)")

model_SPO <- lm(copynum ~ gene, data = mapdata_SPO)

shapiro_test(residuals(model_SPO))

kruskal.test(copynum ~ gene, data = mapdata_SPO)

pairwise.wilcox.test(mapdata_SPO$copynum, mapdata_SPO$gene,
                 p.adjust.method = "BH")


######
mapdata_AO <- mapdata_long %>%
filter(Region == "[AO] Arctic Ocean (MRGID:1906)")

model_SPO <- lm(copynum ~ gene, data = mapdata_AO)

shapiro_test(residuals(model_SPO))

kruskal.test(copynum ~ gene, data = mapdata_AO)

pairwise.wilcox.test(mapdata_AO$copynum, mapdata_AO$gene,
                 p.adjust.method = "BH")


#statistical test for coefficient of variation across all Regions
library(cvequality)

genes_cv_test <- 
with(mapdata_long, 
     asymptotic_test(copynum,
                     gene))
```

```{r}
#making a table
library(kableExtra)

Catmodeltable <- read_excel("Catmodeltable.xlsx")
modeltable <- kable(Catmodeltable, "latex",
  booktabs = TRUE) %>%
  kable_styling(latex_options = c("striped", "scale_down")) %>%
  kable_styling(font_size = 20) %>%
  as_image()

kable(mtcars, "latex", booktabs = T) %>%
  kable_styling(latex_options = c("striped", "scale_down")) %>%
  row_spec(1, color = "red") %>%
  as_image()


#############################

#Making a map with copy numbers: separate DCM and SRF
library(leaflet)

###making map with normalized new data
library(leaflet)
mapdata_allsamples <- catmodels_norm %>%
  filter(depth == "SRF") %>%
  group_by(station) %>%
  reframe(KatGst = sum(normKatG),
            KatEst = sum(normKatE),
            CatMnst = sum(normCatMn),
          Rubst = sum(normrub),
            latitude = latitude,
            longitude = longitude,
            samplepres = 0.1) %>%
  unique()

#making a map for KatG
mapdata_long <- mapdata_allsamples %>%
  pivot_longer(2:5, names_to = "gene", values_to = "copynum")

KatG <- mapdata_long %>%
  filter(gene == "KatGst")

my_map <- leaflet(KatG) %>% 
          addTiles() %>% 
  addProviderTiles("Esri") %>%
          addCircles(lng = KatG$longitude, lat = KatG$latitude,
                 weight = 1, 
                 radius = sqrt(KatG$copynum)*2e5)
             
#making a map for KatE
KatE <- mapdata_long %>%
  filter(gene == "KatEst")

library(leaflegend)

my_map <- leaflet(KatE) %>% 
          addTiles() %>% 
  addProviderTiles("Esri") %>%
          addCircles(lng = KatE$longitude, lat = KatE$latitude,
                 weight = 1, 
                 radius = sqrt(KatE$copynum)*2000000) %>%
  addLegendSize(
    values = KatE$copynum,
   
    color = 'darkblue',
    fillColor = 'darkblue',
    baseSize = 1,
    opacity = 1,
    title = 'Copy number',
    shape = 'circle',
    orientation = 'horizontal',
    position = 'bottomright',
    breaks = 5)
             
#making a map for CatMn

CatMn <- mapdata_long %>%
  filter(gene == "CatMnst")


my_map <- leaflet(CatMn) %>% 
          addTiles() %>% 
  addProviderTiles("Esri") %>%
          addCircles(lng = CatMn$longitude, lat = CatMn$latitude,
                 weight = 1, 
                 radius = sqrt(CatMn$copynum)*20000000)

#map for rubisco
Rub <- mapdata_long %>%
  filter(gene == "Rubst")


my_map <- leaflet(Rub) %>% 
          addTiles() %>% 
  addProviderTiles("Esri") %>%
          addCircles(lng = Rub$longitude, lat = Rub$latitude,
                 weight = 1, 
                 radius = sqrt(Rub$copynum)*3e5)
             
```

```{r}
#new MAGs
#preparing dataset
genomes_summary <- read_excel("~/Desktop/Tara Oceans hmm/genomes-summary.xlsx") 
genomes_size_depth <- read_excel("~/Desktop/Tara Oceans hmm/genomes_size_depth.xlsx")
genomes_size_depth <- janitor::row_to_names(genomes_size_depth, 1) 


catalase_genomes <- read.csv("~/Desktop/Tara Oceans hmm/catalase_genomes.csv")
###############################################################################
#filtering genomes down

genomes_summary_filtered <- genomes_summary %>%
  dplyr::rename("size.fraction" = "size fraction",
         "genome" = "Genome",
         "Completeness" = "CheckM Completeness",
         "Contamination" = "CheckM Contamination") %>%
  mutate(completeness_minus_redundancy = Completeness-Contamination) %>%
  dplyr::filter(grepl('TARA', genome),
                size.fraction == "0.22-3" | size.fraction == "0.45-0.8" | size.fraction == "0.22-0.45" | size.fraction == "0.22-1.6",
                depth <= 10,
                completeness_minus_redundancy >= 70) 

###################################
# genomes_summary_filtered_deep_small <- genomes_summary %>%
#   dplyr::rename("size.fraction" = "size fraction",
#          "genome" = "Genome",
#          "Completeness" = "CheckM Completeness",
#          "Contamination" = "CheckM Contamination") %>%
#   mutate(completeness_minus_redundancy = Completeness-Contamination) %>%
#   dplyr::filter(size.fraction == "0.22-3" | size.fraction == "0.45-0.8" | size.fraction == "0.22-0.45" | size.fraction == "0.22-1.6" | size.fraction == "0.1-0.22" | size.fraction == "0.2-0.8",
#                 depth <= 2100 | depth >= 4100,
#                 completeness_minus_redundancy >= 70) 
# 
# ###################
# 
# genomes_summary_filtered_deep_large <- genomes_summary %>%
#   dplyr::rename("size.fraction" = "size fraction",
#          "genome" = "Genome",
#          "Completeness" = "CheckM Completeness",
#          "Contamination" = "CheckM Contamination") %>%
#   filter(size.fraction == "0.8-20") %>%
#   mutate(completeness_minus_redundancy = Completeness-Contamination) %>%
#                filter(completeness_minus_redundancy >= 70) 
# 
# 
# 
# chisq.test(michelledata_copies_alldepths_updated_cladetest$KatG_char, michelledata_copies_alldepths_updated_cladetest$clade)


##############################################################################################################
KatG_filtered <- read.delim("~/Desktop/Tara Oceans hmm/KatG/KatG_filtered.txt")

KatG_TARA <- KatG_filtered

KatG_TARA_copies <- KatG_TARA %>%
  group_by(genome) %>%
  dplyr::count() %>%
  dplyr::rename("n_KatG" = "n")

KatE_filtered <- read.delim("~/Desktop/Tara Oceans hmm/KatE/KatE_filtered.txt")

KatE_TARA <- KatE_filtered

KatE_TARA_copies <- KatE_TARA %>%
  group_by(genome) %>%
  dplyr::count() %>%
  dplyr::rename("n_KatE" = "n")

CatMn_filtered <- read.delim("~/Desktop/Tara Oceans hmm/Manganese_catalase/CatMn_filtered.txt")

CatMn_TARA <- CatMn_filtered

CatMn_TARA_copies <- CatMn_TARA %>%
  group_by(genome) %>%
  dplyr::count() %>%
  dplyr::rename("n_CatMn" = "n")

#merging
KatGE_copies <- merge(KatG_TARA_copies,
                  KatE_TARA_copies,
                  by = "genome",
                  all.x = TRUE,
                  all.y = TRUE)

KatGEMn_copies <- merge(KatGE_copies,
                  CatMn_TARA_copies,
                  by = "genome",
                  all.x = TRUE,
                  all.y = TRUE)

KatGEMn_copies[is.na(KatGEMn_copies)] <- 0


catalases <- left_join(genomes_summary_filtered,
                       KatGEMn_copies,
                       by = "genome")

# catalases_deep_small <- left_join(genomes_summary_filtered_deep_small,
#                        KatGEMn_copies,
#                        by = "genome")
# 
# 
# catalases_deep_large <- left_join(genomes_summary_filtered_deep_large,
#                        KatGEMn_copies,
#                        by = "genome")

catalases$n_KatG[is.na(catalases$n_KatG)] <- 0
catalases$n_KatE[is.na(catalases$n_KatE)] <- 0
catalases$n_CatMn[is.na(catalases$n_CatMn)] <- 0

# catalases_deep_small$n_KatG[is.na(catalases_deep_small$n_KatG)] <- 0
# catalases_deep_small$n_KatE[is.na(catalases_deep_small$n_KatE)] <- 0
# catalases_deep_small$n_CatMn[is.na(catalases_deep_small$n_CatMn)] <- 0
# 
# catalases_deep_large$n_KatG[is.na(catalases_deep_large$n_KatG)] <- 0
# catalases_deep_large$n_KatE[is.na(catalases_deep_large$n_KatE)] <- 0
# catalases_deep_large$n_CatMn[is.na(catalases_deep_large$n_CatMn)] <- 0
# 
# #checking for significance
# deep_large <- catalases_deep_large %>%
#   transform(KatG_char = as.character(n_KatG),
#   KatE_char = as.character(n_KatE)) %>%
#   select(genome, size.fraction, n_KatG, n_KatE, KatG_char, KatE_char)
# 
# deep_small <- catalases_deep_small %>%
#   transform(KatG_char = as.character(n_KatG),
#   KatE_char = as.character(n_KatE)) %>%
#   select(genome, size.fraction, n_KatG, n_KatE, KatG_char, KatE_char)
# 
# malaspina_data_deep <- rbind(deep_large, deep_small) %>%
#   mutate(size_fraction = case_when(size.fraction == "0.8-20" ~ "large",
#                            size.fraction != "0.8-20" ~ "small"))
# 
# malaspina_data_deep_summary <- malaspina_data_deep %>%
#   group_by(size_fraction) %>%
#   summarise(katG = mean(n_KatG),
#             katE = mean(n_KatE))
# 
# fishers.test(malaspina_data_deep$KatG_char, malaspina_data_deep$size_fraction)
# 
# ggplot(malaspina_data_deep, aes(x=size_fraction, y=n_KatE)) + 
#     geom_boxplot(fill="grey") + theme_bw() + xlab("gene") + ylab("katE copy number per genome") + coord_flip() + theme(text = element_text(size = 12), panel.background = element_blank(),
# panel.grid.major = element_blank(), 
#         panel.grid.minor = element_blank(),
#         panel.border = element_rect(colour = "black", fill=NA, linewidth=1))
# 
# ggplot(malaspina_data_deep, aes(x=size_fraction, y=n_KatG)) + 
#     geom_boxplot(fill="grey") + theme_bw() + xlab("gene") + ylab("katG copy number per genome") + coord_flip() + theme(text = element_text(size = 12), panel.background = element_blank(),
# panel.grid.major = element_blank(), 
#         panel.grid.minor = element_blank(),
#         panel.border = element_rect(colour = "black", fill=NA, linewidth=1))
# 
# #counting up the number of MAGs carrying each type of catalase
# #counting the orders
# 



catalases_MAG_count_byOrder <- catalases_MAG_count_order %>%
  filter(nMAGswithorwithoutcat >= 20)
#counts for percentages 
MAGs_counts_for_percentages_order <- catalases %>%
  dplyr::rename("Genome_size" = "Genome size") %>%
  select(Order, Family, Genus, Genome_size, n_KatG, n_KatE, n_CatMn) %>%
  
  mutate(nKatG = case_when(n_KatG >= 1 ~ 1,
                           n_KatG == 0 ~ 0),
         nKatE = case_when(n_KatE >= 1 ~ 1,
                           n_KatE == 0 ~ 0),
         nCatMn = case_when(n_CatMn >= 1 ~ 1,
                           n_CatMn == 0 ~ 0),
         genepres = case_when(n_KatG == "0" & n_KatE == "0" & n_CatMn == "0" ~ "none",
           n_KatG == "0" & n_KatE > 0 & n_CatMn == "0" ~ "KatE",
                               n_KatG > 0 & n_KatE == "0" & n_CatMn == "0" ~ "KatG",
                            n_KatG == "0" & n_KatE == "0" & n_CatMn > 0 ~ "CatMn",
                            n_KatG > 0 & n_KatE == "0" & n_CatMn > 0 ~ "KatGCatMn",
                              n_KatG > 0 & n_KatE > 0 & n_CatMn == "0"  ~ "KatGKatE",
                            n_KatG == "0" & n_KatE > 0 & n_CatMn > 0 ~ "KatECatMn",
                            n_KatG > 0 & n_KatE > 0 & n_CatMn > 0 ~ "allcats")) %>%
  group_by(genepres) %>%
  tally() 

MAGs_counts_for_filteringtaxa_order <- catalases %>%
  dplyr::rename("Genome_size" = "Genome size") %>%
  mutate(nKatG = case_when(n_KatG >= 1 ~ 1,
                           n_KatG == 0 ~ 0),
         nKatE = case_when(n_KatE >= 1 ~ 1,
                           n_KatE == 0 ~ 0),
         nCatMn = case_when(n_CatMn >= 1 ~ 1,
                           n_CatMn == 0 ~ 0),
         genepres = case_when(n_KatG == "0" & n_KatE == "0" & n_CatMn == "0" ~ "none",
           n_KatG == "0" & n_KatE > 0 & n_CatMn == "0" ~ "KatE",
                               n_KatG > 0 & n_KatE == "0" & n_CatMn == "0" ~ "KatG",
                            n_KatG == "0" & n_KatE == "0" & n_CatMn > 0 ~ "CatMn",
                            n_KatG > 0 & n_KatE == "0" & n_CatMn > 0 ~ "KatGCatMn",
                              n_KatG > 0 & n_KatE > 0 & n_CatMn == "0"  ~ "KatGKatE",
                            n_KatG == "0" & n_KatE > 0 & n_CatMn > 0 ~ "KatECatMn",
                            n_KatG > 0 & n_KatE > 0 & n_CatMn > 0 ~ "allcats")) 
 
 
katE_MAGs <- MAGs_counts_for_filteringtaxa %>%
  filter(genepres == "KatE" | genepres == "KatGKatE" | genepres == "allcats")

# #counting up total number of genomes in each family
# catalases_taxa <- catalases %>%
#   group_by(Family) %>%
#   count() 

#grouping the counts of the genes and the number of MAGs in the set

catalases_alltaxa <- catalases_MAG_count_byFam %>%
  mutate(propKatG = nMAGswithKatG/nMAGswithorwithoutcat,
         propKatE = nMAGswithKatE/nMAGswithorwithoutcat,
         propCatMn = nMAGswithCatMn/nMAGswithorwithoutcat) %>%
  group_by(Family) %>%
  mutate(totalprop = propKatG + propKatE + propCatMn)

catalases_alltaxa_forplot <- catalases_alltaxa %>%
  arrange(desc(totalprop)) %>%
  drop_na(Family)

#
catalases_MAG_count <- catalases %>%
  dplyr::rename("Genome_size" = "Genome size") %>%
  select(Family, Genus, Genome_size, n_KatG, n_KatE, n_CatMn) %>%
  group_by(Family) %>%
  mutate(nKatG = case_when(n_KatG >= 1 ~ 1,
                           n_KatG == 0 ~ 0),
         nKatE = case_when(n_KatE >= 1 ~ 1,
                           n_KatE == 0 ~ 0),
         nCatMn = case_when(n_CatMn >= 1 ~ 1,
                           n_CatMn == 0 ~ 0)) %>%
  dplyr::summarise(totalKatGcopies = sum(n_KatG),
          totalKatEcopies = sum(n_KatE),
          totalCatMncopies = sum(n_CatMn),
          nMAGswithKatG = sum(nKatG),
          nMAGswithKatE = sum(nKatE),
          nMAGswithCatMn = sum(nCatMn),
          n = n(),
          totalcat = nMAGswithKatG + nMAGswithKatE + nMAGswithCatMn) %>%
  dplyr::rename("nMAGswithorwithoutcat" = "n")


catalases_MAG_count_byFam <- catalases_MAG_count %>%
  filter(nMAGswithorwithoutcat >= 20,
         totalcat != 0)

#counts for percentages 
MAGs_counts_for_percentages <- catalases %>%
  dplyr::rename("Genome_size" = "Genome size") %>%
  select(Family, Genus, Genome_size, n_KatG, n_KatE, n_CatMn) %>%
  
  mutate(nKatG = case_when(n_KatG >= 1 ~ 1,data
                           n_KatG == 0 ~ 0),
         nKatE = case_when(n_KatE >= 1 ~ 1,
                           n_KatE == 0 ~ 0),
         nCatMn = case_when(n_CatMn >= 1 ~ 1,
                           n_CatMn == 0 ~ 0),
         genepres = case_when(n_KatG == "0" & n_KatE == "0" & n_CatMn == "0" ~ "none",
           n_KatG == "0" & n_KatE > 0 & n_CatMn == "0" ~ "KatE",
                               n_KatG > 0 & n_KatE == "0" & n_CatMn == "0" ~ "KatG",
                            n_KatG == "0" & n_KatE == "0" & n_CatMn > 0 ~ "CatMn",
                            n_KatG > 0 & n_KatE == "0" & n_CatMn > 0 ~ "KatGCatMn",
                              n_KatG > 0 & n_KatE > 0 & n_CatMn == "0"  ~ "KatGKatE",
                            n_KatG == "0" & n_KatE > 0 & n_CatMn > 0 ~ "KatECatMn",
                            n_KatG > 0 & n_KatE > 0 & n_CatMn > 0 ~ "allcats")) %>%
  group_by(genepres) %>%
  tally() 
    
3474

MAGs_counts_for_filteringtaxa <- catalases %>%
  dplyr::rename("Genome_size" = "Genome size") %>%
  mutate(nKatG = case_when(n_KatG >= 1 ~ 1,
                           n_KatG == 0 ~ 0),
         nKatE = case_when(n_KatE >= 1 ~ 1,
                           n_KatE == 0 ~ 0),
         nCatMn = case_when(n_CatMn >= 1 ~ 1,
                           n_CatMn == 0 ~ 0),
         genepres = case_when(n_KatG == "0" & n_KatE == "0" & n_CatMn == "0" ~ "none",
           n_KatG == "0" & n_KatE > 0 & n_CatMn == "0" ~ "KatE",
                               n_KatG > 0 & n_KatE == "0" & n_CatMn == "0" ~ "KatG",
                            n_KatG == "0" & n_KatE == "0" & n_CatMn > 0 ~ "CatMn",
                            n_KatG > 0 & n_KatE == "0" & n_CatMn > 0 ~ "KatGCatMn",
                              n_KatG > 0 & n_KatE > 0 & n_CatMn == "0"  ~ "KatGKatE",
                            n_KatG == "0" & n_KatE > 0 & n_CatMn > 0 ~ "KatECatMn",
                            n_KatG > 0 & n_KatE > 0 & n_CatMn > 0 ~ "allcats")) 
 
 
katE_MAGs <- MAGs_counts_for_filteringtaxa %>%
  filter(genepres == "KatE" | genepres == "KatGKatE" | genepres == "allcats")

# #counting up total number of genomes in each family
# catalases_taxa <- catalases %>%
#   group_by(Family) %>%
#   count() 

#grouping the counts of the genes and the number of MAGs in the set

catalases_alltaxa <- catalases_MAG_count_byFam %>%
  mutate(propKatG = nMAGswithKatG/nMAGswithorwithoutcat,
         propKatE = nMAGswithKatE/nMAGswithorwithoutcat,
         propCatMn = nMAGswithCatMn/nMAGswithorwithoutcat) %>%
  group_by(Family) %>%
  mutate(totalprop = propKatG + propKatE + propCatMn)

catalases_alltaxa_forplot <- catalases_alltaxa %>%
  arrange(desc(totalprop)) %>%
  drop_na(Family)


############NEWNAMES

#reading in new name dictionary
allFamilies_withtaxonomy_updated <- read_excel("allFamilies_withtaxonomy_updated.xlsx")

catalases_newnames <- left_join(catalases,
                                allFamilies_withtaxonomy_updated,
                                by = "Family")

#now i will export and delete Alteromonadaceae from the taxa that are not Alteromonas
#I will also split up Prochlorococcus and 
write_xlsx(catalases_newnames, "catalases_newnames.xlsx")

#reading in
catalases_newnames_updated <- read_excel("catalases_newnames_updated.xlsx")

#merging rel abundance with this

#renaming SCG123456789_taxa_updated_filt 
SCG123456789_taxa_updated_filt_rename <- SCG123456789_taxa_updated_filt %>%
  dplyr::rename("New_name2" = "...11")

SCG123456789_taxa_updated_filt_rename$New_name2[SCG123456789_taxa_updated_filt_rename$New_name2 == "Synechococcales-Synechococcaceae"] <- "Synechococcales-Synechococcus"
SCG123456789_taxa_updated_filt_rename$New_name2[SCG123456789_taxa_updated_filt_rename$New_name2 == "Synechococcales-Prochloraceae"] <- "Synechococcales-Prochlorococcus"


relabundance_prop <- left_join(catalases_newnames_updated,
                               SCG123456789_taxa_updated_filt_rename,
                               by = "New_name2")
####
catalases_MAG_count_order <- catalases %>%
  dplyr::rename("Genome_size" = "Genome size") %>%
  select(Order, Family, Genus, Genome_size, n_KatG, n_KatE, n_CatMn) %>%
  group_by(Order) %>%
  mutate(nKatG = case_when(n_KatG >= 1 ~ 1,
                           n_KatG == 0 ~ 0),
         nKatE = case_when(n_KatE >= 1 ~ 1,
                           n_KatE == 0 ~ 0),
         nCatMn = case_when(n_CatMn >= 1 ~ 1,
                           n_CatMn == 0 ~ 0)) %>%
  dplyr::summarise(totalKatGcopies = sum(n_KatG),
          totalKatEcopies = sum(n_KatE),
          totalCatMncopies = sum(n_CatMn),
          nMAGswithKatG = sum(nKatG),
          nMAGswithKatE = sum(nKatE),
          nMAGswithCatMn = sum(nCatMn),
          n = n(),
          totalcat = nMAGswithKatG + nMAGswithKatE + nMAGswithCatMn) %>%
  dplyr::rename("nMAGswithorwithoutcat" = "n")

catalases_MAG_count_byorder <- catalases_MAG_count_order %>%
  filter(nMAGswithorwithoutcat >= 20)

catalases_MAG_count <- relabundance_prop %>%
  dplyr::rename("Genome_size" = "Genome size") %>%
  select(New_name2, Genome_size, n_KatG, n_KatE, n_CatMn, row_median) %>%
  group_by(New_name2) %>%
  mutate(nKatG = case_when(n_KatG >= 1 ~ 1,
                           n_KatG == 0 ~ 0),
         nKatE = case_when(n_KatE >= 1 ~ 1,
                           n_KatE == 0 ~ 0),
         nCatMn = case_when(n_CatMn >= 1 ~ 1,
                           n_CatMn == 0 ~ 0)) %>%
  dplyr::summarise(totalKatGcopies = sum(n_KatG),
          totalKatEcopies = sum(n_KatE),
          totalCatMncopies = sum(n_CatMn),
          nMAGswithKatG = sum(nKatG),
          nMAGswithKatE = sum(nKatE),
          nMAGswithCatMn = sum(nCatMn),
          row_median = row_median,
          n = n(),
          totalcat = nMAGswithKatG + nMAGswithKatE + nMAGswithCatMn) %>%
  dplyr::rename("nMAGswithorwithoutcat" = "n")



catalases_MAG_count_bynewname <- catalases_MAG_count %>%
  filter(nMAGswithorwithoutcat >= 20)

#counts for percentages 
MAGs_counts_for_percentages <- relabundance_prop %>%
  dplyr::rename("Genome_size" = "Genome size") %>%
  select(New_name2, Genome_size, n_KatG, n_KatE, n_CatMn, row_median) %>%
  mutate(nKatG = case_when(n_KatG >= 1 ~ 1,
                           n_KatG == 0 ~ 0),
         nKatE = case_when(n_KatE >= 1 ~ 1,
                           n_KatE == 0 ~ 0),
         nCatMn = case_when(n_CatMn >= 1 ~ 1,
                           n_CatMn == 0 ~ 0),
         genepres = case_when(n_KatG == "0" & n_KatE == "0" & n_CatMn == "0" ~ "none",
           n_KatG == "0" & n_KatE > 0 & n_CatMn == "0" ~ "KatE",
                               n_KatG > 0 & n_KatE == "0" & n_CatMn == "0" ~ "KatG",
                            n_KatG == "0" & n_KatE == "0" & n_CatMn > 0 ~ "CatMn",
                            n_KatG > 0 & n_KatE == "0" & n_CatMn > 0 ~ "KatGCatMn",
                              n_KatG > 0 & n_KatE > 0 & n_CatMn == "0"  ~ "KatGKatE",
                            n_KatG == "0" & n_KatE > 0 & n_CatMn > 0 ~ "KatECatMn",
                            n_KatG > 0 & n_KatE > 0 & n_CatMn > 0 ~ "allcats")) %>%
  group_by(New_name2, genepres) %>%
  tally() 
    


MAGs_counts_for_filteringtaxa <- catalases %>%
  dplyr::rename("Genome_size" = "Genome size") %>%
  mutate(nKatG = case_when(n_KatG >= 1 ~ 1,
                           n_KatG == 0 ~ 0),
         nKatE = case_when(n_KatE >= 1 ~ 1,
                           n_KatE == 0 ~ 0),
         nCatMn = case_when(n_CatMn >= 1 ~ 1,
                           n_CatMn == 0 ~ 0),
         genepres = case_when(n_KatG == "0" & n_KatE == "0" & n_CatMn == "0" ~ "none",
           n_KatG == "0" & n_KatE > 0 & n_CatMn == "0" ~ "KatE",
                               n_KatG > 0 & n_KatE == "0" & n_CatMn == "0" ~ "KatG",
                            n_KatG == "0" & n_KatE == "0" & n_CatMn > 0 ~ "CatMn",
                            n_KatG > 0 & n_KatE == "0" & n_CatMn > 0 ~ "KatGCatMn",
                              n_KatG > 0 & n_KatE > 0 & n_CatMn == "0"  ~ "KatGKatE",
                            n_KatG == "0" & n_KatE > 0 & n_CatMn > 0 ~ "KatECatMn",
                            n_KatG > 0 & n_KatE > 0 & n_CatMn > 0 ~ "allcats")) 
 
 
katE_MAGs <- MAGs_counts_for_filteringtaxa %>%
  filter(genepres == "KatE" | genepres == "KatGKatE" | genepres == "allcats")

# #counting up total number of genomes in each family
# catalases_taxa <- catalases %>%
#   group_by(Family) %>%
#   count() 

#grouping the counts of the genes and the number of MAGs in the set

#NEED TO GO IN AND ADD THE REL ABUNDANCE BACK IN
catalases_alltaxa <- catalases_MAG_count_bynewname %>%
  mutate(propKatG = nMAGswithKatG/nMAGswithorwithoutcat,
         propKatE = nMAGswithKatE/nMAGswithorwithoutcat,
         propCatMn = nMAGswithCatMn/nMAGswithorwithoutcat) %>%
  group_by(New_name2) %>%
  mutate(totalprop = propKatG + propKatE + propCatMn)

catalases_alltaxa_forplot <- catalases_alltaxa %>%
  arrange(desc(row_median)) %>%
  distinct()

catalases_alltaxa_forplot_filt <- catalases_alltaxa_forplot[-c(19:22,24),]

catalases_alltaxa_forplot_filt[is.na(catalases_alltaxa_forplot_filt)] <- 0

#######################

#set of Families

#table grouping by order

setofFamilies <- catalases %>%
  select(Phyla, Class, Order, Family) %>%
  unique()

#merging with the allFamilies dataset
allFamilies <- read_excel("allFamilies.xlsx")
allFamilies_withtaxonomy <- left_join(allFamilies,
                                      setofFamilies,
                                      by = "Family")
write_xlsx(allFamilies_withtaxonomy, "allFamilies_withtaxonomy.xlsx")
#preparing two long datasets

catalases_alltaxa_forplot_genepres <- left_join(catalases_alltaxa_forplot,
          MAGs_counts_for_percentages, 
          by = "New_name")

# catalases_alltaxa_forplot_copies <- catalases_alltaxa_forplot %>%
#   select(New_name, totalKatGcopies, totalKatEcopies, totalCatMncopies, totalprop) %>%
#   pivot_longer(2:4, names_to = "gene", values_to = "copynum")

# catalases_alltaxa_forplot_prop <- catalases_alltaxa_forplot %>%
#   select(New_name, propKatG, propKatE, totalprop) %>%
#   pivot_longer(2:3, names_to = "gene", values_to = "proportion")



# catalases_alltaxa_forplot_nMAGs <- catalases_alltaxa_forplot %>%
# select(New_name, nMAGswithKatG, nMAGswithKatE, totalprop, nMAGswithorwithoutcat) %>%
#   pivot_longer(2:3, names_to = "gene", values_to = "copynum")

####free living surf
catalases_alltaxa_forplot_new <- catalases_newnames %>%
  mutate(nKatG = case_when(n_KatG >= 1 ~ 1,
                           n_KatG == 0 ~ 0),
         nKatE = case_when(n_KatE >= 1 ~ 1,
                           n_KatE == 0 ~ 0),
         nCatMn = case_when(n_CatMn >= 1 ~ 1,
                           n_CatMn == 0 ~ 0),
         genepres = case_when(n_KatG == "0" & n_KatE == "0" & n_CatMn == "0" ~ "none",
           n_KatG == "0" & n_KatE > 0 & n_CatMn == "0" ~ "KatE",
                               n_KatG > 0 & n_KatE == "0" & n_CatMn == "0" ~ "KatG",
                            n_KatG == "0" & n_KatE == "0" & n_CatMn > 0 ~ "CatMn",
                            n_KatG > 0 & n_KatE == "0" & n_CatMn > 0 ~ "KatGCatMn",
                              n_KatG > 0 & n_KatE > 0 & n_CatMn == "0"  ~ "KatGKatE",
                            n_KatG == "0" & n_KatE > 0 & n_CatMn > 0 ~ "KatECatMn",
                            n_KatG > 0 & n_KatE > 0 & n_CatMn > 0 ~ "allcats"),
         totalcat = n_KatG + nKatE + n_KatE) %>%
  group_by(New_name, genepres) %>%
  tally() %>%
  dplyr::rename("nMAGs" = "n") 


catalases_alltaxa_forplot_new_notgrouped <- catalases_newnames %>%
  mutate(nKatG = case_when(n_KatG >= 1 ~ 1,
                           n_KatG == 0 ~ 0),
         nKatE = case_when(n_KatE >= 1 ~ 1,
                           n_KatE == 0 ~ 0),
         nCatMn = case_when(n_CatMn >= 1 ~ 1,
                           n_CatMn == 0 ~ 0),
         genepres = case_when(n_KatG == "0" & n_KatE == "0" & n_CatMn == "0" ~ "none",
           n_KatG == "0" & n_KatE > 0 & n_CatMn == "0" ~ "KatE",
                               n_KatG > 0 & n_KatE == "0" & n_CatMn == "0" ~ "KatG",
                            n_KatG == "0" & n_KatE == "0" & n_CatMn > 0 ~ "CatMn",
                            n_KatG > 0 & n_KatE == "0" & n_CatMn > 0 ~ "KatGCatMn",
                              n_KatG > 0 & n_KatE > 0 & n_CatMn == "0"  ~ "KatGKatE",
                            n_KatG == "0" & n_KatE > 0 & n_CatMn > 0 ~ "KatECatMn",
                            n_KatG > 0 & n_KatE > 0 & n_CatMn > 0 ~ "allcats"),
         totalcat = n_KatG + nKatE + n_KatE)

#katEMAGs
catalases_alltaxa_forplot_new_notgrouped_katE <- catalases_alltaxa_forplot_new_notgrouped %>%
  filter(genepres == "KatGKatE" | genepres == "KatECatMn" | genepres == "allcats" | genepres == "KatE")
#katGMAGs

catalases_alltaxa_forplot_new_notgrouped_katG <- catalases_alltaxa_forplot_new_notgrouped %>%
filter(genepres == "KatGKatE" | genepres == "KatGCatMn" | genepres == "allcats" | genepres == "KatG")
#####
####repeat for other size fractions
catalases_MAG_count_deep_small <- catalases_deep_small %>%
  mutate(nKatG = case_when(n_KatG >= 1 ~ 1,
                           n_KatG == 0 ~ 0),
         nKatE = case_when(n_KatE >= 1 ~ 1,
                           n_KatE == 0 ~ 0),
         nCatMn = case_when(n_CatMn >= 1 ~ 1,
                           n_CatMn == 0 ~ 0),
         genepres = case_when(nKatG == "0" & nKatE == "0" & nCatMn == "0" ~ "none",
           nKatG == "0" & nKatE > 0 & nCatMn == "0" ~ "KatE",
                               nKatG > 0 & nKatE == "0" & nCatMn == "0" ~ "KatG",
                            nKatG == "0" & nKatE == "0" & nCatMn > 0 ~ "CatMn",
                            nKatG > 0 & nKatE == "0" & nCatMn > 0 ~ "KatGCatMn",
                              nKatG > 0 & nKatE > 0 & nCatMn == "0"  ~ "KatGKatE",
                            nKatG == "0" & nKatE > 0 & nCatMn > 0 ~ "KatECatMn",
                            nKatG > 0 & nKatE > 0 & nCatMn > 0 ~ "allcats"),
         totalcat = nKatG + nKatE + nCatMn) %>%
  group_by(Family, genepres) %>%
  tally() %>%
  dplyr::rename("nMAGs" = "n") %>%
  filter(genepres != "none")

catalases_MAG_count_byFam_deep_small <- catalases_MAG_count_deep_small %>%
  group_by(Family) %>%
  mutate(totalMAGs = sum(nMAGs)) %>%
  filter(totalMAGs != 1)
#counting up total number of genomes in each family
catalases_taxa_deep_small <- catalases_deep_small %>%
  group_by(Family) %>%
  dplyr::count() 

#grouping the counts of the genes and the number of MAGs in the set

catalases_alltaxa_deep_small <- left_join(catalases_MAG_count_byFam_deep_small,
catalases_taxa_deep_small,
by = "Family") %>%
  mutate(propMAGs = nMAGs/n) %>%
  group_by(Family) %>%
  mutate(totalprop = sum(propMAGs))

catalases_alltaxa_forplot_deep_small <- catalases_alltaxa_deep_small %>%
  arrange(desc(totalprop)) %>%
  drop_na(Family)

#########

catalases_MAG_count_deep_large <- catalases_deep_large %>%
  mutate(nKatG = case_when(n_KatG >= 1 ~ 1,
                           n_KatG == 0 ~ 0),
         nKatE = case_when(n_KatE >= 1 ~ 1,
                           n_KatE == 0 ~ 0),
         nCatMn = case_when(n_CatMn >= 1 ~ 1,
                           n_CatMn == 0 ~ 0),
         genepres = case_when(nKatG == "0" & nKatE == "0" & nCatMn == "0" ~ "none",
           nKatG == "0" & nKatE > 0 & nCatMn == "0" ~ "KatE",
                               nKatG > 0 & nKatE == "0" & nCatMn == "0" ~ "KatG",
                            nKatG == "0" & nKatE == "0" & nCatMn > 0 ~ "CatMn",
                            nKatG > 0 & nKatE == "0" & nCatMn > 0 ~ "KatGCatMn",
                              nKatG > 0 & nKatE > 0 & nCatMn == "0"  ~ "KatGKatE",
                            nKatG == "0" & nKatE > 0 & nCatMn > 0 ~ "KatECatMn",
                            nKatG > 0 & nKatE > 0 & nCatMn > 0 ~ "allcats"),
         totalcat = nKatG + nKatE + nCatMn) %>%
  group_by(Family, genepres) %>%
  tally() %>%
  dplyr::rename("nMAGs" = "n") %>%
  filter(genepres != "none")

catalases_MAG_count_byFam_deep_large <- catalases_MAG_count_deep_large %>%
  group_by(Family) %>%
  mutate(totalMAGs = sum(nMAGs)) %>%
  filter(totalMAGs != 1)
#counting up total number of genomes in each family
catalases_taxa_deep_large <- catalases_deep_large %>%
  group_by(Family) %>%
  dplyr::count() 

#grouping the counts of the genes and the number of MAGs in the set

catalases_alltaxa_deep_large <- left_join(catalases_MAG_count_byFam_deep_large,
catalases_taxa_deep_large,
by = "Family") %>%
  mutate(propMAGs = nMAGs/n) %>%
  group_by(Family) %>%
  mutate(totalprop = sum(propMAGs))

catalases_alltaxa_forplot_deep_large <- catalases_alltaxa_deep_large %>%
  arrange(desc(totalprop)) %>%
  drop_na(Family)

#remove the all catalases one and rename it as having katG and katE
catalases_alltaxa_forplot_genepres$genepres[catalases_alltaxa_forplot_genepres$genepres=="allcats"]<- "KatGKatE"
catalases_alltaxa_forplot_genepres$genepres[catalases_alltaxa_forplot_genepres$genepres=="KatGKatE"]<- "katG and katE"
catalases_alltaxa_forplot_genepres$genepres[catalases_alltaxa_forplot_genepres$genepres=="KatE"]<- "katE"
catalases_alltaxa_forplot_genepres$genepres[catalases_alltaxa_forplot_genepres$genepres=="KatG"]<- "katG"
catalases_alltaxa_forplot_genepres$genepres[catalases_alltaxa_forplot_genepres$genepres=="none"]<- "No catalase"

catalases_alltaxa_forplot_genepres_renamed <- catalases_alltaxa_forplot_genepres %>%
  dplyr::rename("Catalase type" = "genepres")


#now merge together the new rel abundance dataset with the MAGs
SCG123456789_taxa_updated_filt

#need to rerun the code for
catalases_alltaxa_forplot_prop

#################
#binding the table with the raw counts of the MAGs
numMAGs <- ggplot(data = catalases_alltaxa_forplot_filt, aes(x = reorder(New_name2, row_median), y = sqrt(row_median))) + coord_flip() +
  geom_col(fill="grey",color="black") + ylab("Metagenome relative abundance (sqrt)") + theme_bw() + scale_y_continuous(expand = expansion(mult = c(0, .1))) + theme(legend.position="none") + theme(text = element_text(size = 12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1),
legend.position = "right",
axis.text.y=element_blank(),
axis.title.y=element_blank(),
axis.ticks.y=element_blank())

proportioncatalase_katG <- ggplot(data = catalases_alltaxa_forplot_filt, aes(x = reorder(New_name2, row_median), y = propKatG)) + coord_flip() +
  geom_col(fill= "blue", color = "black") + ylab("Proportion of MAGs") + xlab("Taxa") + theme_bw() + scale_y_continuous(expand = expansion(mult = c(0, .1))) + theme(text = element_text(size = 12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1),
strip.text = element_blank(),
legend.position = "none") 

proportioncatalase_katE <- ggplot(data = catalases_alltaxa_forplot_filt, aes(x = reorder(New_name2, row_median), y = propKatE)) + coord_flip() +
  geom_col(fill= "darkgreen", color = "black") + ylab("Proportion of MAGs") + xlab("") + theme_bw() + scale_y_continuous(expand = expansion(mult = c(0, .1))) + theme(text = element_text(size = 12), panel.background = element_blank(),
                                                                                                                                                                          axis.text.y = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1),
strip.text = element_blank(),
legend.position = "none") 


library(gridExtra)
grid.arrange(proportioncatalase_katG, proportioncatalase_katE, numMAGs, ncol = 3)


#calculate more props
catalases_alltaxa_forplot_genepres_allMAGs <- catalases_alltaxa_forplot_genepres %>%
  mutate(percentofallMAGs=nMAGswithorwithoutcat/3474)
  
  
  

library(gridExtra)
grid.arrange(totalcopies, nMAGswithcat, propFamwithcat, ncol = 3)


#deep small

b <- ggplot(data = catalases_alltaxa_forplot_deep_small, aes(x = reorder(Family, totalprop), y = nMAGs)) + coord_flip() +
  geom_col(aes(fill = genepres), color = "black") + ylab("Number of MAGs") + xlab("Taxa") + theme(text=element_text(size=12)) + theme_classic() + 
  scale_y_continuous(expand = expansion(mult = c(0, .1))) + theme(legend.position="none") 

c <- ggplot(data = catalases_alltaxa_forplot_deep_small, aes(x = reorder(Family, totalprop), y = propMAGs)) + coord_flip() +
  geom_col(aes(fill = genepres), color = "black") + ylab("Proportion of MAGs") + xlab("Taxa") + theme(text=element_text(size=12)) + theme_classic() + 
  scale_y_continuous(expand = expansion(mult = c(0, .1))) + guides(fill=guide_legend(title="Catalase type")) 

library(gridExtra)
grid.arrange(b, c, ncol = 2)

#making a pie chart with the prop of each taxa that carries each catalase
catalases_alltaxa_forplot_filt <- read_excel("catalases_alltaxa_forplot_filt.xlsx")

piechartdatakatG <- catalases_alltaxa_forplot_filt %>%
 filter(propKatG!=0) %>%
  arrange(desc(propKatG))

piekatG <- ggplot(piechartdatakatG, aes(x = "", y = propKatG, fill = (New_name2))) +
  geom_bar(width = 1, stat = "identity") +
  theme_void() +
  geom_col(color = "black") +
  coord_polar("y", start = 0) +
   theme(plot.title = element_text(hjust = 0.5),
      legend.text = element_text(size = 20)) 

piechartdatakatE <- catalases_alltaxa_forplot_filt %>%
 filter(propKatE!=0) %>%
  arrange(desc(propKatE))

piekatE <- ggplot(piechartdatakatE, aes(x = "", y = propKatE, fill = (New_name2))) +
  geom_bar(width = 1, stat = "identity") +
  theme_void() +
  geom_col(color = "black") +
  coord_polar("y", start = 0) +
   theme(plot.title = element_text(hjust = 0.5),
      legend.text = element_text(size = 20)) 
#deep large

b <- ggplot(data = catalases_alltaxa_forplot_deep_large, aes(x = reorder(Family, totalprop), y = nMAGs)) + coord_flip() +
  geom_col(aes(fill = genepres), color = "black") + ylab("Number of MAGs") + xlab("Taxa") + theme(text=element_text(size=12)) + theme_classic() + 
  scale_y_continuous(expand = expansion(mult = c(0, .1))) + theme(legend.position="none") 

c <- ggplot(data = catalases_alltaxa_forplot_deep_large, aes(x = reorder(Family, totalprop), y = propMAGs)) + coord_flip() +
  geom_col(aes(fill = genepres), color = "black") + ylab("Proportion of MAGs") + xlab("Taxa") + theme(text=element_text(size=12)) + theme_classic() + 
  scale_y_continuous(expand = expansion(mult = c(0, .1))) + guides(fill=guide_legend(title="Catalase type")) 

library(gridExtra)
grid.arrange(b, c, ncol = 2)
##############
#merging
relabundance_prop <- left_join(SCG123456789_taxa_updated_filt
catalases_alltaxa_forplot





##########
b <- ggplot(data = catalases_alltaxa_forplot, aes(x = reorder(Family, totalprop), y = nMAGs)) + coord_flip() +
  geom_col(aes(fill = genepres), color = "black") + ylab("Number of MAGs") + xlab("Taxa") + theme(text=element_text(size=12)) + theme_classic() + 
  scale_y_continuous(expand = expansion(mult = c(0, .1))) + theme(legend.position="none") 

c <- ggplot(data = catalases_alltaxa_forplot, aes(x = reorder(Family, totalprop), y = propMAGs)) + coord_flip() +
  geom_col(aes(fill = genepres), color = "black") + ylab("Proportion of MAGs") + xlab("Taxa") + theme(text=element_text(size=12)) + theme_classic() + 
  scale_y_continuous(expand = expansion(mult = c(0, .1))) + guides(fill=guide_legend(title="Catalase type")) 

b <- ggplot(data = catalases_alltaxa_forplot, aes(x = reorder(Family, totalprop), y = nMAGs)) + coord_flip() +
  geom_col(aes(fill = genepres), color = "black") + ylab("Number of MAGs") + xlab("Taxa") + theme(text=element_text(size=12)) + theme_classic() + 
  scale_y_continuous(expand = expansion(mult = c(0, .1))) + theme(legend.position="none") 


library(gridExtra)
grid.arrange(b, c, ncol = 2)

###############################
#venn diagram

KatG <- pull(catalases[!catalases$n_KatG == 0 ,"genome"])

Manganese_catalase <- pull(catalases[!catalases$n_CatMn == 0 ,"genome"])

KatE <- pull(catalases[!catalases$n_KatE == 0 ,"genome"])

input <- list(KatG=KatG, KatE=KatE, Manganese_catalase=Manganese_catalase)

library(ggvenn)
  
ggvenn(input)

KatG <- pull(catalases_deep_small[!catalases_deep_small$n_KatG == 0 ,"genome"])

Manganese_catalase <- pull(catalases_deep_small[!catalases_deep_small$n_CatMn == 0 ,"genome"])

KatE <- pull(catalases_deep_small[!catalases_deep_small$n_KatE == 0 ,"genome"])

input <- list(KatG=KatG, KatE=KatE, Manganese_catalase=Manganese_catalase)

library(ggvenn)
  
ggvenn(input)

KatG <- pull(catalases_deep_large[!catalases_deep_large$n_KatG == 0 ,"genome"])

Manganese_catalase <- pull(catalases_deep_large[!catalases_deep_large$n_CatMn == 0 ,"genome"])

KatE <- pull(catalases_deep_large[!catalases_deep_large$n_KatE == 0 ,"genome"])

input <- list(KatG=KatG, KatE=KatE, Manganese_catalase=Manganese_catalase)

library(ggvenn)
  
ggvenn(input)

#need to pivot tables

catalases_deep_large_long <- catalases_deep_large %>%
select(genome, n_KatG, n_KatE, n_CatMn) %>%
  pivot_longer(2:4, names_to = "gene", values_to = "copynum") 

catalases_deep_large_long <- catalases_deep_small %>%
select(genome, n_KatG, n_KatE, n_CatMn) %>%
  pivot_longer(2:4, names_to = "gene", values_to = "copynum") 

#now box plots
ggplot(catalases_deep_large_long, aes(x=gene, y=copynum)) + 
    geom_boxplot(fill="grey") + theme_bw() + xlab("gene") + ylab("copynum") + coord_flip() + theme(text = element_text(size = 12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1))

#use catalases

catalases <- read_excel("~/Desktop/Dyhrman Lab/Desktop_DyhrmanLab/catalases.xlsx")

KatG_catmn_KatE_genome_size_alltaxa <- catalases %>%
  mutate(totalcat = n_KatG + n_KatE + n_CatMn) %>%
  dplyr::rename("Genome.size" = "Genome size",
                "GC.content" = "GC Content") %>%
  filter("Genome.size" != 0)
  

KatG_catmn_KatE_genome_size_alltaxa_filt <- KatG_catmn_KatE_genome_size_alltaxa %>%
  filter(n_KatG < 3,
         n_KatE < 3)

KatG_catmn_KatE_genome_size_alltaxa_katE <- KatG_catmn_KatE_genome_size_alltaxa %>%
  filter(n_KatE < 3)
#want to add new column with genome size and gc content categories
library(ggridges)

p <- ggplot(KatG_catmn_KatE_genome_size_alltaxa_katG, aes(x=nKatG, y=GC.content)) + 
  geom_violin()

hannah_GC_content <- ggplot(KatG_catmn_KatE_genome_size_alltaxa_katG, aes(x=as.character(n_KatG), y=GC.content)) + 
    geom_boxplot(fill="grey") + theme_bw() + xlab("katG copy number per genome") + ylab("GC content (proportion)") + coord_flip() + theme(text = element_text(size = 12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) 

hannah_GC_content_E <- ggplot(KatG_catmn_KatE_genome_size_alltaxa_katE, aes(x=as.character(n_KatE), y=GC.content)) + 
    geom_boxplot(fill="grey") + theme_bw() + xlab("katE copy number per genome") + ylab("GC content (proportion)") + coord_flip() + theme(text = element_text(size = 12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) 


hannah_katG <- ggplot(KatG_catmn_KatE_genome_size_alltaxa_katG, aes(x=as.character(n_KatG), y=Genome.size)) + 
    geom_boxplot(fill="grey") + theme_bw() + xlab("katG copy number per genome") + ylab("Genome length (bp)") + coord_flip() + theme(text = element_text(size = 12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) 

hannah_katE <- ggplot(KatG_catmn_KatE_genome_size_alltaxa_katE, aes(x=as.character(n_KatE), y=Genome.size)) + 
    geom_boxplot(fill="grey") + theme_bw() + xlab("katE copy number per genome") + ylab("Genome length (bp)") + coord_flip() + theme(text = element_text(size = 12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) 


# hannah_allcats<- ggplot(KatG_catmn_KatE_genome_size_noalt, aes(x=as.character(totalcat), y=Genome.size)) + 
#     geom_boxplot(fill="grey") + theme_bw() + xlab("all catalases copies") + ylab("Genome length (bp)") + coord_flip() + theme(text = element_text(size = 12), panel.background = element_blank(),
# panel.grid.major = element_blank(), 
#         panel.grid.minor = element_blank(),
#         panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) 
library(gridExtra)
grid.arrange(hannah_GC_content, hannah_katG, hannah_GC_content_E, hannah_katE, ncol = 2)

##############
###################

#large_deep
KatG_catmn_KatE_genome_size_noalt_deep_large <- catalases_deep_large %>%
  filter(Family != "Alteromonadaceae") %>%
  mutate(totalcat = n_KatG + n_KatE + n_CatMn) %>% 
  rename("Genome.size" = "Genome size")
#repeating box plot


hannah_katGdeep_large <- ggplot(KatG_catmn_KatE_genome_size_noalt_deep_large, aes(x=as.character(n_KatG), y=Genome.size)) + 
    geom_boxplot(fill="grey") + theme_bw() + xlab("katG copies") + ylab("Genome length (bp)") + coord_flip() + theme(text = element_text(size = 12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) 

hannah_katEdeep_large <- ggplot(KatG_catmn_KatE_genome_size_noalt_deep_large, aes(x=as.character(n_KatE), y=Genome.size)) + 
    geom_boxplot(fill="grey") + theme_bw() + xlab("katE copies") + ylab("Genome length (bp)") + coord_flip() + theme(text = element_text(size = 12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) 


hannah_allcatsdeep_large <- ggplot(KatG_catmn_KatE_genome_size_noalt_deep_large, aes(x=as.character(totalcat), y=Genome.size)) + 
    geom_boxplot(fill="grey") + theme_bw() + xlab("all catalases copies") + ylab("Genome length (bp)") + coord_flip() + theme(text = element_text(size = 12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) 


#small_deep

#repeating box plot
KatG_catmn_KatE_genome_size_noalt_deep_small <- catalases_deep_small %>%
  filter(Family != "Alteromonadaceae") %>%
  mutate(totalcat = n_KatG + n_KatE + n_CatMn) %>% 
  rename("Genome.size" = "Genome size")


hannah_katGdeep_small <- ggplot(KatG_catmn_KatE_genome_size_noalt_deep_small, aes(x=as.character(n_KatG), y=Genome.size)) + 
    geom_boxplot(fill="grey") + theme_bw() + xlab("katG copies") + ylab("Genome length (bp)") + coord_flip() + theme(text = element_text(size = 12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) 

hannah_katEdeep_small <- ggplot(KatG_catmn_KatE_genome_size_noalt_deep_small, aes(x=as.character(n_KatE), y=Genome.size)) + 
    geom_boxplot(fill="grey") + theme_bw() + xlab("katE copies") + ylab("Genome length (bp)") + coord_flip() + theme(text = element_text(size = 12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) 


hannah_allcatsdeep_small <- ggplot(KatG_catmn_KatE_genome_size_noalt_deep_small, aes(x=as.character(totalcat), y=Genome.size)) + 
    geom_boxplot(fill="grey") + theme_bw() + xlab("all catalases copies") + ylab("Genome length (bp)") + coord_flip() + theme(text = element_text(size = 12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) 



#Michelle work
Genome_sizes <- read_excel("Genome_sizes.xlsx") %>%
  rename("Genome" = "...1",
         "Genome_Size" = "Geome Size")

Alt_catanalysis <- read_excel("Alt_catanalysis.xlsx")

allgenomes_comconabove70 <- read.csv("~/Desktop/Tara Oceans hmm/allgenomes_comconabove70.csv") %>%
  rename("Genome" = "GenomeName_raw")

michelledata <- left_join(allgenomes_comconabove70,
                           Genome_sizes,
                           by = "Genome")

# Catalase_Totals <- read.csv("~/Desktop/Dyhrman Lab/Desktop_DyhrmanLab/Catalase_Totals.csv") %>%
#   rename("Genome" = "genomesnames.name")

michelledata_copies <- michelledata %>%
  transform(Depth = as.numeric(Depth)) %>%
  dplyr::rename("KatG" = "K03782",
         "KatE" = "K03781") %>%
  mutate(KatGKatEratio = as.character(KatG/KatE),
         totalchar = as.character(total),
         KatGchar = as.character(KatG),
         KatEchar = as.character(KatE),
         genepres = case_when(KatG > KatE ~ "moreKatG",
                             KatG < KatE ~ "moreKatE",
                             KatG == KatE ~ "same",
                        KatG == 0 & KatE > 0 ~ "noKatG",
                        KatG > 0 & KatE == 0 ~ "noKatE",
                        KatG == 0 & KatE == 0 ~ "none"
                        ),
         depth = case_when(Depth <= 20  ~ "SRF",
                           Depth> 20  ~ "deep"
                            
         ))

###############

#export the extras and copy paste to the updated dataset
write_xlsx(michelledata_copies, "michelledata_copies.xlsx")

michelledata_copies_updated_recent <- read_excel("michelledata_copies_updated_recent.xlsx")

#turning the 0 genome size to NAs

michelledata_copies_updated_recent$Genome_Size[michelledata_copies_updated_recent$Genome_Size == "0"] <- NA

#preparing metadata for supplemental table 1

allgenomes_formerging <- read_excel("~/Downloads/Catalase/allgenomes_formerging.xlsx")

only214genomes_IDs <- michelledata_copies_updated_recent %>%
  select(Genome)

only214genomes_metadata <- left_join(only214genomes_IDs,
                                     allgenomes_formerging,
                                     by = "Genome")


# michelledata_copies_alldepths_updated <- read_excel("michelledata_copies_alldepths_updated.xlsx")
# 
# 
# michelledata_copies_clades1 <- left_join(michelledata_copies,
#                                          michelledata_copies_alldepths_updated,
#                                          by = "Genome")
# 
# write_xlsx(michelledata_copies_clades1, "michelledata_copies_clades1.xlsx")
# 
# 
# michelledata_copies_clades1_updated <- read_excel("michelledata_copies_clades1_updated.xlsx")

#now can check the datasets and the statistics


#learn which statistics

#There was significant variation in katG and katE complement between species in Altermonas with A. macleodii and A. australica clades having significantly different numbers of copies of katG (Figure 1B,C; SI Fig 2A). A. macleodii had fewer copies of katG than A. australica (p < 0.0001) (Figure 1B,C). 

#selecting for only the australica and macleodii
michelledata_copies_updated_recent_species <- michelledata_copies_updated_recent %>%
  drop_na(Species)

#comparing katE to species
model_mydata <- lm(KatE ~ Species, data = michelledata_copies_updated_recent_species)
#not normally distributed because p < 0.05
shapiro.test(residuals(model_mydata))

kruskal.test(KatE ~ Species, data = michelledata_copies_updated_recent_species)

#which groups
pairwise.wilcox.test(michelledata_copies_updated_recent_species$KatE,michelledata_copies_updated_recent_species$Species, p.adjust.method = "BH")

#extra numbers (ie number of samples from each species)

numspecies <- michelledata_copies_updated_recent_species %>%
  group_by(Species) %>%
  summarise(n = n())

#comparing katG to species
model_mydata <- lm(KatG ~ Species, data = michelledata_copies_updated_recent_species)
#not normally distributed because p < 0.05
shapiro.test(residuals(model_mydata))

kruskal.test(KatG ~ Species, data = michelledata_copies_updated_recent_species)

#which groups
pairwise.wilcox.test(michelledata_copies_updated_recent_species$KatG,michelledata_copies_updated_recent_species$Species, p.adjust.method = "BH")

#this is significant, so I will make a boxplot to show this significant difference

#for katG
species_katG <- ggplot(michelledata_copies_updated_recent_species, aes(x=Species, y=KatG)) + 
    geom_boxplot(fill="grey") + theme_bw() + xlab("Species") + ylab("katG copy number per genome") + theme(text = element_text(size = 12), panel.background = element_blank(),
                                                                                                   panel.grid.major = element_blank(), 
                                                                                                   panel.grid.minor = element_blank(),
                                                                                                   panel.border = element_rect(colour = "black", fill=NA, linewidth=1))

###now for katE
species_katE <- ggplot(michelledata_copies_updated_recent_species, aes(x=Species, y=KatE)) + 
    geom_boxplot(fill="grey") + theme_bw() + xlab("Species") + ylab("katE copy number per genome") + theme(text = element_text(size = 12), panel.background = element_blank(),
                                                                                                   panel.grid.major = element_blank(), 
                                                                                                   panel.grid.minor = element_blank(),
                                                                                                   panel.border = element_rect(colour = "black", fill=NA, linewidth=1))

#In Alteromonas as a whole, smaller genome sizes were associated with fewer copies of katG (1 versus 2), however smaller genomes tended to have more copies of katE (2 versus 1) (Table 1, SI Figure 1). 

#genome size when filter just those two groups
#is there a significant difference in the genome sizes between two groups?
michelledata_copies_updated_recent_species_genome <- michelledata_copies_updated_recent_species %>%
  drop_na(Genome_Size) %>%
filter(Genome_Size > 80008)


model_mydata <- lm(Genome_Size ~ Species, data = michelledata_copies_updated_recent_species_genome)
#not normally distributed because p < 0.05
shapiro.test(residuals(model_mydata))

kruskal.test(Genome_Size ~ Species, data = michelledata_copies_updated_recent_species_genome)

#not significant but I will make a boxplot anyway

species_genomesize <- ggplot(michelledata_copies_updated_recent_species_genome, aes(x=Species, y=Genome_Size)) + 
    geom_boxplot(fill="grey") + theme_bw() + xlab("Species") + ylab("Genome size") + theme(text = element_text(size = 12), panel.background = element_blank(),
                                                                                                   panel.grid.major = element_blank(), 
                                                                                                   panel.grid.minor = element_blank(),
                                                                                                   panel.border = element_rect(colour = "black", fill=NA, linewidth=1))


#pulling out medians
genomesize_species_stats <- michelledata_copies_updated_recent_species_genome %>%
  drop_na(Genome_Size) %>%
  #also dont want the weird number
  filter(Genome_Size > 80008) %>%
  group_by(Species) %>%
  summarise(median = median(Genome_Size))

#now making a final figure with all of the species level supp figures
library(gridExtra)
grid.arrange(species_katE, species_katG, species_genomesize, ncol = 1)
#######genome size compared to katG
#all genomes

michelledata_copies_updated_recent_genome_katG <- michelledata_copies_updated_recent %>%
  drop_na(Genome_Size) %>%
  filter(Genome_Size > 80008,
         KatG != 4)
  


model_mydata <- lm(Genome_Size ~ KatGchar, data = michelledata_copies_updated_recent_genome_katG)
#not normally distributed because p < 0.05
shapiro.test(residuals(model_mydata))

kruskal.test(Genome_Size ~ KatGchar, data = michelledata_copies_updated_recent_genome_katG)

pairwise.wilcox.test(michelledata_copies_updated_recent_genome_katG$Genome_Size,michelledata_copies_updated_recent_genome_katG$KatGchar, p.adjust.method = "BH")


#repeat for katE
model_mydata <- lm(Genome_Size ~ KatEchar, data = michelledata_copies_updated_recent_genome_katE)
#not normally distributed because p < 0.05
shapiro.test(residuals(model_mydata))

kruskal.test(Genome_Size ~ KatEchar, data = michelledata_copies_updated_recent_genome_katE)

pairwise.wilcox.test(michelledata_copies_updated_recent_genome_katE$Genome_Size,michelledata_copies_updated_recent_genome_katE$KatEchar, p.adjust.method = "BH")

#calculate the median differences

library(rstatix)
test <- aov(Genome_Size ~ KatGchar, data = michelledata_copies_updated_recent_species_genome) %>%
    tukey_hsd()
#not significant but I will make a boxplot anyway

allgenomes_katG <- ggplot(michelledata_copies_updated_recent_genome_katG, aes(x=Genome_Size, y=KatGchar)) + 
    geom_boxplot(fill="grey") + theme_bw() + xlab("Genome size") + ylab("katG copy number per genome") + theme(text = element_text(size = 12), panel.background = element_blank(),
                                                                                                   panel.grid.major = element_blank(), 
                                                                                                   panel.grid.minor = element_blank(),
                                                                                                   panel.border = element_rect(colour = "black", fill=NA, linewidth=1))


allgenomes_katE <- ggplot(michelledata_copies_updated_recent_genome_katE, aes(x=Genome_Size, y=KatEchar)) + 
    geom_boxplot(fill="grey") + theme_bw() + xlab("Genome size") + ylab("katE copy number per genome") + theme(text = element_text(size = 12), panel.background = element_blank(),
                                                                                                   panel.grid.major = element_blank(), 
                                                                                                   panel.grid.minor = element_blank(),
                                                                                                   panel.border = element_rect(colour = "black", fill=NA, linewidth=1))

grid.arrange(allgenomes_katG, allgenomes_katE, ncol = 1)
########repeat but for all genomes
model_mydata <- lm(Genome_Size ~ KatGchar, data = michelledata_copies_updated_recent_species_genome)
#not normally distributed because p < 0.05
shapiro.test(residuals(model_mydata))

kruskal.test(Genome_Size ~ KatGchar, data = michelledata_copies_updated_recent_species_genome)

#repeat for katE
model_mydata <- lm(Genome_Size ~ KatEchar, data = michelledata_copies_updated_recent_species_genome)
#not normally distributed because p < 0.05
shapiro.test(residuals(model_mydata))

kruskal.test(Genome_Size ~ KatGchar, data = michelledata_copies_updated_recent_species_genome)
library(rstatix)
test <- aov(Genome_Size ~ KatGchar, data = michelledata_copies_updated_recent_species_genome) %>%
    tukey_hsd()

#getting some stats
#getting genome size medians for each katE copy number
genomesize_species_stats_katEgenomesize <- michelledata_copies_updated_recent_species_genome %>%
  group_by(KatEchar) %>%
  summarise(median = median(Genome_Size),
            n = n())

#getting genome size medians for katG

genomesize_species_stats_katGgenomesize <- michelledata_copies_updated_recent_species_genome %>%
  group_by(KatGchar) %>%
  summarise(median = median(Genome_Size),
            n = n())



############
#making the final density plot

########repeating the plots but for the two Species 
michelledata_copies_twospecies_long <- michelledata_copies_updated_recent_species_genome %>%
  select(Genome, KatG, KatE, Species) %>%
  pivot_longer(2:3, names_to = "gene", values_to = "copies")

##############
ggplot(data=michelledata_copies_twospecies_long, aes(x=copies, group=gene, fill=gene)) +
  geom_density(alpha = 0.4) +
    theme_classic() + scale_x_continuous(expand = expansion(mult = c(0, .1))) + scale_y_continuous(expand = expansion(mult = c(0, .1))) + scale_fill_manual(
                        values=c("green", "blue"), , labels=c("katE", "katG")) + xlab("catalase copy number per genome") + ylab("Density") + facet_wrap(~Species, ncol = 1)


###############

michelle_katG <- ggplot(michelledata_copies, aes(x=as.character(KatGchar), y=Genome_Size)) + 
    geom_boxplot(fill="grey") + theme_bw() + xlab("Alteromonas katG copies") + ylab("Genome length (bp)") + coord_flip() + theme(text = element_text(size = 12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) 

michelle_katE <- ggplot(michelledata_copies, aes(x=as.character(KatEchar), y=Genome_Size)) + 
    geom_boxplot(fill="grey") + theme_bw() + xlab("Alteromonas katE copies") + ylab("Genome length (bp)") + coord_flip() + theme(text = element_text(size = 12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) 

michelle_allcat <- ggplot(michelledata_copies, aes(x=as.character(totalchar), y=Genome_Size)) + 
    geom_boxplot(fill="grey") + theme_bw() + xlab("Alteromonas catalase copies") + ylab("Genome length (bp)") + coord_flip() + theme(text = element_text(size = 12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) 

grid.arrange(hannah_katG, michelle_katG, hannah_katE, michelle_katE, hannah_allcats, michelle_allcat)

#statistics
#my data
KatG_catmn_KatE_genome_size_noalt_stats <- KatG_catmn_KatE_genome_size_noalt %>%
  transform(n_KatG = as.character(n_KatG),
            n_KatE = as.character(n_KatG),
            n_CatMn = as.character(n_CatMn))
model_mydata_katG <- lm(Genome.size ~ as.character(n_KatG), data = KatG_catmn_KatE_genome_size_noalt_stats)



  
```

```{r}
#new MAGs analysis

#import alignment files 

katG_align <- read.delim("~/Desktop/Tara Oceans hmm/KatG/katG_align.tsv", header=FALSE, comment.char="#")

katG_align_clean_names <- katG_align[7:13] %>%
  rename("Gene_ID" = "V13",
         "start1" = "V7", 
         "end1" = "V8", 
         "start2" = "V9", 
         "end2" = "V10") %>%
  select(-V11,-V12)
  

#changing the names of MAGs so that they match the list of MAGs that are in the small size fraction
abundance_matrix_KatG_MAGs <- read.delim("~/Downloads/OceanGeneAtlas_Data_20240213-02_49_KatG_MAGs/abundance_matrix_KatG_MAGs.csv") 

abundance_matrix_KatG_MAGs <- abundance_matrix_KatG_MAGs %>%
  janitor::row_to_names(1) %>%
   separate(col=taxonomy, into=c('Biota', 'Taxa1', 'Taxa2', 'Taxa3', 'Taxa4', 'Taxa5', 'Taxa6', 'Taxa7', 'Taxa8', 'Taxa9'), sep=';')

abundance_matrix_KatG_MAGs_tomerge <- abundance_matrix_KatG_MAGs[1:11]

##merging 

KatG_align_taxa <- left_join(abundance_matrix_KatG_MAGs_tomerge,
                                        katG_align_clean_names,
                                        by = "Gene_ID") %>%
  extract(Gene_ID, c("shortID", "Gene_ID", "what"), "(.*)_([^_]+)_([^_]+)") %>%
  select(-Gene_ID, -what) 

#now before I check for duplicate hits, I will filter the dataset for the MAGs in the right size fraction
smallMAGs_bymetaGset <- read_excel("~/Desktop/Tara Oceans hmm/smallMAGs_bymetaGset.xlsx") 

smallMAGs_bymetaGset_SRF <- smallMAGs_bymetaGset %>%
select("Relative distribution MAGs", contains("05M")) %>%
  rename("shortID" = "Relative distribution MAGs")

#merging
KatG_small <- left_join(KatG_align_taxa,
                        smallMAGs_bymetaGset_SRF,
                        by = "shortID")
#need to remove one of the:
  
#removing
KatG_small_dropped <- KatG_small %>%
drop_na(ANE_004_05M)

#TARA_ANE_MAG_00049 has identical hits, so I'll drop
KatG_small_copies_dropped <- KatG_small_dropped[-17,]

#######
katE_align <- read.delim("~/Desktop/Tara Oceans hmm/KatE/katE_align.tsv", header=FALSE, comment.char="#")

katE_align_clean_names <- katE_align[7:13] %>%
  rename("Gene_ID" = "V13",
         "start1" = "V7", 
         "end1" = "V8", 
         "start2" = "V9", 
         "end2" = "V10") %>%
  select(-V11,-V12)

abundance_matrix_KatE <- read.delim("~/Downloads/OceanGeneAtlas_Data_20231129-01_48_KatE_MAGs/abundance_matrix.csv")

abundance_matrix_KatE_MAGs <- abundance_matrix_KatE %>%
  janitor::row_to_names(1) %>%
   separate(col=taxonomy, into=c('Biota', 'Taxa1', 'Taxa2', 'Taxa3', 'Taxa4', 'Taxa5', 'Taxa6', 'Taxa7', 'Taxa8', 'Taxa9'), sep=';')

abundance_matrix_KatE_MAGs_tomerge <- abundance_matrix_KatE_MAGs[1:11]

##merging

KatE_align_taxa <- left_join(abundance_matrix_KatE_MAGs_tomerge,
                                        katE_align_clean_names,
                                        by = "Gene_ID") %>%
  extract(Gene_ID, c("shortID", "Gene_ID", "what"), "(.*)_([^_]+)_([^_]+)") %>%
  select(-Gene_ID, -what) 

#merging
KatE_small <- left_join(KatE_align_taxa,
                        smallMAGs_bymetaGset_SRF,
                        by = "shortID")
#need to remove one of the:
  
#removing
KatE_small_dropped <- KatE_small %>%
drop_na(ANE_004_05M)

#TARA_ANE_MAG_00049 has identical hits, so I'll drop
KatE_small_copies_dropped <- KatE_small_dropped

####catmn
catmn_align <- read.delim("~/Desktop/Tara Oceans hmm/Manganese_catalase/catmn_align.tsv", header=FALSE, comment.char="#")

catmn_align_clean_names <- catmn_align[7:13] %>%
  rename("Gene_ID" = "V13",
         "start1" = "V7", 
         "end1" = "V8", 
         "start2" = "V9", 
         "end2" = "V10") %>%
  select(-V11,-V12)

abundance_matrix_CatMn <- read.delim("~/Downloads/OceanGeneAtlas_Data_20231129-01_53_Cat_Mn_MAGs/abundance_matrix.csv")

abundance_matrix_catmn_MAGs <- abundance_matrix_CatMn %>%
  janitor::row_to_names(1) %>%
   separate(col=taxonomy, into=c('Biota', 'Taxa1', 'Taxa2', 'Taxa3', 'Taxa4', 'Taxa5', 'Taxa6', 'Taxa7', 'Taxa8', 'Taxa9'), sep=';')

abundance_matrix_catmn_MAGs_tomerge <- abundance_matrix_catmn_MAGs[1:11]

##merging 

catmn_align_taxa <- left_join(abundance_matrix_catmn_MAGs_tomerge,
                                        catmn_align_clean_names,
                                        by = "Gene_ID") %>%
  extract(Gene_ID, c("shortID", "Gene_ID", "what"), "(.*)_([^_]+)_([^_]+)") %>%
  select(-Gene_ID, -what) 

#merging
catmn_small <- left_join(catmn_align_taxa,
                        smallMAGs_bymetaGset_SRF,
                        by = "shortID")
#
catmn_small_dropped <- catmn_small %>%
drop_na(ANE_004_05M)

#TARA_ANE_MAG_00049 has identical hits, so I'll drop
catmn_small_copies_dropped <- catmn_small_dropped


#####rubisco
rub_align <- read.delim("~/Desktop/Tara Oceans hmm/rbcL/rub_align.tsv", header=FALSE, comment.char="#")

rub_align_clean_names <- rub_align[7:13] %>%
  rename("Gene_ID" = "V13",
         "start1" = "V7", 
         "end1" = "V8", 
         "start2" = "V9", 
         "end2" = "V10") %>%
  select(-V11,-V12)

abundance_matrix_rub <- read.delim("~/Downloads/OceanGeneAtlas_Data_20240315-15_38_Rub_MAGs/abundance_matrix.csv")

abundance_matrix_rub_MAGs <- abundance_matrix_rub %>%
  janitor::row_to_names(1) %>%
   separate(col=taxonomy, into=c('Biota', 'Taxa1', 'Taxa2', 'Taxa3', 'Taxa4', 'Taxa5', 'Taxa6', 'Taxa7', 'Taxa8', 'Taxa9'), sep=';')

abundance_matrix_rub_MAGs_tomerge <- abundance_matrix_rub_MAGs[1:11]

##merging 

rub_align_taxa <- left_join(abundance_matrix_rub_MAGs_tomerge,
                                        rub_align_clean_names,
                                        by = "Gene_ID") %>%
  extract(Gene_ID, c("shortID", "Gene_ID", "what"), "(.*)_([^_]+)_([^_]+)") %>%
  select(-Gene_ID, -what) 

#merging
rub_small <- left_join(rub_align_taxa,
                        smallMAGs_bymetaGset_SRF,
                        by = "shortID")
#
rub_small_dropped <- rub_small %>%
drop_na(ANE_004_05M)

#TARA_ANE_MAG_00049 has identical hits, so I'll drop
rub_small_copies_dropped <- rub_small_dropped

#merging everything together and figuring out which samples are on the surface
KatG_MAGs <- KatG_small_copies_dropped %>%
  distinct(shortID) %>%
  mutate(KatG = 1)

KatE_MAGs <- KatE_small_copies_dropped %>%
  distinct(shortID) %>%
  mutate(KatE = 1)

catmn_MAGs <- catmn_small_copies_dropped %>%
  distinct(shortID) %>%
  mutate(catmn = 1)

rub_MAGs <- rub_small_copies_dropped %>%
  distinct(shortID) %>%
  mutate(rub = 1)

KatG_catmn_forvenn <- merge(KatG_MAGs,
                  catmn_MAGs,
                  by = "shortID",
                  all.x = TRUE,
                  all.y = TRUE)

KatG_catmn_KatE_forvenn <- merge(KatG_catmn_forvenn,
                  KatE_MAGs,
                  by = "shortID",
                  all.x = TRUE,
                  all.y = TRUE)

KatG_catmn_KatE_rub_forvenn <- merge(KatG_catmn_KatE_forvenn,
                  rub_MAGs,
                  by = "shortID",
                  all.x = TRUE,
                  all.y = TRUE)

KatG_catmn_KatE_rub_forvenn[is.na(KatG_catmn_KatE_rub_forvenn)] <- 0


#import reads table 
small_MAGs_reads <- read_excel("~/Desktop/Tara Oceans hmm/small_MAGs_reads.xlsx")
MAGs_reads <- small_MAGs_reads %>%
  rename("shortID" = "Number of reads MAGs") %>%
  select(-shortID) %>%
   rowwise() %>%
  mutate(reads_SRF = sum(c_across(ends_with("05M")), na.rm = T),
         reads_DCM = sum(c_across(!ends_with("05M")), na.rm = T))

IDs <- small_MAGs_reads %>%
  rename("shortID" = "Number of reads MAGs") %>%
  select(shortID)

MAGs_reads_IDs <- cbind(IDs, MAGs_reads) %>%
  mutate(depth = case_when(reads_SRF > reads_DCM ~ "SRF",
                           reads_SRF <= reads_DCM ~ "DCM"))


genes_small <- left_join(KatG_catmn_KatE_rub_forvenn,
                        smallMAGs_bymetaGset_SRF,
                        by = "shortID") %>%
  rowwise() %>%
  mutate(rel_abundance = sum(c_across(ends_with("05M")), na.rm = T)) %>%
  arrange(rel_abundance)

library(ggvenn)
clean <- genes_small_1cutoff %>%
  select(shortID, KatG, catmn, KatE, rub)


KatG <- pull(genes_small[!genes_small$KatG == 0 ,"shortID"])

Manganese_catalase <- pull(genes_small[!genes_small$catmn == 0 ,"shortID"])

KatE <- pull(genes_small[!genes_small$KatE == 0 ,"shortID"])

rub <- pull(genes_small[!genes_small$rub == 0 ,"shortID"])

input <- list(KatG=KatG, KatE=KatE, rub=rub)

library(ggvenn)
  
ggvenn(input)

ggVennDiagram(input)

#who are the taxa in each category
#use this dataset
mergedtaxa_updated <- read_excel("mergedtaxa_updated.xlsx") %>%
  select(shortID, Selected_taxa)

#merging this with the venn diagram data
genes_small_tomerge <- genes_small %>%
  select(shortID, KatG, catmn, KatE, rub)


smallMAGlist_taxa <- left_join(genes_small_tomerge,
                          mergedtaxa_updated,
                          by = "shortID") %>%
   mutate(genepres = case_when(KatG == "0" & KatE == "0" & catmn == "0" ~ "none",
                               KatG == "0" & KatE > 0 & catmn == "0" ~ "KatE",
                               KatG > 0 & KatE == "0" & catmn == "0" ~ "KatG",
                            KatG == "0" & KatE == "0" & catmn > 0 ~ "CatMn",
                            KatG > 0 & KatE == "0" & catmn > 0 ~ "KatGCatMn",
                              KatG > 0 & KatE > 0 & catmn == "0"  ~ "KatGKatE",
                            KatG == "0" & KatE > 0 & catmn > 0 ~ "KatECatMn",
                            KatG > 0 & KatE > 0 & catmn > 0 ~ "allcats"))

write_xlsx(smallMAGlist_taxa, "smallMAGlist_taxa.xlsx")
#working on taxa dataset
smallMAGlist_taxa_updated <- read_excel("smallMAGlist_taxa_updated.xlsx")
#norub


#pies
catalaseMAGs_taxa_KatE <- smallMAGlist_taxa_updated %>%
  filter(genepres == "KatE") %>%
  distinct(shortID, .keep_all = TRUE)

taxa_KatE <- catalaseMAGs_taxa_KatE %>%
  group_by(Selected_taxa) %>%
  tally() %>%
  arrange(desc(n))

ggplot(taxa_KatE, aes(x = "", y = n, fill = fct_inorder(Selected_taxa))) +
  geom_bar(width = 1, stat = "identity") +  
  theme_void() +
  geom_col(color = "black") +
  coord_polar("y", start = 0) +
  guides(fill = guide_legend(title = "taxa")) + ggtitle("KatE") + theme(plot.title = element_text(hjust = 0.5),
      legend.text = element_text(size = 20)) 

#
catalaseMAGs_taxa_KatGKatE <- smallMAGlist_taxa_updated %>%
  filter(genepres == "KatGKatE") %>%
  distinct(shortID, .keep_all = TRUE)

taxa_KatGKatE <- catalaseMAGs_taxa_KatGKatE %>%
  group_by(Selected_taxa) %>%
  tally() %>%
  arrange(desc(n))

ggplot(taxa_KatGKatE, aes(x = "", y = n, fill = fct_inorder(Selected_taxa))) +
  geom_bar(width = 1, stat = "identity") +  
  theme_void() +
  geom_col(color = "black") +
  coord_polar("y", start = 0) +
  guides(fill = guide_legend(title = "taxa")) + ggtitle("KatGKatE") + theme(plot.title = element_text(hjust = 0.5),
      legend.text = element_text(size = 20)) 
#
catalaseMAGs_taxa_KatG <- smallMAGlist_taxa_updated %>%
  filter(genepres == "KatG") %>%
  distinct(shortID, .keep_all = TRUE)

taxa_KatG <- catalaseMAGs_taxa_KatG %>%
  group_by(Selected_taxa) %>%
  tally() %>%
  arrange(desc(n))

taxa_KatG$Selected_taxa[c(11:71)] <- "Other"

top10taxa_KatG <- taxa_KatG %>%
    group_by(Selected_taxa) %>%
    summarise(n = sum(n)) %>%
    arrange(desc(n))

ggplot(taxa_KatG, aes(x = "", y = n, fill = fct_inorder(Selected_taxa))) +
  geom_bar(width = 1, stat = "identity") +  
  theme_void() +
  geom_col(color = "black") +
  coord_polar("y", start = 0) +
  guides(fill = guide_legend(title = "taxa")) + ggtitle("KatG") + theme(plot.title = element_text(hjust = 0.5),
      legend.text = element_text(size = 20)) 

catalaseMAGs_taxa_KatGCatMn <- smallMAGlist_taxa_updated %>%
  filter(genepres == "KatGCatMn") %>%
  distinct(shortID, .keep_all = TRUE)

taxa_KatGCatMn <- catalaseMAGs_taxa_KatGCatMn %>%
  group_by(Selected_taxa) %>%
  tally() %>%
  arrange(desc(n))

ggplot(taxa_KatGCatMn, aes(x = "", y = n, fill = fct_inorder(Selected_taxa))) +
  geom_bar(width = 1, stat = "identity") +  
  theme_void() +
  geom_col(color = "black") +
  coord_polar("y", start = 0) +
  guides(fill = guide_legend(title = "taxa")) + ggtitle("KatGCatMn") + theme(plot.title = element_text(hjust = 0.5),
      legend.text = element_text(size = 20)) 

catalaseMAGs_taxa_KatGCatMn <- smallMAGlist_taxa_updated %>%
  filter(genepres == "KatGCatMn") %>%
  distinct(shortID, .keep_all = TRUE)

taxa_KatGCatMn <- catalaseMAGs_taxa_KatGCatMn %>%
  group_by(Selected_taxa) %>%
  tally() %>%
  arrange(desc(n))

catalaseMAGs_taxa_KatE <- smallMAGlist_taxa_updated %>%
  filter(genepres == "KatE") %>%
  distinct(shortID, .keep_all = TRUE)

taxa_KatE<- catalaseMAGs_taxa_KatE %>%
  group_by(Selected_taxa) %>%
  tally() %>%
  arrange(desc(n))

##making dataset
##allcats
catalaseMAGs_taxa_KatG <- smallMAGlist_taxa_updated %>%
  filter(genepres == "KatG") %>%
  distinct(shortID, .keep_all = TRUE)

catupsetplot_KatG_taxa <- catalaseMAGs_taxa_KatG %>%
mutate(taxa = case_when(Selected_taxa == "Flavobacteriaceae" ~ "Flavobacteriaceae",
       Selected_taxa == "MGIIA" ~ "MGIIA",
       Selected_taxa == "Pelagibacteraceae" ~ "Pelagibacteraceae",
       Selected_taxa == "MB11C04" ~ "MB11C04",
       Selected_taxa == "Rhodobacteraceae" ~ "Rhodobacteraceae",
       Selected_taxa == "TMED127" ~ "TMED127",
       Selected_taxa == "Saccharospirillaceae" ~ "Saccharospirillaceae",
       Selected_taxa == "UBA8229" ~ "UBA8229",
       Selected_taxa == "Cyclobacteriaceae" ~ "Cyclobacteriaceae",
       Selected_taxa == "SAR86" ~ "SAR86"))

catupsetplot_KatG_taxa[is.na(catupsetplot_KatG_taxa)] <- "Other"

catalaseMAGs_taxa_notKatG <- smallMAGlist_taxa_updated %>%
  filter(genepres != "KatG") %>%
  distinct(shortID, .keep_all = TRUE)

catupsetplot_notKatG_taxa <- catalaseMAGs_taxa_notKatG %>%
mutate(taxa = Selected_taxa)

#binding datasets together
alltaxa_upset <- rbind(catupsetplot_KatG_taxa, catupsetplot_notKatG_taxa) %>%
  filter(genepres != "none")

complete_taxa <- left_join(smallMAGlist_taxa_updated, MAGscompleteness,
                   by = "shortID") %>%
  filter(Completion_Minus_Red >= 70)

complete_taxa_plot <- complete_taxa %>%
  filter(genepres != "none")

ggplot(data = complete_taxa_plot, aes(x = fct_rev(fct_infreq(Selected_taxa)))) + coord_flip() +
  geom_bar(aes(fill = genepres)) + xlab("Number of MAGs") + ylab("Taxa") + theme(text=element_text(size=12)) + theme_classic() + 
  scale_y_continuous(expand = expansion(mult = c(0, .1)))


KatG <- pull(complete_taxa[!complete_taxa$KatG == 0 ,"shortID"])

Manganese_catalase <- pull(complete_taxa[!complete_taxa$catmn == 0 ,"shortID"])

KatE <- pull(complete_taxa[!complete_taxa$KatE == 0 ,"shortID"])

rub <- pull(complete_taxa[!complete_taxa$rub == 0 ,"shortID"])

input <- list(KatG=KatG, KatE=KatE, Manganese_catalase=Manganese_catalase)

library(ggvenn)
  
ggvenn(input)


#looking at all of the small
smallMAGlist_taxa_updated

#take list of 1888 MAGs
All_MAGs <- read_excel("All_MAGs.xlsx") %>%
  rename("shortID" = "Genome_Id") %>%
  select(shortID, total_length...2, Portion_ALL_Arctic, Portion_ALL_Atlantic, Portion_ALL_Indian,
         Portion_ALL_MED, Portion_ALL_Pacific, Portion_ALL_RED, Portion_ALL_Southern,
         Completion_Minus_Red)
#take list of small MAGs
smallMAGs_bymetaGset <- read_excel("~/Desktop/Tara Oceans hmm/smallMAGs_bymetaGset.xlsx") %>%
  rename("shortID" = "Relative distribution MAGs") %>%
  select(shortID)

#inner join
allsmallMAGs <- inner_join(smallMAGs_bymetaGset, All_MAGs,
              by = "shortID")

#taxonomy
mergedtaxa_updated <- read_excel("mergedtaxa_updated.xlsx") 

allsmallMAGs_taxa <- left_join(allsmallMAGs,
                               mergedtaxa_updated,
                               by = "shortID")


                  smallMAGlist_taxa_updated             
#merge with my dataset by short ID
 allsmallMAGs_cat <- merge(smallMAGlist_taxa_updated,
                  allsmallMAGs,
                  by = "shortID",
                  all.x = TRUE,
                  all.y = TRUE)
 
allsmallMAGs_cat[is.na(allsmallMAGs_cat)] <- 0
        
                  
#regular bar plot
smallMAGlist_taxa_updated_plot <- smallMAGlist_taxa_updated %>%
  filter(genepres != "none")

ggplot(df, aes(x = group, y = count, fill = group)) +
  geom_bar(stat = "identity")

#################
gene = colnames(alltaxa_upset)[2:4]
alltaxa_upset[gene] = alltaxa_upset[gene] >= 1

upset(alltaxa_upset, gene, name='gene', width_ratio=0.1)

upset(
    alltaxa_upset, 
    gene,
    width_ratio=0.1,
    themes=upset_modify_themes(
        list('intersections_matrix'=theme(text=element_text(size=12)),
             'overall_sizes'=theme(text=element_text(size=12), axis.text.x=element_text(angle = 90)
    ),
    'intersection_sizes' =theme(text=element_text(size=12)))),
    annotations = list(
        'Proportion of MAGs'=(
            ggplot(mapping=aes(fill=taxa))
            + geom_bar(stat='count', position='fill', color = "black")
            + scale_y_continuous(labels=scales::percent_format())
            + scale_fill_manual(values=c(
                '#fff7fb','grey','#d0d1e6','#a6bddb','#74a9cf','#023858',
'#ffffd9','#e5f5e0','#a1d99b','#7fcdbb','#41b6c4','#1d91c0','#225ea8','#253494','#081d58',
'#d4b9da','#c994c7','#df65b0','magenta4', '#e7298a','#ce1256','#980043','#67001f', 'blue','brown'
            ))
            + ylab('Proportion of MAGs'))) 
    ) + patchwork::plot_layout(heights=c(0.5, 0.5, 0.2))

upset(
    alltaxa_upset, 
    gene,
    width_ratio=0.1,
    themes=upset_modify_themes(
        list('intersections_matrix'=theme(text=element_text(size=12)),
             'overall_sizes'=theme(text=element_text(size=12), axis.text.x=element_text(angle = 90)
    )),
    'intersection_sizes' =theme(text=element_text(size=12))),
    annotations = list(
        'Proportion of MAGs'=(
            ggplot(mapping=aes(fill=taxa))
            + geom_bar(stat='count', position='fill', color = "black")
            + scale_y_continuous(labels=scales::percent_format())
            + ylab('Proportion of MAGs'))) 
     + patchwork::plot_layout(heights=c(0.5, 0.5, 0.2)))

#all of the small MAGs in the database

allsmallMAGs_taxa_done <- read_excel("allsmallMAGs_taxa_done.xlsx") %>%
  filter(Completion_Minus_Red >= 70) %>%
  select(shortID, Selected_taxa)

all_taxa <- allsmallMAGs_taxa_done %>%
  group_by(Selected_taxa) %>%
  tally() %>%
  arrange(desc(n))

#counting the total number of MAGs in each taxonomic group

#repeat counting the total number of MAGs in each taxonomic group and at each genepres
MAGscompleteness <- read_excel("MAGscompleteness.xlsx")
MAGscompleteness <- MAGscompleteness %>%
  rename("shortID" = "Genome_Id")


smallMAGlist_taxa_updated_plot <- left_join(smallMAGlist_taxa_updated,
                                            MAGscompleteness,
                                            by = "shortID") %>%
  filter(Completion_Minus_Red >= 70,
         genepres != "none") 

cat_taxa <- left_join(smallMAGlist_taxa_updated_plot,
                      allsmallMAGs_taxa_done,
                      by = "shortID") %>%
  group_by(Selected_taxa.y, genepres) %>%
  tally() %>%
  arrange(desc(n)) %>%
  rename("ncatMAGs" = "n",
         "Selected_taxa" = "Selected_taxa.y") 

#merge two together
smallMAGs_alltaxa <- merge(cat_taxa,
                  all_taxa,
                  by = "Selected_taxa",
                  all.x = TRUE,
                  all.y = TRUE)

smallMAGs_alltaxa$genepres[is.na(smallMAGs_alltaxa$genepres)] <- "none"
smallMAGs_alltaxa$ncatMAGs[is.na(smallMAGs_alltaxa$ncatMAGs)] <- 0

smallMAGs_alltaxa_prop <- smallMAGs_alltaxa %>%
  mutate(propMAGs = ncatMAGs/n) %>%
  filter(genepres != "none") %>%
  group_by(Selected_taxa) %>%
  mutate(total_prop = sum(propMAGs),
         totalcat = sum(ncatMAGs)) %>%
           arrange(desc(total_prop)) %>%
  filter(totalcat > 1)

#binding the table with the raw counts of the MAGs

b <- ggplot(data = smallMAGs_alltaxa_prop, aes(x = reorder(Selected_taxa, total_prop), y = ncatMAGs)) + coord_flip() +
  geom_col(aes(fill = genepres), color = "black") + ylab("Number of MAGs") + xlab("Taxa") + theme(text=element_text(size=12)) + theme_classic() + 
  scale_y_continuous(expand = expansion(mult = c(0, .1))) + theme(legend.position="none") + scale_fill_manual(breaks = c("KatE", "KatG", "KatGKatE"),
                        values=c("purple", "#1d91c0", "darkgreen"))

c <- ggplot(data = smallMAGs_alltaxa_prop, aes(x = reorder(Selected_taxa, total_prop), y = propMAGs)) + coord_flip() +
  geom_col(aes(fill = genepres), color = "black") + ylab("Proportion of MAGs") + xlab("Taxa") + theme(text=element_text(size=12)) + theme_classic() + 
  scale_y_continuous(expand = expansion(mult = c(0, .1))) + scale_fill_manual(limits = c("KatE", "KatG", "KatGKatE"),
                        values=c("purple", "#1d91c0", "darkgreen"), labels=c("katE", "katG", "katG and katE")) + guides(fill=guide_legend(title="Catalase type")) 

   
```

```{r}
#MAGS work
MAGscompleteness <- read_excel("MAGscompleteness.xlsx")
MAGscompleteness <- MAGscompleteness %>%
  rename("shortID" = "Genome_Id")

  # filter(Biota != "Archaea")


#importing MAGs
abundance_matrix_KatG_MAGs <- read.delim("~/Downloads/OceanGeneAtlas_Data_20240213-02_49_KatG_MAGs/abundance_matrix_KatG_MAGs.csv") 

environmental_parameters_catalase_MAGs <- read.delim("~/Downloads/OceanGeneAtlas_Data_20240213-02_49_KatG_MAGs/environmental_parameters_KatG_MAGs.csv") %>%
  select(-OGA_ID)


abundance_matrix_KatG_MAGs <- abundance_matrix_KatG_MAGs %>%
  janitor::row_to_names(1) %>%
  extract(Gene_ID, c("shortID", "Gene_ID", "what"), "(.*)_([^_]+)_([^_]+)") %>%
  select(-Gene_ID, -what) %>%
   separate(col=taxonomy, into=c('Biota', 'Taxa1', 'Taxa2', 'Taxa3', 'Taxa4', 'Taxa5', 'Taxa6', 'Taxa7', 'Taxa8', 'Taxa9'), sep=';')



#merge with MAG completion table
catalase_MAGcompl <- left_join(abundance_matrix_KatG_MAGs, MAGscompleteness, 
                               by="shortID")

#filter for completeness
catalase_MAGcompl_filt <- catalase_MAGcompl %>%
  filter(Completion_Minus_Red > 70)

########drop check 
# filteredMAGs <- anti_join(catalase_MAGcompl, catalase_MAGcompl_filt, by = c("Gene_ID" = "Gene_ID"))

#rotate so have gene abundance column
catalase_taxa_rotated <- catalase_MAGcompl_filt %>%
  pivot_longer(12:911, names_to = "sample_ID", values_to = "catalase_abundance") 

#filtering the MAGs for SRF and 0.22-3µm size fraction
#catalase environmental parameters
catalase_enviro_SRF <- left_join(catalase_taxa_rotated,
                            environmental_parameters_catalase_MAGs,
                            by = "sample_ID")

#instead of summing by region, I want to AVG by the sample ID to combine samples with the same shortID
catalase <- catalase_enviro_SRF %>%
  transform(catalase_abundance = as.numeric(catalase_abundance)) %>%
              group_by(shortID, sample_ID) %>%
              summarise(catalase_abundance = sum(catalase_abundance))

#now that I have filtered and unique short IDs with their %reads, I can change the reads greater than 0.0001 for each sample to 1 (meaning hit)
catalase_hits <- catalase %>%
mutate(catalasehit = case_when(catalase_abundance  > 0 ~1,
                           catalase_abundance  == 0 ~0))

KatG_enviro <- left_join(catalase_hits,
                         environmental_parameters_catalase_MAGs,
                         by = "sample_ID")

#run catalases here
abundance_matrix_KatE <- read.delim("~/Downloads/OceanGeneAtlas_Data_20231129-01_48_KatE_MAGs/abundance_matrix.csv")

environmental_parameters_KatE <- read.delim("~/Downloads/OceanGeneAtlas_Data_20231129-01_48_KatE_MAGs/environmental_parameters.csv")



abundance_matrix_KatE_new <- abundance_matrix_KatE %>%
  janitor::row_to_names(1) %>%
  extract(Gene_ID, c("shortID", "Gene_ID", "what"), "(.*)_([^_]+)_([^_]+)") %>%
  select(-Gene_ID, -what) %>%
   separate(col=taxonomy, into=c('Biota', 'Taxa1', 'Taxa2', 'Taxa3', 'Taxa4', 'Taxa5', 'Taxa6', 'Taxa7', 'Taxa8', 'Taxa9'), sep=';') 
  # filter(Biota != "Archaea")


#joining with completeness
abundance_matrix_KatE_compl <- left_join(abundance_matrix_KatE_new, MAGscompleteness, 
                           by="shortID")


#filtering out the low qual MAGs
KatE_filt <- abundance_matrix_KatE_compl %>%
  filter(Completion_Minus_Red > 70)

#reformatting
KatE_rotated <- KatE_filt %>%
  pivot_longer(12:764, names_to = "sample_ID", values_to = "KatE_abundance") 

#merge with environmental data
KatE_enviro <- left_join(KatE_rotated,
                         environmental_parameters_KatE,
                         by = "sample_ID")

#IMPORTANT STEP! only interested in the samples in my size fraction and depth
KatE_enviro_SRF <- KatE_enviro %>%
  filter(depth == "SRF",
         size.fraction == "[0.22-3µm]" | size.fraction == "[0.8-5µm]")

#instead of summing by region, I want to sum by the sample ID to combine samples with the same shortID
KatE <- KatE_enviro_SRF %>%
  transform(KatE_abundance = as.numeric(KatE_abundance)) %>%
  group_by(shortID, sample_ID) %>%
  summarise(KatE_abundance = sum(KatE_abundance))

#now that I have filtered and unique short IDs with their %reads, I can change the reads greater than 0.0001 for each sample to 1 (meaning hit)
KatE_hits <- KatE %>%
mutate(KatEhit = case_when(KatE_abundance  > 0 ~1,
                           KatE_abundance  == 0 ~0))

KatE_enviro <- left_join(KatE_hits,
                         environmental_parameters_KatE,
                         by = "sample_ID")

#####################################
abundance_matrix_CatMn <- read.delim("~/Downloads/OceanGeneAtlas_Data_20231129-01_53_Cat_Mn_MAGs/abundance_matrix.csv")
environmental_parameters_CatMn <- read.delim("~/Downloads/OceanGeneAtlas_Data_20231129-01_53_Cat_Mn_MAGs/environmental_parameters.csv")



abundance_matrix_CatMn_new <- abundance_matrix_CatMn %>%
  janitor::row_to_names(1) %>%
  extract(Gene_ID, c("shortID", "Gene_ID", "what"), "(.*)_([^_]+)_([^_]+)") %>%
  select(-Gene_ID, -what) %>%
   separate(col=taxonomy, into=c('Biota', 'Taxa1', 'Taxa2', 'Taxa3', 'Taxa4', 'Taxa5', 'Taxa6', 'Taxa7', 'Taxa8', 'Taxa9'), sep=';') 
  # filter(Biota != "Archaea")


#joining with completeness
abundance_matrix_CatMn_compl <- left_join(abundance_matrix_CatMn_new, MAGscompleteness, 
                           by="shortID")


#filtering out the low qual MAGs
Cat_Mn_filt <- abundance_matrix_CatMn_compl %>%
  filter(Completion_Minus_Red > 70)

#reformatting
Cat_Mn_rotated <- Cat_Mn_filt %>%
  pivot_longer(12:313, names_to = "sample_ID", values_to = "Cat_Mn_abundance") 

#merge with environmental data
Cat_Mn_enviro <- left_join(Cat_Mn_rotated,
                         environmental_parameters_CatMn,
                         by = "sample_ID")

#IMPORTANT STEP! only interested in the samples in my size fraction and depth
Cat_Mn_enviro_SRF <- Cat_Mn_enviro %>%
  filter(depth == "SRF",
         size.fraction == "[0.22-3µm]" | size.fraction == "[0.8-5µm]")

#instead of summing by region, I want to sum by the sample ID to combine samples with the same shortID
Cat_Mn <- Cat_Mn_enviro_SRF %>%
  mutate(Cat_Mn_abundance = as.numeric(Cat_Mn_abundance)) %>%
  group_by(shortID, sample_ID) %>%
  summarise(Cat_Mn_abundance = sum(Cat_Mn_abundance))

#now that I have filtered and unique short IDs with their %reads, I can change the reads greater than 0.0001 for each sample to 1 (meaning hit)
Cat_Mn_hits <- Cat_Mn %>%
mutate(CatMnhit = case_when(Cat_Mn_abundance  > 0 ~1,
                           Cat_Mn_abundance  == 0 ~0))

CatMn_enviro <- left_join(Cat_Mn_hits,
                        environmental_parameters_CatMn,
                        by = "sample_ID")
  

#now, I can count up the hits on each level: sample, station, and region and then divide hits on those levels
# CatMnhits_withenviro <- left_join(Cat_Mn_hits,
#                                  environmental_parameters_CatMn,
#                                  by = "sample_ID")

##rubisco
MAGscompleteness <- read_excel("MAGscompleteness.xlsx")
MAGscompleteness <- MAGscompleteness %>%
  rename("shortID" = "Genome_Id")

abundance_matrix_rub <- abundance_matrix <- read.delim("~/Downloads/OceanGeneAtlas_Data_20240315-15_38_Rub_MAGs/abundance_matrix.csv")

environmental_parameters_rub <- read.delim("~/Downloads/OceanGeneAtlas_Data_20240315-15_38_Rub_MAGs/environmental_parameters.csv")



abundance_matrix_rub_new <- abundance_matrix_rub %>%
  janitor::row_to_names(1) %>%
  extract(Gene_ID, c("shortID", "Gene_ID", "what"), "(.*)_([^_]+)_([^_]+)") %>%
  select(-Gene_ID, -what) %>%
   separate(col=taxonomy, into=c('Biota', 'Taxa1', 'Taxa2', 'Taxa3', 'Taxa4', 'Taxa5', 'Taxa6', 'Taxa7', 'Taxa8', 'Taxa9'), sep=';') 
  # filter(Biota != "Archaea")


#joining with completeness
abundance_matrix_rub_compl <- left_join(abundance_matrix_rub_new, MAGscompleteness, 
                           by="shortID")


#filtering out the low qual MAGs
rub_filt <- abundance_matrix_rub_compl %>%
  filter(Completion_Minus_Red > 70)

#reformatting
rub_rotated <- rub_filt %>%
  pivot_longer(12:803, names_to = "sample_ID", values_to = "rub_abundance") 

#merge with environmental data
rub_enviro <- left_join(rub_rotated,
                         environmental_parameters_rub,
                         by = "sample_ID")

#IMPORTANT STEP! only interested in the samples in my size fraction and depth
rub_enviro_SRF <- rub_enviro %>%
  filter(depth == "SRF",
         size.fraction == "[0.22-3µm]" | size.fraction == "[0.8-5µm]")

#instead of summing by region, I want to sum by the sample ID to combine samples with the same shortID
rub <- rub_enviro_SRF %>%
  mutate(rub_abundance = as.numeric(rub_abundance)) %>%
  group_by(shortID, sample_ID) %>%
  summarise(rub_abundance = sum(rub_abundance))

#now that I have filtered and unique short IDs with their %reads, I can change the reads greater than 0.0001 for each sample to 1 (meaning hit)
rub_hits <- rub %>%
mutate(rubhit = case_when(rub_abundance  > 0 ~1,
                          rub_abundance  == 0 ~0))

rub_enviro <- left_join(rub_hits,
                        environmental_parameters_rub,
                        by = "sample_ID")
  
```

```{r}
#venn diagram
#including all MAGs
KatGhits_forvenn <- catalase_hits %>%
  select(shortID, catalasehit) %>%
  group_by(shortID) %>%
  summarise(catalasehit = sum(catalasehit))

CatMnhits_forvenn <- Cat_Mn_hits %>%
select(shortID, CatMnhit) %>%
  group_by(shortID) %>%
  summarise(CatMnhit = sum(CatMnhit))

KatE_forvenn <- KatE_hits %>%
select(shortID, KatEhit) %>%
  group_by(shortID) %>%
  summarise(KatEhit = sum(KatEhit))

rub_forvenn <- rub_hits %>%
select(shortID, rubhit) %>%
  group_by(shortID) %>%
  summarise(rubhit = sum(rubhit))

#combine datasets
cat_catmn_forvenn <- merge(KatGhits_forvenn,
                  CatMnhits_forvenn,
                  by = "shortID",
                  all.x = TRUE,
                  all.y = TRUE)

cat_catmn_KatE_forvenn <- merge(cat_catmn_forvenn,
                  KatE_forvenn,
                  by = "shortID",
                  all.x = TRUE,
                  all.y = TRUE)

cat_catmn_KatE_rub_forvenn <- merge(cat_catmn_KatE_forvenn,
                  rub_forvenn,
                  by = "shortID",
                  all.x = TRUE,
                  all.y = TRUE)


cat_catmn_KatE_rub_forvenn[is.na(cat_catmn_KatE_rub_forvenn)] <- 0

#venn diagram
library(ggVennDiagram)

#venn diagram across all regions
KatG <- cat_catmn_KatE_rub_forvenn[!cat_catmn_KatE_rub_forvenn$catalasehit == 0 ,"shortID"]
Manganese_catalase <- cat_catmn_KatE_rub_forvenn[!cat_catmn_KatE_rub_forvenn$CatMnhit == 0 ,"shortID"]
KatE <- cat_catmn_KatE_rub_forvenn[!cat_catmn_KatE_rub_forvenn$KatEhit == 0 ,"shortID"]
rub <- cat_catmn_KatE_rub_forvenn[!cat_catmn_KatE_rub_forvenn$rubhit == 0 ,"shortID"]


input <- list(KatG = KatG, KatE = KatE, Manganese_catalase = Manganese_catalase)

input <- list(KatG = KatG, KatE = KatE, rub = rub)


library(ggVennDiagram)
  ggVennDiagram(input, set_size = 5, set_color = "black", label_alpha = 0) + 
    labs(fill="Gene presence") +
    scale_fill_gradient(low = "#F4FAFE", high = "#4981BF") +
    scale_color_manual(values = c("black","black","black", "black")) +
    scale_x_continuous(expand = expansion(mult = .2)) 

library(ggvenn)
  
names(input) <- c("KatG","KatE","Rubisco")
ggvenn(
  input, 
  fill_color = c("blue", "purple", "darkgreen"),
  stroke_size = 0.5, text_size = 6, set_name_size = 6,
  digits = 
)


```

```{r}
#michelle data

Genome_sizes <- read_excel("Genome_sizes.xlsx") %>%
  rename("Genome" = "...1",
         "Genome_Size" = "Geome Size")

Alt_catanalysis <- read_excel("Alt_catanalysis.xlsx")

michelledata <- inner_join(Genome_sizes,
                           Alt_catanalysis,
                           by = "Genome")

Catalase_Totals <- read.csv("~/Desktop/Dyhrman Lab/Desktop_DyhrmanLab/Catalase_Totals.csv") %>%
  rename("Genome" = "genomesnames.name")


michelledata_copies <- inner_join(michelledata,
              Catalase_Totals,
              by = "Genome") %>%
  transform(Depth = as.numeric(Depth)) %>%
  select(-KatG, -KatE) %>%
  rename("KatG" = "K03782",
         "KatE" = "K03781") %>%
  filter(Genome_Size != 0,
         Depth <= 10,
         Completeness >= 0) %>%
  mutate(KatGKatEratio = as.character(KatG/KatE),
         totalchar = as.character(total),
         KatGchar = as.character(KatG),
         KatEchar = as.character(KatE),
         genepres = case_when(KatG > KatE ~ "moreKatG",
                             KatG < KatE ~ "moreKatE",
                             KatG == KatE ~ "same",
                        KatG == 0 & KatE > 0 ~ "noKatG",
                        KatG > 0 & KatE == 0 ~ "noKatE",
                        KatG == 0 & KatE == 0 ~ "none"
                        ),
         oneorother = case_when(KatG > 0 & KatE > 0 ~ "both",
                                KatG > 0 & KatE == 0 ~ "KatG",
                                KatG == 0 & KatE > 0 ~ "KatE"
         ))

# cats_MAGlength_taxa <- left_join(cats_MAGlength,
#                                  mergedtaxa_updated,
#                                  by = "shortID")) %>%
#   drop_na(Depth)

#ask if variables are normally distributed
library(ggpubr)
library(rstatix)
#redoing all of the tests to check work

michelledata_copies_clades1_updated_lat <- michelledata_copies_clades1_updated %>%
  drop_na(latitude) %>%
  transform(latitude = as.numeric(latitude))

michelledata_copies_clades1_updated_depth <- michelledata_copies_clades1_updated %>%
  drop_na(Depth.x) %>%
  transform(depth = as.numeric(Depth.x))

michelledata_copies_clades1_updated_sizefraction <- michelledata_copies_clades1_updated %>%
  drop_na(Size.Fraction) 


#do all of the tests
model_depth_all <- lm(KatGchar.y ~ Depth.x, data = michelledata_copies_clades1_updated_depth)

shapiro_test(residuals(model_depth_all))

kruskal.test(KatGchar.y ~ Depth.x, data = michelledata_copies_clades1_updated_depth)

#finding out which groups
pairwise.wilcox.test(michelledata_copies_clades1_updated_depth$Depth.x, michelledata_copies_clades1_updated_depth$KatGchar.y,
                 p.adjust.method = "BH")

#########
model_depth_all_katE <- lm(KatEchar.y ~ Depth.x, data = michelledata_copies_clades1_updated_depth)

shapiro_test(residuals(model_depth_all_katE))

kruskal.test(KatEchar.y ~ Depth.x, data = michelledata_copies_clades1_updated_depth)

#finding out which groups
pairwise.wilcox.test(michelledata_copies_clades1_updated_depth$Depth.x, michelledata_copies_clades1_updated_depth$KatEchar.y,
                 p.adjust.method = "BH")
#################
model_depth_all_genomesize_katG <- lm(KatGchar.y ~ Genome_Size.x, data = michelledata_copies_clades1_updated)

shapiro_test(residuals(model_depth_all_genomesize_katG))

kruskal.test(KatGchar.y ~ Genome_Size.x, data = michelledata_copies_clades1_updated_depth)

#finding out which groups
pairwise.wilcox.test(michelledata_copies_clades1_updated$Genome_Size.x, michelledata_copies_clades1_updated$KatGchar.y,
                 p.adjust.method = "BH")

##########
model_depth_all_genomesize_katE <- lm(KatEchar.y ~ Genome_Size.x, data = michelledata_copies_clades1_updated)

shapiro_test(residuals(model_depth_all_genomesize_katE))

#finding out which groups
pairwise.wilcox.test(michelledata_copies_clades1_updated$Genome_Size.x, michelledata_copies_clades1_updated$KatEchar.y,
                 p.adjust.method = "BH")

################

model_depth_all_lat_katG <- lm(KatGchar.y ~ latitude, data = michelledata_copies_clades1_updated_lat)

shapiro_test(residuals(model_depth_all_lat_katG))

pairwise.wilcox.test(michelledata_copies_clades1_updated_lat$latitude, michelledata_copies_clades1_updated_lat$KatGchar.y,
                 p.adjust.method = "BH")

###
model_depth_all_lat_katE <- lm(KatEchar.y ~ latitude, data = michelledata_copies_clades1_updated_lat)

shapiro_test(residuals(model_depth_all_lat_katE))

pairwise.wilcox.test(michelledata_copies_clades1_updated_lat$latitude, michelledata_copies_clades1_updated_lat$KatEchar.y,
                 p.adjust.method = "BH")

####
model_depth_all_sizefraction_katG <- lm(KatG.x ~ Size.Fraction, data = michelledata_copies_clades1_updated_sizefraction)

shapiro_test(residuals(model_depth_all_sizefraction_katG))

pairwise.wilcox.test(michelledata_copies_clades1_updated_sizefraction$KatG.x, michelledata_copies_clades1_updated_sizefraction$Size.Fraction,
                 p.adjust.method = "BH")

##
model_depth_all_sizefraction_katE <- lm(KatE.x ~ Size.Fraction, data = michelledata_copies_clades1_updated_sizefraction)

shapiro_test(residuals(model_depth_all_sizefraction_katE))

pairwise.wilcox.test(michelledata_copies_clades1_updated_sizefraction$KatE.x, michelledata_copies_clades1_updated_sizefraction$Size.Fraction,
                 p.adjust.method = "BH")


##########looking at just two species
michelledata_copies_clades1_updated_2species <- michelledata_copies_clades1_updated %>%
  dplyr::filter(Species == "Alteromonas australica" | Species == "Alteromonas macleodii")

model_depth_all_species_katG <- lm(KatG.x ~ Species, data = michelledata_copies_clades1_updated_2species)

shapiro_test(residuals(model_depth_all_species_katG))

pairwise.wilcox.test(michelledata_copies_clades1_updated_2species$KatG.x, michelledata_copies_clades1_updated_2species$Species,
                 p.adjust.method = "BH")

#######
model_depth_all_species_katE <- lm(KatE.x ~ Species, data = michelledata_copies_clades1_updated_2species)

shapiro_test(residuals(model_depth_all_species_katE))

pairwise.wilcox.test(michelledata_copies_clades1_updated_2species$KatE.x, michelledata_copies_clades1_updated_2species$Species, p.adjust.method = "BH")

michelledata_copies_clades1_updated_2species_lat <- michelledata_copies_clades1_updated_2species %>%
  transform(latitude = as.numeric(latitude)) %>%
  drop_na(latitude)
#######
model_depth_all_species_lat <- lm(latitude ~ Species, data = michelledata_copies_clades1_updated_2species_lat)

shapiro_test(residuals(model_depth_all_species_lat))

pairwise.wilcox.test(michelledata_copies_clades1_updated_2species_lat$latitude, michelledata_copies_clades1_updated_2species_lat$Species,
                 p.adjust.method = "BH")
#####

michelledata_copies_clades1_updated_2species_depth <- michelledata_copies_clades1_updated_2species %>%
  transform(latitude = as.numeric(Depth.x)) %>%
  drop_na(Depth.x)


model_depth_all_species_depth <- lm(Depth.x ~ Species, data = michelledata_copies_clades1_updated_2species_depth)

shapiro_test(residuals(model_depth_all_species_depth))

pairwise.wilcox.test(michelledata_copies_clades1_updated_2species_depth$Depth.x, michelledata_copies_clades1_updated_2species_depth$Species,
                 p.adjust.method = "BH")

#####
model_depth_all_species_genome <- lm(Genome_Size.x ~ Species, data = michelledata_copies_clades1_updated_2species_depth)

shapiro_test(residuals(model_depth_all_species_genome))

pairwise.wilcox.test(michelledata_copies_clades1_updated_2species$Genome_Size.x, michelledata_copies_clades1_updated_2species$Species,
                 p.adjust.method = "BH")

###########
model_depth_all_species_genome <- lm(latitude ~ Species, data = michelledata_copies_clades1_updated_2species_lat)

shapiro_test(residuals(model_depth_all_species_genome))

pairwise.wilcox.test(michelledata_copies_clades1_updated_2species_lat$latitude, michelledata_copies_clades1_updated_2species_lat$Species,
                 p.adjust.method = "BH")

#medians
michelledata_copies_clades1_updated_2species_median_katG <- michelledata_copies_clades1_updated %>%
  group_by(KatGchar.x) %>%
  dplyr::summarise(katG_med = median(Genome_Size.x))

  
michelledata_copies_clades1_updated_2species_median_katE <- michelledata_copies_clades1_updated %>%
  group_by(KatEchar.x) %>%
  dplyr::summarise(katE_med = median(Genome_Size.x))


#############   
michelledata_copies_clades1_updated_2species_median_latkatG <- michelledata_copies_clades1_updated_lat %>%
  group_by(KatGchar.x) %>%
  dplyr::summarise(katG_med = median(latitude))

michelledata_copies_clades1_updated_2species_median_latkatE <- michelledata_copies_clades1_updated_lat %>%
  group_by(KatEchar.x) %>%
  dplyr::summarise(katE_med = median(latitude))
                   
michelledata_copies_clades1_updated_2species_median_latkatE <- michelledata_copies_clades1_updated_sizefraction %>%
  group_by(Size.Fraction) %>%
  dplyr::summarise(katG_med = mean(KatG.x))
                
#######ggplots for all of the comparisons
#preparing ggplots for figure

#need to have one column called gene and then a column called copy number and then the depth and genome size and latitude to easily plot katE and katG next to each other

#pivoting the depth table for all species


katG_depth_all <- ggplot(michelledata_copies_clades1_updated_depth, aes(x=KatGchar.x, y=Depth.x)) + 
    geom_boxplot(fill="grey") + theme_bw() + xlab("katG copies per genome") + ylab("depth") + coord_flip() + theme(text = element_text(size = 12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1))

katE_depth_all <- ggplot(michelledata_copies_clades1_updated_depth, aes(x=KatEchar.x, y=Depth.x)) + 
    geom_boxplot(fill="grey") + theme_bw() + xlab("katE copies per genome") + ylab("depth") + coord_flip() + theme(text = element_text(size = 12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1))

########
katG_genomesize_all <- ggplot(michelledata_copies_clades1_updated, aes(x=KatGchar.x, y=Genome_Size.x)) + 
    geom_boxplot(fill="grey") + theme_bw() + xlab("katG copies per genome") + ylab("Genome length (bp)") + coord_flip() + theme(text = element_text(size = 12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) 

katE_genomesize_all <- ggplot(michelledata_copies_clades1_updated, aes(x=KatEchar.x, y=Genome_Size.x)) + 
    geom_boxplot(fill="grey") + theme_bw() + xlab("katE copies per genome") + ylab("Genome length (bp)") + coord_flip() + theme(text = element_text(size = 12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) 

katG_latitude_all <- ggplot(michelledata_copies_clades1_updated_lat, aes(x=KatGchar.x, y=latitude)) + 
    geom_boxplot(fill="grey") + theme_bw() + xlab("katG copies per genome") + ylab("latitude") + coord_flip() + theme(text = element_text(size = 12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) 

katE_latitude_all <- ggplot(michelledata_copies_clades1_updated_lat, aes(x=KatEchar.x, y=latitude)) + 
    geom_boxplot(fill="grey") + theme_bw() + xlab("katE copies per genome") + ylab("latitude") + coord_flip() + theme(text = element_text(size = 12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) 

katG_sizefraction_all <- ggplot(michelledata_copies_clades1_updated_sizefraction, aes(x=(KatG.x), y=Size.Fraction)) + 
    geom_boxplot(fill="grey") + theme_bw() + xlab("katG copies per genome") + ylab("size fraction") + coord_flip() + theme(text = element_text(size = 12), axis.text.x=element_text(angle = 90),                                   panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) 

katG_species_ <- ggplot(michelledata_copies_clades1_updated_sizefraction, aes(x=(Depth.x), y=Size.Fraction)) + 
    geom_boxplot(fill="grey") + theme_bw() + xlab("katG copies per genome") + ylab("size fraction") + coord_flip() + theme(text = element_text(size = 12), axis.text.x=element_text(angle = 90),                                   panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) 
##making the grid

grid.arrange(katG_genomesize_all, katE_genomesize_all, katG_latitude_all, katE_latitude_all, katG_sizefraction_all,  ncol = 2)


mydata_katG <- ggplot(KatG_catmn_KatE_genome_size_noalt, aes(x=as.character(KatG_copy), y=total_length)) + 
    geom_boxplot(fill="grey") + theme_bw() + xlab("All taxa katG copies") + ylab("Genome length (bp)") + coord_flip() + theme(text = element_text(size = 12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) 

grid.arrange(mydata_allcat, michelle_allcat, mydata_katG, michelle_katG, mydata_katE, michelle_katE, ncol = 2)                                         






###################
abundance_matrix_KatG_MAGs <- read.delim("~/Downloads/OceanGeneAtlas_Data_20240213-02_49_KatG_MAGs/abundance_matrix_KatG_MAGs.csv") 

environmental_parameters_catalase_MAGs <- read.delim("~/Downloads/OceanGeneAtlas_Data_20240213-02_49_KatG_MAGs/environmental_parameters_KatG_MAGs.csv") %>%
  select(-OGA_ID)


abundance_matrix_KatG_hitcount <- abundance_matrix_KatG_MAGs %>%
  janitor::row_to_names(1) %>%
  mutate(fullID = Gene_ID) %>%
  extract(Gene_ID, c("shortID", "Gene_ID", "what"), "(.*)_([^_]+)_([^_]+)") %>%
  select(-Gene_ID, -what) 



#merge with MAG completion table
catalase_MAGcompl_hitcount <- merge(abundance_matrix_KatG_hitcount, MAGscompleteness, 
                               by="shortID",
#                   by = "sample_ID",
#                   all.x = TRUE,
#                   all.y = TRUE))

#filter for completeness
catalase_MAGcompl_filt_hitcount <- catalase_MAGcompl_hitcount %>%
  filter(Completion_Minus_Red > 70)

########drop check 
# filteredMAGs <- anti_join(catalase_MAGcompl, catalase_MAGcompl_filt, by = c("Gene_ID" = "Gene_ID"))

#rotate so have gene abundance column
catalase_taxa_rotated <- catalase_MAGcompl_filt_hitcount %>%
  pivot_longer(3:902, names_to = "sample_ID", values_to = "catalase_abundance") 

#filtering the MAGs for SRF and 0.22-3µm size fraction
#catalase environmental parameters
catalase_enviro <- left_join(catalase_taxa_rotated,
                            environmental_parameters_catalase_MAGs,
                            by = "sample_ID")

catalase_enviro_SRF <- catalase_enviro %>%
  filter(depth == "SRF",
         size.fraction == "[0.22-3µm]" | size.fraction == "[0.8-5µm]")

#group by the fullID and sum catalase abundance
KatGMAGs_formichelleplot <- catalase_enviro_SRF %>%
  transform(catalase_abundance = as.numeric(catalase_abundance)) %>%
  group_by(fullID) %>%
  summarise(KatGabundance = sum(catalase_abundance),
            shortID = shortID) %>%
  distinct(fullID, .keep_all = TRUE) %>%
  group_by(shortID) %>%
  tally() %>%
  rename("nKatGhitsbyMAG" = "n")

#########################################
#repeat for KatE
abundance_matrix_KatE <- read.delim("~/Downloads/OceanGeneAtlas_Data_20231129-01_48_KatE_MAGs/abundance_matrix.csv")

environmental_parameters_KatE <- read.delim("~/Downloads/OceanGeneAtlas_Data_20231129-01_48_KatE_MAGs/environmental_parameters.csv")



abundance_matrix_KatE_hitcount <- abundance_matrix_KatE %>%
  janitor::row_to_names(1) %>%
  mutate(fullID = Gene_ID) %>%
  extract(Gene_ID, c("shortID", "Gene_ID", "what"), "(.*)_([^_]+)_([^_]+)") %>%
  select(-Gene_ID, -what) 


#joining with completeness
abundance_matrix_KatE_compl_hitcount <- left_join(abundance_matrix_KatE_hitcount, MAGscompleteness, 
                           by="shortID")


#filtering out the low qual MAGs
KatE_filt_hitcount <- abundance_matrix_KatE_compl_hitcount %>%
  filter(Completion_Minus_Red > 70)

#reformatting
KatE_rotated <- KatE_filt_hitcount %>%
  pivot_longer(3:755, names_to = "sample_ID", values_to = "KatE_abundance") 

#merge with environmental data
KatE_enviro <- left_join(KatE_rotated,
                         environmental_parameters_KatE,
                         by = "sample_ID")

#IMPORTANT STEP! only interested in the samples in my size fraction and depth
KatE_enviro_SRF <- KatE_enviro %>%
  filter(depth == "SRF",
         size.fraction == "[0.22-3µm]" | size.fraction == "[0.8-5µm]")

KatEMAGs_formichelleplot <- KatE_enviro_SRF %>%
transform(KatE_abundance = as.numeric(KatE_abundance)) %>%
  group_by(fullID) %>%
  summarise(KatEabundance = sum(KatE_abundance),
            shortID = shortID) %>%
  distinct(fullID, .keep_all = TRUE) %>%
  group_by(shortID) %>%
  tally() %>%
  rename("nKatEhitsbyMAG" = "n")

##repeat for CatMn
abundance_matrix_CatMn <- read.delim("~/Downloads/OceanGeneAtlas_Data_20231129-01_53_Cat_Mn_MAGs/abundance_matrix.csv")
environmental_parameters_CatMn <- read.delim("~/Downloads/OceanGeneAtlas_Data_20231129-01_53_Cat_Mn_MAGs/environmental_parameters.csv")



abundance_matrix_CatMn_hitcount <- abundance_matrix_CatMn %>%
  janitor::row_to_names(1) %>%
  mutate(fullID = Gene_ID) %>%
  extract(Gene_ID, c("shortID", "Gene_ID", "what"), "(.*)_([^_]+)_([^_]+)") %>%
  select(-Gene_ID, -what) 


#joining with completeness
abundance_matrix_CatMn_compl_hitcount <- left_join(abundance_matrix_CatMn_hitcount, MAGscompleteness, 
                           by="shortID")


#filtering out the low qual MAGs
Cat_Mn_filt_hitcount <- abundance_matrix_CatMn_compl_hitcount %>%
  filter(Completion_Minus_Red > 70)

#reformatting
Cat_Mn_rotated <- Cat_Mn_filt_hitcount %>%
  pivot_longer(3:304, names_to = "sample_ID", values_to = "Cat_Mn_abundance") 

#merge with environmental data
Cat_Mn_enviro <- left_join(Cat_Mn_rotated,
                         environmental_parameters_CatMn,
                         by = "sample_ID")

#IMPORTANT STEP! only interested in the samples in my size fraction and depth
Cat_Mn_enviro_SRF <- Cat_Mn_enviro %>%
  filter(depth == "SRF",
         size.fraction == "[0.22-3µm]" | size.fraction == "[0.8-5µm]")

#instead of summing by region, I want to sum by the sample ID to combine samples with the same shortID
CatMnMAGs_formichelleplot <- Cat_Mn_enviro_SRF %>%
  transform(Cat_Mn_abundance = as.numeric(Cat_Mn_abundance)) %>%
  group_by(fullID) %>%
  summarise(CatMnabundance = sum(Cat_Mn_abundance),
            shortID = shortID) %>%
  distinct(fullID, .keep_all = TRUE) %>%
  group_by(shortID) %>%
  tally() %>%
  rename("nCatMnhitsbyMAG" = "n")

#merging the datasets together
KatG_hit_tomerge <- KatGMAGs_formichelleplot %>%
  select(shortID, nKatGhitsbyMAG) 
KatE_hit_tomerge <- KatEMAGs_formichelleplot %>%
  select(shortID, nKatEhitsbyMAG) 
CatMn_hit_tomerge <- CatMnMAGs_formichelleplot %>%
  select(shortID, nCatMnhitsbyMAG) 
KatG_KatE <- merge(KatG_hit_tomerge,
                  KatE_hit_tomerge,
                  by = "shortID",
                  all.x = TRUE,
                  all.y = TRUE)

cats <- merge(KatG_KatE,
                  CatMn_hit_tomerge,
                  by = "shortID",
                  all.x = TRUE,
                  all.y = TRUE)

MAGslist <- read_excel("MAGscompleteness.xlsx") %>%
   filter(Completion_Minus_Red > 70) %>%
  rename("shortID" = "Genome_Id")

cats_allMAGs <- merge(cats,
                  MAGslist,
                  by = "shortID",
                  all.x = TRUE,
                  all.y = TRUE)

cats_allMAGs[is.na(cats_allMAGs)] <- 0

#merging dataset with genome size dataset
MAGlength <- read_excel("MAGlength.xlsx") %>%
  rename("shortID" = "Genome_Id")

cats_MAGlength <- left_join(cats_allMAGs,
                            MAGlength,
                            by = "shortID") %>%
  mutate(totalcat = nKatGhitsbyMAG + nKatEhitsbyMAG + nCatMnhitsbyMAG,
         totalcatcar = as.character(totalcat),
         KatGchar = as.character(nKatGhitsbyMAG),
         KatEchar = as.character(nKatEhitsbyMAG),
        KatGKatEratio = as.character(nKatGhitsbyMAG/nKatEhitsbyMAG),
        genepres = case_when(nKatGhitsbyMAG > nKatEhitsbyMAG ~ "moreKatG",
                             nKatGhitsbyMAG < nKatEhitsbyMAG ~ "moreKatE",
                             nKatGhitsbyMAG == nKatEhitsbyMAG ~ "same",
                        nKatGhitsbyMAG == 0 & nKatEhitsbyMAG > 0 ~ "noKatG",
                        nKatGhitsbyMAG > 0 & nKatEhitsbyMAG == 0 ~ "noKatE",
                        nKatGhitsbyMAG == 0 & nKatEhitsbyMAG == 0 ~ "none"
                        ),
        nocat = case_when(totalcatcar == 0 ~ 1),
        
         oneorother = case_when(nKatGhitsbyMAG > 0 & nKatEhitsbyMAG > 0 ~ "both",
                                nKatGhitsbyMAG > 0 & nKatEhitsbyMAG == 0 ~ "KatG",
                                nKatGhitsbyMAG == 0 & nKatEhitsbyMAG > 0 ~ "KatE"))
         

cats_MAGlength$nocat[is.na(cats_MAGlength$nocat)] <- 0


#statistical tests
#normally distributed
model_mydata <- lm(total_length ~ totalcatcar, data = cats_MAGlength)
#not normally distributed
shapiro_test(residuals(model_mydata))

kruskal.test(total_length ~ totalcatcar, data = cats_MAGlength)

#finding out which groups
pairwise.wilcox.test(cats_MAGlength$total_length, cats_MAGlength$totalcatcar,
                 p.adjust.method = "BH")

#repeat for the gene presence question
model2_mydata <- lm(total_length ~ genepres, data = cats_MAGlength)
#not normally distributed
shapiro_test(residuals(model2_mydata))

kruskal.test(total_length ~ genepres, data = cats_MAGlength)

#finding out which groups
pairwise.wilcox.test(cats_MAGlength$total_length, cats_MAGlength$genepres,
                 p.adjust.method = "BH")

#KatG only
model3_mydata <- lm(total_length ~ KatGchar, data = cats_MAGlength)
#not normally distributed
shapiro_test(residuals(model3_mydata))

kruskal.test(total_length ~ KatGchar, data = cats_MAGlength)


#KatE only
model4_mydata <- lm(total_length ~ KatEchar, data = cats_MAGlength)
#not normally distributed
shapiro_test(residuals(model4_mydata))

kruskal.test(total_length ~ KatEchar, data = cats_MAGlength)


#finding out which groups
pairwise.wilcox.test(cats_MAGlength$total_length, cats_MAGlength$KatEchar,
                 p.adjust.method = "BH")


mergedtaxa_updated <- read_excel("mergedtaxa_updated.xlsx")

cats_MAGlength_taxa <- left_join(cats_MAGlength,
                                 mergedtaxa_updated,
                                 by = "shortID")

     
#alt only 

cats_MAGlength_taxa_alt <- cats_MAGlength_taxa %>%
  filter(Selected_taxa == "Alteromonadaceae")

###############
library(PieGlyph)
ggplot(data = cats_MAGlength, aes(x = total_length, y = totalcat))+
  geom_pie_glyph(slices = c("nKatGhitsbyMAG", "nKatEhitsbyMAG", "nCatMnhitsbyMAG", "nocat"), radius = 0.2)+
  theme_classic()

ggplot(cats_MAGlength_taxa_alt, aes(x=genepres, y=total_length)) + 
    geom_boxplot() + theme_classic() + xlab("Gene presence") + ylab("Genome length (bp)") + coord_flip() + theme(legend.text = element_text(size = 15),
                                                                                       legend.title = element_text(size = 20),
                                                                                       axis.text = element_text(size = 15),
                                                                                       axis.title = element_text(size = 20)) 

ggplot(data = cats_MAGlength_taxa_alt, aes(x = total_length, y = totalcat))+
  geom_pie_glyph(slices = c("nKatGhitsbyMAG", "nKatEhitsbyMAG", "nCatMnhitsbyMAG"), radius = 0.2)+
  theme_classic()

ggplot(cats_MAGlength_taxa_alt, aes(x=as.character(totalcatcar), y=total_length)) + 
    geom_boxplot()

#top taxa
cats_MAGlength_taxa_order <- cats_MAGlength_taxa %>%
  arrange(desc(totalcat))

#counting up the number of taxa belonging to each category 
cats_MAGlength_taxa_counts <- cats_MAGlength_taxa %>%
  select(totalcatcar, Selected_taxa) %>%
  group_by(totalcatcar, Selected_taxa) %>%
    tally()

#working on each category

cats_MAGlength_taxa_counts_6 <- cats_MAGlength_taxa_counts %>%
  filter(totalcatcar == 6)
         
copy6 <- cats_MAGlength_taxa_counts_6 %>%
  arrange(desc(n))

# top10taxa_KatGrub <- taxa_KatGrub 
#   
# top10taxa_KatGrub$Selected_taxa[c(11:20)] <- "Other"
# 
# top10taxa_KatGrub <- top10taxa_KatGrub %>%
#   group_by(Selected_taxa) %>%
#   summarise(n = sum(n)) %>%
#   arrange(desc(n))

ggplot(copy6, aes(x = "", y = n, fill = fct_inorder(Selected_taxa))) +
  geom_bar(width = 1, stat = "identity") +  
  theme_void() +
  geom_col(color = "black") +
  coord_polar("y", start = 0) +
  guides(fill = guide_legend(title = "taxa")) + ggtitle("6") + theme(plot.title = element_text(hjust = 0.5),
      legend.text = element_text(size = 20)) 
########
#5

cats_MAGlength_taxa_counts_4 <- cats_MAGlength_taxa_counts %>%
  filter(totalcatcar == 4)
         
copy4 <- cats_MAGlength_taxa_counts_4 %>%
  arrange(desc(n))

top10taxa_4 <- copy4 

top10taxa_4$Selected_taxa[c(11)] <- "Other"

top10taxa_4 <- top10taxa_4 %>%
  group_by(Selected_taxa) %>%
  summarise(n = sum(n)) %>%
  arrange(desc(n))

ggplot(top10taxa_4, aes(x = "", y = n, fill = fct_inorder(Selected_taxa))) +
  geom_bar(width = 1, stat = "identity") +  
  theme_void() +
  geom_col(color = "black") +
  coord_polar("y", start = 0) +
  guides(fill = guide_legend(title = "taxa")) + ggtitle("4") + theme(plot.title = element_text(hjust = 0.5),
      legend.text = element_text(size = 20)) 

######3

cats_MAGlength_taxa_counts_3 <- cats_MAGlength_taxa_counts %>%
  filter(totalcatcar == 3)
         
copy3 <- cats_MAGlength_taxa_counts_3 %>%
  arrange(desc(n))

top10taxa_3 <- copy3 

top10taxa_3$Selected_taxa[c(11:22)] <- "Other"

top10taxa_3 <- top10taxa_3 %>%
  group_by(Selected_taxa) %>%
  summarise(n = sum(n)) %>%
  arrange(desc(n))

ggplot(top10taxa_3, aes(x = "", y = n, fill = fct_inorder(Selected_taxa))) +
  geom_bar(width = 1, stat = "identity") +  
  theme_void() +
  geom_col(color = "black") +
  coord_polar("y", start = 0) +
  guides(fill = guide_legend(title = "taxa")) + ggtitle("3") + theme(plot.title = element_text(hjust = 0.5),
      legend.text = element_text(size = 20)) 

######2

cats_MAGlength_taxa_counts_2 <- cats_MAGlength_taxa_counts %>%
  filter(totalcatcar == 2)
         
copy2 <- cats_MAGlength_taxa_counts_2 %>%
  arrange(desc(n))

top10taxa_2 <- copy2

top10taxa_2$Selected_taxa[c(11:86)] <- "Other"

top10taxa_2 <- top10taxa_2 %>%
  group_by(Selected_taxa) %>%
  summarise(n = sum(n)) %>%
  arrange(desc(n))

ggplot(top10taxa_2, aes(x = "", y = n, fill = fct_inorder(Selected_taxa))) +
  geom_bar(width = 1, stat = "identity") +  
  theme_void() +
  geom_col(color = "black") +
  coord_polar("y", start = 0) +
  guides(fill = guide_legend(title = "taxa")) + ggtitle("2") + theme(plot.title = element_text(hjust = 0.5),
      legend.text = element_text(size = 20)) 

######1

cats_MAGlength_taxa_counts_1 <- cats_MAGlength_taxa_counts %>%
  filter(totalcatcar == 1)
         
copy1 <- cats_MAGlength_taxa_counts_1 %>%
  arrange(desc(n))

top10taxa_1 <- copy1

top10taxa_1$Selected_taxa[c(11:165)] <- "Other"

top10taxa_1 <- top10taxa_1 %>%
  group_by(Selected_taxa) %>%
  summarise(n = sum(n)) %>%
  arrange(desc(n))

ggplot(top10taxa_1, aes(x = "", y = n, fill = fct_inorder(Selected_taxa))) +
  geom_bar(width = 1, stat = "identity") +  
  theme_void() +
  geom_col(color = "black") +
  coord_polar("y", start = 0) +
  guides(fill = guide_legend(title = "taxa")) + ggtitle("1") + theme(plot.title = element_text(hjust = 0.5),
      legend.text = element_text(size = 20)) 


##quick check


ggplot(allsmallMAGs_cat, aes(x=genepres, y=total_length...2)) + 
    geom_boxplot() + theme_classic() + xlab("Gene presence") + ylab("Genome length (bp)") + coord_flip() + theme(legend.text = element_text(size = 15),
                                                                                       legend.title = element_text(size = 20),
                                                                                       axis.text = element_text(size = 15),
                                                                                       axis.title = element_text(size = 20)) 


#normally distributed
model_mydata <- lm(total_length ~ as.character(totalcat), data = KatG_catmn_KatE_genome_size_noalt)
#not normally distributed
shapiro.test(residuals(model_mydata))

kruskal.test(total_length ~ as.character(totalcat), data = KatG_catmn_KatE_genome_size_noalt)

#finding out which groups
pairwise.wilcox.test(KatG_catmn_KatE_genome_size_noalt$total_length, KatG_catmn_KatE_genome_size_noalt$totalcatcar,
                 p.adjust.method = "BH")

#repeat for the gene presence question
model2_mydata <- lm(total_length ~ genepres, data = cats_MAGlength)
#not normally distributed
shapiro_test(residuals(model2_mydata))

kruskal.test(total_length ~ genepres, data = cats_MAGlength)

#finding out which groups
pairwise.wilcox.test(cats_MAGlength$total_length, cats_MAGlength$genepres,
                 p.adjust.method = "BH")

#KatG only
model3_mydata <- lm(total_length ~ KatGchar, data = cats_MAGlength)
#not normally distributed
shapiro_test(residuals(model3_mydata))

kruskal.test(total_length ~ KatGchar, data = cats_MAGlength)

```

```{r}
psbO_new1 <- mgcv::gam(log10(normpsbO) ~ s(normCatMn) + s(normKatG) + s(normKatE) + s(Temp_co) + s(Sal_co) + s(Coast_km) + s(PAR) + s(NO3_NO2_co) + s(PO4_co) + s(Iron_5m) + s(POC), data=catmodels_notArctic_new_clean_SRF, method = "REML")

psbO_new2 <- mgcv::gam(log10(normpsbO) ~  s(normKatG) + s(normKatE) + s(Temp_co) + s(Sal_co) + s(Coast_km) + s(PAR) + s(NO3_NO2_co) + s(PO4_co) + s(Iron_5m) + s(POC), data=catmodels_notArctic_new_clean_SRF, method = "REML")

psbO_new3 <- mgcv::gam(log10(normpsbO) ~ s(normKatG) + s(normKatE) + s(Temp_co) + s(Coast_km) + s(PAR) + s(NO3_NO2_co) + s(PO4_co) + s(Iron_5m) + s(POC), data=catmodels_notArctic_new_clean_SRF, method = "REML")

psbO_new4 <- mgcv::gam(log10(normpsbO) ~ s(normKatG) + s(normKatE) + s(Temp_co) + s(Coast_km) + s(NO3_NO2_co) + s(PO4_co) + s(Iron_5m) + s(POC), data=catmodels_notArctic_new_clean_SRF, method = "REML")

psbO_new5 <- mgcv::gam(log10(normpsbO) ~ s(normKatG) + s(normKatE) + s(Temp_co) + s(Coast_km) + s(PO4_co) + s(Iron_5m) + s(POC), data=catmodels_notArctic_new_clean_SRF, method = "REML")

psbO_new6 <- mgcv::gam(log10(normpsbO) ~ s(normKatG) + s(normKatE) + s(Temp_co) + s(Coast_km) + s(Iron_5m) + s(POC), data=catmodels_notArctic_new_clean_SRF, method = "REML")

psbO_new7 <- mgcv::gam(log10(normpsbO) ~ s(normKatG) + s(normKatE) + s(Temp_co) + s(Coast_km) +  s(POC), data=catmodels_notArctic_new_clean_SRF, method = "REML")

psbO_new8 <- mgcv::gam(log10(normpsbO) ~ s(normKatG) + s(normKatE) + s(Temp_co) + s(Coast_km) +  s(POC), data=catmodels_notArctic_new_clean_SRF, method = "REML")

psbO_new9 <- mgcv::gam(log10(normpsbO) ~ normKatG + s(normKatE) + s(Temp_co) + s(Coast_km) +  s(POC), data=catmodels_notArctic_new_clean_SRF, method = "REML")

psbO_new10 <- mgcv::gam(log10(normpsbO) ~ normKatG + normKatE + s(Temp_co) + s(Coast_km) +  s(POC), data=catmodels_notArctic_new_clean_SRF, method = "REML")

psbO_new11 <- mgcv::gam(log10(normpsbO) ~ normKatG + normKatE + Temp_co + s(Coast_km) +  s(POC), data=catmodels_notArctic_new_clean_SRF, method = "REML")

psbO_new12 <- mgcv::gam(log10(normpsbO) ~ normKatG + normKatE + Temp_co + Coast_km +  s(POC), data=catmodels_notArctic_new_clean_SRF, method = "REML")

psbO_new13 <- mgcv::gam(log10(normpsbO) ~ normKatG + normKatE + Temp_co + Coast_km +  s(POC), data=catmodels_notArctic_new_clean_SRF, method = "REML")

catmodels_notArctic_new_clean_SRF_KatE <- catmodels_notArctic_new_clean_SRF %>%
  arrange(desc(normKatE))

catmodels_notArctic_new_clean_SRF_KatE <- catmodels_notArctic_new_clean_SRF_KatE[-1,]

catmodels_notArctic_new_clean_SRF_KatE_POC <- catmodels_notArctic_new_clean_SRF_KatE %>%
  arrange(desc(POC))

catmodels_notArctic_new_clean_SRF_KatE_POC <- catmodels_notArctic_new_clean_SRF_KatE_POC[-c(1:2),]

psbO_new13 <- mgcv::gam(log10(normpsbO) ~ normKatG + normKatE + Temp_co + Coast_km +  s(POC), data=catmodels_notArctic_new_clean_SRF_KatE_POC, method = "REML")

plot(ggeffects::ggpredict(psbO_new13, terms = "normKatG"), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("Iron concentration (µg/L) at 5 meters") + ylab("katG\ncopy number per genome")

#######katG

catmodels_notArctic_new_clean_SRF_catmn <- catmodels_notArctic_new_clean_SRF%>%
  arrange(desc(normCatMn))

catmodels_notArctic_new_clean_SRF_catmn <- catmodels_notArctic_new_clean_SRF_catmn[-c(1:2),]

katG_new1 <- mgcv::gam(log10(normKatG) ~ s(normCatMn) + s(normpsbO) + s(normKatE) + s(Temp_co) + s(Sal_co) + s(Coast_km) + s(PAR) + s(NO3_NO2_co) + s(PO4_co) + s(Iron_5m) + s(POC), data=catmodels_notArctic_new_clean_SRF, method = "REML")

katG_new2 <- mgcv::gam(log10(normKatG) ~ s(normpsbO) + s(normKatE) + s(Temp_co) + s(Sal_co) + s(Coast_km) + s(PAR) + s(NO3_NO2_co) + s(PO4_co) + s(Iron_5m) + s(POC), data=catmodels_notArctic_new_clean_SRF, method = "REML")

katG_new3 <- mgcv::gam(log10(normKatG) ~ s(normpsbO) + s(Temp_co) + s(Sal_co) + s(Coast_km) + s(PAR) + s(NO3_NO2_co) + s(PO4_co) + s(Iron_5m) + s(POC), data=catmodels_notArctic_new_clean_SRF, method = "REML")

katG_new4 <- mgcv::gam(log10(normKatG) ~ s(normpsbO) + s(Temp_co) + s(Sal_co) + s(PAR) + s(NO3_NO2_co) + s(PO4_co) + s(Iron_5m) + s(POC), data=catmodels_notArctic_new_clean_SRF, method = "REML")

katG_new5 <- mgcv::gam(log10(normKatG) ~ s(normpsbO) + s(Temp_co) + s(Sal_co) + s(NO3_NO2_co) + s(PO4_co) + s(Iron_5m) + s(POC), data=catmodels_notArctic_new_clean_SRF, method = "REML")

katG_new6 <- mgcv::gam(log10(normKatG) ~ s(normpsbO) + s(Temp_co) + s(Sal_co) + s(PO4_co) + s(Iron_5m) + s(POC), data=catmodels_notArctic_new_clean_SRF, method = "REML")

katG_new7 <- mgcv::gam(log10(normKatG) ~ s(normpsbO) + s(Temp_co) + s(Sal_co) + s(Iron_5m) + s(POC), data=catmodels_notArctic_new_clean_SRF, method = "REML")

katG_new8 <- mgcv::gam(log10(normKatG) ~ s(normpsbO) + s(Temp_co) + s(Sal_co) + s(Iron_5m) + s(POC), data=catmodels_notArctic_new_clean_SRF, method = "REML")

katG_new9 <- mgcv::gam(log10(normKatG) ~ normpsbO + s(Temp_co) + s(Sal_co) + s(Iron_5m) + s(POC), data=catmodels_notArctic_new_clean_SRF, method = "REML")

katG_new10 <- mgcv::gam(log10(normKatG) ~ normpsbO + s(Temp_co) + s(Sal_co) + Iron_5m + s(POC), data=catmodels_notArctic_new_clean_SRF, method = "REML")

katG_new11 <- mgcv::gam(log10(normKatG) ~ normpsbO + s(Temp_co) + s(Sal_co) + Iron_5m + s(POC), data=catmodels_notArctic_new_clean_SRF, method = "REML")

katG_new11 <- mgcv::gam(log10(normKatG) ~ normpsbO + s(Temp_co) + s(Sal_co) + Iron_5m + s(POC), data=catmodels_notArctic_new_clean_SRF_sal_POC, method = "REML")


#remove Sal and POC outliers
catmodels_notArctic_new_clean_SRF_sal <- catmodels_notArctic_new_clean_SRF %>%
  arrange((Sal_co))

catmodels_notArctic_new_clean_SRF_sal <- catmodels_notArctic_new_clean_SRF_sal[-1,]

catmodels_notArctic_new_clean_SRF_sal_POC <- catmodels_notArctic_new_clean_SRF_sal %>%
  arrange(desc(POC))

catmodels_notArctic_new_clean_SRF_sal_POC <- catmodels_notArctic_new_clean_SRF_sal_POC[-c(1:2),]

#############katE models
catmodels_notArctic_new_clean_SRF_KatE <- catmodels_notArctic_new_clean_SRF %>%
  arrange(desc(normKatE))

catmodels_notArctic_new_clean_SRF_KatE <- catmodels_notArctic_new_clean_SRF_KatE[-1,]

katE_new1 <- mgcv::gam(log10(normKatE) ~ s(normCatMn) + s(normpsbO) + s(normKatG) + s(Temp_co) + s(Sal_co) + s(Coast_km) + s(PAR) + s(NO3_NO2_co) + s(PO4_co) + s(Iron_5m) + s(POC), data=catmodels_notArctic_new_clean_SRF_KatE, method = "REML")

katE_new2 <- mgcv::gam(log10(normKatE) ~ s(normCatMn) + s(normpsbO) + s(normKatG) + s(Sal_co) + s(Coast_km) + s(PAR) + s(NO3_NO2_co) + s(PO4_co) + s(Iron_5m) + s(POC), data=catmodels_notArctic_new_clean_SRF_KatE, method = "REML")

katE_new3 <- mgcv::gam(log10(normKatE) ~ s(normCatMn) + s(normpsbO) + s(normKatG) + s(Sal_co) + s(Coast_km) + s(PAR) + s(PO4_co) + s(Iron_5m) + s(POC), data=catmodels_notArctic_new_clean_SRF_KatE, method = "REML")

katE_new4 <- mgcv::gam(log10(normKatE) ~ s(normCatMn) + s(normpsbO) + s(normKatG) + s(Sal_co) + s(Coast_km) + s(PAR) + s(Iron_5m) + s(POC), data=catmodels_notArctic_new_clean_SRF_KatE, method = "REML")

#checking if poc is important
catmodels_notArctic_new_clean_SRF_KatE_POC <- catmodels_notArctic_new_clean_SRF_KatE %>%
  arrange(desc(POC))

catmodels_notArctic_new_clean_SRF_KatE_POC <- catmodels_notArctic_new_clean_SRF_KatE_POC[-1,]

katE_new5 <- mgcv::gam(log10(normKatE) ~ s(normCatMn) + s(normpsbO) + s(normKatG) + s(Sal_co) + s(Coast_km) + s(PAR) + s(POC), data=catmodels_notArctic_new_clean_SRF_KatE, method = "REML")

katE_new6 <- mgcv::gam(log10(normKatE) ~ s(normCatMn) + s(normpsbO) + s(normKatG) + s(Sal_co) + s(Coast_km) + s(PAR) + s(POC), data=catmodels_notArctic_new_clean_SRF_KatE, method = "REML")

katE_new7 <- mgcv::gam(log10(normKatE) ~ s(normCatMn) + s(normpsbO) + s(normKatG) + Sal_co + s(Coast_km) + s(PAR) + s(POC), data=catmodels_notArctic_new_clean_SRF_KatE, method = "REML")

katE_new8 <- mgcv::gam(log10(normKatE) ~ s(normCatMn) + s(normpsbO) + s(normKatG) + Sal_co + Coast_km + s(PAR) + s(POC), data=catmodels_notArctic_new_clean_SRF_KatE, method = "REML")

katE_new9 <- mgcv::gam(log10(normKatE) ~ s(normCatMn) + s(normpsbO) + s(normKatG) + Sal_co + Coast_km + s(PAR) + s(POC), data=catmodels_notArctic_new_clean_SRF_KatE, method = "REML")

catmodels_notArctic_new_clean_SRF_KatE_POC_Sal <- catmodels_notArctic_new_clean_SRF_KatE_POC %>%
  arrange(Sal_co)

catmodels_notArctic_new_clean_SRF_KatE_POC_Sal <- catmodels_notArctic_new_clean_SRF_KatE_POC_Sal[-1,]


#I will also need to remove a catmn outlier
catmodels_notArctic_new_clean_SRF_KatE_POC_Sal_CatMn <- catmodels_notArctic_new_clean_SRF_KatE_POC_Sal %>%
  arrange(desc(normCatMn))

catmodels_notArctic_new_clean_SRF_KatE_POC_Sal_CatMn <- catmodels_notArctic_new_clean_SRF_KatE_POC_Sal_CatMn[-1,]

katE_new10 <- mgcv::gam(log10(normKatE) ~ s(normCatMn) + s(normpsbO) + s(normKatG) + Sal_co + Coast_km + s(PAR) + s(POC), data=catmodels_notArctic_new_clean_SRF_KatE_POC_Sal_CatMn, method = "REML")


####catmn
catmodels_notArctic_clean_SRF_CatMn <- catmodels_notArctic_new_clean_SRF %>%
  arrange(desc(normCatMn))

catmodels_notArctic_clean_SRF_CatMn <- catmodels_notArctic_clean_SRF_CatMn[-c(1:2),]


catmn_new1 <- mgcv::gam((normCatMn) ~ s(normKatE) + s(normpsbO) + s(normKatG) + s(Temp_co) + s(Sal_co) + s(Coast_km) + s(PAR) + s(NO3_NO2_co) + s(PO4_co) + s(Iron_5m) + s(POC), data=catmodels_notArctic_clean_SRF_CatMn, method = "REML")

catmn_new2 <- mgcv::gam((normCatMn) ~ s(normKatE) + s(normKatG) + s(Temp_co) + s(Sal_co) + s(Coast_km) + s(PAR) + s(NO3_NO2_co) + s(PO4_co) + s(Iron_5m) + s(POC), data=catmodels_notArctic_clean_SRF_CatMn, method = "REML")











catmodels_notArctic_clean_SRF_forlogcatmn <- catmodels_notArctic_new_clean_SRF

catmodels_notArctic_clean_SRF_forlogcatmn$normCatMn[catmodels_notArctic_clean_SRF_forlogcatmn$normCatMn == 0] <- 0.0000000001

catmodels_notArctic_clean_SRF_forlogcatmn_CatMn <- catmodels_notArctic_clean_SRF_forlogcatmn %>%
  arrange(desc(normCatMn))

catmodels_notArctic_clean_SRF_forlogcatmn_CatMn <- catmodels_notArctic_clean_SRF_forlogcatmn_CatMn[-c(1:2),]

catmn_new1 <- mgcv::gam(log10(normCatMn) ~ s(normKatE) + s(normpsbO) + s(normKatG) + s(Temp_co) + s(Sal_co) + s(Coast_km) + s(PAR) + s(NO3_NO2_co) + s(PO4_co) + s(Iron_5m) + s(POC), data=catmodels_notArctic_clean_SRF_forlogcatmn_CatMn, method = "REML")

catmn_new2 <- mgcv::gam(log10(normCatMn) ~ s(normKatE) + s(normKatG) + s(Temp_co) + s(Sal_co) + s(Coast_km) + s(PAR) + s(NO3_NO2_co) + s(PO4_co) + s(Iron_5m) + s(POC), data=catmodels_notArctic_clean_SRF_forlogcatmn_CatMn, method = "REML")

catmn_new3 <- mgcv::gam(log10(normCatMn) ~ s(normKatE) + s(normKatG) + s(Sal_co) + s(Coast_km) + s(PAR) + s(NO3_NO2_co) + s(PO4_co) + s(Iron_5m) + s(POC), data=catmodels_notArctic_clean_SRF_forlogcatmn_CatMn, method = "REML")

catmn_new4 <- mgcv::gam(log10(normCatMn) ~ s(normKatE) + s(normKatG) + s(Coast_km) + s(PAR) + s(NO3_NO2_co) + s(PO4_co) + s(Iron_5m) + s(POC), data=catmodels_notArctic_clean_SRF_forlogcatmn_CatMn, method = "REML")

##checking if I can drop POC
catmodels_notArctic_clean_SRF_forlogcatmn_CatMn_POC <- catmodels_notArctic_clean_SRF_forlogcatmn_CatMn %>%
  arrange(desc(POC))

catmodels_notArctic_clean_SRF_forlogcatmn_CatMn_POC <- catmodels_notArctic_clean_SRF_forlogcatmn_CatMn_POC[-c(1),]

catmn_new5 <- mgcv::gam(log10(normCatMn) ~ s(normKatE) + s(normKatG) + s(Coast_km) + s(NO3_NO2_co) + s(Iron_5m) + s(POC), data=catmodels_notArctic_clean_SRF_forlogcatmn_CatMn, method = "REML")

catmn_new6 <- mgcv::gam(log10(normCatMn) ~ normKatE + s(normKatG) + s(Coast_km) + s(NO3_NO2_co) + s(Iron_5m) + s(POC), data=catmodels_notArctic_clean_SRF_forlogcatmn_CatMn, method = "REML")

catmn_new7 <- mgcv::gam(log10(normCatMn) ~ normKatE + s(normKatG) + Coast_km + s(NO3_NO2_co) + s(Iron_5m) + s(POC), data=catmodels_notArctic_clean_SRF_forlogcatmn_CatMn, method = "REML")

catmn_new8 <- mgcv::gam(log10(normCatMn) ~ normKatE + s(normKatG) + Coast_km + s(NO3_NO2_co) + Iron_5m + s(POC), data=catmodels_notArctic_clean_SRF_forlogcatmn_CatMn, method = "REML")

catmn_new10 <- mgcv::gam(log10(normCatMn) ~ normKatE + s(normKatG) + Coast_km + s(NO3_NO2_co) + Iron_5m + s(POC), data=catmodels_notArctic_clean_SRF_forlogcatmn_CatMn_POC_KatE, method = "REML")

catmodels_notArctic_clean_SRF_forlogcatmn_CatMn_POC_KatE <- catmodels_notArctic_clean_SRF_forlogcatmn_CatMn_POC %>%
  arrange(desc(normKatE))

catmodels_notArctic_clean_SRF_forlogcatmn_CatMn_POC_KatE <- catmodels_notArctic_clean_SRF_forlogcatmn_CatMn_POC_KatE[-c(1),]


########
plot(ggeffects::ggpredict(KatE_enviro7, terms = "Coast_km"), show_data = TRUE, colors = "darkgrey", dot_size = 2, line_size = 1) + theme_bw() + ggtitle("") + theme(text=element_text(size=12), panel.background = element_blank(),
panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA, linewidth=1)) + xlab("Distance from the coast (km)") + ylab("katE\ncopy number per genome")
################extra models
check <- mgcv::gam(log10(normKatG) ~ s(Coast_km) + s(Iron_5m) + s(proDarwin_umolC.kg) + s(sar11) + s(manganese_dissolved_nmol.kg) + s(CO2_co), data=catmodels_notArctic_new_clean_SRF_extra, method = "REML")

katG_extra1 <- mgcv::gam(log10(normKatG) ~ s(Temp_co) + s(Sal_co) + s(Coast_km) + s(PAR) + s(NO3_NO2_co) + s(PO4_co) + s(Iron_5m) + s(POC) + s(proDarwin_umolC.kg) + s(sar11) + s(manganese_dissolved_nmol.kg) + s(CO2_co), data=catmodels_notArctic_new_clean_SRF_extra, method = "REML")

katG_extra2 <- mgcv::gam(log10(normKatG) ~ s(Sal_co) + s(Coast_km) + s(PAR) + s(NO3_NO2_co) + s(PO4_co) + s(Iron_5m) + s(POC) + s(proDarwin_umolC.kg) + s(sar11) + s(manganese_dissolved_nmol.kg) + s(CO2_co), data=catmodels_notArctic_new_clean_SRF_extra, method = "REML")

katG_extra3 <- mgcv::gam(log10(normKatG) ~ s(Sal_co) + s(PAR) + s(NO3_NO2_co) + s(PO4_co) + s(Iron_5m) + s(POC) + s(proDarwin_umolC.kg) + s(sar11) + s(manganese_dissolved_nmol.kg) + s(CO2_co), data=catmodels_notArctic_new_clean_SRF_extra, method = "REML")

katG_extra4 <- mgcv::gam(log10(normKatG) ~ s(Sal_co) + s(PAR) + s(PO4_co) + s(Iron_5m) + s(POC) + s(proDarwin_umolC.kg) + s(sar11) + s(manganese_dissolved_nmol.kg) + s(CO2_co), data=catmodels_notArctic_new_clean_SRF_extra, method = "REML")

katG_extra5 <- mgcv::gam(log10(normKatG) ~ s(Sal_co) + s(PAR) + s(PO4_co) + s(Iron_5m) + s(proDarwin_umolC.kg) + s(sar11) + s(manganese_dissolved_nmol.kg) + s(CO2_co), data=catmodels_notArctic_new_clean_SRF_extra, method = "REML")

katG_extra6 <- mgcv::gam(log10(normKatG) ~ s(Sal_co) + s(PAR) + s(PO4_co) + s(Iron_5m) + s(proDarwin_umolC.kg) + s(sar11) + s(manganese_dissolved_nmol.kg) + s(CO2_co), data=catmodels_notArctic_new_clean_SRF_extra, method = "REML")

katG_extra7 <- mgcv::gam(log10(normKatG) ~ s(Sal_co) + s(PAR) + PO4_co + s(Iron_5m) + s(proDarwin_umolC.kg) + s(sar11) + s(manganese_dissolved_nmol.kg) + s(CO2_co), data=catmodels_notArctic_new_clean_SRF_extra, method = "REML")

katG_extra8 <- mgcv::gam(log10(normKatG) ~ s(Sal_co) + s(PAR) + PO4_co + s(Iron_5m) + s(proDarwin_umolC.kg) + s(sar11) + manganese_dissolved_nmol.kg + s(CO2_co), data=catmodels_notArctic_new_clean_SRF_extra, method = "REML")

katG_extra9 <- mgcv::gam(log10(normKatG) ~ s(Sal_co) + s(PAR) + PO4_co + s(Iron_5m) + s(proDarwin_umolC.kg) + s(sar11) + manganese_dissolved_nmol.kg + CO2_co, data=catmodels_notArctic_new_clean_SRF_extra, method = "REML")
######
catmodels_notArctic_new_clean_SRF_extra_Sal <- catmodels_notArctic_new_clean_SRF_extra %>%
  arrange((Sal_co))

catmodels_notArctic_new_clean_SRF_extra_Sal <- catmodels_notArctic_new_clean_SRF_extra_Sal[-c(1),]



#removing sal outlier and refitting the model
katG_extra9 <- mgcv::gam(log10(normKatG) ~ s(Sal_co) + s(PAR) + PO4_co + s(Iron_5m) + s(proDarwin_umolC.kg) + s(sar11) + manganese_dissolved_nmol.kg + CO2_co, data=catmodels_notArctic_new_clean_SRF_extra_Sal, method = "REML")

###building from the original model

katG_new11_extra <- mgcv::gam(log10(normKatG) ~ normpsbO + s(Temp_co) + s(Sal_co) + Iron_5m + s(POC) + s(proDarwin_umolC.kg) + s(sar11) + s(manganese_dissolved_nmol.kg) + s(CO2_co), data=catmodels_notArctic_new_clean_SRF_extra, method = "REML")

katG_new12_extra <- mgcv::gam(log10(normKatG) ~ normpsbO + s(Temp_co) + s(Sal_co) + Iron_5m + s(POC) + s(proDarwin_umolC.kg) + s(sar11) + s(manganese_dissolved_nmol.kg) + s(CO2_co), data=catmodels_notArctic_new_clean_SRF_extra, method = "REML")

katG_new13_extra <- mgcv::gam(log10(normKatG) ~ normpsbO + s(Temp_co) + s(Sal_co) + Iron_5m + s(POC) + s(proDarwin_umolC.kg) + s(sar11) + manganese_dissolved_nmol.kg + s(CO2_co), data=catmodels_notArctic_new_clean_SRF_extra, method = "REML")

katG_new14_extra <- mgcv::gam(log10(normKatG) ~ normpsbO + s(Temp_co) + s(Sal_co) + Iron_5m + s(POC) + s(proDarwin_umolC.kg) + s(sar11) + manganese_dissolved_nmol.kg + s(CO2_co), data=catmodels_notArctic_new_clean_SRF_extra, method = "REML")

#need to remove a POC and a Salinity outlier then refit

catmodels_notArctic_new_clean_SRF_extra_Sal_POC <- catmodels_notArctic_new_clean_SRF_extra_Sal %>%
  arrange(desc(POC))

catmodels_notArctic_new_clean_SRF_extra_Sal_POC <- catmodels_notArctic_new_clean_SRF_extra_Sal_POC[-1,]

katG_new14_extra <- mgcv::gam(log10(normKatG) ~ normpsbO + s(Temp_co) + s(Sal_co) + Iron_5m + s(POC) + s(proDarwin_umolC.kg) + s(sar11) + manganese_dissolved_nmol.kg + s(CO2_co), data=catmodels_notArctic_new_clean_SRF_extra_Sal_POC, method = "REML")


################
katG_biomass <- mgcv::gam(log10(normKatG) ~ s(normpsbO) + s(normKatE) + s(normCatMn) + (proDarwin_umolC.kg) + s(sar11), data=catmodels_notArctic_new_clean_SRF_extra, method = "REML")

katG_biomass2 <- mgcv::gam(log10(normKatG) ~ s(normKatE) + s(normCatMn) + (proDarwin_umolC.kg) + s(sar11), data=catmodels_notArctic_new_clean_SRF_extra, method = "REML")

katG_biomass3 <- mgcv::gam(log10(normKatG) ~ s(normKatE) + (proDarwin_umolC.kg) + s(sar11), data=catmodels_notArctic_new_clean_SRF_extra, method = "REML")

katG_biomass4 <- mgcv::gam(log10(normKatG) ~ s(normKatE) + s(sar11), data=catmodels_notArctic_new_clean_SRF_extra, method = "REML")

katG_biomass5 <- mgcv::gam(log10(normKatG) ~ s(normKatE) + s(sar11), data=catmodels_notArctic_new_clean_SRF_extra, method = "REML")

catmodels_notArctic_new_clean_SRF_extra_KatE <- catmodels_notArctic_new_clean_SRF_extra %>%
  arrange(desc(normKatE))

catmodels_notArctic_new_clean_SRF_extra_KatE <- catmodels_notArctic_new_clean_SRF_extra_KatE[-1,]

katG_biomass6 <- mgcv::gam(log10(normKatG) ~ s(normKatE) + s(sar11), data=catmodels_notArctic_new_clean_SRF_extra, method = "REML")

katG_biomass7 <- mgcv::gam(log10(normKatG) ~ s(normKatE) + sar11, data=catmodels_notArctic_new_clean_SRF_extra_KatE, method = "REML")
##################
katG_abiotic <- mgcv::gam(log10(normKatG) ~ s(Temp_co) + s(Sal_co) + s(Coast_km) + s(PAR) + s(NO3_NO2_co) + s(PO4_co) + s(Iron_5m) + s(POC) + s(manganese_dissolved_nmol.kg) + s(CO2_co), data=catmodels_notArctic_new_clean_SRF_extra, method = "REML")

katG_abiotic2 <- mgcv::gam(log10(normKatG) ~ s(Temp_co) + s(Sal_co) + s(Coast_km) + s(PAR) + s(PO4_co) + s(Iron_5m) + s(POC) + s(manganese_dissolved_nmol.kg) + s(CO2_co), data=catmodels_notArctic_new_clean_SRF_extra, method = "REML")

katG_abiotic3 <- mgcv::gam(log10(normKatG) ~ s(Temp_co) + s(Sal_co) + s(Coast_km) + s(PAR) + s(PO4_co) + s(Iron_5m)  + s(manganese_dissolved_nmol.kg) + s(CO2_co), data=catmodels_notArctic_new_clean_SRF_extra, method = "REML")

katG_abiotic4 <- mgcv::gam(log10(normKatG) ~ s(Temp_co) + s(Sal_co) + s(Coast_km) + s(PAR) + s(PO4_co) + s(Iron_5m)  + s(manganese_dissolved_nmol.kg) + s(CO2_co), data=catmodels_notArctic_new_clean_SRF_extra, method = "REML")

katG_abiotic5 <- mgcv::gam(log10(normKatG) ~ s(Sal_co) + s(Coast_km) + s(PAR) + s(PO4_co) + s(Iron_5m)  + s(manganese_dissolved_nmol.kg) + s(CO2_co), data=catmodels_notArctic_new_clean_SRF_extra, method = "REML")

katG_abiotic6 <- mgcv::gam(log10(normKatG) ~ s(Sal_co) + s(PAR) + s(PO4_co) + s(Iron_5m)  + s(manganese_dissolved_nmol.kg) + s(CO2_co), data=catmodels_notArctic_new_clean_SRF_extra, method = "REML")

katG_abiotic7 <- mgcv::gam(log10(normKatG) ~ s(Sal_co) + s(PO4_co) + s(Iron_5m) + s(manganese_dissolved_nmol.kg) + s(CO2_co), data=catmodels_notArctic_new_clean_SRF_extra, method = "REML")

katG_abiotic8 <- mgcv::gam(log10(normKatG) ~ s(Sal_co) + s(PO4_co) + s(Iron_5m) + s(manganese_dissolved_nmol.kg), data=catmodels_notArctic_new_clean_SRF_extra, method = "REML")

katG_abiotic9 <- mgcv::gam(log10(normKatG) ~ s(Sal_co) + PO4_co + s(Iron_5m) + s(manganese_dissolved_nmol.kg), data=catmodels_notArctic_new_clean_SRF_extra, method = "REML")

katG_abiotic10 <- mgcv::gam(log10(normKatG) ~ s(Sal_co) + PO4_co + s(Iron_5m) + s(manganese_dissolved_nmol.kg), data=catmodels_notArctic_new_clean_SRF_extra_Sal, method = "REML")

#####katE
katE_all <- mgcv::gam(log10(normKatE) ~ s(normCatMn) + s(normpsbO) + s(normKatG) + s(Temp_co) + s(Sal_co) + s(Coast_km) + s(PAR) + s(NO3_NO2_co) + s(PO4_co) + s(Iron_5m) + s(POC) + s(proDarwin_umolC.kg) + s(sar11) + manganese_dissolved_nmol.kg + s(CO2_co), data=catmodels_notArctic_new_clean_SRF_extra_KatE, method = "REML")

katE_all2 <- mgcv::gam(log10(normKatE) ~ s(normCatMn) + s(normpsbO) + s(normKatG) + s(Temp_co) + s(Sal_co) + s(PAR) + s(NO3_NO2_co) + s(PO4_co) + s(Iron_5m) + s(POC) + s(proDarwin_umolC.kg) + s(sar11) + s(manganese_dissolved_nmol.kg) + s(CO2_co), data=catmodels_notArctic_new_clean_SRF_extra_KatE, method = "REML")

katE_all3 <- mgcv::gam(log10(normKatE) ~ s(normCatMn) + s(normpsbO) + s(normKatG) + s(Temp_co) + s(Sal_co) + s(PAR) + s(NO3_NO2_co) + s(PO4_co) + s(Iron_5m) + s(POC) + s(proDarwin_umolC.kg) + s(sar11) + s(manganese_dissolved_nmol.kg), data=catmodels_notArctic_new_clean_SRF_extra_KatE, method = "REML")

####can't go any further, so I think there are too may predictors here. I will try again but remove the other catalase genes
katE_all <- mgcv::gam(log10(normKatE) ~ s(Temp_co) + s(Sal_co) + s(Coast_km) + s(PAR) + s(NO3_NO2_co) + s(PO4_co) + s(Iron_5m) + s(POC) + s(proDarwin_umolC.kg) + s(sar11) + s(manganese_dissolved_nmol.kg) + s(CO2_co), data=catmodels_notArctic_new_clean_SRF_extra_KatE, method = "REML")

katE_all2 <- mgcv::gam(log10(normKatE) ~ s(Temp_co) + s(Sal_co) + s(NO3_NO2_co) + s(PO4_co) + s(PAR) + s(Iron_5m) + s(POC) + s(proDarwin_umolC.kg) + s(sar11) + s(manganese_dissolved_nmol.kg) + s(CO2_co), data=catmodels_notArctic_new_clean_SRF_extra_KatE, method = "REML")

katE_all3 <- mgcv::gam(log10(normKatE) ~ s(Temp_co) + s(Sal_co) + s(NO3_NO2_co) + s(PAR) + s(Iron_5m) + s(POC) + s(proDarwin_umolC.kg) + s(sar11) + s(manganese_dissolved_nmol.kg) + s(CO2_co), data=catmodels_notArctic_new_clean_SRF_extra_KatE, method = "REML")

katE_all4 <- mgcv::gam(log10(normKatE) ~ s(Temp_co) + s(Sal_co) + s(NO3_NO2_co) + s(PAR) + s(Iron_5m) + s(POC) + s(proDarwin_umolC.kg) + s(sar11) + s(manganese_dissolved_nmol.kg) + s(CO2_co), data=catmodels_notArctic_new_clean_SRF_extra_KatE, method = "REML")

#######
katE_abiotic <- mgcv::gam(log10(normKatE) ~ s(Temp_co) + s(Sal_co) + s(Coast_km) + s(PAR) + s(NO3_NO2_co) + s(PO4_co) + s(Iron_5m) + s(POC) + s(manganese_dissolved_nmol.kg) + s(CO2_co), data=catmodels_notArctic_new_clean_SRF_extra_KatE, method = "REML")

katE_abiotic2 <- mgcv::gam(log10(normKatE) ~ s(Sal_co) + s(Coast_km) + s(PAR) + s(NO3_NO2_co) + s(PO4_co) + s(Iron_5m) + s(POC) + s(manganese_dissolved_nmol.kg) + s(CO2_co), data=catmodels_notArctic_new_clean_SRF_extra_KatE, method = "REML")

katE_abiotic3 <- mgcv::gam(log10(normKatE) ~  s(Coast_km) + s(PAR) + s(NO3_NO2_co) + s(PO4_co) + s(Iron_5m) + s(POC) + s(manganese_dissolved_nmol.kg) + s(CO2_co), data=catmodels_notArctic_new_clean_SRF_extra_KatE, method = "REML")

katE_abiotic4 <- mgcv::gam(log10(normKatE) ~  s(Coast_km) + s(PAR) + s(PO4_co) + s(Iron_5m) + s(POC) + s(manganese_dissolved_nmol.kg) + s(CO2_co), data=catmodels_notArctic_new_clean_SRF_extra_KatE, method = "REML")

katE_abiotic5 <- mgcv::gam(log10(normKatE) ~  s(Coast_km) + s(PAR) + s(PO4_co)  + s(POC) + s(manganese_dissolved_nmol.kg) + s(CO2_co), data=catmodels_notArctic_new_clean_SRF_extra_KatE, method = "REML")

katE_abiotic6 <- mgcv::gam(log10(normKatE) ~  s(Coast_km) + s(PAR) + s(PO4_co) + s(manganese_dissolved_nmol.kg) + s(CO2_co), data=catmodels_notArctic_new_clean_SRF_extra_KatE, method = "REML")

katE_abiotic7 <- mgcv::gam(log10(normKatE) ~  s(Coast_km) + s(PAR) + s(PO4_co) + s(manganese_dissolved_nmol.kg) + s(CO2_co), data=catmodels_notArctic_new_clean_SRF_extra_KatE, method = "REML")

katE_abiotic8 <- mgcv::gam(log10(normKatE) ~  s(Coast_km) + s(PAR) + s(PO4_co)  + s(CO2_co), data=catmodels_notArctic_new_clean_SRF_extra_KatE, method = "REML")

katE_abiotic9 <- mgcv::gam(log10(normKatE) ~s(Coast_km) + s(PAR) + s(PO4_co), data=catmodels_notArctic_new_clean_SRF_extra_KatE, method = "REML")

katE_abiotic10 <- mgcv::gam(log10(normKatE) ~Coast_km + s(PAR) + s(PO4_co), data=catmodels_notArctic_new_clean_SRF_extra_KatE, method = "REML")

katE_abiotic11 <- mgcv::gam(log10(normKatE) ~Coast_km + s(PAR) + PO4_co, data=catmodels_notArctic_new_clean_SRF_extra_KatE, method = "REML")


catmodels_notArctic_new_clean_SRF_extra_KatE_POC <- catmodels_notArctic_new_clean_SRF_extra_KatE %>%
  arrange(desc(Sal_co))

catmodels_notArctic_new_clean_SRF_extra_KatE_POC <- catmodels_notArctic_new_clean_SRF_extra_KatE_POC[-1,]

##katE biotic factors
katE_biotic <- mgcv::gam(log10(normKatE) ~ s(normpsbO) + s(normKatG) + s(normCatMn) + s(proDarwin_umolC.kg) + s(sar11), data=catmodels_notArctic_new_clean_SRF_extra_KatE, method = "REML")

katE_biotic2 <- mgcv::gam(log10(normKatE) ~ s(normpsbO) + s(normKatG) + s(normCatMn) + s(proDarwin_umolC.kg) + s(sar11), data=catmodels_notArctic_new_clean_SRF_extra_KatE, method = "REML")

katE_biotic3 <- mgcv::gam(log10(normKatE) ~ s(normpsbO) + s(normKatG) + s(normCatMn) + proDarwin_umolC.kg + s(sar11), data=catmodels_notArctic_new_clean_SRF_extra_KatE, method = "REML")

katE_biotic4 <- mgcv::gam(log10(normKatE) ~ s(normpsbO) + s(normKatG) + s(normCatMn) + proDarwin_umolC.kg + s(sar11), data=catmodels_notArctic_new_clean_SRF_extra_KatE, method = "REML")

#need to remove the catmn outliers
catmodels_notArctic_new_clean_SRF_extra_KatE_CatMn <- catmodels_notArctic_new_clean_SRF_extra_KatE %>%
  arrange(desc(normCatMn))

catmodels_notArctic_new_clean_SRF_extra_KatE_CatMn <- catmodels_notArctic_new_clean_SRF_extra_KatE_CatMn[-c(1:2),]

katE_biotic4 <- mgcv::gam(log10(normKatE) ~ s(normpsbO) + s(normKatG) + s(normCatMn) + proDarwin_umolC.kg + s(sar11), data=catmodels_notArctic_new_clean_SRF_extra_KatE_CatMn, method = "REML")

###only biomass
katE_biomass <- mgcv::gam(log10(normKatE) ~ s(proDarwin_umolC.kg) + s(sar11), data=catmodels_notArctic_new_clean_SRF_extra_KatE, method = "REML")

katE_biomass2 <- mgcv::gam(log10(normKatE) ~ s(sar11), data=catmodels_notArctic_new_clean_SRF_extra_KatE, method = "REML")

katE_biomass3 <- mgcv::gam(log10(normKatE) ~ 1, data=catmodels_notArctic_new_clean_SRF_extra_KatE, method = "REML")

##########model reduction with only new predictors to build off of best model
catmodels_notArctic_new_clean_SRF_extra_KatE_CatMn_POC <- catmodels_notArctic_new_clean_SRF_extra_KatE_CatMn %>%
  arrange(desc(POC))

catmodels_notArctic_new_clean_SRF_extra_KatE_CatMn_POC <- catmodels_notArctic_new_clean_SRF_extra_KatE_CatMn_POC[-1,]

katE_new10_extra <- mgcv::gam(log10(normKatE) ~ s(normCatMn) + s(normpsbO) + s(normKatG) + Sal_co + Coast_km + s(PAR) + s(POC) + s(proDarwin_umolC.kg) + s(sar11) + s(manganese_dissolved_nmol.kg) + s(CO2_co), data=catmodels_notArctic_new_clean_SRF_extra_KatE_CatMn_POC, method = "REML")

katE_new10_extra2 <- mgcv::gam(log10(normKatE) ~ s(normCatMn) + s(normpsbO) + s(normKatG) + Sal_co + Coast_km + s(PAR) + s(POC) + s(proDarwin_umolC.kg) + s(sar11) + s(manganese_dissolved_nmol.kg) + s(CO2_co), data=catmodels_notArctic_new_clean_SRF_extra_KatE_CatMn_POC, method = "REML")

katE_new10_extra3 <- mgcv::gam(log10(normKatE) ~ s(normCatMn) + s(normpsbO) + s(normKatG) + Sal_co + Coast_km + s(PAR) + s(POC) + proDarwin_umolC.kg + s(sar11) + s(manganese_dissolved_nmol.kg) + s(CO2_co), data=catmodels_notArctic_new_clean_SRF_extra_KatE_CatMn_POC, method = "REML")

katE_new10_extra4 <- mgcv::gam(log10(normKatE) ~ s(normCatMn) + s(normpsbO) + s(normKatG) + Sal_co + Coast_km + s(PAR) + s(POC) + proDarwin_umolC.kg + s(sar11) + s(manganese_dissolved_nmol.kg) + s(CO2_co), data=catmodels_notArctic_new_clean_SRF_extra_KatE_CatMn_POC, method = "REML")

####CatMn abiotic

catmodels_notArctic_clean_SRF_forlogcatmn <- catmodels_notArctic_new_clean_SRF_extra

catmodels_notArctic_clean_SRF_forlogcatmn$normCatMn[catmodels_notArctic_clean_SRF_forlogcatmn$normCatMn == 0] <- 0.0000000001

catmodels_notArctic_clean_SRF_forlogcatmn_CatMn <- catmodels_notArctic_clean_SRF_forlogcatmn %>%
  arrange(desc(normCatMn))

catmodels_notArctic_clean_SRF_forlogcatmn_CatMn <- catmodels_notArctic_clean_SRF_forlogcatmn_CatMn[-c(1:2),]


catmn_abiotic <- mgcv::gam(log10(normCatMn) ~ s(Temp_co) + s(Sal_co) + s(Coast_km) + s(PAR) + s(NO3_NO2_co) + s(PO4_co) + s(Iron_5m) + s(POC) + s(manganese_dissolved_nmol.kg) + s(CO2_co), data=catmodels_notArctic_clean_SRF_forlogcatmn_CatMn, method = "REML")

catmn_abiotic2 <- mgcv::gam(log10(normCatMn) ~ s(Sal_co) + s(Coast_km) + s(PAR) + s(NO3_NO2_co) + s(PO4_co) + s(Iron_5m) + s(POC) + s(manganese_dissolved_nmol.kg) + s(CO2_co), data=catmodels_notArctic_clean_SRF_forlogcatmn_CatMn, method = "REML")

catmn_abiotic3 <- mgcv::gam(log10(normCatMn) ~ s(Sal_co) + s(PAR) + s(NO3_NO2_co) + s(PO4_co) + s(Iron_5m) + s(POC) + s(manganese_dissolved_nmol.kg) + s(CO2_co), data=catmodels_notArctic_clean_SRF_forlogcatmn_CatMn, method = "REML")

catmn_abiotic4 <- mgcv::gam(log10(normCatMn) ~ s(PAR) + s(NO3_NO2_co) + s(PO4_co) + s(Iron_5m) + s(POC) + s(manganese_dissolved_nmol.kg) + s(CO2_co), data=catmodels_notArctic_clean_SRF_forlogcatmn_CatMn, method = "REML")

catmn_abiotic5 <- mgcv::gam(log10(normCatMn) ~ s(PAR) + s(NO3_NO2_co) + s(Iron_5m) + s(POC) + s(manganese_dissolved_nmol.kg) + s(CO2_co), data=catmodels_notArctic_clean_SRF_forlogcatmn_CatMn, method = "REML")

catmn_abiotic6 <- mgcv::gam(log10(normCatMn) ~ s(NO3_NO2_co) + s(Iron_5m) + s(POC) + s(manganese_dissolved_nmol.kg) + s(CO2_co), data=catmodels_notArctic_clean_SRF_forlogcatmn_CatMn, method = "REML")

catmn_abiotic7 <- mgcv::gam(log10(normCatMn) ~ s(NO3_NO2_co) + s(Iron_5m) + s(manganese_dissolved_nmol.kg) + s(CO2_co), data=catmodels_notArctic_clean_SRF_forlogcatmn_CatMn, method = "REML")

catmn_abiotic8 <- mgcv::gam(log10(normCatMn) ~ s(NO3_NO2_co) + s(Iron_5m) + s(manganese_dissolved_nmol.kg) + s(CO2_co), data=catmodels_notArctic_clean_SRF_forlogcatmn_CatMn, method = "REML")

catmn_abiotic9 <- mgcv::gam(log10(normCatMn) ~ NO3_NO2_co + s(Iron_5m) + s(manganese_dissolved_nmol.kg) + s(CO2_co), data=catmodels_notArctic_clean_SRF_forlogcatmn_CatMn, method = "REML")

catmn_abiotic10 <- mgcv::gam(log10(normCatMn) ~ NO3_NO2_co + s(Iron_5m) + manganese_dissolved_nmol.kg + s(CO2_co), data=catmodels_notArctic_clean_SRF_forlogcatmn_CatMn, method = "REML")

catmn_abiotic11 <- mgcv::gam(log10(normCatMn) ~ NO3_NO2_co + s(Iron_5m) + manganese_dissolved_nmol.kg + CO2_co, data=catmodels_notArctic_clean_SRF_forlogcatmn_CatMn, method = "REML")

########

catmn_biotic <- mgcv::gam(log10(normCatMn) ~ s(normpsbO) + s(normKatG) + s(normKatE) + s(proDarwin_umolC.kg) + s(sar11), data=catmodels_notArctic_clean_SRF_forlogcatmn_CatMn, method = "REML")

catmn_biotic2 <- mgcv::gam(log10(normCatMn) ~ s(normpsbO) + s(normKatE) + s(proDarwin_umolC.kg) + s(sar11), data=catmodels_notArctic_clean_SRF_forlogcatmn_CatMn, method = "REML")

catmn_biotic3 <-mgcv::gam(log10(normCatMn) ~ s(normpsbO) + s(normKatE) + s(proDarwin_umolC.kg) + s(sar11), data=catmodels_notArctic_clean_SRF_forlogcatmn_CatMn, method = "REML")

catmn_biotic4 <-mgcv::gam(log10(normCatMn) ~ s(normpsbO) + s(normKatE) + s(proDarwin_umolC.kg) + s(sar11), data=catmodels_notArctic_clean_SRF_forlogcatmn_CatMn, method = "REML")

catmodels_notArctic_clean_SRF_forlogcatmn_CatMn_KatE <- catmodels_notArctic_clean_SRF_forlogcatmn_CatMn %>%
  arrange(desc(normKatE))

catmodels_notArctic_clean_SRF_forlogcatmn_CatMn_KatE <- catmodels_notArctic_clean_SRF_forlogcatmn_CatMn_KatE[-1,]

catmn_biotic4 <-mgcv::gam(log10(normCatMn) ~ s(normpsbO) + s(normKatE) + s(proDarwin_umolC.kg) + s(sar11), data=catmodels_notArctic_clean_SRF_forlogcatmn_CatMn_KatE, method = "REML")


#count of taxa with no catalases

SAR202 <- catalases %>%
  mutate(sumcatalase = n_KatG + n_KatE + n_CatMn) %>%
  group_by(Order) %>%
  filter(Order== "UBA6926" | Order== "UBA2985"| Order== "UBA6952"| Order== "UBA2963"| Order== "UBA3495"| Order== "GCA-2717565"| Order== "UBA1127"| Order== "SAR202"| Order== "UBA1151") %>%
  summarise(totalcat = sum(sumcatalase),
            n = n())
```
